{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20b8d80b-9eb0-4846-bb21-d80d9882e4c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script false --no-raise-error\n",
    "#Copy this code and run it in the terminal everytime you create a new directory or file\n",
    "#Gives permission for everyone to work with the files created\n",
    "#Run at the end of a jupyter script, the output is too messy\n",
    "!cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork \n",
    "!chmod -R 777 . "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a29ef54-a06b-4be8-8d13-ac484ca8d09a",
   "metadata": {},
   "source": [
    "# Download cDNA library \n",
    "From ensembl: https://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/ and unzip."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "359a5dbb-293e-4af1-9006-f155830ee599",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "File 'data/Homo_sapiens.GRCh38.cdna.all.fa.gz' already there; not retrieving.\n",
      "gzip: data/Homo_sapiens.GRCh38.cdna.all.fa already exists;\tnot overwritten\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# -nc makes sure that if the file is already there, it won't be overwritten.\n",
    "# -nc can be changed to -N, if it is of interest to import newer versions\n",
    "# -P will save it to the directory specified, in this case data/cDNA_library\n",
    "wget -nc ftp://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz -P data/\n",
    "gzip -d data/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "rm data/Homo_sapiens.GRCh38.cdna.all.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea4c3af5-18a2-4d28-8772-13d6a876ca6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ea8812a-8e96-48cb-a70c-de2ecd4169c6",
   "metadata": {},
   "source": [
    "# Build Bowtie index\n",
    "This takes around 30 minutes to run on one kernel, so don't run it if you don't have to."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "41402daf-553c-485b-b530-cf740db5a805",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"data/Bowtie_index/hg38_cdna_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  data/Homo_sapiens.GRCh38.cdna.all.fa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:03\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:02\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:32\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:04\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:03\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 40214484 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:37\n",
      "Returning block of 40214485 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 66480457 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:42\n",
      "Returning block of 66480458 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 68586474 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:48\n",
      "Returning block of 68586475 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 49097504 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:00\n",
      "Returning block of 49097505 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 65349089 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:38\n",
      "Returning block of 65349090 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 28965054 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:06\n",
      "Returning block of 28965055 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 72809500 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:53\n",
      "Returning block of 72809501 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 3519685 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:07\n",
      "Returning block of 3519686 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/Bowtie_index/hg38_cdna_index.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/Bowtie_index/hg38_cdna_index.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:18:19\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:03\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:32\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:03\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:04\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 69612405 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:50\n",
      "Returning block of 69612406 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 48012854 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:56\n",
      "Returning block of 48012855 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 53601703 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:07\n",
      "Returning block of 53601704 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 50816782 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:02\n",
      "Returning block of 50816783 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 49006051 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:52\n",
      "Returning block of 49006052 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 25671179 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:02\n",
      "Returning block of 25671180 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 58667016 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:22\n",
      "Returning block of 58667017 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 39634257 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:35\n",
      "Returning block of 39634258 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/Bowtie_index/hg38_cdna_index.rev.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/Bowtie_index/hg38_cdna_index.rev.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:18:14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Renaming data/Bowtie_index/hg38_cdna_index.3.bt2.tmp to data/Bowtie_index/hg38_cdna_index.3.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.4.bt2.tmp to data/Bowtie_index/hg38_cdna_index.4.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.1.bt2.tmp to data/Bowtie_index/hg38_cdna_index.1.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.2.bt2.tmp to data/Bowtie_index/hg38_cdna_index.2.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.rev.1.bt2.tmp to data/Bowtie_index/hg38_cdna_index.rev.1.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.rev.2.bt2.tmp to data/Bowtie_index/hg38_cdna_index.rev.2.bt2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir data/Bowtie_index\n",
    "bowtie2-build data/Homo_sapiens.GRCh38.cdna.all.fa data/Bowtie_index/hg38_cdna_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "48a43be5-1743-404d-9785-b9f5e201722f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "hg38_cdna_index.1.bt2  hg38_cdna_index.3.bt2  hg38_cdna_index.rev.1.bt2\n",
      "hg38_cdna_index.2.bt2  hg38_cdna_index.4.bt2  hg38_cdna_index.rev.2.bt2\n",
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "ERR950158_trimmed_1.fastq.gz  ERR950178_trimmed_2.fastq.gz\n",
      "ERR950158_trimmed_2.fastq.gz  ERR950179_trimmed_1.fastq.gz\n",
      "ERR950159_trimmed_1.fastq.gz  ERR950179_trimmed_2.fastq.gz\n",
      "ERR950159_trimmed_2.fastq.gz  ERR950180_trimmed_1.fastq.gz\n",
      "ERR950160_trimmed_1.fastq.gz  ERR950180_trimmed_2.fastq.gz\n",
      "ERR950160_trimmed_2.fastq.gz  ERR950181_trimmed_1.fastq.gz\n",
      "ERR950161_trimmed_1.fastq.gz  ERR950181_trimmed_2.fastq.gz\n",
      "ERR950161_trimmed_2.fastq.gz  ERR950182_trimmed_1.fastq.gz\n",
      "ERR950162_trimmed_1.fastq.gz  ERR950182_trimmed_2.fastq.gz\n",
      "ERR950162_trimmed_2.fastq.gz  ERR950186_trimmed_1.fastq.gz\n",
      "ERR950163_trimmed_1.fastq.gz  ERR950186_trimmed_2.fastq.gz\n",
      "ERR950163_trimmed_2.fastq.gz  ERR950187_trimmed_1.fastq.gz\n",
      "ERR950177_trimmed_1.fastq.gz  ERR950187_trimmed_2.fastq.gz\n",
      "ERR950177_trimmed_2.fastq.gz  fastqc_subset_trimmed\n",
      "ERR950178_trimmed_1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!ls data/Bowtie_index\n",
    "!ls data/subset_trimmed_RNA-seq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bc571cc3-585c-444a-a60a-a7b35d590787",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning ERR950158_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950158_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950158_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950158_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950158_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950158_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950158_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950159_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950159_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950159_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950159_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950159_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950159_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950159_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950160_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950160_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950160_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950160_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950160_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950160_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950160_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950161_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950161_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950161_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950161_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950161_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950161_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950161_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950162_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950162_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950162_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950162_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950162_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950162_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950162_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950163_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950163_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950163_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950163_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950163_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950163_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950163_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950177_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950177_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950177_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950177_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950177_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950177_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950177_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950178_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950178_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950178_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950178_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950178_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950178_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950178_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950179_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950179_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950179_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950179_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950179_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950179_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950179_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950180_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950180_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950180_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950180_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950180_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950180_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950180_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950181_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950181_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950181_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950181_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950181_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950181_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950181_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950182_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950182_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950182_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950182_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950182_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950182_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950182_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950186_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950186_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950186_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950186_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950186_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950186_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950186_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950187_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950187_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950187_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950187_trimmed.bam\" : No such file or directory\n",
      "samtools index: failed to open \"alignment_Bowtie2_subset/ERR950187_trimmed.bam\": No such file or directory\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950187_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950187_trimmed\n",
      "-----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p alignment_Bowtie2_subset\n",
    "\n",
    "# Loop over all FASTQ files ending in _1.fastq.gz in the subset_trimmed_RNA-seq folder\n",
    "for R1 in data/subset_trimmed_RNA-seq/*_1.fastq.gz; do\n",
    "  \n",
    "  # Extract the base name \n",
    "  BN=$(basename \"$R1\")\n",
    "  \n",
    "  # Remove the _1.fastq.gz suffix to get the sample prefix \n",
    "  SAMPLE=${BN%%_1.fastq.gz}\n",
    "  \n",
    "  # Construct the path to the mate 2 file \n",
    "  R2=\"data/subset_trimmed_RNA-seq/${SAMPLE}_2.fastq.gz\"\n",
    "  \n",
    "  echo \"Aligning $SAMPLE with Bowtie2...\"\n",
    "  \n",
    "  # Run Bowtie2 in local alignment mode for the paired-end reads\n",
    "  bowtie2 --local \n",
    "    -x data/hg38_cdna_index \n",
    "    -1 \"$R1\" \n",
    "    -2 \"$R2\" \n",
    "    -S alignment_Bowtie2_subset/${SAMPLE}.sam \n",
    "    --threads 4\n",
    "  \n",
    "  # Convert the SAM to a sorted BAM and create an index using samtools\n",
    "  samtools view -bS alignment_Bowtie2_subset/${SAMPLE}.sam | samtools sort -o alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  samtools index alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  \n",
    "  # Remove the SAM file to save disk space\n",
    "  rm alignment_Bowtie2_subset/${SAMPLE}.sam\n",
    "  \n",
    "  echo \"Done aligning $SAMPLE\"\n",
    "  echo \"-----------------------------------------\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4f529bb9-c1bd-4769-9490-a0e799b23fac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "total 1464931\n",
      "-rw-r--r-- 1 nobody nogroup  83164222 Feb 25 20:59 ERR950158_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   6049096 Feb 26 10:08 ERR950158_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  82982403 Feb 25 21:02 ERR950159_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5178176 Feb 26 10:08 ERR950159_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  83563426 Feb 25 21:05 ERR950160_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5964464 Feb 26 10:08 ERR950160_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  80792165 Feb 25 21:08 ERR950161_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5849536 Feb 26 10:08 ERR950161_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  69514274 Feb 25 21:11 ERR950162_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   4070304 Feb 26 10:08 ERR950162_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  84502479 Feb 25 21:14 ERR950163_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5878184 Feb 26 10:08 ERR950163_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  79448036 Feb 25 21:18 ERR950177_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5347400 Feb 26 10:08 ERR950177_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  81779036 Feb 25 21:21 ERR950178_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5591712 Feb 26 10:08 ERR950178_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  82932454 Feb 25 21:24 ERR950179_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5707192 Feb 26 10:08 ERR950179_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  82446219 Feb 25 21:27 ERR950180_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5620408 Feb 26 10:08 ERR950180_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  82012189 Feb 25 21:31 ERR950181_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5575584 Feb 26 10:08 ERR950181_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  80420591 Feb 25 21:34 ERR950182_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5504848 Feb 26 10:08 ERR950182_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup  81349387 Feb 25 21:38 ERR950186_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   5316016 Feb 26 10:09 ERR950186_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup       616 Feb 26 10:29 ERR950187.log\n",
      "-rw-r--r-- 1 nobody nogroup  76373426 Feb 25 21:41 ERR950187_trimmed.bam\n",
      "-rw-r--r-- 1 nobody nogroup   4899720 Feb 26 10:09 ERR950187_trimmed.bam.bai\n",
      "-rw-r--r-- 1 nobody nogroup 292248577 Feb 26 10:29 ERR950187_trimmed_1.sam\n"
     ]
    }
   ],
   "source": [
    "!ls -l data/alignment_Bowtie2_subset/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8553eeaa-f81f-4dfc-9e51-e65e897bd0df",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "998450 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "998450 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "450566 + 0 mapped (45.13% : N/A)\n",
      "450566 + 0 primary mapped (45.13% : N/A)\n",
      "998450 + 0 paired in sequencing\n",
      "499225 + 0 read1\n",
      "499225 + 0 read2\n",
      "422530 + 0 properly paired (42.32% : N/A)\n",
      "438398 + 0 with itself and mate mapped\n",
      "12168 + 0 singletons (1.22% : N/A)\n",
      "11068 + 0 with mate mapped to a different chr\n",
      "4064 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samtools flagstat alignment_Bowtie2_subset/ERR950187_trimmed.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3885de00-1837-489b-9ea9-3fb8e6e992a1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "Version 2.0.8\n",
      "\n",
      "Usage: featureCounts [options] -a <annotation_file> -o <output_file> input_file1 [input_file2] ... \n",
      "\n",
      "## Mandatory arguments:\n",
      "\n",
      "  -a <string>         Name of an annotation file. GTF/GFF format by default. See\n",
      "                      -F option for more format information. Inbuilt annotations\n",
      "                      (SAF format) is available in 'annotation' directory of the\n",
      "                      package. Gzipped file is also accepted.\n",
      "\n",
      "  -o <string>         Name of output file including read counts. A separate file\n",
      "                      including summary statistics of counting results is also\n",
      "                      included in the output ('<string>.summary'). Both files\n",
      "                      are in tab delimited format.\n",
      "\n",
      "  input_file1 [input_file2] ...   A list of SAM or BAM format files. They can be\n",
      "                      either name or location sorted. If no files provided,\n",
      "                      <stdin> input is expected. Location-sorted paired-end reads\n",
      "                      are automatically sorted by read names.\n",
      "\n",
      "## Optional arguments:\n",
      "# Annotation\n",
      "\n",
      "  -F <string>         Specify format of the provided annotation file. Acceptable\n",
      "                      formats include 'GTF' (or compatible GFF format) and\n",
      "                      'SAF'. 'GTF' by default.  For SAF format, please refer to\n",
      "                      Users Guide.\n",
      "\n",
      "  -t <string>         Specify feature type(s) in a GTF annotation. If multiple\n",
      "                      types are provided, they should be separated by ',' with\n",
      "                      no space in between. 'exon' by default. Rows in the\n",
      "                      annotation with a matched feature will be extracted and\n",
      "                      used for read mapping. \n",
      "\n",
      "  -g <string>         Specify attribute type in GTF annotation. 'gene_id' by \n",
      "                      default. Meta-features used for read counting will be \n",
      "                      extracted from annotation using the provided value.\n",
      "\n",
      "  --extraAttributes   Extract extra attribute types from the provided GTF\n",
      "                      annotation and include them in the counting output. These\n",
      "                      attribute types will not be used to group features. If\n",
      "                      more than one attribute type is provided they should be\n",
      "                      separated by comma.\n",
      "\n",
      "  -A <string>         Provide a chromosome name alias file to match chr names in\n",
      "                      annotation with those in the reads. This should be a two-\n",
      "                      column comma-delimited text file. Its first column should\n",
      "                      include chr names in the annotation and its second column\n",
      "                      should include chr names in the reads. Chr names are case\n",
      "                      sensitive. No column header should be included in the\n",
      "                      file.\n",
      "\n",
      "# Level of summarization\n",
      "\n",
      "  -f                  Perform read counting at feature level (eg. counting \n",
      "                      reads for exons rather than genes).\n",
      "\n",
      "# Overlap between reads and features\n",
      "\n",
      "  -O                  Assign reads to all their overlapping meta-features (or \n",
      "                      features if -f is specified).\n",
      "\n",
      "  --minOverlap <int>  Minimum number of overlapping bases in a read that is\n",
      "                      required for read assignment. 1 by default. Number of\n",
      "                      overlapping bases is counted from both reads if paired\n",
      "                      end. If a negative value is provided, then a gap of up\n",
      "                      to specified size will be allowed between read and the\n",
      "                      feature that the read is assigned to.\n",
      "\n",
      "  --fracOverlap <float> Minimum fraction of overlapping bases in a read that is\n",
      "                      required for read assignment. Value should be within range\n",
      "                      [0,1]. 0 by default. Number of overlapping bases is\n",
      "                      counted from both reads if paired end. Both this option\n",
      "                      and '--minOverlap' option need to be satisfied for read\n",
      "                      assignment.\n",
      "\n",
      "  --fracOverlapFeature <float> Minimum fraction of overlapping bases in a\n",
      "                      feature that is required for read assignment. Value\n",
      "                      should be within range [0,1]. 0 by default.\n",
      "\n",
      "  --largestOverlap    Assign reads to a meta-feature/feature that has the \n",
      "                      largest number of overlapping bases.\n",
      "\n",
      "  --nonOverlap <int>  Maximum number of non-overlapping bases in a read (or a\n",
      "                      read pair) that is allowed when being assigned to a\n",
      "                      feature. No limit is set by default.\n",
      "\n",
      "  --nonOverlapFeature <int> Maximum number of non-overlapping bases in a feature\n",
      "                      that is allowed in read assignment. No limit is set by\n",
      "                      default.\n",
      "\n",
      "  --readExtension5 <int> Reads are extended upstream by <int> bases from their\n",
      "                      5' end.\n",
      "\n",
      "  --readExtension3 <int> Reads are extended upstream by <int> bases from their\n",
      "                      3' end.\n",
      "\n",
      "  --read2pos <5:3>    Reduce reads to their 5' most base or 3' most base. Read\n",
      "                      counting is then performed based on the single base the \n",
      "                      read is reduced to.\n",
      "\n",
      "# Multi-mapping reads\n",
      "\n",
      "  -M                  Multi-mapping reads will also be counted. For a multi-\n",
      "                      mapping read, all its reported alignments will be \n",
      "                      counted. The 'NH' tag in BAM/SAM input is used to detect \n",
      "                      multi-mapping reads.\n",
      "\n",
      "# Fractional counting\n",
      "\n",
      "  --fraction          Assign fractional counts to features. This option must\n",
      "                      be used together with '-M' or '-O' or both. When '-M' is\n",
      "                      specified, each reported alignment from a multi-mapping\n",
      "                      read (identified via 'NH' tag) will carry a fractional\n",
      "                      count of 1/x, instead of 1 (one), where x is the total\n",
      "                      number of alignments reported for the same read. When '-O'\n",
      "                      is specified, each overlapping feature will receive a\n",
      "                      fractional count of 1/y, where y is the total number of\n",
      "                      features overlapping with the read. When both '-M' and\n",
      "                      '-O' are specified, each alignment will carry a fractional\n",
      "                      count of 1/(x*y).\n",
      "\n",
      "# Read filtering\n",
      "\n",
      "  -Q <int>            The minimum mapping quality score a read must satisfy in\n",
      "                      order to be counted. For paired-end reads, at least one\n",
      "                      end should satisfy this criteria. 0 by default.\n",
      "\n",
      "  --splitOnly         Count split alignments only (ie. alignments with CIGAR\n",
      "                      string containing 'N'). An example of split alignments is\n",
      "                      exon-spanning reads in RNA-seq data.\n",
      "\n",
      "  --nonSplitOnly      If specified, only non-split alignments (CIGAR strings do\n",
      "                      not contain letter 'N') will be counted. All the other\n",
      "                      alignments will be ignored.\n",
      "\n",
      "  --primary           Count primary alignments only. Primary alignments are \n",
      "                      identified using bit 0x100 in SAM/BAM FLAG field.\n",
      "\n",
      "  --ignoreDup         Ignore duplicate reads in read counting. Duplicate reads \n",
      "                      are identified using bit Ox400 in BAM/SAM FLAG field. The \n",
      "                      whole read pair is ignored if one of the reads is a \n",
      "                      duplicate read for paired end data.\n",
      "\n",
      "# Strandness\n",
      "\n",
      "  -s <int or string>  Perform strand-specific read counting. A single integer\n",
      "                      value (applied to all input files) or a string of comma-\n",
      "                      separated values (applied to each corresponding input\n",
      "                      file) should be provided. Possible values include:\n",
      "                      0 (unstranded), 1 (stranded) and 2 (reversely stranded).\n",
      "                      Default value is 0 (ie. unstranded read counting carried\n",
      "                      out for all input files).\n",
      "\n",
      "# Exon-exon junctions\n",
      "\n",
      "  -J                  Count number of reads supporting each exon-exon junction.\n",
      "                      Junctions were identified from all the exon-spanning reads\n",
      "                      in the input (containing 'N' in CIGAR string). Counting\n",
      "                      results are saved to a file named '<output_file>.jcounts'\n",
      "\n",
      "  -G <string>         Provide the name of a FASTA-format file that contains the\n",
      "                      reference sequences used in read mapping that produced the\n",
      "                      provided SAM/BAM files. This optional argument can be used\n",
      "                      with '-J' option to improve read counting for junctions.\n",
      "\n",
      "# Parameters specific to paired end reads\n",
      "\n",
      "  -p                  If specified, libraries are assumed to contain paired-end\n",
      "                      reads. For any library that contains paired-end reads, the\n",
      "                      'countReadPairs' parameter controls if read pairs or reads\n",
      "                      should be counted.\n",
      "\n",
      "  --countReadPairs    If specified, fragments (or templates) will be counted\n",
      "                      instead of reads. This option is only applicable for\n",
      "                      paired-end reads. For single-end data, it is ignored.\n",
      "\n",
      "  -B                  Only count read pairs that have both ends aligned.\n",
      "\n",
      "  -P                  Check validity of paired-end distance when counting read \n",
      "                      pairs. Use -d and -D to set thresholds.\n",
      "\n",
      "  -d <int>            Minimum fragment/template length, 50 by default.\n",
      "\n",
      "  -D <int>            Maximum fragment/template length, 600 by default.\n",
      "\n",
      "  -C                  Do not count read pairs that have their two ends mapping \n",
      "                      to different chromosomes or mapping to same chromosome \n",
      "                      but on different strands.\n",
      "\n",
      "  --donotsort         Do not sort reads in BAM/SAM input. Note that reads from \n",
      "                      the same pair are required to be located next to each \n",
      "                      other in the input.\n",
      "\n",
      "# Number of CPU threads\n",
      "\n",
      "  -T <int>            Number of the threads. 1 by default.\n",
      "\n",
      "# Read groups\n",
      "\n",
      "  --byReadGroup       Assign reads by read group. \"RG\" tag is required to be\n",
      "                      present in the input BAM/SAM files.\n",
      "                      \n",
      "\n",
      "# Long reads\n",
      "\n",
      "  -L                  Count long reads such as Nanopore and PacBio reads. Long\n",
      "                      read counting can only run in one thread and only reads\n",
      "                      (not read-pairs) can be counted. There is no limitation on\n",
      "                      the number of 'M' operations allowed in a CIGAR string in\n",
      "                      long read counting.\n",
      "\n",
      "# Assignment results for each read\n",
      "\n",
      "  -R <format>         Output detailed assignment results for each read or read-\n",
      "                      pair. Results are saved to a file that is in one of the\n",
      "                      following formats: CORE, SAM and BAM. See Users Guide for\n",
      "                      more info about these formats.\n",
      "\n",
      "  --Rpath <string>    Specify a directory to save the detailed assignment\n",
      "                      results. If unspecified, the directory where counting\n",
      "                      results are saved is used.\n",
      "\n",
      "# Miscellaneous\n",
      "\n",
      "  --tmpDir <string>   Directory under which intermediate files are saved (later\n",
      "                      removed). By default, intermediate files will be saved to\n",
      "                      the directory specified in '-o' argument.\n",
      "\n",
      "  --verbose           Output verbose information for debugging, such as un-\n",
      "                      matched chromosome/contig names.\n",
      "\n",
      "  -v                  Output version of the program.\n",
      "\n",
      "bash: line 2: -a: command not found\n",
      "bash: line 3: -o: command not found\n",
      "bash: line 4: -p: command not found\n",
      "bash: line 5: -T: command not found\n",
      "bash: line 6: ERR950158_trimmed.bam: command not found\n",
      "bash: line 7: ERR950158_trimmed.bam.bai: command not found\n",
      "bash: line 8: ERR950159_trimmed.bam: command not found\n",
      "bash: line 9: ERR950159_trimmed.bam.bai: command not found\n",
      "bash: line 10: ERR950160_trimmed.bam: command not found\n",
      "bash: line 11: ERR950160_trimmed.bam.bai: command not found\n",
      "bash: line 12: ERR950161_trimmed.bam: command not found\n",
      "bash: line 13: ERR950161_trimmed.bam.bai: command not found\n",
      "bash: line 14: ERR950162_trimmed.bam: command not found\n",
      "bash: line 15: ERR950162_trimmed.bam.bai: command not found\n",
      "bash: line 16: ERR950163_trimmed.bam: command not found\n",
      "bash: line 17: ERR950163_trimmed.bam.bai: command not found\n",
      "bash: line 18: ERR950177_trimmed.bam: command not found\n",
      "bash: line 19: ERR950177_trimmed.bam.bai: command not found\n",
      "bash: line 20: ERR950178_trimmed.bam: command not found\n",
      "bash: line 21: ERR950178_trimmed.bam.bai: command not found\n",
      "bash: line 22: ERR950179_trimmed.bam: command not found\n",
      "bash: line 23: ERR950179_trimmed.bam.bai: command not found\n",
      "bash: line 24: ERR950180_trimmed.bam: command not found\n",
      "bash: line 25: ERR950180_trimmed.bam.bai: command not found\n",
      "bash: line 26: ERR950181_trimmed.bam: command not found\n",
      "bash: line 27: ERR950181_trimmed.bam.bai: command not found\n",
      "bash: line 28: ERR950182_trimmed.bam: command not found\n",
      "bash: line 29: ERR950182_trimmed.bam.bai: command not found\n",
      "bash: line 30: ERR950186_trimmed.bam: command not found\n",
      "bash: line 31: ERR950186_trimmed.bam.bai: command not found\n",
      "bash: line 32: ERR950187.log: command not found\n",
      "bash: line 33: ERR950187_trimmed.bam: command not found\n",
      "bash: line 34: ERR950187_trimmed.bam.bai: command not found\n",
      "bash: line 35: ERR950187_trimmed_1.sam: command not found\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'featureCounts \\n  -a data/Homo_sapiens.GRCh38.113.gtf    # Path to your GTF annotation\\n  -o results/counts.txt                  # Output file\\n  -p -B -C                               # Options for paired-end data\\n  -T 4                                   # Threads\\n  ERR950158_trimmed.bam\\nERR950158_trimmed.bam.bai\\nERR950159_trimmed.bam\\nERR950159_trimmed.bam.bai\\nERR950160_trimmed.bam\\nERR950160_trimmed.bam.bai\\nERR950161_trimmed.bam\\nERR950161_trimmed.bam.bai\\nERR950162_trimmed.bam\\nERR950162_trimmed.bam.bai\\nERR950163_trimmed.bam\\nERR950163_trimmed.bam.bai\\nERR950177_trimmed.bam\\nERR950177_trimmed.bam.bai\\nERR950178_trimmed.bam\\nERR950178_trimmed.bam.bai\\nERR950179_trimmed.bam\\nERR950179_trimmed.bam.bai\\nERR950180_trimmed.bam\\nERR950180_trimmed.bam.bai\\nERR950181_trimmed.bam\\nERR950181_trimmed.bam.bai\\nERR950182_trimmed.bam\\nERR950182_trimmed.bam.bai\\nERR950186_trimmed.bam\\nERR950186_trimmed.bam.bai\\nERR950187.log\\nERR950187_trimmed.bam\\nERR950187_trimmed.bam.bai\\nERR950187_trimmed_1.sam\\n'' returned non-zero exit status 127.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mbash\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mfeatureCounts \u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m  -a data/Homo_sapiens.GRCh38.113.gtf    # Path to your GTF annotation\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m  -o results/counts.txt                  # Output file\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m  -p -B -C                               # Options for paired-end data\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m  -T 4                                   # Threads\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m  ERR950158_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950158_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950159_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950159_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950160_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950160_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950161_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950161_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950162_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950162_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950163_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950163_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950177_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950177_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950178_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950178_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950179_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950179_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950180_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950180_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950181_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950181_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950182_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950182_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950186_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950186_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950187.log\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950187_trimmed.bam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950187_trimmed.bam.bai\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mERR950187_trimmed_1.sam\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2543\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2541\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2542\u001b[0m     args \u001b[38;5;241m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2543\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2545\u001b[0m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[1;32m   2546\u001b[0m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[1;32m   2547\u001b[0m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[1;32m   2548\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic\u001b[38;5;241m.\u001b[39mMAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:159\u001b[0m, in \u001b[0;36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    157\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    158\u001b[0m     line \u001b[38;5;241m=\u001b[39m script\n\u001b[0;32m--> 159\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:336\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m args\u001b[38;5;241m.\u001b[39mraise_error \u001b[38;5;129;01mand\u001b[39;00m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    332\u001b[0m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    333\u001b[0m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    334\u001b[0m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    335\u001b[0m     rc \u001b[38;5;241m=\u001b[39m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m9\u001b[39m\n\u001b[0;32m--> 336\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'featureCounts \\n  -a data/Homo_sapiens.GRCh38.113.gtf    # Path to your GTF annotation\\n  -o results/counts.txt                  # Output file\\n  -p -B -C                               # Options for paired-end data\\n  -T 4                                   # Threads\\n  ERR950158_trimmed.bam\\nERR950158_trimmed.bam.bai\\nERR950159_trimmed.bam\\nERR950159_trimmed.bam.bai\\nERR950160_trimmed.bam\\nERR950160_trimmed.bam.bai\\nERR950161_trimmed.bam\\nERR950161_trimmed.bam.bai\\nERR950162_trimmed.bam\\nERR950162_trimmed.bam.bai\\nERR950163_trimmed.bam\\nERR950163_trimmed.bam.bai\\nERR950177_trimmed.bam\\nERR950177_trimmed.bam.bai\\nERR950178_trimmed.bam\\nERR950178_trimmed.bam.bai\\nERR950179_trimmed.bam\\nERR950179_trimmed.bam.bai\\nERR950180_trimmed.bam\\nERR950180_trimmed.bam.bai\\nERR950181_trimmed.bam\\nERR950181_trimmed.bam.bai\\nERR950182_trimmed.bam\\nERR950182_trimmed.bam.bai\\nERR950186_trimmed.bam\\nERR950186_trimmed.bam.bai\\nERR950187.log\\nERR950187_trimmed.bam\\nERR950187_trimmed.bam.bai\\nERR950187_trimmed_1.sam\\n'' returned non-zero exit status 127."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "featureCounts \n",
    "  -a data/Homo_sapiens.GRCh38.113.gtf    # Path to your GTF annotation\n",
    "  -o results/counts.txt                  # Output file\n",
    "  -p -B -C                               # Options for paired-end data\n",
    "  -T 4                                   # Threads\n",
    "  ERR950158_trimmed.bam\n",
    "ERR950158_trimmed.bam.bai\n",
    "ERR950159_trimmed.bam\n",
    "ERR950159_trimmed.bam.bai\n",
    "ERR950160_trimmed.bam\n",
    "ERR950160_trimmed.bam.bai\n",
    "ERR950161_trimmed.bam\n",
    "ERR950161_trimmed.bam.bai\n",
    "ERR950162_trimmed.bam\n",
    "ERR950162_trimmed.bam.bai\n",
    "ERR950163_trimmed.bam\n",
    "ERR950163_trimmed.bam.bai\n",
    "ERR950177_trimmed.bam\n",
    "ERR950177_trimmed.bam.bai\n",
    "ERR950178_trimmed.bam\n",
    "ERR950178_trimmed.bam.bai\n",
    "ERR950179_trimmed.bam\n",
    "ERR950179_trimmed.bam.bai\n",
    "ERR950180_trimmed.bam\n",
    "ERR950180_trimmed.bam.bai\n",
    "ERR950181_trimmed.bam\n",
    "ERR950181_trimmed.bam.bai\n",
    "ERR950182_trimmed.bam\n",
    "ERR950182_trimmed.bam.bai\n",
    "ERR950186_trimmed.bam\n",
    "ERR950186_trimmed.bam.bai\n",
    "ERR950187.log\n",
    "ERR950187_trimmed.bam\n",
    "ERR950187_trimmed.bam.bai\n",
    "ERR950187_trimmed_1.sam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8879a271-b191-4f8b-87e4-5c5ca4e0439c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "chmod: changing permissions of '.': Operation not permitted\n",
      "chmod: changing permissions of './Bowtie.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './containers': Operation not permitted\n",
      "chmod: changing permissions of './containers/group_Bjork_rstudio.sh': Operation not permitted\n",
      "chmod: changing permissions of './containers/Bjork_container.sif': Operation not permitted\n",
      "chmod: changing permissions of './containers/Bjork_container.def': Operation not permitted\n",
      "chmod: changing permissions of './containers/group_Bjork_jupyter.sh': Operation not permitted\n",
      "chmod: changing permissions of './containers/container.sif': Operation not permitted\n",
      "chmod: changing permissions of './containers/container.def': Operation not permitted\n",
      "chmod: changing permissions of './containers/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './containers/.ipynb_checkpoints/Bjork_container-checkpoint.def': Operation not permitted\n",
      "chmod: changing permissions of './RNA-seq-list.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950158_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950163_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950187_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950162_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950179_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950178_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950158_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950177_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950186_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950187_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950161_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950163_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950158_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950186_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950162_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950160_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950180_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950181_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950159_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950181_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950182_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950178_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950180_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950182_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950159_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950161_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950177_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950179_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950160_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950163_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950180_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950181_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950178_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950177_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950187_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950162_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950161_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950160_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950180_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950163_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950182_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950158_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950186_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950187_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950159_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950162_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950179_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950178_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950186_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950177_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950160_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950161_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950181_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950159_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950182_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950179_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './CN.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './subset_trimmed_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950159_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950162_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed_1.sam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950178_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950179_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950181_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950163_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950182_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950163_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950179_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950180_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950177_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950159_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950161_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187.log': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950162_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950160_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950160_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950186_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950181_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950180_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950182_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950158_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950177_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950186_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950158_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950178_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950161_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './Preprocessing.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './DESeq2.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.git': Operation not permitted\n",
      "chmod: changing permissions of './.git/HEAD': Operation not permitted\n",
      "chmod: changing permissions of './.git/description': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/66': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/66/452897458e05dbd938d918f6397505146dd8bb': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/00': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/00/4ac72bc224eb0cd2a221ca987e7f20a9f91ce5': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/info': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ca': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ca/4a1b2996370a666697b0cd1b3f367aef80918a': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/c6': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/c6/ff6019645459b3793345e3f75280c0b2c66db3': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/0b': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/0b/c5f58cbf1b08a2d2c24dd1306192a1f59b517f': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/cd': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/cd/500414444e8f65d2d6e03d9f6c18894d317a5e': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ac': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ac/1cc772c92ebb0f64586ffb19c1f07901ebc465': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/pack': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-merge-commit.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-commit.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/fsmonitor-watchman.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/commit-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-rebase.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-receive.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/prepare-commit-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/update.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-push.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/sendemail-validate.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-applypatch.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/push-to-checkout.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/post-update.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/applypatch-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/FETCH_HEAD': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/bbt045_project': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/bbt045_project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/project': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/info': Operation not permitted\n",
      "chmod: changing permissions of './.git/info/exclude': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/tags': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/heads': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/bbt045_project': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/bbt045_project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/project': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/branches': Operation not permitted\n",
      "chmod: changing permissions of './CN_TESTAR': Operation not permitted\n",
      "chmod: changing permissions of './CN_TESTAR/Homo_sapiens.GRCh38.dna.toplevel.fa.gz.fai': Operation not permitted\n",
      "chmod: changing permissions of './gene_ccounts_DESeq': Operation not permitted\n",
      "chmod: changing permissions of './Log.out': Operation not permitted\n",
      "chmod: changing permissions of './STAR.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950186_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950182_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950161_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950181_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950177_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950187_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950159_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950179_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950162_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950162_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950178_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950187_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950178_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950163_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950181_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950163_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950186_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950158_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950160_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950179_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950158_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950159_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950182_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950180_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950161_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950180_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950177_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950160_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/edgeR_STAR-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/Log-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/RNA-seq-list-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/Preprocessing-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/DESeq2-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/STAR-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/CN-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.4.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.2.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/Untitled.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.3.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.1.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950177_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950178_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950186_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950187_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950163_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_length_distribution_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950182_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950181_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950178_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950159_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950182_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950160_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950161_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950177_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950158_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950158_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950161_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950163_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950162_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950181_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950180_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950179_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950179_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950162_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950187_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950186_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950160_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950180_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950159_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950158_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950158_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950180_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950182_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950181_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950186_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950179_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950162_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950162_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950160_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/test.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950159_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950163_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950163_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950161_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950178_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950187_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950178_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950180_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950160_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950177_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950181_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950186_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/test.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950179_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950161_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/ERR950158_trimmed.counts.txt-checkpoint.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/test.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/ERR950158_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950177_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950159_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950187_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950182_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbList.fromGTF.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrNameLength.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/exonInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/transcriptInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/exonGeTrInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrStart.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/_STARtmp': Operation not permitted\n",
      "chmod: cannot read directory './data/STAR_index/_STARtmp': Permission denied\n",
      "chmod: changing permissions of './data/STAR_index/chrLength.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrName.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/SAindex': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbList.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/geneInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/genomeParameters.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/Genome': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/SA': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbInfo.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/merged_counts_matrix.csv': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.rev.1.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950158_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950158_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950182_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950181_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950179_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950162_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950160_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950187_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950181_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950186_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950163_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950161_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950178_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950160_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950163_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950180_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950187_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950182_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950179_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950180_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950161_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950177_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950162_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950186_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950159_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950159_trimmed_idxstats-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950158_trimmed_idxstats-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950158_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950159_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/data': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/data/gene_counts_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950177_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950159_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950178_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950178_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950181_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950177_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950161_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950179_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950180_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950162_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950163_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950160_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950158_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950186_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950181_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950162_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950182_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950180_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950182_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950158_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950177_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950163_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950186_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950159_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950179_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950159_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950163_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950178_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950162_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950158_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950160_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950180_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950177_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950181_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950160_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950179_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950186_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950161_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950161_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950178_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950159_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/merged_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints/ERR950158_trimmed_htseq_counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints/merged_counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950182_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950178_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950186_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950160_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950179_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950180_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950181_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950182_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950161_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950162_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950187_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950158_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950159_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950163_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_ReadsPerGene.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_Log-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_Log.progress-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950180_trimmed_SJ.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950180_trimmed_ReadsPerGene.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950177_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/hg38_cdna_index.rev.2.bt2': Operation not permitted\n",
      "chmod: changing permissions of './data/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './edgeR_STAR.ipynb': Operation not permitted\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'chmod -R 777 . \\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mbash\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mchmod -R 777 . \u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2543\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2541\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2542\u001b[0m     args \u001b[38;5;241m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2543\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2545\u001b[0m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[1;32m   2546\u001b[0m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[1;32m   2547\u001b[0m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[1;32m   2548\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic\u001b[38;5;241m.\u001b[39mMAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:159\u001b[0m, in \u001b[0;36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    157\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    158\u001b[0m     line \u001b[38;5;241m=\u001b[39m script\n\u001b[0;32m--> 159\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:336\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m args\u001b[38;5;241m.\u001b[39mraise_error \u001b[38;5;129;01mand\u001b[39;00m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    332\u001b[0m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    333\u001b[0m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    334\u001b[0m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    335\u001b[0m     rc \u001b[38;5;241m=\u001b[39m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m9\u001b[39m\n\u001b[0;32m--> 336\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'chmod -R 777 . \\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "chmod -R 777 . "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
