{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20b8d80b-9eb0-4846-bb21-d80d9882e4c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script false --no-raise-error\n",
    "#Copy this code and run it in the terminal everytime you create a new directory or file\n",
    "#Gives permission for everyone to work with the files created\n",
    "#Run at the end of a jupyter script, the output is too messy\n",
    "!cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork \n",
    "!chmod -R 777 . "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a29ef54-a06b-4be8-8d13-ac484ca8d09a",
   "metadata": {},
   "source": [
    "# Download cDNA library \n",
    "From ensembl: https://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/ and unzip."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "359a5dbb-293e-4af1-9006-f155830ee599",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "File 'data/Homo_sapiens.GRCh38.cdna.all.fa.gz' already there; not retrieving.\n",
      "gzip: data/Homo_sapiens.GRCh38.cdna.all.fa already exists;\tnot overwritten\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# -nc makes sure that if the file is already there, it won't be overwritten.\n",
    "# -nc can be changed to -N, if it is of interest to import newer versions\n",
    "# -P will save it to the directory specified, in this case data/cDNA_library\n",
    "wget -nc ftp://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz -P data/\n",
    "gzip -d data/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "rm data/Homo_sapiens.GRCh38.cdna.all.fa.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ea8812a-8e96-48cb-a70c-de2ecd4169c6",
   "metadata": {},
   "source": [
    "# Build Bowtie index\n",
    "This takes around 30 minutes to run on one kernel, so don't run it if you don't have to."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "41402daf-553c-485b-b530-cf740db5a805",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"data/Bowtie_index/hg38_cdna_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  data/Homo_sapiens.GRCh38.cdna.all.fa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:03\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:02\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:32\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:04\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:03\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 40214484 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:37\n",
      "Returning block of 40214485 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 66480457 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:42\n",
      "Returning block of 66480458 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 68586474 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:48\n",
      "Returning block of 68586475 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 49097504 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:00\n",
      "Returning block of 49097505 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 65349089 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:38\n",
      "Returning block of 65349090 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 28965054 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:06\n",
      "Returning block of 28965055 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 72809500 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:53\n",
      "Returning block of 72809501 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 3519685 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:07\n",
      "Returning block of 3519686 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/Bowtie_index/hg38_cdna_index.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/Bowtie_index/hg38_cdna_index.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:18:19\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:03\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:32\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:03\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:04\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 69612405 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:50\n",
      "Returning block of 69612406 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 48012854 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:56\n",
      "Returning block of 48012855 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 53601703 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:07\n",
      "Returning block of 53601704 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 50816782 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:02\n",
      "Returning block of 50816783 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 49006051 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:52\n",
      "Returning block of 49006052 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 25671179 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:02\n",
      "Returning block of 25671180 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 58667016 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:22\n",
      "Returning block of 58667017 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 39634257 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:35\n",
      "Returning block of 39634258 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/Bowtie_index/hg38_cdna_index.rev.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/Bowtie_index/hg38_cdna_index.rev.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:18:14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Renaming data/Bowtie_index/hg38_cdna_index.3.bt2.tmp to data/Bowtie_index/hg38_cdna_index.3.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.4.bt2.tmp to data/Bowtie_index/hg38_cdna_index.4.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.1.bt2.tmp to data/Bowtie_index/hg38_cdna_index.1.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.2.bt2.tmp to data/Bowtie_index/hg38_cdna_index.2.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.rev.1.bt2.tmp to data/Bowtie_index/hg38_cdna_index.rev.1.bt2\n",
      "Renaming data/Bowtie_index/hg38_cdna_index.rev.2.bt2.tmp to data/Bowtie_index/hg38_cdna_index.rev.2.bt2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir data/Bowtie_index\n",
    "bowtie2-build data/Homo_sapiens.GRCh38.cdna.all.fa data/Bowtie_index/hg38_cdna_index"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c5ca7d8-87a4-4226-90ee-cd8989fd11c9",
   "metadata": {},
   "source": [
    "# Run Bowtie2\n",
    "Started ~14:00, ended ~14:50 with one core.\n",
    "\n",
    "Started ~15:00, ended ~14:20 with two cores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bc571cc3-585c-444a-a60a-a7b35d590787",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning ERR950158_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499532 reads; of these:\n",
      "  499532 (100.00%) were paired; of these:\n",
      "    261310 (52.31%) aligned concordantly 0 times\n",
      "    62637 (12.54%) aligned concordantly exactly 1 time\n",
      "    175585 (35.15%) aligned concordantly >1 times\n",
      "    ----\n",
      "    261310 pairs aligned concordantly 0 times; of these:\n",
      "      1885 (0.72%) aligned discordantly 1 time\n",
      "    ----\n",
      "    259425 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      518850 mates make up the pairs; of these:\n",
      "        486970 (93.86%) aligned 0 times\n",
      "        6480 (1.25%) aligned exactly 1 time\n",
      "        25400 (4.90%) aligned >1 times\n",
      "51.26% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950158_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950159_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499704 reads; of these:\n",
      "  499704 (100.00%) were paired; of these:\n",
      "    303889 (60.81%) aligned concordantly 0 times\n",
      "    67440 (13.50%) aligned concordantly exactly 1 time\n",
      "    128375 (25.69%) aligned concordantly >1 times\n",
      "    ----\n",
      "    303889 pairs aligned concordantly 0 times; of these:\n",
      "      5498 (1.81%) aligned discordantly 1 time\n",
      "    ----\n",
      "    298391 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      596782 mates make up the pairs; of these:\n",
      "        562995 (94.34%) aligned 0 times\n",
      "        7124 (1.19%) aligned exactly 1 time\n",
      "        26663 (4.47%) aligned >1 times\n",
      "43.67% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950159_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950160_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "498925 reads; of these:\n",
      "  498925 (100.00%) were paired; of these:\n",
      "    262484 (52.61%) aligned concordantly 0 times\n",
      "    71038 (14.24%) aligned concordantly exactly 1 time\n",
      "    165403 (33.15%) aligned concordantly >1 times\n",
      "    ----\n",
      "    262484 pairs aligned concordantly 0 times; of these:\n",
      "      2581 (0.98%) aligned discordantly 1 time\n",
      "    ----\n",
      "    259903 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      519806 mates make up the pairs; of these:\n",
      "        485428 (93.39%) aligned 0 times\n",
      "        7622 (1.47%) aligned exactly 1 time\n",
      "        26756 (5.15%) aligned >1 times\n",
      "51.35% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950160_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950161_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "498850 reads; of these:\n",
      "  498850 (100.00%) were paired; of these:\n",
      "    275872 (55.30%) aligned concordantly 0 times\n",
      "    64315 (12.89%) aligned concordantly exactly 1 time\n",
      "    158663 (31.81%) aligned concordantly >1 times\n",
      "    ----\n",
      "    275872 pairs aligned concordantly 0 times; of these:\n",
      "      2081 (0.75%) aligned discordantly 1 time\n",
      "    ----\n",
      "    273791 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      547582 mates make up the pairs; of these:\n",
      "        518496 (94.69%) aligned 0 times\n",
      "        5861 (1.07%) aligned exactly 1 time\n",
      "        23225 (4.24%) aligned >1 times\n",
      "48.03% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950161_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950162_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "481513 reads; of these:\n",
      "  481513 (100.00%) were paired; of these:\n",
      "    297188 (61.72%) aligned concordantly 0 times\n",
      "    86273 (17.92%) aligned concordantly exactly 1 time\n",
      "    98052 (20.36%) aligned concordantly >1 times\n",
      "    ----\n",
      "    297188 pairs aligned concordantly 0 times; of these:\n",
      "      3000 (1.01%) aligned discordantly 1 time\n",
      "    ----\n",
      "    294188 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      588376 mates make up the pairs; of these:\n",
      "        566809 (96.33%) aligned 0 times\n",
      "        6705 (1.14%) aligned exactly 1 time\n",
      "        14862 (2.53%) aligned >1 times\n",
      "41.14% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950162_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950163_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "498470 reads; of these:\n",
      "  498470 (100.00%) were paired; of these:\n",
      "    280228 (56.22%) aligned concordantly 0 times\n",
      "    61874 (12.41%) aligned concordantly exactly 1 time\n",
      "    156368 (31.37%) aligned concordantly >1 times\n",
      "    ----\n",
      "    280228 pairs aligned concordantly 0 times; of these:\n",
      "      2132 (0.76%) aligned discordantly 1 time\n",
      "    ----\n",
      "    278096 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      556192 mates make up the pairs; of these:\n",
      "        524390 (94.28%) aligned 0 times\n",
      "        6905 (1.24%) aligned exactly 1 time\n",
      "        24897 (4.48%) aligned >1 times\n",
      "47.40% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950163_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950177_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "493727 reads; of these:\n",
      "  493727 (100.00%) were paired; of these:\n",
      "    232338 (47.06%) aligned concordantly 0 times\n",
      "    74666 (15.12%) aligned concordantly exactly 1 time\n",
      "    186723 (37.82%) aligned concordantly >1 times\n",
      "    ----\n",
      "    232338 pairs aligned concordantly 0 times; of these:\n",
      "      3172 (1.37%) aligned discordantly 1 time\n",
      "    ----\n",
      "    229166 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      458332 mates make up the pairs; of these:\n",
      "        413445 (90.21%) aligned 0 times\n",
      "        10203 (2.23%) aligned exactly 1 time\n",
      "        34684 (7.57%) aligned >1 times\n",
      "58.13% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950177_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950178_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "495553 reads; of these:\n",
      "  495553 (100.00%) were paired; of these:\n",
      "    283061 (57.12%) aligned concordantly 0 times\n",
      "    57864 (11.68%) aligned concordantly exactly 1 time\n",
      "    154628 (31.20%) aligned concordantly >1 times\n",
      "    ----\n",
      "    283061 pairs aligned concordantly 0 times; of these:\n",
      "      1988 (0.70%) aligned discordantly 1 time\n",
      "    ----\n",
      "    281073 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      562146 mates make up the pairs; of these:\n",
      "        534649 (95.11%) aligned 0 times\n",
      "        5828 (1.04%) aligned exactly 1 time\n",
      "        21669 (3.85%) aligned >1 times\n",
      "46.06% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950178_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950179_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499543 reads; of these:\n",
      "  499543 (100.00%) were paired; of these:\n",
      "    263435 (52.74%) aligned concordantly 0 times\n",
      "    64256 (12.86%) aligned concordantly exactly 1 time\n",
      "    171852 (34.40%) aligned concordantly >1 times\n",
      "    ----\n",
      "    263435 pairs aligned concordantly 0 times; of these:\n",
      "      2068 (0.79%) aligned discordantly 1 time\n",
      "    ----\n",
      "    261367 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      522734 mates make up the pairs; of these:\n",
      "        487698 (93.30%) aligned 0 times\n",
      "        6839 (1.31%) aligned exactly 1 time\n",
      "        28197 (5.39%) aligned >1 times\n",
      "51.19% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950179_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950180_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "496509 reads; of these:\n",
      "  496509 (100.00%) were paired; of these:\n",
      "    272032 (54.79%) aligned concordantly 0 times\n",
      "    66108 (13.31%) aligned concordantly exactly 1 time\n",
      "    158369 (31.90%) aligned concordantly >1 times\n",
      "    ----\n",
      "    272032 pairs aligned concordantly 0 times; of these:\n",
      "      2204 (0.81%) aligned discordantly 1 time\n",
      "    ----\n",
      "    269828 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      539656 mates make up the pairs; of these:\n",
      "        506779 (93.91%) aligned 0 times\n",
      "        7866 (1.46%) aligned exactly 1 time\n",
      "        25011 (4.63%) aligned >1 times\n",
      "48.97% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950180_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950181_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499494 reads; of these:\n",
      "  499494 (100.00%) were paired; of these:\n",
      "    259505 (51.95%) aligned concordantly 0 times\n",
      "    74619 (14.94%) aligned concordantly exactly 1 time\n",
      "    165370 (33.11%) aligned concordantly >1 times\n",
      "    ----\n",
      "    259505 pairs aligned concordantly 0 times; of these:\n",
      "      7452 (2.87%) aligned discordantly 1 time\n",
      "    ----\n",
      "    252053 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      504106 mates make up the pairs; of these:\n",
      "        472498 (93.73%) aligned 0 times\n",
      "        5794 (1.15%) aligned exactly 1 time\n",
      "        25814 (5.12%) aligned >1 times\n",
      "52.70% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950181_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950182_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499213 reads; of these:\n",
      "  499213 (100.00%) were paired; of these:\n",
      "    257178 (51.52%) aligned concordantly 0 times\n",
      "    74321 (14.89%) aligned concordantly exactly 1 time\n",
      "    167714 (33.60%) aligned concordantly >1 times\n",
      "    ----\n",
      "    257178 pairs aligned concordantly 0 times; of these:\n",
      "      2838 (1.10%) aligned discordantly 1 time\n",
      "    ----\n",
      "    254340 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      508680 mates make up the pairs; of these:\n",
      "        473492 (93.08%) aligned 0 times\n",
      "        8413 (1.65%) aligned exactly 1 time\n",
      "        26775 (5.26%) aligned >1 times\n",
      "52.58% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950182_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950186_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "498509 reads; of these:\n",
      "  498509 (100.00%) were paired; of these:\n",
      "    249406 (50.03%) aligned concordantly 0 times\n",
      "    78249 (15.70%) aligned concordantly exactly 1 time\n",
      "    170854 (34.27%) aligned concordantly >1 times\n",
      "    ----\n",
      "    249406 pairs aligned concordantly 0 times; of these:\n",
      "      5881 (2.36%) aligned discordantly 1 time\n",
      "    ----\n",
      "    243525 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      487050 mates make up the pairs; of these:\n",
      "        454211 (93.26%) aligned 0 times\n",
      "        6884 (1.41%) aligned exactly 1 time\n",
      "        25955 (5.33%) aligned >1 times\n",
      "54.44% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950186_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950187_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "499225 reads; of these:\n",
      "  499225 (100.00%) were paired; of these:\n",
      "    287960 (57.68%) aligned concordantly 0 times\n",
      "    60107 (12.04%) aligned concordantly exactly 1 time\n",
      "    151158 (30.28%) aligned concordantly >1 times\n",
      "    ----\n",
      "    287960 pairs aligned concordantly 0 times; of these:\n",
      "      2318 (0.80%) aligned discordantly 1 time\n",
      "    ----\n",
      "    285642 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      571284 mates make up the pairs; of these:\n",
      "        547884 (95.90%) aligned 0 times\n",
      "        5741 (1.00%) aligned exactly 1 time\n",
      "        17659 (3.09%) aligned >1 times\n",
      "45.13% overall alignment rate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950187_trimmed\n",
      "-----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p data/alignment_Bowtie2_subset\n",
    "\n",
    "# Loop over all FASTQ files ending in _1.fastq.gz in the subset_trimmed_RNA-seq folder\n",
    "for R1 in data/subset_trimmed_RNA-seq/*_1.fastq.gz; do\n",
    "  \n",
    "  # Extract the base name \n",
    "  BN=$(basename \"$R1\")\n",
    "  \n",
    "  # Remove the _1.fastq.gz suffix to get the sample prefix \n",
    "  SAMPLE=${BN%%_1.fastq.gz}\n",
    "  \n",
    "  # Construct the path to the mate 2 file \n",
    "  R2=\"data/subset_trimmed_RNA-seq/${SAMPLE}_2.fastq.gz\"\n",
    "  \n",
    "  echo \"Aligning $SAMPLE with Bowtie2...\"\n",
    "  \n",
    "  # Run Bowtie2 in local alignment mode for the paired-end reads\n",
    "  bowtie2 --local \\\n",
    "    -x data/Bowtie_index/hg38_cdna_index \\\n",
    "    -1 \"$R1\" -2 \"$R2\" \\\n",
    "    -S data/alignment_Bowtie2_subset/${SAMPLE}.sam \\\n",
    "    --threads 4\n",
    "  \n",
    "  # Convert the SAM to a sorted BAM and create an index using samtools\n",
    "  samtools view -bS data/alignment_Bowtie2_subset/${SAMPLE}.sam | samtools sort -o data/alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  samtools index data/alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  \n",
    "  # Remove the SAM file to save disk space\n",
    "  rm data/alignment_Bowtie2_subset/${SAMPLE}.sam\n",
    "  \n",
    "  echo \"Done aligning $SAMPLE\"\n",
    "  echo \"-----------------------------------------\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8553eeaa-f81f-4dfc-9e51-e65e897bd0df",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "998450 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "998450 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "450566 + 0 mapped (45.13% : N/A)\n",
      "450566 + 0 primary mapped (45.13% : N/A)\n",
      "998450 + 0 paired in sequencing\n",
      "499225 + 0 read1\n",
      "499225 + 0 read2\n",
      "422530 + 0 properly paired (42.32% : N/A)\n",
      "438398 + 0 with itself and mate mapped\n",
      "12168 + 0 singletons (1.22% : N/A)\n",
      "11068 + 0 with mate mapped to a different chr\n",
      "4064 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samtools flagstat data/alignment_Bowtie2_subset/ERR950187_trimmed.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eaaa9ad7-ccba-4e4b-a204-030d6f861f3c",
   "metadata": {},
   "source": [
    "# Download .gtf file for summarization\n",
    "https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens\n",
    "/Homo_sapiens.GRCh38.113.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4e0fa967-04c0-4562-999c-a3719938bafb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "--2025-03-04 21:02:22--  https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 64141785 (61M) [application/x-gzip]\n",
      "Saving to: 'data/Homo_sapiens.GRCh38.113.gtf.gz'\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0% 1.06M 58s\n",
      "    50K .......... .......... .......... .......... ..........  0% 2.07M 44s\n",
      "   100K .......... .......... .......... .......... ..........  0%  112M 29s\n",
      "   150K .......... .......... .......... .......... ..........  0%  162M 22s\n",
      "   200K .......... .......... .......... .......... ..........  0% 2.09M 23s\n",
      "   250K .......... .......... .......... .......... ..........  0%  142M 20s\n",
      "   300K .......... .......... .......... .......... ..........  0%  165M 17s\n",
      "   350K .......... .......... .......... .......... ..........  0%  137M 15s\n",
      "   400K .......... .......... .......... .......... ..........  0% 2.16M 16s\n",
      "   450K .......... .......... .......... .......... ..........  0%  124M 15s\n",
      "   500K .......... .......... .......... .......... ..........  0%  156M 13s\n",
      "   550K .......... .......... .......... .......... ..........  0%  153M 12s\n",
      "   600K .......... .......... .......... .......... ..........  1%  144M 11s\n",
      "   650K .......... .......... .......... .......... ..........  1%  177M 11s\n",
      "   700K .......... .......... .......... .......... ..........  1%  150M 10s\n",
      "   750K .......... .......... .......... .......... ..........  1%  167M 9s\n",
      "   800K .......... .......... .......... .......... ..........  1%  171M 9s\n",
      "   850K .......... .......... .......... .......... ..........  1% 2.27M 10s\n",
      "   900K .......... .......... .......... .......... ..........  1%  154M 9s\n",
      "   950K .......... .......... .......... .......... ..........  1%  138M 9s\n",
      "  1000K .......... .......... .......... .......... ..........  1%  162M 8s\n",
      "  1050K .......... .......... .......... .......... ..........  1%  152M 8s\n",
      "  1100K .......... .......... .......... .......... ..........  1%  146M 8s\n",
      "  1150K .......... .......... .......... .......... ..........  1%  165M 7s\n",
      "  1200K .......... .......... .......... .......... ..........  1%  145M 7s\n",
      "  1250K .......... .......... .......... .......... ..........  2%  177M 7s\n",
      "  1300K .......... .......... .......... .......... ..........  2%  173M 7s\n",
      "  1350K .......... .......... .......... .......... ..........  2%  146M 6s\n",
      "  1400K .......... .......... .......... .......... ..........  2%  173M 6s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  156M 6s\n",
      "  1500K .......... .......... .......... .......... ..........  2%  171M 6s\n",
      "  1550K .......... .......... .......... .......... ..........  2%  166M 6s\n",
      "  1600K .......... .......... .......... .......... ..........  2%  149M 5s\n",
      "  1650K .......... .......... .......... .......... ..........  2%  170M 5s\n",
      "  1700K .......... .......... .......... .......... ..........  2%  150M 5s\n",
      "  1750K .......... .......... .......... .......... ..........  2% 2.62M 6s\n",
      "  1800K .......... .......... .......... .......... ..........  2%  225M 5s\n",
      "  1850K .......... .......... .......... .......... ..........  3%  183M 5s\n",
      "  1900K .......... .......... .......... .......... ..........  3%  228M 5s\n",
      "  1950K .......... .......... .......... .......... ..........  3%  201M 5s\n",
      "  2000K .......... .......... .......... .......... ..........  3%  234M 5s\n",
      "  2050K .......... .......... .......... .......... ..........  3%  182M 5s\n",
      "  2100K .......... .......... .......... .......... ..........  3%  197M 5s\n",
      "  2150K .......... .......... .......... .......... ..........  3%  244M 5s\n",
      "  2200K .......... .......... .......... .......... ..........  3%  199M 4s\n",
      "  2250K .......... .......... .......... .......... ..........  3%  218M 4s\n",
      "  2300K .......... .......... .......... .......... ..........  3%  240M 4s\n",
      "  2350K .......... .......... .......... .......... ..........  3%  206M 4s\n",
      "  2400K .......... .......... .......... .......... ..........  3%  261M 4s\n",
      "  2450K .......... .......... .......... .......... ..........  3%  214M 4s\n",
      "  2500K .......... .......... .......... .......... ..........  4%  248M 4s\n",
      "  2550K .......... .......... .......... .......... ..........  4%  245M 4s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  214M 4s\n",
      "  2650K .......... .......... .......... .......... ..........  4%  269M 4s\n",
      "  2700K .......... .......... .......... .......... ..........  4%  214M 4s\n",
      "  2750K .......... .......... .......... .......... ..........  4%  263M 4s\n",
      "  2800K .......... .......... .......... .......... ..........  4%  253M 4s\n",
      "  2850K .......... .......... .......... .......... ..........  4%  229M 4s\n",
      "  2900K .......... .......... .......... .......... ..........  4%  249M 3s\n",
      "  2950K .......... .......... .......... .......... ..........  4%  210M 3s\n",
      "  3000K .......... .......... .......... .......... ..........  4%  252M 3s\n",
      "  3050K .......... .......... .......... .......... ..........  4%  250M 3s\n",
      "  3100K .......... .......... .......... .......... ..........  5%  222M 3s\n",
      "  3150K .......... .......... .......... .......... ..........  5%  250M 3s\n",
      "  3200K .......... .......... .......... .......... ..........  5%  214M 3s\n",
      "  3250K .......... .......... .......... .......... ..........  5%  251M 3s\n",
      "  3300K .......... .......... .......... .......... ..........  5%  264M 3s\n",
      "  3350K .......... .......... .......... .......... ..........  5%  215M 3s\n",
      "  3400K .......... .......... .......... .......... ..........  5%  247M 3s\n",
      "  3450K .......... .......... .......... .......... ..........  5%  210M 3s\n",
      "  3500K .......... .......... .......... .......... ..........  5%  256M 3s\n",
      "  3550K .......... .......... .......... .......... ..........  5% 2.82M 3s\n",
      "  3600K .......... .......... .......... .......... ..........  5%  248M 3s\n",
      "  3650K .......... .......... .......... .......... ..........  5%  375M 3s\n",
      "  3700K .......... .......... .......... .......... ..........  5%  343M 3s\n",
      "  3750K .......... .......... .......... .......... ..........  6%  353M 3s\n",
      "  3800K .......... .......... .......... .......... ..........  6%  399M 3s\n",
      "  3850K .......... .......... .......... .......... ..........  6%  345M 3s\n",
      "  3900K .......... .......... .......... .......... ..........  6%  372M 3s\n",
      "  3950K .......... .......... .......... .......... ..........  6%  362M 3s\n",
      "  4000K .......... .......... .......... .......... ..........  6%  391M 3s\n",
      "  4050K .......... .......... .......... .......... ..........  6%  401M 3s\n",
      "  4100K .......... .......... .......... .......... ..........  6% 4.98M 3s\n",
      "  4150K .......... .......... .......... .......... ..........  6%  371M 3s\n",
      "  4200K .......... .......... .......... .......... ..........  6%  315M 3s\n",
      "  4250K .......... .......... .......... .......... ..........  6%  368M 3s\n",
      "  4300K .......... .......... .......... .......... ..........  6%  400M 3s\n",
      "  4350K .......... .......... .......... .......... ..........  7%  300M 3s\n",
      "  4400K .......... .......... .......... .......... ..........  7%  391M 3s\n",
      "  4450K .......... .......... .......... .......... ..........  7%  329M 3s\n",
      "  4500K .......... .......... .......... .......... ..........  7%  380M 3s\n",
      "  4550K .......... .......... .......... .......... ..........  7%  375M 3s\n",
      "  4600K .......... .......... .......... .......... ..........  7%  324M 3s\n",
      "  4650K .......... .......... .......... .......... ..........  7%  385M 3s\n",
      "  4700K .......... .......... .......... .......... ..........  7%  328M 2s\n",
      "  4750K .......... .......... .......... .......... ..........  7%  396M 2s\n",
      "  4800K .......... .......... .......... .......... ..........  7%  395M 2s\n",
      "  4850K .......... .......... .......... .......... ..........  7%  331M 2s\n",
      "  4900K .......... .......... .......... .......... ..........  7%  358M 2s\n",
      "  4950K .......... .......... .......... .......... ..........  7%  331M 2s\n",
      "  5000K .......... .......... .......... .......... ..........  8%  395M 2s\n",
      "  5050K .......... .......... .......... .......... ..........  8%  377M 2s\n",
      "  5100K .......... .......... .......... .......... ..........  8%  325M 2s\n",
      "  5150K .......... .......... .......... .......... ..........  8%  386M 2s\n",
      "  5200K .......... .......... .......... .......... ..........  8%  328M 2s\n",
      "  5250K .......... .......... .......... .......... ..........  8%  371M 2s\n",
      "  5300K .......... .......... .......... .......... ..........  8%  376M 2s\n",
      "  5350K .......... .......... .......... .......... ..........  8%  312M 2s\n",
      "  5400K .......... .......... .......... .......... ..........  8%  370M 2s\n",
      "  5450K .......... .......... .......... .......... ..........  8%  345M 2s\n",
      "  5500K .......... .......... .......... .......... ..........  8%  377M 2s\n",
      "  5550K .......... .......... .......... .......... ..........  8%  382M 2s\n",
      "  5600K .......... .......... .......... .......... ..........  9% 5.45M 2s\n",
      "  5650K .......... .......... .......... .......... ..........  9%  323M 2s\n",
      "  5700K .......... .......... .......... .......... ..........  9%  313M 2s\n",
      "  5750K .......... .......... .......... .......... ..........  9%  337M 2s\n",
      "  5800K .......... .......... .......... .......... ..........  9%  345M 2s\n",
      "  5850K .......... .......... .......... .......... ..........  9%  305M 2s\n",
      "  5900K .......... .......... .......... .......... ..........  9%  391M 2s\n",
      "  5950K .......... .......... .......... .......... ..........  9%  317M 2s\n",
      "  6000K .......... .......... .......... .......... ..........  9%  375M 2s\n",
      "  6050K .......... .......... .......... .......... ..........  9%  366M 2s\n",
      "  6100K .......... .......... .......... .......... ..........  9%  255M 2s\n",
      "  6150K .......... .......... .......... .......... ..........  9%  399M 2s\n",
      "  6200K .......... .......... .......... .......... ..........  9%  337M 2s\n",
      "  6250K .......... .......... .......... .......... .......... 10%  408M 2s\n",
      "  6300K .......... .......... .......... .......... .......... 10%  411M 2s\n",
      "  6350K .......... .......... .......... .......... .......... 10%  356M 2s\n",
      "  6400K .......... .......... .......... .......... .......... 10%  384M 2s\n",
      "  6450K .......... .......... .......... .......... .......... 10%  348M 2s\n",
      "  6500K .......... .......... .......... .......... .......... 10%  411M 2s\n",
      "  6550K .......... .......... .......... .......... .......... 10%  402M 2s\n",
      "  6600K .......... .......... .......... .......... .......... 10%  351M 2s\n",
      "  6650K .......... .......... .......... .......... .......... 10%  429M 2s\n",
      "  6700K .......... .......... .......... .......... .......... 10%  351M 2s\n",
      "  6750K .......... .......... .......... .......... .......... 10%  395M 2s\n",
      "  6800K .......... .......... .......... .......... .......... 10%  400M 2s\n",
      "  6850K .......... .......... .......... .......... .......... 11%  342M 2s\n",
      "  6900K .......... .......... .......... .......... .......... 11%  413M 2s\n",
      "  6950K .......... .......... .......... .......... .......... 11%  362M 2s\n",
      "  7000K .......... .......... .......... .......... .......... 11%  415M 2s\n",
      "  7050K .......... .......... .......... .......... .......... 11%  399M 2s\n",
      "  7100K .......... .......... .......... .......... .......... 11%  361M 2s\n",
      "  7150K .......... .......... .......... .......... .......... 11%  415M 2s\n",
      "  7200K .......... .......... .......... .......... .......... 11%  334M 2s\n",
      "  7250K .......... .......... .......... .......... .......... 11% 7.18M 2s\n",
      "  7300K .......... .......... .......... .......... .......... 11%  369M 2s\n",
      "  7350K .......... .......... .......... .......... .......... 11%  382M 2s\n",
      "  7400K .......... .......... .......... .......... .......... 11% 3.85M 2s\n",
      "  7450K .......... .......... .......... .......... .......... 11%  293M 2s\n",
      "  7500K .......... .......... .......... .......... .......... 12%  355M 2s\n",
      "  7550K .......... .......... .......... .......... .......... 12%  455M 2s\n",
      "  7600K .......... .......... .......... .......... .......... 12%  345M 2s\n",
      "  7650K .......... .......... .......... .......... .......... 12%  397M 2s\n",
      "  7700K .......... .......... .......... .......... .......... 12%  365M 2s\n",
      "  7750K .......... .......... .......... .......... .......... 12%  403M 2s\n",
      "  7800K .......... .......... .......... .......... .......... 12%  426M 2s\n",
      "  7850K .......... .......... .......... .......... .......... 12%  361M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 12%  409M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 12%  376M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 12%  425M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 12%  437M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 13%  386M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 13%  431M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 13%  395M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 13%  461M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 13%  471M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 13%  395M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 13%  453M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 13%  384M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 13%  461M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 13%  418M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 13%  364M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 13%  425M 1s\n",
      "  8700K .......... .......... .......... .......... .......... 13%  341M 1s\n",
      "  8750K .......... .......... .......... .......... .......... 14%  434M 1s\n",
      "  8800K .......... .......... .......... .......... .......... 14%  409M 1s\n",
      "  8850K .......... .......... .......... .......... .......... 14%  360M 1s\n",
      "  8900K .......... .......... .......... .......... .......... 14%  421M 1s\n",
      "  8950K .......... .......... .......... .......... .......... 14%  365M 1s\n",
      "  9000K .......... .......... .......... .......... .......... 14%  428M 1s\n",
      "  9050K .......... .......... .......... .......... .......... 14%  410M 1s\n",
      "  9100K .......... .......... .......... .......... .......... 14%  338M 1s\n",
      "  9150K .......... .......... .......... .......... .......... 14%  411M 1s\n",
      "  9200K .......... .......... .......... .......... .......... 14%  365M 1s\n",
      "  9250K .......... .......... .......... .......... .......... 14%  440M 1s\n",
      "  9300K .......... .......... .......... .......... .......... 14%  410M 1s\n",
      "  9350K .......... .......... .......... .......... .......... 15%  346M 1s\n",
      "  9400K .......... .......... .......... .......... .......... 15% 7.99M 1s\n",
      "  9450K .......... .......... .......... .......... .......... 15%  313M 1s\n",
      "  9500K .......... .......... .......... .......... .......... 15%  350M 1s\n",
      "  9550K .......... .......... .......... .......... .......... 15%  366M 1s\n",
      "  9600K .......... .......... .......... .......... .......... 15%  332M 1s\n",
      "  9650K .......... .......... .......... .......... .......... 15%  397M 1s\n",
      "  9700K .......... .......... .......... .......... .......... 15%  366M 1s\n",
      "  9750K .......... .......... .......... .......... .......... 15%  409M 1s\n",
      "  9800K .......... .......... .......... .......... .......... 15%  424M 1s\n",
      "  9850K .......... .......... .......... .......... .......... 15%  356M 1s\n",
      "  9900K .......... .......... .......... .......... .......... 15%  415M 1s\n",
      "  9950K .......... .......... .......... .......... .......... 15%  378M 1s\n",
      " 10000K .......... .......... .......... .......... .......... 16%  410M 1s\n",
      " 10050K .......... .......... .......... .......... .......... 16%  419M 1s\n",
      " 10100K .......... .......... .......... .......... .......... 16%  358M 1s\n",
      " 10150K .......... .......... .......... .......... .......... 16%  455M 1s\n",
      " 10200K .......... .......... .......... .......... .......... 16%  356M 1s\n",
      " 10250K .......... .......... .......... .......... .......... 16%  416M 1s\n",
      " 10300K .......... .......... .......... .......... .......... 16%  394M 1s\n",
      " 10350K .......... .......... .......... .......... .......... 16% 4.63M 1s\n",
      " 10400K .......... .......... .......... .......... .......... 16%  395M 1s\n",
      " 10450K .......... .......... .......... .......... .......... 16%  354M 1s\n",
      " 10500K .......... .......... .......... .......... .......... 16%  379M 1s\n",
      " 10550K .......... .......... .......... .......... .......... 16%  449M 1s\n",
      " 10600K .......... .......... .......... .......... .......... 17%  358M 1s\n",
      " 10650K .......... .......... .......... .......... .......... 17%  458M 1s\n",
      " 10700K .......... .......... .......... .......... .......... 17%  389M 1s\n",
      " 10750K .......... .......... .......... .......... .......... 17%  443M 1s\n",
      " 10800K .......... .......... .......... .......... .......... 17%  451M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 17%  339M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 17%  436M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 17%  376M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 17%  469M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 17%  429M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 17%  387M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 17%  474M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 17%  400M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 18%  324M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 18%  457M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 18%  398M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 18%  468M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 18%  422M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 18%  467M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 18% 5.78M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 18%  259M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 18%  394M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 18%  344M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 18%  414M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 18%  473M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 18%  342M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 19%  414M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 19%  369M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 19%  405M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 19%  483M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 19%  392M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 19%  472M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 19%  397M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 19%  459M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 19%  488M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 19%  396M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 19%  460M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 19%  388M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 20%  451M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 20%  486M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 20%  403M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 20%  492M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 20%  413M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 20%  478M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 20%  459M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 20% 4.98M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 20%  369M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 20%  346M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 20%  444M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 20%  362M 1s\n",
      " 13100K .......... .......... .......... .......... .......... 20%  349M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 21%  402M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 21%  379M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 21%  480M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 21%  434M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 21%  371M 1s\n",
      " 13400K .......... .......... .......... .......... .......... 21%  451M 1s\n",
      " 13450K .......... .......... .......... .......... .......... 21%  360M 1s\n",
      " 13500K .......... .......... .......... .......... .......... 21%  491M 1s\n",
      " 13550K .......... .......... .......... .......... .......... 21%  434M 1s\n",
      " 13600K .......... .......... .......... .......... .......... 21%  375M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 21%  482M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 21%  411M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 22%  447M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 22%  483M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 22%  425M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 22%  453M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 22%  419M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 22%  457M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 22%  446M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 22%  406M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 22%  492M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 22%  385M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 22%  448M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 22%  446M 1s\n",
      " 14350K .......... .......... .......... .......... .......... 22%  400M 1s\n",
      " 14400K .......... .......... .......... .......... .......... 23%  485M 1s\n",
      " 14450K .......... .......... .......... .......... .......... 23% 6.39M 1s\n",
      " 14500K .......... .......... .......... .......... .......... 23%  373M 1s\n",
      " 14550K .......... .......... .......... .......... .......... 23%  363M 1s\n",
      " 14600K .......... .......... .......... .......... .......... 23%  348M 1s\n",
      " 14650K .......... .......... .......... .......... .......... 23%  435M 1s\n",
      " 14700K .......... .......... .......... .......... .......... 23%  377M 1s\n",
      " 14750K .......... .......... .......... .......... .......... 23%  396M 1s\n",
      " 14800K .......... .......... .......... .......... .......... 23%  456M 1s\n",
      " 14850K .......... .......... .......... .......... .......... 23%  372M 1s\n",
      " 14900K .......... .......... .......... .......... .......... 23%  426M 1s\n",
      " 14950K .......... .......... .......... .......... .......... 23%  375M 1s\n",
      " 15000K .......... .......... .......... .......... .......... 24%  460M 1s\n",
      " 15050K .......... .......... .......... .......... .......... 24%  459M 1s\n",
      " 15100K .......... .......... .......... .......... .......... 24%  391M 1s\n",
      " 15150K .......... .......... .......... .......... .......... 24%  479M 1s\n",
      " 15200K .......... .......... .......... .......... .......... 24%  147M 1s\n",
      " 15250K .......... .......... .......... .......... .......... 24% 4.66M 1s\n",
      " 15300K .......... .......... .......... .......... .......... 24%  381M 1s\n",
      " 15350K .......... .......... .......... .......... .......... 24%  375M 1s\n",
      " 15400K .......... .......... .......... .......... .......... 24%  427M 1s\n",
      " 15450K .......... .......... .......... .......... .......... 24%  362M 1s\n",
      " 15500K .......... .......... .......... .......... .......... 24%  423M 1s\n",
      " 15550K .......... .......... .......... .......... .......... 24%  415M 1s\n",
      " 15600K .......... .......... .......... .......... .......... 24%  409M 1s\n",
      " 15650K .......... .......... .......... .......... .......... 25%  422M 1s\n",
      " 15700K .......... .......... .......... .......... .......... 25%  377M 1s\n",
      " 15750K .......... .......... .......... .......... .......... 25%  394M 1s\n",
      " 15800K .......... .......... .......... .......... .......... 25%  499M 1s\n",
      " 15850K .......... .......... .......... .......... .......... 25%  364M 1s\n",
      " 15900K .......... .......... .......... .......... .......... 25%  436M 1s\n",
      " 15950K .......... .......... .......... .......... .......... 25%  384M 1s\n",
      " 16000K .......... .......... .......... .......... .......... 25%  458M 1s\n",
      " 16050K .......... .......... .......... .......... .......... 25%  444M 1s\n",
      " 16100K .......... .......... .......... .......... .......... 25%  394M 1s\n",
      " 16150K .......... .......... .......... .......... .......... 25%  457M 1s\n",
      " 16200K .......... .......... .......... .......... .......... 25%  413M 1s\n",
      " 16250K .......... .......... .......... .......... .......... 26%  469M 1s\n",
      " 16300K .......... .......... .......... .......... .......... 26%  466M 1s\n",
      " 16350K .......... .......... .......... .......... .......... 26%  407M 1s\n",
      " 16400K .......... .......... .......... .......... .......... 26%  439M 1s\n",
      " 16450K .......... .......... .......... .......... .......... 26%  402M 1s\n",
      " 16500K .......... .......... .......... .......... .......... 26%  440M 1s\n",
      " 16550K .......... .......... .......... .......... .......... 26%  494M 1s\n",
      " 16600K .......... .......... .......... .......... .......... 26%  397M 1s\n",
      " 16650K .......... .......... .......... .......... .......... 26%  466M 1s\n",
      " 16700K .......... .......... .......... .......... .......... 26%  396M 1s\n",
      " 16750K .......... .......... .......... .......... .......... 26%  485M 1s\n",
      " 16800K .......... .......... .......... .......... .......... 26%  493M 1s\n",
      " 16850K .......... .......... .......... .......... .......... 26%  381M 1s\n",
      " 16900K .......... .......... .......... .......... .......... 27%  456M 1s\n",
      " 16950K .......... .......... .......... .......... .......... 27% 6.41M 1s\n",
      " 17000K .......... .......... .......... .......... .......... 27%  402M 1s\n",
      " 17050K .......... .......... .......... .......... .......... 27%  356M 1s\n",
      " 17100K .......... .......... .......... .......... .......... 27%  358M 1s\n",
      " 17150K .......... .......... .......... .......... .......... 27%  407M 1s\n",
      " 17200K .......... .......... .......... .......... .......... 27%  355M 1s\n",
      " 17250K .......... .......... .......... .......... .......... 27%  459M 1s\n",
      " 17300K .......... .......... .......... .......... .......... 27%  415M 1s\n",
      " 17350K .......... .......... .......... .......... .......... 27%  387M 1s\n",
      " 17400K .......... .......... .......... .......... .......... 27%  453M 1s\n",
      " 17450K .......... .......... .......... .......... .......... 27%  373M 1s\n",
      " 17500K .......... .......... .......... .......... .......... 28%  492M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 28%  471M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 28%  375M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 28%  421M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 28%  399M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 28%  463M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 28%  450M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 28%  397M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 28%  312M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 28%  405M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 28%  427M 1s\n",
      " 18050K .......... .......... .......... .......... .......... 28%  448M 1s\n",
      " 18100K .......... .......... .......... .......... .......... 28%  397M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 29%  463M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 29%  425M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 29% 5.10M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 29%  272M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 29%  360M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 29%  433M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 29%  345M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 29%  430M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 29%  325M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 29%  276M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 29%  448M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 29%  388M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 30%  149M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 30%  392M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 30%  274M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 30%  498M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 30%  287M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 30%  384M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 30%  349M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 30%  362M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 30%  431M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 30%  388M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 30% 39.1M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 30%  401M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 30%  391M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 31%  432M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 31%  371M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 31%  418M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 31% 7.55M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 31%  263M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 31%  394M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 31%  344M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 31%  392M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 31%  431M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 31%  389M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 31%  421M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 31%  384M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 32%  446M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 32%  468M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 32%  396M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 32%  413M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 32%  376M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 32%  429M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 32%  501M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 32%  390M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 32%  471M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 32%  369M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 32%  428M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 32%  441M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 32%  364M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 33%  456M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 33%  412M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 33%  463M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 33%  494M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 33%  402M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 33%  475M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 33%  410M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 33%  487M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 33% 5.01M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 33%  320M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 33%  383M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 33%  349M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 34%  447M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 34%  450M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 34%  366M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 34%  429M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 34%  384M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 34%  455M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 34%  432M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 34%  351M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 34%  445M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 34%  390M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 34%  473M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 34%  447M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 34%  364M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 35%  457M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 35%  415M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 35%  434M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 35%  448M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 35%  401M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 35%  481M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 35%  422M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 35%  122M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 35%  129M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 35%  145M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 35%  429M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 35%  421M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 36% 7.45M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 36%  344M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 36%  332M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 36%  445M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 36%  339M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 36%  401M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 36%  404M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 36%  358M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 36%  454M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 36%  366M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 36%  418M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 36%  420M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 36%  359M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 37%  453M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 37%  350M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 37%  425M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 37%  456M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 37%  372M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 37%  456M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 37%  352M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 37%  434M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 37%  438M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 37%  398M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 37%  462M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 37% 72.3M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 37% 4.93M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 38%  371M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 38%  373M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 38%  388M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 38%  345M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 38%  424M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 38%  426M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 38%  373M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 38%  456M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 38%  386M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 38%  432M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 38%  454M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 38%  401M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 39%  447M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 39%  368M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 39%  452M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 39%  459M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 39%  412M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 39%  475M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 39%  412M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 39%  449M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 39%  498M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 39%  393M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 39%  442M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 39%  369M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 39%  412M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 40%  474M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 40%  381M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 40%  439M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 40%  403M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 40%  489M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 40%  470M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 40%  403M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 40% 83.0M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 40% 7.93M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 40%  384M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 40%  424M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 40%  353M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 41%  413M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 41%  358M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 41%  471M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 41%  400M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 41%  366M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 41%  432M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 41%  374M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 41%  445M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 41%  436M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 41%  392M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 41%  434M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 41%  382M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 41%  421M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 42%  415M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 42%  375M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 42%  452M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 42%  412M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 42%  458M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 42% 89.5M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 42%  357M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 42%  470M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 42%  396M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 42% 4.91M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 42%  375M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 42%  335M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 43%  462M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 43%  339M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 43%  445M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 43%  449M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 43%  396M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 43%  492M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 43%  379M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 43%  430M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 43%  466M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 43%  295M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 43%  472M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 43%  418M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 43%  498M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 44%  458M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 44%  428M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 44%  111M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 44%  372M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 44%  424M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 44%  456M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 44%  396M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 44%  474M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 44%  417M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 44%  465M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 44%  473M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 44%  403M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 45%  469M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 45%  120M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 45%  413M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 45%  482M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 45%  408M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 45%  444M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 45% 8.04M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 45%  373M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 45%  425M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 45%  355M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 45%  416M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 45%  311M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 45%  415M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 46%  453M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 46%  362M 0s\n",
      " 28900K .......... .......... .......... .......... .......... 46%  435M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 46%  414M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 46%  460M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 46%  471M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 46%  218M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 46%  419M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 46%  298M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 46%  447M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 46%  446M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 46%  389M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 47%  465M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 47% 89.8M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 47%  465M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 47%  543M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 47%  109M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 47% 67.5M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 47%  294M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 47%  463M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 47% 5.48M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 47%  334M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 47%  427M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 47%  350M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 47%  468M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 48%  411M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 48%  399M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 48%  447M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 48%  393M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 48%  416M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 48%  472M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 48%  405M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 48%  432M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 48%  415M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 48%  107M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 48%  410M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 48%  399M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 49%  426M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 49%  402M 0s\n",
      " 30750K .......... .......... .......... .......... .......... 49%  457M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 49%  456M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 49%  408M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 49%  467M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 49% 80.7M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 49%  438M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 49%  427M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 49%  399M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 49%  474M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 49%  392M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 49%  463M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 50% 7.98M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 50%  304M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 50%  446M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 50%  365M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 50%  430M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 50%  414M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 50%  367M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 50%  435M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 50%  383M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 50%  414M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 50%  471M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 50%  401M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 51%  412M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 51%  363M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 51%  446M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 51%  429M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 51%  405M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 51%  408M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 51%  395M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 51%  458M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 51%  455M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 51%  145M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 51%  413M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 51%  395M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 51%  259M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 52%  434M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 52%  391M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 52% 68.6M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 52%  245M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 52% 5.43M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 52%  337M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 52%  360M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 52%  419M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 52%  367M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 52%  426M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 52%  456M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 52%  324M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 53%  459M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 53%  405M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 53%  452M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 53%  192M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 53%  389M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 53%  317M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 53%  317M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 53%  484M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 53%  123M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 53%  341M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 53%  457M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 53%  390M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 53%  480M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 54% 72.1M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 54%  350M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 54%  430M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 54%  384M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 54%  454M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 54%  450M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 54%  393M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 54% 76.9M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 54% 8.86M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 54%  385M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 54%  408M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 54%  349M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 54%  397M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 55%  390M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 55%  434M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 55%  428M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 55%  384M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 55%  465M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 55%  409M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 55%  329M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 55%  423M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 55%  352M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 55%  434M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 55%  399M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 55%  429M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 56%  452M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 56%  356M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 56%  440M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 56%  361M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 56%  464M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 56%  415M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 56%  385M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 56%  319M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 56%  386M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 56%  470M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 56%  451M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 56%  167M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 56%  175M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 57%  111M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 57% 57.6M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 57% 57.6M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 57% 6.44M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 57%  419M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 57%  376M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 57%  313M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 57%  434M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 57%  388M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 57%  430M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 57%  248M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 57%  202M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 58%  450M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 58%  197M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 58%  430M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 58%  398M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 58%  441M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 58%  512M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 58%  138M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 58% 65.5M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 58%  362M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 58%  433M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 58%  501M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 58%  394M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 58%  466M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 59%  380M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 59%  493M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 59%  497M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 59% 79.5M 0s\n",
      " 37150K .......... .......... .......... .......... .......... 59%  468M 0s\n",
      " 37200K .......... .......... .......... .......... .......... 59% 8.92M 0s\n",
      " 37250K .......... .......... .......... .......... .......... 59%  131M 0s\n",
      " 37300K .......... .......... .......... .......... .......... 59%  384M 0s\n",
      " 37350K .......... .......... .......... .......... .......... 59%  342M 0s\n",
      " 37400K .......... .......... .......... .......... .......... 59%  429M 0s\n",
      " 37450K .......... .......... .......... .......... .......... 59%  400M 0s\n",
      " 37500K .......... .......... .......... .......... .......... 59%  480M 0s\n",
      " 37550K .......... .......... .......... .......... .......... 60%  443M 0s\n",
      " 37600K .......... .......... .......... .......... .......... 60%  370M 0s\n",
      " 37650K .......... .......... .......... .......... .......... 60%  484M 0s\n",
      " 37700K .......... .......... .......... .......... .......... 60%  388M 0s\n",
      " 37750K .......... .......... .......... .......... .......... 60%  442M 0s\n",
      " 37800K .......... .......... .......... .......... .......... 60%  483M 0s\n",
      " 37850K .......... .......... .......... .......... .......... 60%  406M 0s\n",
      " 37900K .......... .......... .......... .......... .......... 60%  440M 0s\n",
      " 37950K .......... .......... .......... .......... .......... 60%  409M 0s\n",
      " 38000K .......... .......... .......... .......... .......... 60%  470M 0s\n",
      " 38050K .......... .......... .......... .......... .......... 60%  480M 0s\n",
      " 38100K .......... .......... .......... .......... .......... 60%  389M 0s\n",
      " 38150K .......... .......... .......... .......... .......... 60%  483M 0s\n",
      " 38200K .......... .......... .......... .......... .......... 61%  388M 0s\n",
      " 38250K .......... .......... .......... .......... .......... 61%  491M 0s\n",
      " 38300K .......... .......... .......... .......... .......... 61%  481M 0s\n",
      " 38350K .......... .......... .......... .......... .......... 61%  385M 0s\n",
      " 38400K .......... .......... .......... .......... .......... 61%  511M 0s\n",
      " 38450K .......... .......... .......... .......... .......... 61%  427M 0s\n",
      " 38500K .......... .......... .......... .......... .......... 61%  276M 0s\n",
      " 38550K .......... .......... .......... .......... .......... 61%  110M 0s\n",
      " 38600K .......... .......... .......... .......... .......... 61%  384M 0s\n",
      " 38650K .......... .......... .......... .......... .......... 61%  455M 0s\n",
      " 38700K .......... .......... .......... .......... .......... 61%  427M 0s\n",
      " 38750K .......... .......... .......... .......... .......... 61% 50.4M 0s\n",
      " 38800K .......... .......... .......... .......... .......... 62% 49.6M 0s\n",
      " 38850K .......... .......... .......... .......... .......... 62% 6.40M 0s\n",
      " 38900K .......... .......... .......... .......... .......... 62%  372M 0s\n",
      " 38950K .......... .......... .......... .......... .......... 62%  338M 0s\n",
      " 39000K .......... .......... .......... .......... .......... 62%  442M 0s\n",
      " 39050K .......... .......... .......... .......... .......... 62%  443M 0s\n",
      " 39100K .......... .......... .......... .......... .......... 62%  151M 0s\n",
      " 39150K .......... .......... .......... .......... .......... 62%  423M 0s\n",
      " 39200K .......... .......... .......... .......... .......... 62%  393M 0s\n",
      " 39250K .......... .......... .......... .......... .......... 62%  322M 0s\n",
      " 39300K .......... .......... .......... .......... .......... 62%  439M 0s\n",
      " 39350K .......... .......... .......... .......... .......... 62%  228M 0s\n",
      " 39400K .......... .......... .......... .......... .......... 62%  529M 0s\n",
      " 39450K .......... .......... .......... .......... .......... 63%  321M 0s\n",
      " 39500K .......... .......... .......... .......... .......... 63%  315M 0s\n",
      " 39550K .......... .......... .......... .......... .......... 63%  416M 0s\n",
      " 39600K .......... .......... .......... .......... .......... 63%  148M 0s\n",
      " 39650K .......... .......... .......... .......... .......... 63%  203M 0s\n",
      " 39700K .......... .......... .......... .......... .......... 63% 97.3M 0s\n",
      " 39750K .......... .......... .......... .......... .......... 63%  455M 0s\n",
      " 39800K .......... .......... .......... .......... .......... 63%  182M 0s\n",
      " 39850K .......... .......... .......... .......... .......... 63%  367M 0s\n",
      " 39900K .......... .......... .......... .......... .......... 63%  510M 0s\n",
      " 39950K .......... .......... .......... .......... .......... 63%  111M 0s\n",
      " 40000K .......... .......... .......... .......... .......... 63%  430M 0s\n",
      " 40050K .......... .......... .......... .......... .......... 64%  457M 0s\n",
      " 40100K .......... .......... .......... .......... .......... 64%  420M 0s\n",
      " 40150K .......... .......... .......... .......... .......... 64% 8.66M 0s\n",
      " 40200K .......... .......... .......... .......... .......... 64%  341M 0s\n",
      " 40250K .......... .......... .......... .......... .......... 64%  342M 0s\n",
      " 40300K .......... .......... .......... .......... .......... 64%  126M 0s\n",
      " 40350K .......... .......... .......... .......... .......... 64%  414M 0s\n",
      " 40400K .......... .......... .......... .......... .......... 64%  473M 0s\n",
      " 40450K .......... .......... .......... .......... .......... 64%  365M 0s\n",
      " 40500K .......... .......... .......... .......... .......... 64%  439M 0s\n",
      " 40550K .......... .......... .......... .......... .......... 64%  457M 0s\n",
      " 40600K .......... .......... .......... .......... .......... 64%  393M 0s\n",
      " 40650K .......... .......... .......... .......... .......... 64%  429M 0s\n",
      " 40700K .......... .......... .......... .......... .......... 65%  395M 0s\n",
      " 40750K .......... .......... .......... .......... .......... 65%  457M 0s\n",
      " 40800K .......... .......... .......... .......... .......... 65%  428M 0s\n",
      " 40850K .......... .......... .......... .......... .......... 65%  402M 0s\n",
      " 40900K .......... .......... .......... .......... .......... 65%  469M 0s\n",
      " 40950K .......... .......... .......... .......... .......... 65%  386M 0s\n",
      " 41000K .......... .......... .......... .......... .......... 65%  453M 0s\n",
      " 41050K .......... .......... .......... .......... .......... 65%  483M 0s\n",
      " 41100K .......... .......... .......... .......... .......... 65%  393M 0s\n",
      " 41150K .......... .......... .......... .......... .......... 65%  475M 0s\n",
      " 41200K .......... .......... .......... .......... .......... 65%  381M 0s\n",
      " 41250K .......... .......... .......... .......... .......... 65%  415M 0s\n",
      " 41300K .......... .......... .......... .......... .......... 66%  509M 0s\n",
      " 41350K .......... .......... .......... .......... .......... 66%  399M 0s\n",
      " 41400K .......... .......... .......... .......... .......... 66%  448M 0s\n",
      " 41450K .......... .......... .......... .......... .......... 66%  418M 0s\n",
      " 41500K .......... .......... .......... .......... .......... 66%  440M 0s\n",
      " 41550K .......... .......... .......... .......... .......... 66%  170M 0s\n",
      " 41600K .......... .......... .......... .......... .......... 66%  309M 0s\n",
      " 41650K .......... .......... .......... .......... .......... 66% 62.5M 0s\n",
      " 41700K .......... .......... .......... .......... .......... 66%  342M 0s\n",
      " 41750K .......... .......... .......... .......... .......... 66%  352M 0s\n",
      " 41800K .......... .......... .......... .......... .......... 66% 57.2M 0s\n",
      " 41850K .......... .......... .......... .......... .......... 66% 6.32M 0s\n",
      " 41900K .......... .......... .......... .......... .......... 66%  233M 0s\n",
      " 41950K .......... .......... .......... .......... .......... 67%  343M 0s\n",
      " 42000K .......... .......... .......... .......... .......... 67%  481M 0s\n",
      " 42050K .......... .......... .......... .......... .......... 67%  135M 0s\n",
      " 42100K .......... .......... .......... .......... .......... 67%  352M 0s\n",
      " 42150K .......... .......... .......... .......... .......... 67%  462M 0s\n",
      " 42200K .......... .......... .......... .......... .......... 67%  376M 0s\n",
      " 42250K .......... .......... .......... .......... .......... 67%  485M 0s\n",
      " 42300K .......... .......... .......... .......... .......... 67%  197M 0s\n",
      " 42350K .......... .......... .......... .......... .......... 67%  257M 0s\n",
      " 42400K .......... .......... .......... .......... .......... 67%  434M 0s\n",
      " 42450K .......... .......... .......... .......... .......... 67%  398M 0s\n",
      " 42500K .......... .......... .......... .......... .......... 67%  456M 0s\n",
      " 42550K .......... .......... .......... .......... .......... 68%  471M 0s\n",
      " 42600K .......... .......... .......... .......... .......... 68%  256M 0s\n",
      " 42650K .......... .......... .......... .......... .......... 68%  201M 0s\n",
      " 42700K .......... .......... .......... .......... .......... 68%  185M 0s\n",
      " 42750K .......... .......... .......... .......... .......... 68% 77.3M 0s\n",
      " 42800K .......... .......... .......... .......... .......... 68%  451M 0s\n",
      " 42850K .......... .......... .......... .......... .......... 68%  119M 0s\n",
      " 42900K .......... .......... .......... .......... .......... 68%  367M 0s\n",
      " 42950K .......... .......... .......... .......... .......... 68%  412M 0s\n",
      " 43000K .......... .......... .......... .......... .......... 68% 59.5M 0s\n",
      " 43050K .......... .......... .......... .......... .......... 68% 9.42M 0s\n",
      " 43100K .......... .......... .......... .......... .......... 68%  353M 0s\n",
      " 43150K .......... .......... .......... .......... .......... 68%  456M 0s\n",
      " 43200K .......... .......... .......... .......... .......... 69%  317M 0s\n",
      " 43250K .......... .......... .......... .......... .......... 69%  348M 0s\n",
      " 43300K .......... .......... .......... .......... .......... 69%  447M 0s\n",
      " 43350K .......... .......... .......... .......... .......... 69%  243M 0s\n",
      " 43400K .......... .......... .......... .......... .......... 69%  337M 0s\n",
      " 43450K .......... .......... .......... .......... .......... 69%  461M 0s\n",
      " 43500K .......... .......... .......... .......... .......... 69%  321M 0s\n",
      " 43550K .......... .......... .......... .......... .......... 69%  447M 0s\n",
      " 43600K .......... .......... .......... .......... .......... 69%  373M 0s\n",
      " 43650K .......... .......... .......... .......... .......... 69%  479M 0s\n",
      " 43700K .......... .......... .......... .......... .......... 69%  360M 0s\n",
      " 43750K .......... .......... .......... .......... .......... 69%  440M 0s\n",
      " 43800K .......... .......... .......... .......... .......... 70%  441M 0s\n",
      " 43850K .......... .......... .......... .......... .......... 70%  388M 0s\n",
      " 43900K .......... .......... .......... .......... .......... 70%  458M 0s\n",
      " 43950K .......... .......... .......... .......... .......... 70%  383M 0s\n",
      " 44000K .......... .......... .......... .......... .......... 70%  431M 0s\n",
      " 44050K .......... .......... .......... .......... .......... 70%  411M 0s\n",
      " 44100K .......... .......... .......... .......... .......... 70%  382M 0s\n",
      " 44150K .......... .......... .......... .......... .......... 70%  478M 0s\n",
      " 44200K .......... .......... .......... .......... .......... 70%  366M 0s\n",
      " 44250K .......... .......... .......... .......... .......... 70%  387M 0s\n",
      " 44300K .......... .......... .......... .......... .......... 70%  445M 0s\n",
      " 44350K .......... .......... .......... .......... .......... 70%  390M 0s\n",
      " 44400K .......... .......... .......... .......... .......... 70%  472M 0s\n",
      " 44450K .......... .......... .......... .......... .......... 71%  391M 0s\n",
      " 44500K .......... .......... .......... .......... .......... 71%  460M 0s\n",
      " 44550K .......... .......... .......... .......... .......... 71%  274M 0s\n",
      " 44600K .......... .......... .......... .......... .......... 71%  458M 0s\n",
      " 44650K .......... .......... .......... .......... .......... 71% 63.4M 0s\n",
      " 44700K .......... .......... .......... .......... .......... 71%  282M 0s\n",
      " 44750K .......... .......... .......... .......... .......... 71% 80.3M 0s\n",
      " 44800K .......... .......... .......... .......... .......... 71%  243M 0s\n",
      " 44850K .......... .......... .......... .......... .......... 71% 47.1M 0s\n",
      " 44900K .......... .......... .......... .......... .......... 71% 7.03M 0s\n",
      " 44950K .......... .......... .......... .......... .......... 71%  344M 0s\n",
      " 45000K .......... .......... .......... .......... .......... 71%  284M 0s\n",
      " 45050K .......... .......... .......... .......... .......... 72%  207M 0s\n",
      " 45100K .......... .......... .......... .......... .......... 72%  255M 0s\n",
      " 45150K .......... .......... .......... .......... .......... 72%  214M 0s\n",
      " 45200K .......... .......... .......... .......... .......... 72%  214M 0s\n",
      " 45250K .......... .......... .......... .......... .......... 72%  231M 0s\n",
      " 45300K .......... .......... .......... .......... .......... 72%  461M 0s\n",
      " 45350K .......... .......... .......... .......... .......... 72%  363M 0s\n",
      " 45400K .......... .......... .......... .......... .......... 72%  470M 0s\n",
      " 45450K .......... .......... .......... .......... .......... 72%  395M 0s\n",
      " 45500K .......... .......... .......... .......... .......... 72%  449M 0s\n",
      " 45550K .......... .......... .......... .......... .......... 72%  457M 0s\n",
      " 45600K .......... .......... .......... .......... .......... 72%  179M 0s\n",
      " 45650K .......... .......... .......... .......... .......... 72%  245M 0s\n",
      " 45700K .......... .......... .......... .......... .......... 73%  144M 0s\n",
      " 45750K .......... .......... .......... .......... .......... 73%  489M 0s\n",
      " 45800K .......... .......... .......... .......... .......... 73% 68.2M 0s\n",
      " 45850K .......... .......... .......... .......... .......... 73%  370M 0s\n",
      " 45900K .......... .......... .......... .......... .......... 73%  412M 0s\n",
      " 45950K .......... .......... .......... .......... .......... 73% 59.3M 0s\n",
      " 46000K .......... .......... .......... .......... .......... 73%  457M 0s\n",
      " 46050K .......... .......... .......... .......... .......... 73% 9.18M 0s\n",
      " 46100K .......... .......... .......... .......... .......... 73%  170M 0s\n",
      " 46150K .......... .......... .......... .......... .......... 73%  239M 0s\n",
      " 46200K .......... .......... .......... .......... .......... 73%  388M 0s\n",
      " 46250K .......... .......... .......... .......... .......... 73%  433M 0s\n",
      " 46300K .......... .......... .......... .......... .......... 73%  445M 0s\n",
      " 46350K .......... .......... .......... .......... .......... 74%  390M 0s\n",
      " 46400K .......... .......... .......... .......... .......... 74%  468M 0s\n",
      " 46450K .......... .......... .......... .......... .......... 74%  384M 0s\n",
      " 46500K .......... .......... .......... .......... .......... 74%  230M 0s\n",
      " 46550K .......... .......... .......... .......... .......... 74%  236M 0s\n",
      " 46600K .......... .......... .......... .......... .......... 74%  376M 0s\n",
      " 46650K .......... .......... .......... .......... .......... 74%  468M 0s\n",
      " 46700K .......... .......... .......... .......... .......... 74%  409M 0s\n",
      " 46750K .......... .......... .......... .......... .......... 74%  505M 0s\n",
      " 46800K .......... .......... .......... .......... .......... 74%  441M 0s\n",
      " 46850K .......... .......... .......... .......... .......... 74%  413M 0s\n",
      " 46900K .......... .......... .......... .......... .......... 74%  446M 0s\n",
      " 46950K .......... .......... .......... .......... .......... 75%  213M 0s\n",
      " 47000K .......... .......... .......... .......... .......... 75%  241M 0s\n",
      " 47050K .......... .......... .......... .......... .......... 75%  455M 0s\n",
      " 47100K .......... .......... .......... .......... .......... 75%  362M 0s\n",
      " 47150K .......... .......... .......... .......... .......... 75%  443M 0s\n",
      " 47200K .......... .......... .......... .......... .......... 75%  411M 0s\n",
      " 47250K .......... .......... .......... .......... .......... 75%  451M 0s\n",
      " 47300K .......... .......... .......... .......... .......... 75%  454M 0s\n",
      " 47350K .......... .......... .......... .......... .......... 75%  172M 0s\n",
      " 47400K .......... .......... .......... .......... .......... 75%  487M 0s\n",
      " 47450K .......... .......... .......... .......... .......... 75%  380M 0s\n",
      " 47500K .......... .......... .......... .......... .......... 75%  460M 0s\n",
      " 47550K .......... .......... .......... .......... .......... 75%  335M 0s\n",
      " 47600K .......... .......... .......... .......... .......... 76%  260M 0s\n",
      " 47650K .......... .......... .......... .......... .......... 76%  484M 0s\n",
      " 47700K .......... .......... .......... .......... .......... 76%  416M 0s\n",
      " 47750K .......... .......... .......... .......... .......... 76%  126M 0s\n",
      " 47800K .......... .......... .......... .......... .......... 76% 41.2M 0s\n",
      " 47850K .......... .......... .......... .......... .......... 76%  368M 0s\n",
      " 47900K .......... .......... .......... .......... .......... 76%  339M 0s\n",
      " 47950K .......... .......... .......... .......... .......... 76% 6.97M 0s\n",
      " 48000K .......... .......... .......... .......... .......... 76%  154M 0s\n",
      " 48050K .......... .......... .......... .......... .......... 76%  526M 0s\n",
      " 48100K .......... .......... .......... .......... .......... 76%  151M 0s\n",
      " 48150K .......... .......... .......... .......... .......... 76%  470M 0s\n",
      " 48200K .......... .......... .......... .......... .......... 77%  188M 0s\n",
      " 48250K .......... .......... .......... .......... .......... 77%  118M 0s\n",
      " 48300K .......... .......... .......... .......... .......... 77%  300M 0s\n",
      " 48350K .......... .......... .......... .......... .......... 77%  204M 0s\n",
      " 48400K .......... .......... .......... .......... .......... 77%  452M 0s\n",
      " 48450K .......... .......... .......... .......... .......... 77%  392M 0s\n",
      " 48500K .......... .......... .......... .......... .......... 77%  453M 0s\n",
      " 48550K .......... .......... .......... .......... .......... 77%  366M 0s\n",
      " 48600K .......... .......... .......... .......... .......... 77%  320M 0s\n",
      " 48650K .......... .......... .......... .......... .......... 77%  182M 0s\n",
      " 48700K .......... .......... .......... .......... .......... 77%  231M 0s\n",
      " 48750K .......... .......... .......... .......... .......... 77%  148M 0s\n",
      " 48800K .......... .......... .......... .......... .......... 77%  229M 0s\n",
      " 48850K .......... .......... .......... .......... .......... 78%  197M 0s\n",
      " 48900K .......... .......... .......... .......... .......... 78%  269M 0s\n",
      " 48950K .......... .......... .......... .......... .......... 78% 65.8M 0s\n",
      " 49000K .......... .......... .......... .......... .......... 78% 9.03M 0s\n",
      " 49050K .......... .......... .......... .......... .......... 78%  231M 0s\n",
      " 49100K .......... .......... .......... .......... .......... 78%  289M 0s\n",
      " 49150K .......... .......... .......... .......... .......... 78%  221M 0s\n",
      " 49200K .......... .......... .......... .......... .......... 78%  201M 0s\n",
      " 49250K .......... .......... .......... .......... .......... 78%  335M 0s\n",
      " 49300K .......... .......... .......... .......... .......... 78%  274M 0s\n",
      " 49350K .......... .......... .......... .......... .......... 78%  204M 0s\n",
      " 49400K .......... .......... .......... .......... .......... 78%  300M 0s\n",
      " 49450K .......... .......... .......... .......... .......... 79%  250M 0s\n",
      " 49500K .......... .......... .......... .......... .......... 79%  182M 0s\n",
      " 49550K .......... .......... .......... .......... .......... 79%  249M 0s\n",
      " 49600K .......... .......... .......... .......... .......... 79%  286M 0s\n",
      " 49650K .......... .......... .......... .......... .......... 79%  345M 0s\n",
      " 49700K .......... .......... .......... .......... .......... 79%  302M 0s\n",
      " 49750K .......... .......... .......... .......... .......... 79%  233M 0s\n",
      " 49800K .......... .......... .......... .......... .......... 79%  184M 0s\n",
      " 49850K .......... .......... .......... .......... .......... 79% 96.7M 0s\n",
      " 49900K .......... .......... .......... .......... .......... 79%  310M 0s\n",
      " 49950K .......... .......... .......... .......... .......... 79%  272M 0s\n",
      " 50000K .......... .......... .......... .......... .......... 79%  341M 0s\n",
      " 50050K .......... .......... .......... .......... .......... 79%  335M 0s\n",
      " 50100K .......... .......... .......... .......... .......... 80%  211M 0s\n",
      " 50150K .......... .......... .......... .......... .......... 80%  171M 0s\n",
      " 50200K .......... .......... .......... .......... .......... 80%  164M 0s\n",
      " 50250K .......... .......... .......... .......... .......... 80%  190M 0s\n",
      " 50300K .......... .......... .......... .......... .......... 80%  144M 0s\n",
      " 50350K .......... .......... .......... .......... .......... 80%  285M 0s\n",
      " 50400K .......... .......... .......... .......... .......... 80%  320M 0s\n",
      " 50450K .......... .......... .......... .......... .......... 80%  286M 0s\n",
      " 50500K .......... .......... .......... .......... .......... 80%  351M 0s\n",
      " 50550K .......... .......... .......... .......... .......... 80%  300M 0s\n",
      " 50600K .......... .......... .......... .......... .......... 80%  295M 0s\n",
      " 50650K .......... .......... .......... .......... .......... 80%  320M 0s\n",
      " 50700K .......... .......... .......... .......... .......... 81%  286M 0s\n",
      " 50750K .......... .......... .......... .......... .......... 81%  252M 0s\n",
      " 50800K .......... .......... .......... .......... .......... 81%  170M 0s\n",
      " 50850K .......... .......... .......... .......... .......... 81%  168M 0s\n",
      " 50900K .......... .......... .......... .......... .......... 81%  181M 0s\n",
      " 50950K .......... .......... .......... .......... .......... 81%  121M 0s\n",
      " 51000K .......... .......... .......... .......... .......... 81% 9.21M 0s\n",
      " 51050K .......... .......... .......... .......... .......... 81% 58.6M 0s\n",
      " 51100K .......... .......... .......... .......... .......... 81%  185M 0s\n",
      " 51150K .......... .......... .......... .......... .......... 81%  232M 0s\n",
      " 51200K .......... .......... .......... .......... .......... 81%  202M 0s\n",
      " 51250K .......... .......... .......... .......... .......... 81%  229M 0s\n",
      " 51300K .......... .......... .......... .......... .......... 81%  218M 0s\n",
      " 51350K .......... .......... .......... .......... .......... 82%  197M 0s\n",
      " 51400K .......... .......... .......... .......... .......... 82%  240M 0s\n",
      " 51450K .......... .......... .......... .......... .......... 82%  169M 0s\n",
      " 51500K .......... .......... .......... .......... .......... 82%  359M 0s\n",
      " 51550K .......... .......... .......... .......... .......... 82%  344M 0s\n",
      " 51600K .......... .......... .......... .......... .......... 82%  298M 0s\n",
      " 51650K .......... .......... .......... .......... .......... 82%  371M 0s\n",
      " 51700K .......... .......... .......... .......... .......... 82%  305M 0s\n",
      " 51750K .......... .......... .......... .......... .......... 82%  339M 0s\n",
      " 51800K .......... .......... .......... .......... .......... 82%  360M 0s\n",
      " 51850K .......... .......... .......... .......... .......... 82%  153M 0s\n",
      " 51900K .......... .......... .......... .......... .......... 82%  213M 0s\n",
      " 51950K .......... .......... .......... .......... .......... 83%  238M 0s\n",
      " 52000K .......... .......... .......... .......... .......... 83% 53.0M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 83% 9.78M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 83%  123M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 83%  219M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 83%  276M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 83%  242M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 83%  166M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 83%  217M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 83%  149M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 83%  257M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 83%  317M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 83%  297M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 84%  291M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 84%  166M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 84%  207M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 84%  304M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 84%  173M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 84%  251M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 84%  242M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 84%  197M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 84%  306M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 84%  251M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 84%  189M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 84%  328M 0s\n",
      " 53200K .......... .......... .......... .......... .......... 85%  225M 0s\n",
      " 53250K .......... .......... .......... .......... .......... 85%  216M 0s\n",
      " 53300K .......... .......... .......... .......... .......... 85%  356M 0s\n",
      " 53350K .......... .......... .......... .......... .......... 85%  219M 0s\n",
      " 53400K .......... .......... .......... .......... .......... 85%  212M 0s\n",
      " 53450K .......... .......... .......... .......... .......... 85%  269M 0s\n",
      " 53500K .......... .......... .......... .......... .......... 85%  241M 0s\n",
      " 53550K .......... .......... .......... .......... .......... 85%  179M 0s\n",
      " 53600K .......... .......... .......... .......... .......... 85%  285M 0s\n",
      " 53650K .......... .......... .......... .......... .......... 85%  335M 0s\n",
      " 53700K .......... .......... .......... .......... .......... 85%  293M 0s\n",
      " 53750K .......... .......... .......... .......... .......... 85%  336M 0s\n",
      " 53800K .......... .......... .......... .......... .......... 85%  168M 0s\n",
      " 53850K .......... .......... .......... .......... .......... 86%  166M 0s\n",
      " 53900K .......... .......... .......... .......... .......... 86%  248M 0s\n",
      " 53950K .......... .......... .......... .......... .......... 86% 9.34M 0s\n",
      " 54000K .......... .......... .......... .......... .......... 86%  388M 0s\n",
      " 54050K .......... .......... .......... .......... .......... 86% 73.8M 0s\n",
      " 54100K .......... .......... .......... .......... .......... 86%  179M 0s\n",
      " 54150K .......... .......... .......... .......... .......... 86%  116M 0s\n",
      " 54200K .......... .......... .......... .......... .......... 86%  340M 0s\n",
      " 54250K .......... .......... .......... .......... .......... 86%  139M 0s\n",
      " 54300K .......... .......... .......... .......... .......... 86%  273M 0s\n",
      " 54350K .......... .......... .......... .......... .......... 86%  321M 0s\n",
      " 54400K .......... .......... .......... .......... .......... 86%  313M 0s\n",
      " 54450K .......... .......... .......... .......... .......... 87%  191M 0s\n",
      " 54500K .......... .......... .......... .......... .......... 87%  275M 0s\n",
      " 54550K .......... .......... .......... .......... .......... 87%  202M 0s\n",
      " 54600K .......... .......... .......... .......... .......... 87%  235M 0s\n",
      " 54650K .......... .......... .......... .......... .......... 87%  482M 0s\n",
      " 54700K .......... .......... .......... .......... .......... 87%  203M 0s\n",
      " 54750K .......... .......... .......... .......... .......... 87%  226M 0s\n",
      " 54800K .......... .......... .......... .......... .......... 87%  413M 0s\n",
      " 54850K .......... .......... .......... .......... .......... 87%  258M 0s\n",
      " 54900K .......... .......... .......... .......... .......... 87%  390M 0s\n",
      " 54950K .......... .......... .......... .......... .......... 87% 69.2M 0s\n",
      " 55000K .......... .......... .......... .......... .......... 87%  169M 0s\n",
      " 55050K .......... .......... .......... .......... .......... 87% 9.91M 0s\n",
      " 55100K .......... .......... .......... .......... .......... 88% 86.4M 0s\n",
      " 55150K .......... .......... .......... .......... .......... 88%  236M 0s\n",
      " 55200K .......... .......... .......... .......... .......... 88%  159M 0s\n",
      " 55250K .......... .......... .......... .......... .......... 88%  230M 0s\n",
      " 55300K .......... .......... .......... .......... .......... 88%  265M 0s\n",
      " 55350K .......... .......... .......... .......... .......... 88%  174M 0s\n",
      " 55400K .......... .......... .......... .......... .......... 88%  183M 0s\n",
      " 55450K .......... .......... .......... .......... .......... 88%  126M 0s\n",
      " 55500K .......... .......... .......... .......... .......... 88%  351M 0s\n",
      " 55550K .......... .......... .......... .......... .......... 88%  186M 0s\n",
      " 55600K .......... .......... .......... .......... .......... 88%  189M 0s\n",
      " 55650K .......... .......... .......... .......... .......... 88%  399M 0s\n",
      " 55700K .......... .......... .......... .......... .......... 89%  321M 0s\n",
      " 55750K .......... .......... .......... .......... .......... 89%  243M 0s\n",
      " 55800K .......... .......... .......... .......... .......... 89%  343M 0s\n",
      " 55850K .......... .......... .......... .......... .......... 89%  170M 0s\n",
      " 55900K .......... .......... .......... .......... .......... 89%  261M 0s\n",
      " 55950K .......... .......... .......... .......... .......... 89%  331M 0s\n",
      " 56000K .......... .......... .......... .......... .......... 89%  366M 0s\n",
      " 56050K .......... .......... .......... .......... .......... 89%  278M 0s\n",
      " 56100K .......... .......... .......... .......... .......... 89%  183M 0s\n",
      " 56150K .......... .......... .......... .......... .......... 89%  406M 0s\n",
      " 56200K .......... .......... .......... .......... .......... 89%  134M 0s\n",
      " 56250K .......... .......... .......... .......... .......... 89%  146M 0s\n",
      " 56300K .......... .......... .......... .......... .......... 89%  350M 0s\n",
      " 56350K .......... .......... .......... .......... .......... 90%  387M 0s\n",
      " 56400K .......... .......... .......... .......... .......... 90%  439M 0s\n",
      " 56450K .......... .......... .......... .......... .......... 90%  399M 0s\n",
      " 56500K .......... .......... .......... .......... .......... 90%  124M 0s\n",
      " 56550K .......... .......... .......... .......... .......... 90%  227M 0s\n",
      " 56600K .......... .......... .......... .......... .......... 90%  194M 0s\n",
      " 56650K .......... .......... .......... .......... .......... 90%  342M 0s\n",
      " 56700K .......... .......... .......... .......... .......... 90% 64.8M 0s\n",
      " 56750K .......... .......... .......... .......... .......... 90%  430M 0s\n",
      " 56800K .......... .......... .......... .......... .......... 90%  440M 0s\n",
      " 56850K .......... .......... .......... .......... .......... 90%  412M 0s\n",
      " 56900K .......... .......... .......... .......... .......... 90% 8.93M 0s\n",
      " 56950K .......... .......... .......... .......... .......... 90%  338M 0s\n",
      " 57000K .......... .......... .......... .......... .......... 91%  447M 0s\n",
      " 57050K .......... .......... .......... .......... .......... 91%  399M 0s\n",
      " 57100K .......... .......... .......... .......... .......... 91%  187M 0s\n",
      " 57150K .......... .......... .......... .......... .......... 91%  184M 0s\n",
      " 57200K .......... .......... .......... .......... .......... 91% 88.6M 0s\n",
      " 57250K .......... .......... .......... .......... .......... 91%  427M 0s\n",
      " 57300K .......... .......... .......... .......... .......... 91%  318M 0s\n",
      " 57350K .......... .......... .......... .......... .......... 91%  209M 0s\n",
      " 57400K .......... .......... .......... .......... .......... 91%  222M 0s\n",
      " 57450K .......... .......... .......... .......... .......... 91%  387M 0s\n",
      " 57500K .......... .......... .......... .......... .......... 91%  181M 0s\n",
      " 57550K .......... .......... .......... .......... .......... 91%  479M 0s\n",
      " 57600K .......... .......... .......... .......... .......... 92%  239M 0s\n",
      " 57650K .......... .......... .......... .......... .......... 92%  102M 0s\n",
      " 57700K .......... .......... .......... .......... .......... 92%  216M 0s\n",
      " 57750K .......... .......... .......... .......... .......... 92%  420M 0s\n",
      " 57800K .......... .......... .......... .......... .......... 92%  215M 0s\n",
      " 57850K .......... .......... .......... .......... .......... 92%  338M 0s\n",
      " 57900K .......... .......... .......... .......... .......... 92%  373M 0s\n",
      " 57950K .......... .......... .......... .......... .......... 92%  113M 0s\n",
      " 58000K .......... .......... .......... .......... .......... 92% 9.05M 0s\n",
      " 58050K .......... .......... .......... .......... .......... 92%  242M 0s\n",
      " 58100K .......... .......... .......... .......... .......... 92%  142M 0s\n",
      " 58150K .......... .......... .......... .......... .......... 92%  308M 0s\n",
      " 58200K .......... .......... .......... .......... .......... 92% 82.2M 0s\n",
      " 58250K .......... .......... .......... .......... .......... 93%  451M 0s\n",
      " 58300K .......... .......... .......... .......... .......... 93%  215M 0s\n",
      " 58350K .......... .......... .......... .......... .......... 93%  315M 0s\n",
      " 58400K .......... .......... .......... .......... .......... 93%  298M 0s\n",
      " 58450K .......... .......... .......... .......... .......... 93%  106M 0s\n",
      " 58500K .......... .......... .......... .......... .......... 93%  533M 0s\n",
      " 58550K .......... .......... .......... .......... .......... 93%  109M 0s\n",
      " 58600K .......... .......... .......... .......... .......... 93%  299M 0s\n",
      " 58650K .......... .......... .......... .......... .......... 93%  234M 0s\n",
      " 58700K .......... .......... .......... .......... .......... 93%  257M 0s\n",
      " 58750K .......... .......... .......... .......... .......... 93%  471M 0s\n",
      " 58800K .......... .......... .......... .......... .......... 93%  442M 0s\n",
      " 58850K .......... .......... .......... .......... .......... 94%  302M 0s\n",
      " 58900K .......... .......... .......... .......... .......... 94%  176M 0s\n",
      " 58950K .......... .......... .......... .......... .......... 94%  387M 0s\n",
      " 59000K .......... .......... .......... .......... .......... 94%  506M 0s\n",
      " 59050K .......... .......... .......... .......... .......... 94%  199M 0s\n",
      " 59100K .......... .......... .......... .......... .......... 94%  206M 0s\n",
      " 59150K .......... .......... .......... .......... .......... 94%  130M 0s\n",
      " 59200K .......... .......... .......... .......... .......... 94%  378M 0s\n",
      " 59250K .......... .......... .......... .......... .......... 94%  349M 0s\n",
      " 59300K .......... .......... .......... .......... .......... 94%  166M 0s\n",
      " 59350K .......... .......... .......... .......... .......... 94%  141M 0s\n",
      " 59400K .......... .......... .......... .......... .......... 94%  210M 0s\n",
      " 59450K .......... .......... .......... .......... .......... 94%  219M 0s\n",
      " 59500K .......... .......... .......... .......... .......... 95%  440M 0s\n",
      " 59550K .......... .......... .......... .......... .......... 95%  394M 0s\n",
      " 59600K .......... .......... .......... .......... .......... 95%  361M 0s\n",
      " 59650K .......... .......... .......... .......... .......... 95%  132M 0s\n",
      " 59700K .......... .......... .......... .......... .......... 95%  171M 0s\n",
      " 59750K .......... .......... .......... .......... .......... 95% 7.68M 0s\n",
      " 59800K .......... .......... .......... .......... .......... 95%  377M 0s\n",
      " 59850K .......... .......... .......... .......... .......... 95%  352M 0s\n",
      " 59900K .......... .......... .......... .......... .......... 95%  460M 0s\n",
      " 59950K .......... .......... .......... .......... .......... 95%  377M 0s\n",
      " 60000K .......... .......... .......... .......... .......... 95%  432M 0s\n",
      " 60050K .......... .......... .......... .......... .......... 95%  412M 0s\n",
      " 60100K .......... .......... .......... .......... .......... 96%  265M 0s\n",
      " 60150K .......... .......... .......... .......... .......... 96%  404M 0s\n",
      " 60200K .......... .......... .......... .......... .......... 96%  191M 0s\n",
      " 60250K .......... .......... .......... .......... .......... 96%  224M 0s\n",
      " 60300K .......... .......... .......... .......... .......... 96%  238M 0s\n",
      " 60350K .......... .......... .......... .......... .......... 96%  215M 0s\n",
      " 60400K .......... .......... .......... .......... .......... 96%  360M 0s\n",
      " 60450K .......... .......... .......... .......... .......... 96%  137M 0s\n",
      " 60500K .......... .......... .......... .......... .......... 96%  197M 0s\n",
      " 60550K .......... .......... .......... .......... .......... 96%  193M 0s\n",
      " 60600K .......... .......... .......... .......... .......... 96%  151M 0s\n",
      " 60650K .......... .......... .......... .......... .......... 96%  372M 0s\n",
      " 60700K .......... .......... .......... .......... .......... 96%  300M 0s\n",
      " 60750K .......... .......... .......... .......... .......... 97%  349M 0s\n",
      " 60800K .......... .......... .......... .......... .......... 97%  376M 0s\n",
      " 60850K .......... .......... .......... .......... .......... 97%  326M 0s\n",
      " 60900K .......... .......... .......... .......... .......... 97%  120M 0s\n",
      " 60950K .......... .......... .......... .......... .......... 97%  218M 0s\n",
      " 61000K .......... .......... .......... .......... .......... 97%  405M 0s\n",
      " 61050K .......... .......... .......... .......... .......... 97% 9.59M 0s\n",
      " 61100K .......... .......... .......... .......... .......... 97%  102M 0s\n",
      " 61150K .......... .......... .......... .......... .......... 97%  319M 0s\n",
      " 61200K .......... .......... .......... .......... .......... 97%  251M 0s\n",
      " 61250K .......... .......... .......... .......... .......... 97% 65.3M 0s\n",
      " 61300K .......... .......... .......... .......... .......... 97%  310M 0s\n",
      " 61350K .......... .......... .......... .......... .......... 98%  314M 0s\n",
      " 61400K .......... .......... .......... .......... .......... 98%  344M 0s\n",
      " 61450K .......... .......... .......... .......... .......... 98%  227M 0s\n",
      " 61500K .......... .......... .......... .......... .......... 98%  228M 0s\n",
      " 61550K .......... .......... .......... .......... .......... 98%  106M 0s\n",
      " 61600K .......... .......... .......... .......... .......... 98%  145M 0s\n",
      " 61650K .......... .......... .......... .......... .......... 98%  171M 0s\n",
      " 61700K .......... .......... .......... .......... .......... 98%  121M 0s\n",
      " 61750K .......... .......... .......... .......... .......... 98%  171M 0s\n",
      " 61800K .......... .......... .......... .......... .......... 98%  191M 0s\n",
      " 61850K .......... .......... .......... .......... .......... 98%  123M 0s\n",
      " 61900K .......... .......... .......... .......... .......... 98%  181M 0s\n",
      " 61950K .......... .......... .......... .......... .......... 98%  119M 0s\n",
      " 62000K .......... .......... .......... .......... .......... 99%  177M 0s\n",
      " 62050K .......... .......... .......... .......... .......... 99%  181M 0s\n",
      " 62100K .......... .......... .......... .......... .......... 99%  147M 0s\n",
      " 62150K .......... .......... .......... .......... .......... 99%  167M 0s\n",
      " 62200K .......... .......... .......... .......... .......... 99%  197M 0s\n",
      " 62250K .......... .......... .......... .......... .......... 99%  330M 0s\n",
      " 62300K .......... .......... .......... .......... .......... 99%  336M 0s\n",
      " 62350K .......... .......... .......... .......... .......... 99%  295M 0s\n",
      " 62400K .......... .......... .......... .......... .......... 99%  350M 0s\n",
      " 62450K .......... .......... .......... .......... .......... 99%  297M 0s\n",
      " 62500K .......... .......... .......... .......... .......... 99%  360M 0s\n",
      " 62550K .......... .......... .......... .......... .......... 99%  345M 0s\n",
      " 62600K .......... .......... .......... ........             100%  308M=0.7s\n",
      "\n",
      "2025-03-04 21:02:23 (89.1 MB/s) - 'data/Homo_sapiens.GRCh38.113.gtf.gz' saved [64141785/64141785]\n",
      "\n",
      "rm: cannot remove 'data/Homo_sapiens.GRCh38.113.gtf.gz': No such file or directory\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b\"# -nc makes sure that if the file is already there, it won't be overwritten.\\n# -nc can be changed to -N, if it is of interest to import newer versions\\n# -P will save it to the directory specified, in this case data/cDNA_library\\nwget -nc https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz -P data/\\ngzip -d data/Homo_sapiens.GRCh38.113.gtf.gz\\nrm data/Homo_sapiens.GRCh38.113.gtf.gz\\n\"' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[23], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mbash\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m# -nc makes sure that if the file is already there, it won\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mt be overwritten.\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m# -nc can be changed to -N, if it is of interest to import newer versions\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m# -P will save it to the directory specified, in this case data/cDNA_library\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mwget -nc https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz -P data/\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mgzip -d data/Homo_sapiens.GRCh38.113.gtf.gz\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43mrm data/Homo_sapiens.GRCh38.113.gtf.gz\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2543\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2541\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2542\u001b[0m     args \u001b[38;5;241m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2543\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2545\u001b[0m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[1;32m   2546\u001b[0m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[1;32m   2547\u001b[0m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[1;32m   2548\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic\u001b[38;5;241m.\u001b[39mMAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:159\u001b[0m, in \u001b[0;36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    157\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    158\u001b[0m     line \u001b[38;5;241m=\u001b[39m script\n\u001b[0;32m--> 159\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:336\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m args\u001b[38;5;241m.\u001b[39mraise_error \u001b[38;5;129;01mand\u001b[39;00m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    332\u001b[0m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    333\u001b[0m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    334\u001b[0m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    335\u001b[0m     rc \u001b[38;5;241m=\u001b[39m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m9\u001b[39m\n\u001b[0;32m--> 336\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b\"# -nc makes sure that if the file is already there, it won't be overwritten.\\n# -nc can be changed to -N, if it is of interest to import newer versions\\n# -P will save it to the directory specified, in this case data/cDNA_library\\nwget -nc https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz -P data/\\ngzip -d data/Homo_sapiens.GRCh38.113.gtf.gz\\nrm data/Homo_sapiens.GRCh38.113.gtf.gz\\n\"' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# -nc makes sure that if the file is already there, it won't be overwritten.\n",
    "# -nc can be changed to -N, if it is of interest to import newer versions\n",
    "# -P will save it to the directory specified, in this case data/cDNA_library\n",
    "wget -nc https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz -P data/\n",
    "gzip -d data/Homo_sapiens.GRCh38.113.gtf.gz\n",
    "rm data/Homo_sapiens.GRCh38.113.gtf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af188c54-f62d-4454-9589-919c77e31b74",
   "metadata": {},
   "source": [
    "# Summarize with featureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3885de00-1837-489b-9ea9-3fb8e6e992a1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 14 BAM files                                     ||\n",
      "||                                                                            ||\n",
      "||                           ERR950158_trimmed.bam                            ||\n",
      "||                           ERR950159_trimmed.bam                            ||\n",
      "||                           ERR950160_trimmed.bam                            ||\n",
      "||                           ERR950161_trimmed.bam                            ||\n",
      "||                           ERR950162_trimmed.bam                            ||\n",
      "||                           ERR950163_trimmed.bam                            ||\n",
      "||                           ERR950177_trimmed.bam                            ||\n",
      "||                           ERR950178_trimmed.bam                            ||\n",
      "||                           ERR950179_trimmed.bam                            ||\n",
      "||                           ERR950180_trimmed.bam                            ||\n",
      "||                           ERR950181_trimmed.bam                            ||\n",
      "||                           ERR950182_trimmed.bam                            ||\n",
      "||                           ERR950186_trimmed.bam                            ||\n",
      "||                           ERR950187_trimmed.bam                            ||\n",
      "||                                                                            ||\n",
      "||             Output file : bowtie_subset_counts.txt                         ||\n",
      "||                 Summary : bowtie_subset_counts.txt.summary                 ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/Bowtie2_gene_counts                         ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 1                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950158_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 999064                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950159_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 999408                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950160_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 997850                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.04 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950161_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 997700                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950162_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 963026                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950163_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 996940                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950177_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 987454                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950178_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 991106                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.04 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950179_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 999086                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950180_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 993018                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950181_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 998988                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950182_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 998426                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950186_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 997018                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950187_trimmed.bam...                                  ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 998450                                               ||\n",
      "||    Successfully assigned alignments : 0 (0.0%)                             ||\n",
      "||    Running time : 0.03 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/Bowtie2_gene_count  ||\n",
      "|| s/bowtie_subset_counts.txt.summary\"                                        ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# path to the GTF annotation, output file, -p to specify paired-end reads \n",
    "# -C to remove pairs that have been mapped to different chromosomes \n",
    "# takes ~30 sek to run on two cores\n",
    "featureCounts \\\n",
    "-t exon \\\n",
    "-g gene_id \\\n",
    "-a data/Homo_sapiens.GRCh38.113.gtf \\\n",
    "-o data/Bowtie2_gene_counts/bowtie_subset_counts.txt \\\n",
    "-p \\\n",
    "data/alignment_Bowtie2_subset/*.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c37ea849-307c-4618-bcf3-356d46ed00ca",
   "metadata": {},
   "source": [
    "# Remember to change access for everyone"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8879a271-b191-4f8b-87e4-5c5ca4e0439c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "chmod: changing permissions of '.': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6789033.out': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6776073.err': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/script_STAR_alignment.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/pre_script.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/script_count_STAR.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/count_STAR.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6784971.out': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6781057.out': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/STAR_alignment_full_data.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6781057.err': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6784971.err': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/merge_counts.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/script_merge_counts.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6776073.out': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/Preprocessing_full_data.sh': Operation not permitted\n",
      "chmod: changing permissions of './Sbatch_scripts/job.6789033.err': Operation not permitted\n",
      "chmod: changing permissions of './Bowtie.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './containers': Operation not permitted\n",
      "chmod: changing permissions of './containers/group_Bjork_rstudio.sh': Operation not permitted\n",
      "chmod: changing permissions of './containers/Bjork_container.sif': Operation not permitted\n",
      "chmod: changing permissions of './containers/Bjork_container.def': Operation not permitted\n",
      "chmod: changing permissions of './containers/group_Bjork_jupyter.sh': Operation not permitted\n",
      "chmod: changing permissions of './containers/container.sif': Operation not permitted\n",
      "chmod: changing permissions of './containers/container.def': Operation not permitted\n",
      "chmod: changing permissions of './containers/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './containers/.ipynb_checkpoints/Bjork_container-checkpoint.def': Operation not permitted\n",
      "chmod: changing permissions of './DESeq2_STAR.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './RNA-seq-list.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950158_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950163_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950187_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950162_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950179_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950178_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950158_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950177_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950186_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950187_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950161_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950163_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950158_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950186_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950162_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950160_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950180_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950181_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950159_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950182_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950178_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950161_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950180_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950160_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950179_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950187_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950163_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950181_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950158_trimmed_R2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950162_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950186_trimmed_R2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950177_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/fastqc/ERR950159_trimmed_R1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950181_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950182_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950178_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950180_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950182_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950159_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950161_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950177_trimmed_R1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950179_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/trimmed/ERR950160_trimmed_R2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950163_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950180_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950181_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950178_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950177_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950187_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950162_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950161_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950160_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950180_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950163_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950182_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950158_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950186_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950187_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950159_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950162_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950182_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950177_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950181_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950187_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950163_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950161_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950159_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950186_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950180_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950160_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950158_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950162_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950179_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/fastqc/ERR950178_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950179_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950178_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950186_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950177_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950160_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950161_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950181_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950159_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950182_2.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './subset_rawdata_RNA-seq/ERR950179_1.fastq.gz.subset.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './CN.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './subset_trimmed_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './STAR.DESeq2_results.csv': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950159_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950162_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed_1.sam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950178_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950179_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950181_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950163_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950182_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950163_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950179_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950180_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950177_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950159_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950161_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950187.log': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950162_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950160_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950160_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950186_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950181_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950180_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950182_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950158_trimmed.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950177_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950186_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950158_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950178_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './alignment_Bowtie2_subset_old/ERR950161_trimmed.bam': Operation not permitted\n",
      "chmod: changing permissions of './Preprocessing.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.git': Operation not permitted\n",
      "chmod: changing permissions of './.git/HEAD': Operation not permitted\n",
      "chmod: changing permissions of './.git/description': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/65': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/65/f0d5ad57dded0513f24684893347e2f9cd1363': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/8d': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/8d/ece6a2f1db99ad634495e78d7449cec0847966': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/66': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/66/452897458e05dbd938d918f6397505146dd8bb': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/00': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/00/4ac72bc224eb0cd2a221ca987e7f20a9f91ce5': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/info': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ca': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ca/4a1b2996370a666697b0cd1b3f367aef80918a': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ae': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ae/d1711414c5166dd7cdb34d5bd83c44d017347a': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/c6': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/c6/ff6019645459b3793345e3f75280c0b2c66db3': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/40': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/40/4dd551326832b742de5d5118ff2269e72cdeaa': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/0b': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/0b/c5f58cbf1b08a2d2c24dd1306192a1f59b517f': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/cd': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/cd/500414444e8f65d2d6e03d9f6c18894d317a5e': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ac': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/ac/1cc772c92ebb0f64586ffb19c1f07901ebc465': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/4b/acbfb9ef2d9c1f51de66b3e52f07d86fd46e37': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/f3': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/f3/e04a8a3f1d698b59c15d0f07436e02cac922d9': Operation not permitted\n",
      "chmod: changing permissions of './.git/objects/pack': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-merge-commit.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-commit.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/fsmonitor-watchman.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/commit-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-rebase.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-receive.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/prepare-commit-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/update.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-push.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/sendemail-validate.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/pre-applypatch.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/push-to-checkout.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/post-update.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/hooks/applypatch-msg.sample': Operation not permitted\n",
      "chmod: changing permissions of './.git/FETCH_HEAD': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/bbt045_project': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/bbt045_project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/project': Operation not permitted\n",
      "chmod: changing permissions of './.git/logs/refs/remotes/project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/info': Operation not permitted\n",
      "chmod: changing permissions of './.git/info/exclude': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/tags': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/heads': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/bbt045_project': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/bbt045_project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/project': Operation not permitted\n",
      "chmod: changing permissions of './.git/refs/remotes/project/main': Operation not permitted\n",
      "chmod: changing permissions of './.git/branches': Operation not permitted\n",
      "chmod: changing permissions of './CN_TESTAR': Operation not permitted\n",
      "chmod: changing permissions of './CN_TESTAR/Homo_sapiens.GRCh38.dna.toplevel.fa.gz.fai': Operation not permitted\n",
      "chmod: changing permissions of './gene_ccounts_DESeq': Operation not permitted\n",
      "chmod: changing permissions of './Log.out': Operation not permitted\n",
      "chmod: changing permissions of './STAR.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950186_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950182_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950161_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950181_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950177_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950187_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950159_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950179_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950162_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950162_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950178_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950187_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950178_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950163_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950181_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950163_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950186_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950158_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950160_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950179_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950158_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950159_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950182_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950180_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950161_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950180_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950177_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './rawdata_RNA-seq/ERR950160_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/edgeR_STAR-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/Log-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/RNA-seq-list-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/Preprocessing-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/DESeq2_STAR-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/STAR.DESeq2_results-checkpoint.csv': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/STAR-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './.ipynb_checkpoints/CN-checkpoint.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_DE_analysis': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_DE_analysis/STAR_metadata.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_DE_analysis/STAR_merged_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/Untitled.ipynb': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950182_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950177_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950181_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950187_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950163_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950161_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950159_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950186_2.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950180_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950160_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950158_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950162_2.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950179_1.fastq.gz.subset_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_rawdata_RNA-seq/fastqc_subset_rawdata/ERR950178_1.fastq.gz.subset_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950177_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950178_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950186_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950187_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950163_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950180_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950178_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950179_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_length_distribution_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950159_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950161_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950182_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950187_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950163_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950158_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950160_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950177_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950186_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950162_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/fastqc_subset_trimmed/ERR950181_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950182_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950181_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950178_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950159_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950182_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950160_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950161_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950177_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950158_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950158_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950161_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950163_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950162_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950181_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950180_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950179_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950179_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950162_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950187_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950186_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950160_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950180_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/subset_trimmed_RNA-seq/ERR950159_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950158_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950158_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950180_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950182_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950181_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950186_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950179_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950162_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950162_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950160_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/test.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950159_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950163_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950163_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950161_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950178_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950187_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950178_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950180_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950160_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950177_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950181_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950186_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/test.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950179_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950161_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/ERR950158_trimmed.counts.txt-checkpoint.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/test.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/.ipynb_checkpoints/ERR950158_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950177_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950159_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950187_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq/ERR950182_trimmed.counts.txt.summary': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbList.fromGTF.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrNameLength.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/exonInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/transcriptInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/exonGeTrInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrStart.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/_STARtmp': Operation not permitted\n",
      "chmod: cannot read directory './data/STAR_index/_STARtmp': Permission denied\n",
      "chmod: changing permissions of './data/STAR_index/chrLength.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/chrName.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/SAindex': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbList.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/geneInfo.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/genomeParameters.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/Genome': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/SA': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_index/sjdbInfo.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/merged_counts_matrix.csv': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950178_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/metadata.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950181_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950177_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950161_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950179_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950180_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950162_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950163_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950160_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950158_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950187_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950186_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950181_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950162_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950182_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950180_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950182_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950187_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950158_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950177_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950163_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950186_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950159_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950179_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950159_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950163_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950178_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950162_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950158_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950160_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950180_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950177_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950187_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950181_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950160_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950179_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950186_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950161_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950161_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/sub_dir_count/ERR950178_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950159_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/merged_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts/ERR950182_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950158_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950158_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950182_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950181_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950179_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950162_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950160_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950187_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950181_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950186_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950163_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950161_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950178_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950160_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950163_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950180_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950187_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950182_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950179_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950180_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950161_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950177_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950162_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950186_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950159_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950159_trimmed_idxstats-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950158_trimmed_idxstats-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950158_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/.ipynb_checkpoints/ERR950159_trimmed.counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/data': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/data/gene_counts_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950177_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950159_trimmed.counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/gene_counts_DESeq_TEST/ERR950178_trimmed_idxstats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950178_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950181_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950177_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950161_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950179_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950180_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950162_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950163_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950160_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950158_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950186_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950181_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950162_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950182_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950180_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950182_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950158_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950177_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950163_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950186_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950159_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950179_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950159_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950163_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950178_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950162_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950158_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950160_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950180_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950177_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950181_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950160_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950179_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950186_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950161_trimmed_sorted.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950161_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/.ipynb_checkpoints/ERR950162_trimmed_htseq-count-checkpoint.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/sub_dir_count/ERR950178_trimmed_htseq-count.log': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950159_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/merged_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints/ERR950158_trimmed_htseq_counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/.ipynb_checkpoints/merged_counts-checkpoint.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/CN_gene_counts_TEST/ERR950182_trimmed_htseq_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950178_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950186_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950160_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950179_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950180_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950181_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950182_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950161_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950162_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950187_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950158_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950159_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950178_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950160_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950181_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950159_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950163_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950187_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_ReadsPerGene.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_Log-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950186_trimmed_Log.progress-checkpoint.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950180_trimmed_SJ.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/.ipynb_checkpoints/ERR950180_trimmed_ReadsPerGene.out-checkpoint.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950162_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950163_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950158_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950177_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR_subset/ERR950177_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950161_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950186_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950182_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950179_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.sortedByCoord.out.bam.bai': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR_subset/ERR950180_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts_TEST': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts_TEST/sub_dir_count': Operation not permitted\n",
      "chmod: changing permissions of './data/STAR_gene_counts_TEST/merged_counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950178_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950186_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950160_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950179_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950180_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950181_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950182_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950161_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950162_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950162_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950187_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950158_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950159_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950181_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950178_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950160_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_Log.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950159_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950163_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950187_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_SJ.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_Aligned.toTranscriptome.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950163_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950158_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950177_trimmed__STARgenome': Operation not permitted\n",
      "chmod: cannot read directory './data/alignment_STAR/ERR950177_trimmed__STARgenome': Permission denied\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950161_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950186_trimmed_ReadsPerGene.out.tab': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950182_trimmed_Log.final.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950179_trimmed_Log.progress.out': Operation not permitted\n",
      "chmod: changing permissions of './data/alignment_STAR/ERR950180_trimmed_Aligned.sortedByCoord.out.bam': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950161_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950160_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950179_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950178_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950180_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950158_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950177_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950177_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950159_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950186_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950163_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950163_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950180_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950181_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950179_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950163_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950182_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950178_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950158_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950162_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950177_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950159_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950161_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950186_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950182_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950181_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950187_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950179_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950162_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950163_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950180_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950186_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950159_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950186_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950159_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950162_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950182_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950178_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950181_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950160_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950160_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950187_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950179_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950158_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950178_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950161_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950160_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950181_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950177_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950161_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950180_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950158_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950187_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950187_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950162_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/rawdata_RNA-seq/fastqc_rawdata/ERR950182_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/.ipynb_checkpoints': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950177_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950178_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950186_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950187_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950163_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950182_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950181_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950178_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950159_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950182_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950160_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950161_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950177_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950158_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950158_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950161_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950163_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950161_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950186_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950163_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950178_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950177_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950177_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950178_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950161_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950178_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950179_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950161_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950160_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950186_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950179_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950179_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950186_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950180_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_report.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950181_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950160_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950180_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950159_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950180_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950182_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950163_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950180_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950162_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950158_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950182_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950158_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950178_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950159_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950179_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950187_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950182_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950159_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950177_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_per_base_sequence_quality_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_sequence_counts_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_sequence_length_distribution_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc.log': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc-status-check-heatmap.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_general_stats.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_citations.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_software_versions.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_fastqc.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_per_base_n_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_sequence_duplication_levels_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_data.json': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/multiqc_sources.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_top_overrepresented_sequences_table.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_overrepresented_sequences_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/multiqc_data/fastqc_adapter_content_plot.txt': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950159_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950158_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950160_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950162_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950181_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950161_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950162_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950187_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950182_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950187_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950187_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950181_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950163_trimmed_2_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950158_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950160_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950177_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950186_trimmed_1_fastqc.html': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950162_trimmed_1_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/fastqc_trimmed/ERR950181_trimmed_2_fastqc.zip': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950163_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950162_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950181_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950180_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950179_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950179_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950162_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950187_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950186_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950160_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950180_trimmed_2.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './data/trimmed_RNA-seq/ERR950159_trimmed_1.fastq.gz': Operation not permitted\n",
      "chmod: changing permissions of './metadata.sh': Operation not permitted\n",
      "chmod: changing permissions of './edgeR_STAR.ipynb': Operation not permitted\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'chmod -R 777 . \\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[25], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mbash\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mchmod -R 777 . \u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2543\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2541\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2542\u001b[0m     args \u001b[38;5;241m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2543\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2545\u001b[0m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[1;32m   2546\u001b[0m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[1;32m   2547\u001b[0m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[1;32m   2548\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic\u001b[38;5;241m.\u001b[39mMAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:159\u001b[0m, in \u001b[0;36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    157\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    158\u001b[0m     line \u001b[38;5;241m=\u001b[39m script\n\u001b[0;32m--> 159\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/IPython/core/magics/script.py:336\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m args\u001b[38;5;241m.\u001b[39mraise_error \u001b[38;5;129;01mand\u001b[39;00m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    332\u001b[0m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    333\u001b[0m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    334\u001b[0m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    335\u001b[0m     rc \u001b[38;5;241m=\u001b[39m p\u001b[38;5;241m.\u001b[39mreturncode \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m9\u001b[39m\n\u001b[0;32m--> 336\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'chmod -R 777 . \\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "chmod -R 777 . "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f9d681a9-59fa-4538-b238-7b33db621c5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "featureCounts: unrecognized option '--help'\n",
      "\n",
      "Version 2.0.8\n",
      "\n",
      "Usage: featureCounts [options] -a <annotation_file> -o <output_file> input_file1 [input_file2] ... \n",
      "\n",
      "## Mandatory arguments:\n",
      "\n",
      "  -a <string>         Name of an annotation file. GTF/GFF format by default. See\n",
      "                      -F option for more format information. Inbuilt annotations\n",
      "                      (SAF format) is available in 'annotation' directory of the\n",
      "                      package. Gzipped file is also accepted.\n",
      "\n",
      "  -o <string>         Name of output file including read counts. A separate file\n",
      "                      including summary statistics of counting results is also\n",
      "                      included in the output ('<string>.summary'). Both files\n",
      "                      are in tab delimited format.\n",
      "\n",
      "  input_file1 [input_file2] ...   A list of SAM or BAM format files. They can be\n",
      "                      either name or location sorted. If no files provided,\n",
      "                      <stdin> input is expected. Location-sorted paired-end reads\n",
      "                      are automatically sorted by read names.\n",
      "\n",
      "## Optional arguments:\n",
      "# Annotation\n",
      "\n",
      "  -F <string>         Specify format of the provided annotation file. Acceptable\n",
      "                      formats include 'GTF' (or compatible GFF format) and\n",
      "                      'SAF'. 'GTF' by default.  For SAF format, please refer to\n",
      "                      Users Guide.\n",
      "\n",
      "  -t <string>         Specify feature type(s) in a GTF annotation. If multiple\n",
      "                      types are provided, they should be separated by ',' with\n",
      "                      no space in between. 'exon' by default. Rows in the\n",
      "                      annotation with a matched feature will be extracted and\n",
      "                      used for read mapping. \n",
      "\n",
      "  -g <string>         Specify attribute type in GTF annotation. 'gene_id' by \n",
      "                      default. Meta-features used for read counting will be \n",
      "                      extracted from annotation using the provided value.\n",
      "\n",
      "  --extraAttributes   Extract extra attribute types from the provided GTF\n",
      "                      annotation and include them in the counting output. These\n",
      "                      attribute types will not be used to group features. If\n",
      "                      more than one attribute type is provided they should be\n",
      "                      separated by comma.\n",
      "\n",
      "  -A <string>         Provide a chromosome name alias file to match chr names in\n",
      "                      annotation with those in the reads. This should be a two-\n",
      "                      column comma-delimited text file. Its first column should\n",
      "                      include chr names in the annotation and its second column\n",
      "                      should include chr names in the reads. Chr names are case\n",
      "                      sensitive. No column header should be included in the\n",
      "                      file.\n",
      "\n",
      "# Level of summarization\n",
      "\n",
      "  -f                  Perform read counting at feature level (eg. counting \n",
      "                      reads for exons rather than genes).\n",
      "\n",
      "# Overlap between reads and features\n",
      "\n",
      "  -O                  Assign reads to all their overlapping meta-features (or \n",
      "                      features if -f is specified).\n",
      "\n",
      "  --minOverlap <int>  Minimum number of overlapping bases in a read that is\n",
      "                      required for read assignment. 1 by default. Number of\n",
      "                      overlapping bases is counted from both reads if paired\n",
      "                      end. If a negative value is provided, then a gap of up\n",
      "                      to specified size will be allowed between read and the\n",
      "                      feature that the read is assigned to.\n",
      "\n",
      "  --fracOverlap <float> Minimum fraction of overlapping bases in a read that is\n",
      "                      required for read assignment. Value should be within range\n",
      "                      [0,1]. 0 by default. Number of overlapping bases is\n",
      "                      counted from both reads if paired end. Both this option\n",
      "                      and '--minOverlap' option need to be satisfied for read\n",
      "                      assignment.\n",
      "\n",
      "  --fracOverlapFeature <float> Minimum fraction of overlapping bases in a\n",
      "                      feature that is required for read assignment. Value\n",
      "                      should be within range [0,1]. 0 by default.\n",
      "\n",
      "  --largestOverlap    Assign reads to a meta-feature/feature that has the \n",
      "                      largest number of overlapping bases.\n",
      "\n",
      "  --nonOverlap <int>  Maximum number of non-overlapping bases in a read (or a\n",
      "                      read pair) that is allowed when being assigned to a\n",
      "                      feature. No limit is set by default.\n",
      "\n",
      "  --nonOverlapFeature <int> Maximum number of non-overlapping bases in a feature\n",
      "                      that is allowed in read assignment. No limit is set by\n",
      "                      default.\n",
      "\n",
      "  --readExtension5 <int> Reads are extended upstream by <int> bases from their\n",
      "                      5' end.\n",
      "\n",
      "  --readExtension3 <int> Reads are extended upstream by <int> bases from their\n",
      "                      3' end.\n",
      "\n",
      "  --read2pos <5:3>    Reduce reads to their 5' most base or 3' most base. Read\n",
      "                      counting is then performed based on the single base the \n",
      "                      read is reduced to.\n",
      "\n",
      "# Multi-mapping reads\n",
      "\n",
      "  -M                  Multi-mapping reads will also be counted. For a multi-\n",
      "                      mapping read, all its reported alignments will be \n",
      "                      counted. The 'NH' tag in BAM/SAM input is used to detect \n",
      "                      multi-mapping reads.\n",
      "\n",
      "# Fractional counting\n",
      "\n",
      "  --fraction          Assign fractional counts to features. This option must\n",
      "                      be used together with '-M' or '-O' or both. When '-M' is\n",
      "                      specified, each reported alignment from a multi-mapping\n",
      "                      read (identified via 'NH' tag) will carry a fractional\n",
      "                      count of 1/x, instead of 1 (one), where x is the total\n",
      "                      number of alignments reported for the same read. When '-O'\n",
      "                      is specified, each overlapping feature will receive a\n",
      "                      fractional count of 1/y, where y is the total number of\n",
      "                      features overlapping with the read. When both '-M' and\n",
      "                      '-O' are specified, each alignment will carry a fractional\n",
      "                      count of 1/(x*y).\n",
      "\n",
      "# Read filtering\n",
      "\n",
      "  -Q <int>            The minimum mapping quality score a read must satisfy in\n",
      "                      order to be counted. For paired-end reads, at least one\n",
      "                      end should satisfy this criteria. 0 by default.\n",
      "\n",
      "  --splitOnly         Count split alignments only (ie. alignments with CIGAR\n",
      "                      string containing 'N'). An example of split alignments is\n",
      "                      exon-spanning reads in RNA-seq data.\n",
      "\n",
      "  --nonSplitOnly      If specified, only non-split alignments (CIGAR strings do\n",
      "                      not contain letter 'N') will be counted. All the other\n",
      "                      alignments will be ignored.\n",
      "\n",
      "  --primary           Count primary alignments only. Primary alignments are \n",
      "                      identified using bit 0x100 in SAM/BAM FLAG field.\n",
      "\n",
      "  --ignoreDup         Ignore duplicate reads in read counting. Duplicate reads \n",
      "                      are identified using bit Ox400 in BAM/SAM FLAG field. The \n",
      "                      whole read pair is ignored if one of the reads is a \n",
      "                      duplicate read for paired end data.\n",
      "\n",
      "# Strandness\n",
      "\n",
      "  -s <int or string>  Perform strand-specific read counting. A single integer\n",
      "                      value (applied to all input files) or a string of comma-\n",
      "                      separated values (applied to each corresponding input\n",
      "                      file) should be provided. Possible values include:\n",
      "                      0 (unstranded), 1 (stranded) and 2 (reversely stranded).\n",
      "                      Default value is 0 (ie. unstranded read counting carried\n",
      "                      out for all input files).\n",
      "\n",
      "# Exon-exon junctions\n",
      "\n",
      "  -J                  Count number of reads supporting each exon-exon junction.\n",
      "                      Junctions were identified from all the exon-spanning reads\n",
      "                      in the input (containing 'N' in CIGAR string). Counting\n",
      "                      results are saved to a file named '<output_file>.jcounts'\n",
      "\n",
      "  -G <string>         Provide the name of a FASTA-format file that contains the\n",
      "                      reference sequences used in read mapping that produced the\n",
      "                      provided SAM/BAM files. This optional argument can be used\n",
      "                      with '-J' option to improve read counting for junctions.\n",
      "\n",
      "# Parameters specific to paired end reads\n",
      "\n",
      "  -p                  If specified, libraries are assumed to contain paired-end\n",
      "                      reads. For any library that contains paired-end reads, the\n",
      "                      'countReadPairs' parameter controls if read pairs or reads\n",
      "                      should be counted.\n",
      "\n",
      "  --countReadPairs    If specified, fragments (or templates) will be counted\n",
      "                      instead of reads. This option is only applicable for\n",
      "                      paired-end reads. For single-end data, it is ignored.\n",
      "\n",
      "  -B                  Only count read pairs that have both ends aligned.\n",
      "\n",
      "  -P                  Check validity of paired-end distance when counting read \n",
      "                      pairs. Use -d and -D to set thresholds.\n",
      "\n",
      "  -d <int>            Minimum fragment/template length, 50 by default.\n",
      "\n",
      "  -D <int>            Maximum fragment/template length, 600 by default.\n",
      "\n",
      "  -C                  Do not count read pairs that have their two ends mapping \n",
      "                      to different chromosomes or mapping to same chromosome \n",
      "                      but on different strands.\n",
      "\n",
      "  --donotsort         Do not sort reads in BAM/SAM input. Note that reads from \n",
      "                      the same pair are required to be located next to each \n",
      "                      other in the input.\n",
      "\n",
      "# Number of CPU threads\n",
      "\n",
      "  -T <int>            Number of the threads. 1 by default.\n",
      "\n",
      "# Read groups\n",
      "\n",
      "  --byReadGroup       Assign reads by read group. \"RG\" tag is required to be\n",
      "                      present in the input BAM/SAM files.\n",
      "                      \n",
      "\n",
      "# Long reads\n",
      "\n",
      "  -L                  Count long reads such as Nanopore and PacBio reads. Long\n",
      "                      read counting can only run in one thread and only reads\n",
      "                      (not read-pairs) can be counted. There is no limitation on\n",
      "                      the number of 'M' operations allowed in a CIGAR string in\n",
      "                      long read counting.\n",
      "\n",
      "# Assignment results for each read\n",
      "\n",
      "  -R <format>         Output detailed assignment results for each read or read-\n",
      "                      pair. Results are saved to a file that is in one of the\n",
      "                      following formats: CORE, SAM and BAM. See Users Guide for\n",
      "                      more info about these formats.\n",
      "\n",
      "  --Rpath <string>    Specify a directory to save the detailed assignment\n",
      "                      results. If unspecified, the directory where counting\n",
      "                      results are saved is used.\n",
      "\n",
      "# Miscellaneous\n",
      "\n",
      "  --tmpDir <string>   Directory under which intermediate files are saved (later\n",
      "                      removed). By default, intermediate files will be saved to\n",
      "                      the directory specified in '-o' argument.\n",
      "\n",
      "  --verbose           Output verbose information for debugging, such as un-\n",
      "                      matched chromosome/contig names.\n",
      "\n",
      "  -v                  Output version of the program.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! featureCounts --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "4c18bbde-b285-4e66-921b-db5cd134bc53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "#!genome-build GRCh38.p14\n",
      "#!genome-version GRCh38\n",
      "#!genome-date 2013-12\n",
      "#!genome-build-accession GCA_000001405.29\n",
      "#!genebuild-last-updated 2024-07\n",
      "1\tensembl_havana\tgene\t3069168\t3438621\t.\t+\t.\tgene_id \"ENSG00000142611\"; gene_version \"17\"; gene_name \"PRDM16\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\";\n",
      "1\thavana\ttranscript\t3069168\t3434342\t.\t+\t.\tgene_id \"ENSG00000142611\"; gene_version \"17\"; transcript_id \"ENST00000511072\"; transcript_version \"5\"; gene_name \"PRDM16\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"PRDM16-206\"; transcript_source \"havana\"; transcript_biotype \"protein_coding\"; tag \"gencode_basic\"; tag \"gencode_primary\"; transcript_support_level \"5\";\n",
      "1\thavana\texon\t3069168\t3069296\t.\t+\t.\tgene_id \"ENSG00000142611\"; gene_version \"17\"; transcript_id \"ENST00000511072\"; transcript_version \"5\"; exon_number \"1\"; gene_name \"PRDM16\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"PRDM16-206\"; transcript_source \"havana\"; transcript_biotype \"protein_coding\"; exon_id \"ENSE00002048533\"; exon_version \"1\"; tag \"gencode_basic\"; tag \"gencode_primary\"; transcript_support_level \"5\";\n",
      "1\thavana\tCDS\t3069260\t3069296\t.\t+\t0\tgene_id \"ENSG00000142611\"; gene_version \"17\"; transcript_id \"ENST00000511072\"; transcript_version \"5\"; exon_number \"1\"; gene_name \"PRDM16\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"PRDM16-206\"; transcript_source \"havana\"; transcript_biotype \"protein_coding\"; protein_id \"ENSP00000426975\"; protein_version \"1\"; tag \"gencode_basic\"; tag \"gencode_primary\"; transcript_support_level \"5\";\n",
      "1\thavana\tstart_codon\t3069260\t3069262\t.\t+\t0\tgene_id \"ENSG00000142611\"; gene_version \"17\"; transcript_id \"ENST00000511072\"; transcript_version \"5\"; exon_number \"1\"; gene_name \"PRDM16\"; gene_source \"ensembl_havana\"; gene_biotype \"protein_coding\"; transcript_name \"PRDM16-206\"; transcript_source \"havana\"; transcript_biotype \"protein_coding\"; tag \"gencode_basic\"; tag \"gencode_primary\"; transcript_support_level \"5\";\n"
     ]
    }
   ],
   "source": [
    "! head -10 data/Homo_sapiens.GRCh38.113.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "00601606-a344-41ff-8aa9-fe32d8145935",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      ">ENST00000390473.1 cdna chromosome:GRCh38:14:22450089:22450139:1 gene:ENSG00000211825.1 gene_biotype:TR_J_gene transcript_biotype:TR_J_gene gene_symbol:TRDJ1 description:T cell receptor delta joining 1 [Source:HGNC Symbol;Acc:HGNC:12257]\n",
      "ACACCGATAAACTCATCTTTGGAAAAGGAACCCGTGTGACTGTGGAACCAA\n",
      ">ENST00000390484.1 cdna chromosome:GRCh38:14:22482287:22482346:1 gene:ENSG00000211836.1 gene_biotype:TR_J_gene transcript_biotype:TR_J_gene gene_symbol:TRAJ54 description:T cell receptor alpha joining 54 [Source:HGNC Symbol;Acc:HGNC:12086]\n",
      "TAATTCAGGGAGCCCAGAAGCTGGTATTTGGCCAAGGAACCAGGCTGACTATCAACCCAA\n",
      ">ENST00000390488.1 cdna chromosome:GRCh38:14:22489488:22489543:1 gene:ENSG00000211840.1 gene_biotype:TR_J_gene transcript_biotype:TR_J_gene gene_symbol:TRAJ49 description:T cell receptor alpha joining 49 [Source:HGNC Symbol;Acc:HGNC:12080]\n",
      "GAACACCGGTAACCAGTTCTATTTTGGGACAGGGACAAGTTTGACGGTCATTCCAA\n",
      ">ENST00000390476.1 cdna chromosome:GRCh38:14:22459098:22459156:1 gene:ENSG00000211828.1 gene_biotype:TR_J_gene transcript_biotype:TR_J_gene gene_symbol:TRDJ3 description:T cell receptor delta joining 3 [Source:HGNC Symbol;Acc:HGNC:12259]\n"
     ]
    }
   ],
   "source": [
    "! head -7 data/Homo_sapiens.GRCh38.cdna.all.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "2a30a6ad-2429-4c54-9635-273e6fcd3a44",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "ERR950158.33795\t101\tENST00000614481.1\t1\t0\t*\t=\t1\t0\tCCAGCCCGGAAAACAAAGTAAGATCCTGTCTCTAAAAATATATATTTAAAAATATAAAATAAATAAGTTAAAGAAGGAGGAAGGAGAGGAAGAAAATTTGG\tCCCFFFFFGFFFHIJIJAAGHGHDFGDHGHIJJJIHICEGHJFJJJJJIIEGHJJIJJG@GHIIIBHAEG@AHHHFFEFB?EE@BDBBBA??ADC######\tYT:Z:UP\n",
      "ERR950158.33795\t153\tENST00000614481.1\t1\t22\t67S34M\t=\t1\t0\tGGAGGAAGGAGAGGAAGAAAATTTGGCTTAGTAGGAAAGTGTTTTTGTACTGGGCAGAAACACTGTGTCAAACTGGGGCAAACAACCTCTTCTTTAGGACT\t>CCDDDDDCC>ACDEEEEEEEFFFFFEEHHHHJJJJJJJJJJJIJIGFEIHJJJIJJIIGHFIJIJIJJJJIGJJJJJIIIGIGGJIJHHGHHFFED=B@@\tAS:i:61\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:28G5\tYT:Z:UP\n",
      "ERR950158.154015\t99\tENST00000390504.1\t1\t24\t50S51M\t=\t1\t136\tGGATCCCCGGAAGGACAGTGAAGGTTTTTGTTAAGGTTTTTGTGTCTGTGTGGATAGCAACTATCAGTTAATCTGGGGCGCTGGGACCAAGCTAATTATAA\tBCCFFFFFHHFHHJGIJJFHHIIIDGHJJJJJIJIJBGIIJJJJJJJJHIGHIGGIHHHFHHFFFFFBF@CCCEDECD@BBDDDB@B?ABBBDCCCDA@4:\tAS:i:102\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYS:i:114\tYT:Z:CP\n",
      "ERR950158.154015\t147\tENST00000390504.1\t1\t24\t15S57M29S\t=\t1\t-136\tGTTTTTGTGTCTGTGTGGATAGCAACTATCAGTTAATCTGGGGCGCTGGGACCAAGCTAATTATAAAGCCAGGTAAGTCTCAGAGATGTGACTGCACTAGA\tBBA@DDDDDDBDDDCC>;EDEEEDEDFFFFFEHHHEEJIJJIJJJJJJJIHJJJIJJIGJJIIGIJJIJJIHJJJIIJIJJJJIGIIJHGHHHFFDDBC@B\tAS:i:114\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:57\tYS:i:102\tYT:Z:CP\n",
      "ERR950158.390189\t73\tENST00000390523.1\t22\t22\t31M70S\t=\t22\t0\tGGGAGTGGGACAAGATTATCAGTAAAACCTGGTAAGTAGGCAATATGTCACTAAAGTAGGAGGCTTAATGTGGCTACTGAGACCCACTAAACTTACTGCAG\t@CCFFBDDHHHHHJHIJJJJJJJJJJJJJJJJHIIJIHIJJIIJIIIJJEGHIGHHFGIJGIIIIHEHGDDG>EEEHC@CDEDDDDCDC>CC@>CC@(:AA\tAS:i:62\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:31\tYT:Z:UP\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error reading file \"data/alignment_Bowtie2_subset/ERR950158_trimmed.bam\": Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "! samtools view data/alignment_Bowtie2_subset/ERR950158_trimmed.bam | head -5 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ef04fd01-0ed2-4192-ab6d-1bdfa6bcc4ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "Usage: head [OPTION]... [FILE]...\n",
      "Print the first 10 lines of each FILE to standard output.\n",
      "With more than one FILE, precede each with a header giving the file name.\n",
      "\n",
      "With no FILE, or when FILE is -, read standard input.\n",
      "\n",
      "Mandatory arguments to long options are mandatory for short options too.\n",
      "  -c, --bytes=[-]NUM       print the first NUM bytes of each file;\n",
      "                             with the leading '-', print all but the last\n",
      "                             NUM bytes of each file\n",
      "  -n, --lines=[-]NUM       print the first NUM lines instead of the first 10;\n",
      "                             with the leading '-', print all but the last\n",
      "                             NUM lines of each file\n",
      "  -q, --quiet, --silent    never print headers giving file names\n",
      "  -v, --verbose            always print headers giving file names\n",
      "  -z, --zero-terminated    line delimiter is NUL, not newline\n",
      "      --help     display this help and exit\n",
      "      --version  output version information and exit\n",
      "\n",
      "NUM may have a multiplier suffix:\n",
      "b 512, kB 1000, K 1024, MB 1000*1000, M 1024*1024,\n",
      "GB 1000*1000*1000, G 1024*1024*1024, and so on for T, P, E, Z, Y.\n",
      "Binary prefixes can be used, too: KiB=K, MiB=M, and so on.\n",
      "\n",
      "GNU coreutils online help: <https://www.gnu.org/software/coreutils/>\n",
      "Report any translation bugs to <https://translationproject.org/team/>\n",
      "Full documentation <https://www.gnu.org/software/coreutils/head>\n",
      "or available locally via: info '(coreutils) head invocation'\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ebef3b4-2b3d-47cb-a7ca-395c1f1c4343",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
