{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "20b8d80b-9eb0-4846-bb21-d80d9882e4c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%script false --no-raise-error\n",
    "#Copy this code and run it in the terminal everytime you create a new directory or file\n",
    "#Gives permission for everyone to work with the files created\n",
    "#Run at the end of a jupyter script, the output is too messy\n",
    "!cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork \n",
    "!chmod -R 777 . "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a29ef54-a06b-4be8-8d13-ac484ca8d09a",
   "metadata": {},
   "source": [
    "# Download cDNA library \n",
    "From ensembl: https://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/ and unzip."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "359a5dbb-293e-4af1-9006-f155830ee599",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "--2025-02-27 10:32:43--  ftp://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
      "           => 'data/Homo_sapiens.GRCh38.cdna.all.fa.gz'\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-113/fasta/homo_sapiens/cdna ... done.\n",
      "==> SIZE Homo_sapiens.GRCh38.cdna.all.fa.gz ... 79685174\n",
      "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.cdna.all.fa.gz ... done.\n",
      "Length: 79685174 (76M) (unauthoritative)\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  774K 1m40s\n",
      "    50K .......... .......... .......... .......... ..........  0% 2.05M 69s\n",
      "   100K .......... .......... .......... .......... ..........  0% 99.0M 46s\n",
      "   150K .......... .......... .......... .......... ..........  0% 17.6M 36s\n",
      "   200K .......... .......... .......... .......... ..........  0% 2.34M 35s\n",
      "   250K .......... .......... .......... .......... ..........  0%  177M 29s\n",
      "   300K .......... .......... .......... .......... ..........  0%  532M 25s\n",
      "   350K .......... .......... .......... .......... ..........  0%  480M 22s\n",
      "   400K .......... .......... .......... .......... ..........  0% 2.11M 23s\n",
      "   450K .......... .......... .......... .......... ..........  0%  611M 21s\n",
      "   500K .......... .......... .......... .......... ..........  0%  216M 19s\n",
      "   550K .......... .......... .......... .......... ..........  0%  365M 18s\n",
      "   600K .......... .......... .......... .......... ..........  0%  160M 16s\n",
      "   650K .......... .......... .......... .......... ..........  0%  432M 15s\n",
      "   700K .......... .......... .......... .......... ..........  0%  531M 14s\n",
      "   750K .......... .......... .......... .......... ..........  1%  653M 13s\n",
      "   800K .......... .......... .......... .......... ..........  1% 16.3M 13s\n",
      "   850K .......... .......... .......... .......... ..........  1% 2.47M 14s\n",
      "   900K .......... .......... .......... .......... ..........  1%  469M 13s\n",
      "   950K .......... .......... .......... .......... ..........  1%  571M 12s\n",
      "  1000K .......... .......... .......... .......... ..........  1%  320M 12s\n",
      "  1050K .......... .......... .......... .......... ..........  1%  499M 11s\n",
      "  1100K .......... .......... .......... .......... ..........  1%  474M 11s\n",
      "  1150K .......... .......... .......... .......... ..........  1%  559M 10s\n",
      "  1200K .......... .......... .......... .......... ..........  1%  527M 10s\n",
      "  1250K .......... .......... .......... .......... ..........  1%  622M 9s\n",
      "  1300K .......... .......... .......... .......... ..........  1%  694M 9s\n",
      "  1350K .......... .......... .......... .......... ..........  1%  516M 9s\n",
      "  1400K .......... .......... .......... .......... ..........  1%  682M 8s\n",
      "  1450K .......... .......... .......... .......... ..........  1%  658M 8s\n",
      "  1500K .......... .......... .......... .......... ..........  1%  676M 8s\n",
      "  1550K .......... .......... .......... .......... ..........  2%  563M 8s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 75.0M 7s\n",
      "  1650K .......... .......... .......... .......... ..........  2% 31.6M 7s\n",
      "  1700K .......... .......... .......... .......... ..........  2% 2.43M 8s\n",
      "  1750K .......... .......... .......... .......... ..........  2%  402M 8s\n",
      "  1800K .......... .......... .......... .......... ..........  2%  418M 8s\n",
      "  1850K .......... .......... .......... .......... ..........  2%  508M 7s\n",
      "  1900K .......... .......... .......... .......... ..........  2%  417M 7s\n",
      "  1950K .......... .......... .......... .......... ..........  2%  382M 7s\n",
      "  2000K .......... .......... .......... .......... ..........  2%  632M 7s\n",
      "  2050K .......... .......... .......... .......... ..........  2%  236M 7s\n",
      "  2100K .......... .......... .......... .......... ..........  2%  596M 6s\n",
      "  2150K .......... .......... .......... .......... ..........  2%  635M 6s\n",
      "  2200K .......... .......... .......... .......... ..........  2%  695M 6s\n",
      "  2250K .......... .......... .......... .......... ..........  2%  501M 6s\n",
      "  2300K .......... .......... .......... .......... ..........  3%  411M 6s\n",
      "  2350K .......... .......... .......... .......... ..........  3%  634M 6s\n",
      "  2400K .......... .......... .......... .......... ..........  3%  562M 6s\n",
      "  2450K .......... .......... .......... .......... ..........  3%  664M 6s\n",
      "  2500K .......... .......... .......... .......... ..........  3%  576M 5s\n",
      "  2550K .......... .......... .......... .......... ..........  3%  666M 5s\n",
      "  2600K .......... .......... .......... .......... ..........  3%  643M 5s\n",
      "  2650K .......... .......... .......... .......... ..........  3%  679M 5s\n",
      "  2700K .......... .......... .......... .......... ..........  3%  573M 5s\n",
      "  2750K .......... .......... .......... .......... ..........  3%  503M 5s\n",
      "  2800K .......... .......... .......... .......... ..........  3%  683M 5s\n",
      "  2850K .......... .......... .......... .......... ..........  3%  649M 5s\n",
      "  2900K .......... .......... .......... .......... ..........  3%  582M 5s\n",
      "  2950K .......... .......... .......... .......... ..........  3%  673M 5s\n",
      "  3000K .......... .......... .......... .......... ..........  3%  663M 5s\n",
      "  3050K .......... .......... .......... .......... ..........  3%  649M 4s\n",
      "  3100K .......... .......... .......... .......... ..........  4%  571M 4s\n",
      "  3150K .......... .......... .......... .......... ..........  4%  689M 4s\n",
      "  3200K .......... .......... .......... .......... ..........  4% 34.3M 4s\n",
      "  3250K .......... .......... .......... .......... ..........  4% 2.43M 5s\n",
      "  3300K .......... .......... .......... .......... ..........  4%  456M 5s\n",
      "  3350K .......... .......... .......... .......... ..........  4%  511M 5s\n",
      "  3400K .......... .......... .......... .......... ..........  4%  535M 4s\n",
      "  3450K .......... .......... .......... .......... ..........  4%  499M 4s\n",
      "  3500K .......... .......... .......... .......... ..........  4%  665M 4s\n",
      "  3550K .......... .......... .......... .......... ..........  4%  575M 4s\n",
      "  3600K .......... .......... .......... .......... ..........  4%  659M 4s\n",
      "  3650K .......... .......... .......... .......... ..........  4%  367M 4s\n",
      "  3700K .......... .......... .......... .......... ..........  4%  531M 4s\n",
      "  3750K .......... .......... .......... .......... ..........  4%  594M 4s\n",
      "  3800K .......... .......... .......... .......... ..........  4%  522M 4s\n",
      "  3850K .......... .......... .......... .......... ..........  5%  530M 4s\n",
      "  3900K .......... .......... .......... .......... ..........  5%  692M 4s\n",
      "  3950K .......... .......... .......... .......... ..........  5%  680M 4s\n",
      "  4000K .......... .......... .......... .......... ..........  5%  546M 4s\n",
      "  4050K .......... .......... .......... .......... ..........  5%  574M 4s\n",
      "  4100K .......... .......... .......... .......... ..........  5%  581M 4s\n",
      "  4150K .......... .......... .......... .......... ..........  5%  448M 4s\n",
      "  4200K .......... .......... .......... .......... ..........  5%  619M 4s\n",
      "  4250K .......... .......... .......... .......... ..........  5%  574M 4s\n",
      "  4300K .......... .......... .......... .......... ..........  5%  573M 4s\n",
      "  4350K .......... .......... .......... .......... ..........  5%  599M 3s\n",
      "  4400K .......... .......... .......... .......... ..........  5%  642M 3s\n",
      "  4450K .......... .......... .......... .......... ..........  5%  458M 3s\n",
      "  4500K .......... .......... .......... .......... ..........  5%  601M 3s\n",
      "  4550K .......... .......... .......... .......... ..........  5%  679M 3s\n",
      "  4600K .......... .......... .......... .......... ..........  5%  691M 3s\n",
      "  4650K .......... .......... .......... .......... ..........  6%  525M 3s\n",
      "  4700K .......... .......... .......... .......... ..........  6% 65.4M 3s\n",
      "  4750K .......... .......... .......... .......... ..........  6% 88.8M 3s\n",
      "  4800K .......... .......... .......... .......... ..........  6% 2.43M 3s\n",
      "  4850K .......... .......... .......... .......... ..........  6%  617M 3s\n",
      "  4900K .......... .......... .......... .......... ..........  6%  549M 3s\n",
      "  4950K .......... .......... .......... .......... ..........  6%  223M 3s\n",
      "  5000K .......... .......... .......... .......... ..........  6%  608M 3s\n",
      "  5050K .......... .......... .......... .......... ..........  6%  536M 3s\n",
      "  5100K .......... .......... .......... .......... ..........  6%  465M 3s\n",
      "  5150K .......... .......... .......... .......... ..........  6%  677M 3s\n",
      "  5200K .......... .......... .......... .......... ..........  6%  399M 3s\n",
      "  5250K .......... .......... .......... .......... ..........  6%  628M 3s\n",
      "  5300K .......... .......... .......... .......... ..........  6%  470M 3s\n",
      "  5350K .......... .......... .......... .......... ..........  6%  700M 3s\n",
      "  5400K .......... .......... .......... .......... ..........  7%  715M 3s\n",
      "  5450K .......... .......... .......... .......... ..........  7%  495M 3s\n",
      "  5500K .......... .......... .......... .......... ..........  7%  566M 3s\n",
      "  5550K .......... .......... .......... .......... ..........  7%  439M 3s\n",
      "  5600K .......... .......... .......... .......... ..........  7%  624M 3s\n",
      "  5650K .......... .......... .......... .......... ..........  7%  633M 3s\n",
      "  5700K .......... .......... .......... .......... ..........  7%  485M 3s\n",
      "  5750K .......... .......... .......... .......... ..........  7%  516M 3s\n",
      "  5800K .......... .......... .......... .......... ..........  7%  724M 3s\n",
      "  5850K .......... .......... .......... .......... ..........  7%  552M 3s\n",
      "  5900K .......... .......... .......... .......... ..........  7%  641M 3s\n",
      "  5950K .......... .......... .......... .......... ..........  7%  404M 3s\n",
      "  6000K .......... .......... .......... .......... ..........  7%  531M 3s\n",
      "  6050K .......... .......... .......... .......... ..........  7%  598M 3s\n",
      "  6100K .......... .......... .......... .......... ..........  7%  546M 3s\n",
      "  6150K .......... .......... .......... .......... ..........  7%  611M 3s\n",
      "  6200K .......... .......... .......... .......... ..........  8%  430M 3s\n",
      "  6250K .......... .......... .......... .......... ..........  8% 78.5M 3s\n",
      "  6300K .......... .......... .......... .......... ..........  8%  184M 3s\n",
      "  6350K .......... .......... .......... .......... ..........  8% 2.36M 3s\n",
      "  6400K .......... .......... .......... .......... ..........  8%  443M 3s\n",
      "  6450K .......... .......... .......... .......... ..........  8%  694M 3s\n",
      "  6500K .......... .......... .......... .......... ..........  8%  274M 3s\n",
      "  6550K .......... .......... .......... .......... ..........  8%  634M 3s\n",
      "  6600K .......... .......... .......... .......... ..........  8%  570M 3s\n",
      "  6650K .......... .......... .......... .......... ..........  8%  207M 3s\n",
      "  6700K .......... .......... .......... .......... ..........  8%  644M 3s\n",
      "  6750K .......... .......... .......... .......... ..........  8%  541M 3s\n",
      "  6800K .......... .......... .......... .......... ..........  8%  516M 3s\n",
      "  6850K .......... .......... .......... .......... ..........  8%  714M 3s\n",
      "  6900K .......... .......... .......... .......... ..........  8%  409M 3s\n",
      "  6950K .......... .......... .......... .......... ..........  8%  736M 3s\n",
      "  7000K .......... .......... .......... .......... ..........  9%  175M 3s\n",
      "  7050K .......... .......... .......... .......... ..........  9%  497M 3s\n",
      "  7100K .......... .......... .......... .......... ..........  9%  734M 3s\n",
      "  7150K .......... .......... .......... .......... ..........  9%  354M 3s\n",
      "  7200K .......... .......... .......... .......... ..........  9%  190M 3s\n",
      "  7250K .......... .......... .......... .......... ..........  9%  581M 2s\n",
      "  7300K .......... .......... .......... .......... ..........  9%  586M 2s\n",
      "  7350K .......... .......... .......... .......... ..........  9%  684M 2s\n",
      "  7400K .......... .......... .......... .......... ..........  9%  388M 2s\n",
      "  7450K .......... .......... .......... .......... ..........  9%  176M 2s\n",
      "  7500K .......... .......... .......... .......... ..........  9%  656M 2s\n",
      "  7550K .......... .......... .......... .......... ..........  9%  356M 2s\n",
      "  7600K .......... .......... .......... .......... ..........  9%  374M 2s\n",
      "  7650K .......... .......... .......... .......... ..........  9%  691M 2s\n",
      "  7700K .......... .......... .......... .......... ..........  9%  274M 2s\n",
      "  7750K .......... .......... .......... .......... .......... 10% 79.4M 2s\n",
      "  7800K .......... .......... .......... .......... .......... 10%  298M 2s\n",
      "  7850K .......... .......... .......... .......... .......... 10%  571M 2s\n",
      "  7900K .......... .......... .......... .......... .......... 10% 2.16M 2s\n",
      "  7950K .......... .......... .......... .......... .......... 10%  494M 2s\n",
      "  8000K .......... .......... .......... .......... .......... 10%  638M 2s\n",
      "  8050K .......... .......... .......... .......... .......... 10%  116M 2s\n",
      "  8100K .......... .......... .......... .......... .......... 10%  404M 2s\n",
      "  8150K .......... .......... .......... .......... .......... 10%  174M 2s\n",
      "  8200K .......... .......... .......... .......... .......... 10%  464M 2s\n",
      "  8250K .......... .......... .......... .......... .......... 10%  672M 2s\n",
      "  8300K .......... .......... .......... .......... .......... 10%  243M 2s\n",
      "  8350K .......... .......... .......... .......... .......... 10%  504M 2s\n",
      "  8400K .......... .......... .......... .......... .......... 10%  211M 2s\n",
      "  8450K .......... .......... .......... .......... .......... 10%  534M 2s\n",
      "  8500K .......... .......... .......... .......... .......... 10%  679M 2s\n",
      "  8550K .......... .......... .......... .......... .......... 11%  135M 2s\n",
      "  8600K .......... .......... .......... .......... .......... 11%  673M 2s\n",
      "  8650K .......... .......... .......... .......... .......... 11%  587M 2s\n",
      "  8700K .......... .......... .......... .......... .......... 11%  494M 2s\n",
      "  8750K .......... .......... .......... .......... .......... 11% 1.90M 2s\n",
      "  8800K .......... .......... .......... .......... .......... 11%  619M 2s\n",
      "  8850K .......... .......... .......... .......... .......... 11%  625M 2s\n",
      "  8900K .......... .......... .......... .......... .......... 11% 62.7M 2s\n",
      "  8950K .......... .......... .......... .......... .......... 11%  619M 2s\n",
      "  9000K .......... .......... .......... .......... .......... 11%  456M 2s\n",
      "  9050K .......... .......... .......... .......... .......... 11%  626M 2s\n",
      "  9100K .......... .......... .......... .......... .......... 11%  675M 2s\n",
      "  9150K .......... .......... .......... .......... .......... 11%  651M 2s\n",
      "  9200K .......... .......... .......... .......... .......... 11% 24.2M 2s\n",
      "  9250K .......... .......... .......... .......... .......... 11%  435M 2s\n",
      "  9300K .......... .......... .......... .......... .......... 12%  641M 2s\n",
      "  9350K .......... .......... .......... .......... .......... 12%  603M 2s\n",
      "  9400K .......... .......... .......... .......... .......... 12%  573M 2s\n",
      "  9450K .......... .......... .......... .......... .......... 12%  557M 2s\n",
      "  9500K .......... .......... .......... .......... .......... 12%  628M 2s\n",
      "  9550K .......... .......... .......... .......... .......... 12%  632M 2s\n",
      "  9600K .......... .......... .......... .......... .......... 12%  641M 2s\n",
      "  9650K .......... .......... .......... .......... .......... 12%  539M 2s\n",
      "  9700K .......... .......... .......... .......... .......... 12%  632M 2s\n",
      "  9750K .......... .......... .......... .......... .......... 12%  634M 2s\n",
      "  9800K .......... .......... .......... .......... .......... 12%  620M 2s\n",
      "  9850K .......... .......... .......... .......... .......... 12%  561M 2s\n",
      "  9900K .......... .......... .......... .......... .......... 12%  665M 2s\n",
      "  9950K .......... .......... .......... .......... .......... 12%  624M 2s\n",
      " 10000K .......... .......... .......... .......... .......... 12%  604M 2s\n",
      " 10050K .......... .......... .......... .......... .......... 12%  544M 2s\n",
      " 10100K .......... .......... .......... .......... .......... 13%  650M 2s\n",
      " 10150K .......... .......... .......... .......... .......... 13%  662M 2s\n",
      " 10200K .......... .......... .......... .......... .......... 13%  622M 2s\n",
      " 10250K .......... .......... .......... .......... .......... 13% 2.53M 2s\n",
      " 10300K .......... .......... .......... .......... .......... 13%  343M 2s\n",
      " 10350K .......... .......... .......... .......... .......... 13%  503M 2s\n",
      " 10400K .......... .......... .......... .......... .......... 13% 87.8M 2s\n",
      " 10450K .......... .......... .......... .......... .......... 13%  312M 2s\n",
      " 10500K .......... .......... .......... .......... .......... 13%  529M 2s\n",
      " 10550K .......... .......... .......... .......... .......... 13%  652M 2s\n",
      " 10600K .......... .......... .......... .......... .......... 13% 25.6M 2s\n",
      " 10650K .......... .......... .......... .......... .......... 13%  281M 2s\n",
      " 10700K .......... .......... .......... .......... .......... 13%  672M 2s\n",
      " 10750K .......... .......... .......... .......... .......... 13%  578M 2s\n",
      " 10800K .......... .......... .......... .......... .......... 13%  444M 2s\n",
      " 10850K .......... .......... .......... .......... .......... 14%  676M 2s\n",
      " 10900K .......... .......... .......... .......... .......... 14%  217M 2s\n",
      " 10950K .......... .......... .......... .......... .......... 14%  556M 2s\n",
      " 11000K .......... .......... .......... .......... .......... 14%  544M 2s\n",
      " 11050K .......... .......... .......... .......... .......... 14%  473M 2s\n",
      " 11100K .......... .......... .......... .......... .......... 14%  691M 2s\n",
      " 11150K .......... .......... .......... .......... .......... 14%  538M 2s\n",
      " 11200K .......... .......... .......... .......... .......... 14%  536M 2s\n",
      " 11250K .......... .......... .......... .......... .......... 14%  598M 2s\n",
      " 11300K .......... .......... .......... .......... .......... 14%  505M 2s\n",
      " 11350K .......... .......... .......... .......... .......... 14%  507M 2s\n",
      " 11400K .......... .......... .......... .......... .......... 14%  489M 2s\n",
      " 11450K .......... .......... .......... .......... .......... 14%  731M 2s\n",
      " 11500K .......... .......... .......... .......... .......... 14%  587M 2s\n",
      " 11550K .......... .......... .......... .......... .......... 14%  620M 2s\n",
      " 11600K .......... .......... .......... .......... .......... 14%  569M 2s\n",
      " 11650K .......... .......... .......... .......... .......... 15%  554M 2s\n",
      " 11700K .......... .......... .......... .......... .......... 15%  626M 2s\n",
      " 11750K .......... .......... .......... .......... .......... 15%  589M 2s\n",
      " 11800K .......... .......... .......... .......... .......... 15% 2.64M 2s\n",
      " 11850K .......... .......... .......... .......... .......... 15%  498M 2s\n",
      " 11900K .......... .......... .......... .......... .......... 15% 79.3M 2s\n",
      " 11950K .......... .......... .......... .......... .......... 15%  385M 2s\n",
      " 12000K .......... .......... .......... .......... .......... 15%  441M 2s\n",
      " 12050K .......... .......... .......... .......... .......... 15%  685M 2s\n",
      " 12100K .......... .......... .......... .......... .......... 15% 42.4M 2s\n",
      " 12150K .......... .......... .......... .......... .......... 15% 51.2M 2s\n",
      " 12200K .......... .......... .......... .......... .......... 15%  586M 2s\n",
      " 12250K .......... .......... .......... .......... .......... 15%  596M 2s\n",
      " 12300K .......... .......... .......... .......... .......... 15%  155M 2s\n",
      " 12350K .......... .......... .......... .......... .......... 15%  528M 2s\n",
      " 12400K .......... .......... .......... .......... .......... 15%  507M 2s\n",
      " 12450K .......... .......... .......... .......... .......... 16%  165M 2s\n",
      " 12500K .......... .......... .......... .......... .......... 16%  519M 2s\n",
      " 12550K .......... .......... .......... .......... .......... 16%  710M 2s\n",
      " 12600K .......... .......... .......... .......... .......... 16%  618M 2s\n",
      " 12650K .......... .......... .......... .......... .......... 16%  519M 2s\n",
      " 12700K .......... .......... .......... .......... .......... 16%  558M 2s\n",
      " 12750K .......... .......... .......... .......... .......... 16%  692M 2s\n",
      " 12800K .......... .......... .......... .......... .......... 16%  574M 2s\n",
      " 12850K .......... .......... .......... .......... .......... 16%  408M 2s\n",
      " 12900K .......... .......... .......... .......... .......... 16%  598M 2s\n",
      " 12950K .......... .......... .......... .......... .......... 16%  494M 2s\n",
      " 13000K .......... .......... .......... .......... .......... 16%  556M 2s\n",
      " 13050K .......... .......... .......... .......... .......... 16%  685M 2s\n",
      " 13100K .......... .......... .......... .......... .......... 16%  616M 2s\n",
      " 13150K .......... .......... .......... .......... .......... 16%  595M 2s\n",
      " 13200K .......... .......... .......... .......... .......... 17%  224M 2s\n",
      " 13250K .......... .......... .......... .......... .......... 17%  387M 2s\n",
      " 13300K .......... .......... .......... .......... .......... 17% 34.4M 2s\n",
      " 13350K .......... .......... .......... .......... .......... 17% 2.90M 2s\n",
      " 13400K .......... .......... .......... .......... .......... 17%  537M 2s\n",
      " 13450K .......... .......... .......... .......... .......... 17%  106M 2s\n",
      " 13500K .......... .......... .......... .......... .......... 17%  243M 2s\n",
      " 13550K .......... .......... .......... .......... .......... 17%  542M 2s\n",
      " 13600K .......... .......... .......... .......... .......... 17%  668M 2s\n",
      " 13650K .......... .......... .......... .......... .......... 17% 27.4M 2s\n",
      " 13700K .......... .......... .......... .......... .......... 17%  143M 2s\n",
      " 13750K .......... .......... .......... .......... .......... 17%  513M 2s\n",
      " 13800K .......... .......... .......... .......... .......... 17%  193M 2s\n",
      " 13850K .......... .......... .......... .......... .......... 17%  403M 2s\n",
      " 13900K .......... .......... .......... .......... .......... 17%  704M 2s\n",
      " 13950K .......... .......... .......... .......... .......... 17%  274M 2s\n",
      " 14000K .......... .......... .......... .......... .......... 18%  705M 2s\n",
      " 14050K .......... .......... .......... .......... .......... 18%  720M 2s\n",
      " 14100K .......... .......... .......... .......... .......... 18%  297M 2s\n",
      " 14150K .......... .......... .......... .......... .......... 18%  592M 2s\n",
      " 14200K .......... .......... .......... .......... .......... 18%  468M 2s\n",
      " 14250K .......... .......... .......... .......... .......... 18%  647M 2s\n",
      " 14300K .......... .......... .......... .......... .......... 18%  564M 2s\n",
      " 14350K .......... .......... .......... .......... .......... 18%  153M 2s\n",
      " 14400K .......... .......... .......... .......... .......... 18%  422M 2s\n",
      " 14450K .......... .......... .......... .......... .......... 18%  612M 2s\n",
      " 14500K .......... .......... .......... .......... .......... 18%  557M 2s\n",
      " 14550K .......... .......... .......... .......... .......... 18%  514M 2s\n",
      " 14600K .......... .......... .......... .......... .......... 18%  529M 2s\n",
      " 14650K .......... .......... .......... .......... .......... 18%  687M 2s\n",
      " 14700K .......... .......... .......... .......... .......... 18% 79.2M 2s\n",
      " 14750K .......... .......... .......... .......... .......... 19%  627M 2s\n",
      " 14800K .......... .......... .......... .......... .......... 19%  720M 2s\n",
      " 14850K .......... .......... .......... .......... .......... 19% 56.9M 2s\n",
      " 14900K .......... .......... .......... .......... .......... 19% 2.86M 2s\n",
      " 14950K .......... .......... .......... .......... .......... 19%  511M 2s\n",
      " 15000K .......... .......... .......... .......... .......... 19%  578M 2s\n",
      " 15050K .......... .......... .......... .......... .......... 19%  143M 2s\n",
      " 15100K .......... .......... .......... .......... .......... 19%  677M 2s\n",
      " 15150K .......... .......... .......... .......... .......... 19% 53.8M 2s\n",
      " 15200K .......... .......... .......... .......... .......... 19% 48.1M 2s\n",
      " 15250K .......... .......... .......... .......... .......... 19%  248M 2s\n",
      " 15300K .......... .......... .......... .......... .......... 19%  312M 2s\n",
      " 15350K .......... .......... .......... .......... .......... 19%  250M 2s\n",
      " 15400K .......... .......... .......... .......... .......... 19%  139M 2s\n",
      " 15450K .......... .......... .......... .......... .......... 19%  494M 2s\n",
      " 15500K .......... .......... .......... .......... .......... 19%  591M 2s\n",
      " 15550K .......... .......... .......... .......... .......... 20%  235M 2s\n",
      " 15600K .......... .......... .......... .......... .......... 20%  141M 2s\n",
      " 15650K .......... .......... .......... .......... .......... 20%  115M 2s\n",
      " 15700K .......... .......... .......... .......... .......... 20%  443M 2s\n",
      " 15750K .......... .......... .......... .......... .......... 20%  583M 2s\n",
      " 15800K .......... .......... .......... .......... .......... 20%  583M 2s\n",
      " 15850K .......... .......... .......... .......... .......... 20%  551M 2s\n",
      " 15900K .......... .......... .......... .......... .......... 20%  502M 2s\n",
      " 15950K .......... .......... .......... .......... .......... 20%  620M 2s\n",
      " 16000K .......... .......... .......... .......... .......... 20%  485M 2s\n",
      " 16050K .......... .......... .......... .......... .......... 20%  655M 2s\n",
      " 16100K .......... .......... .......... .......... .......... 20%  578M 2s\n",
      " 16150K .......... .......... .......... .......... .......... 20%  709M 2s\n",
      " 16200K .......... .......... .......... .......... .......... 20%  605M 2s\n",
      " 16250K .......... .......... .......... .......... .......... 20%  179M 2s\n",
      " 16300K .......... .......... .......... .......... .......... 21%  367M 2s\n",
      " 16350K .......... .......... .......... .......... .......... 21% 63.7M 2s\n",
      " 16400K .......... .......... .......... .......... .......... 21%  390M 2s\n",
      " 16450K .......... .......... .......... .......... .......... 21% 2.86M 2s\n",
      " 16500K .......... .......... .......... .......... .......... 21%  452M 2s\n",
      " 16550K .......... .......... .......... .......... .......... 21%  551M 2s\n",
      " 16600K .......... .......... .......... .......... .......... 21%  600M 2s\n",
      " 16650K .......... .......... .......... .......... .......... 21%  581M 2s\n",
      " 16700K .......... .......... .......... .......... .......... 21% 27.8M 2s\n",
      " 16750K .......... .......... .......... .......... .......... 21%  127M 2s\n",
      " 16800K .......... .......... .......... .......... .......... 21%  524M 2s\n",
      " 16850K .......... .......... .......... .......... .......... 21%  559M 2s\n",
      " 16900K .......... .......... .......... .......... .......... 21%  170M 2s\n",
      " 16950K .......... .......... .......... .......... .......... 21%  236M 2s\n",
      " 17000K .......... .......... .......... .......... .......... 21%  455M 2s\n",
      " 17050K .......... .......... .......... .......... .......... 21%  276M 2s\n",
      " 17100K .......... .......... .......... .......... .......... 22%  554M 2s\n",
      " 17150K .......... .......... .......... .......... .......... 22%  110M 2s\n",
      " 17200K .......... .......... .......... .......... .......... 22%  144M 2s\n",
      " 17250K .......... .......... .......... .......... .......... 22%  326M 2s\n",
      " 17300K .......... .......... .......... .......... .......... 22%  184M 2s\n",
      " 17350K .......... .......... .......... .......... .......... 22%  590M 2s\n",
      " 17400K .......... .......... .......... .......... .......... 22%  279M 2s\n",
      " 17450K .......... .......... .......... .......... .......... 22%  611M 2s\n",
      " 17500K .......... .......... .......... .......... .......... 22%  598M 1s\n",
      " 17550K .......... .......... .......... .......... .......... 22%  572M 1s\n",
      " 17600K .......... .......... .......... .......... .......... 22%  685M 1s\n",
      " 17650K .......... .......... .......... .......... .......... 22%  560M 1s\n",
      " 17700K .......... .......... .......... .......... .......... 22%  689M 1s\n",
      " 17750K .......... .......... .......... .......... .......... 22%  305M 1s\n",
      " 17800K .......... .......... .......... .......... .......... 22%  547M 1s\n",
      " 17850K .......... .......... .......... .......... .......... 23%  672M 1s\n",
      " 17900K .......... .......... .......... .......... .......... 23% 60.9M 1s\n",
      " 17950K .......... .......... .......... .......... .......... 23%  684M 1s\n",
      " 18000K .......... .......... .......... .......... .......... 23% 2.77M 2s\n",
      " 18050K .......... .......... .......... .......... .......... 23%  430M 2s\n",
      " 18100K .......... .......... .......... .......... .......... 23%  123M 1s\n",
      " 18150K .......... .......... .......... .......... .......... 23%  553M 1s\n",
      " 18200K .......... .......... .......... .......... .......... 23% 30.3M 1s\n",
      " 18250K .......... .......... .......... .......... .......... 23%  463M 1s\n",
      " 18300K .......... .......... .......... .......... .......... 23%  513M 1s\n",
      " 18350K .......... .......... .......... .......... .......... 23%  545M 1s\n",
      " 18400K .......... .......... .......... .......... .......... 23%  239M 1s\n",
      " 18450K .......... .......... .......... .......... .......... 23%  230M 1s\n",
      " 18500K .......... .......... .......... .......... .......... 23%  624M 1s\n",
      " 18550K .......... .......... .......... .......... .......... 23%  361M 1s\n",
      " 18600K .......... .......... .......... .......... .......... 23%  440M 1s\n",
      " 18650K .......... .......... .......... .......... .......... 24%  117M 1s\n",
      " 18700K .......... .......... .......... .......... .......... 24%  220M 1s\n",
      " 18750K .......... .......... .......... .......... .......... 24%  362M 1s\n",
      " 18800K .......... .......... .......... .......... .......... 24%  425M 1s\n",
      " 18850K .......... .......... .......... .......... .......... 24%  491M 1s\n",
      " 18900K .......... .......... .......... .......... .......... 24%  315M 1s\n",
      " 18950K .......... .......... .......... .......... .......... 24%  551M 1s\n",
      " 19000K .......... .......... .......... .......... .......... 24%  505M 1s\n",
      " 19050K .......... .......... .......... .......... .......... 24%  656M 1s\n",
      " 19100K .......... .......... .......... .......... .......... 24%  552M 1s\n",
      " 19150K .......... .......... .......... .......... .......... 24%  451M 1s\n",
      " 19200K .......... .......... .......... .......... .......... 24%  677M 1s\n",
      " 19250K .......... .......... .......... .......... .......... 24%  697M 1s\n",
      " 19300K .......... .......... .......... .......... .......... 24%  407M 1s\n",
      " 19350K .......... .......... .......... .......... .......... 24%  185M 1s\n",
      " 19400K .......... .......... .......... .......... .......... 24%  104M 1s\n",
      " 19450K .......... .......... .......... .......... .......... 25%  384M 1s\n",
      " 19500K .......... .......... .......... .......... .......... 25% 2.40M 1s\n",
      " 19550K .......... .......... .......... .......... .......... 25%  355M 1s\n",
      " 19600K .......... .......... .......... .......... .......... 25% 82.4M 1s\n",
      " 19650K .......... .......... .......... .......... .......... 25%  467M 1s\n",
      " 19700K .......... .......... .......... .......... .......... 25%  653M 1s\n",
      " 19750K .......... .......... .......... .......... .......... 25% 27.6M 1s\n",
      " 19800K .......... .......... .......... .......... .......... 25%  492M 1s\n",
      " 19850K .......... .......... .......... .......... .......... 25%  545M 1s\n",
      " 19900K .......... .......... .......... .......... .......... 25%  666M 1s\n",
      " 19950K .......... .......... .......... .......... .......... 25%  198M 1s\n",
      " 20000K .......... .......... .......... .......... .......... 25%  583M 1s\n",
      " 20050K .......... .......... .......... .......... .......... 25%  185M 1s\n",
      " 20100K .......... .......... .......... .......... .......... 25%  652M 1s\n",
      " 20150K .......... .......... .......... .......... .......... 25%  661M 1s\n",
      " 20200K .......... .......... .......... .......... .......... 26%  107M 1s\n",
      " 20250K .......... .......... .......... .......... .......... 26%  133M 1s\n",
      " 20300K .......... .......... .......... .......... .......... 26%  516M 1s\n",
      " 20350K .......... .......... .......... .......... .......... 26%  590M 1s\n",
      " 20400K .......... .......... .......... .......... .......... 26%  687M 1s\n",
      " 20450K .......... .......... .......... .......... .......... 26%  145M 1s\n",
      " 20500K .......... .......... .......... .......... .......... 26%  573M 1s\n",
      " 20550K .......... .......... .......... .......... .......... 26%  489M 1s\n",
      " 20600K .......... .......... .......... .......... .......... 26%  569M 1s\n",
      " 20650K .......... .......... .......... .......... .......... 26%  725M 1s\n",
      " 20700K .......... .......... .......... .......... .......... 26%  307M 1s\n",
      " 20750K .......... .......... .......... .......... .......... 26%  470M 1s\n",
      " 20800K .......... .......... .......... .......... .......... 26%  405M 1s\n",
      " 20850K .......... .......... .......... .......... .......... 26%  573M 1s\n",
      " 20900K .......... .......... .......... .......... .......... 26%  235M 1s\n",
      " 20950K .......... .......... .......... .......... .......... 26% 63.3M 1s\n",
      " 21000K .......... .......... .......... .......... .......... 27%  620M 1s\n",
      " 21050K .......... .......... .......... .......... .......... 27% 2.87M 1s\n",
      " 21100K .......... .......... .......... .......... .......... 27%  581M 1s\n",
      " 21150K .......... .......... .......... .......... .......... 27%  289M 1s\n",
      " 21200K .......... .......... .......... .......... .......... 27%  412M 1s\n",
      " 21250K .......... .......... .......... .......... .......... 27% 21.8M 1s\n",
      " 21300K .......... .......... .......... .......... .......... 27%  447M 1s\n",
      " 21350K .......... .......... .......... .......... .......... 27%  550M 1s\n",
      " 21400K .......... .......... .......... .......... .......... 27%  523M 1s\n",
      " 21450K .......... .......... .......... .......... .......... 27%  584M 1s\n",
      " 21500K .......... .......... .......... .......... .......... 27%  511M 1s\n",
      " 21550K .......... .......... .......... .......... .......... 27%  552M 1s\n",
      " 21600K .......... .......... .......... .......... .......... 27%  203M 1s\n",
      " 21650K .......... .......... .......... .......... .......... 27%  651M 1s\n",
      " 21700K .......... .......... .......... .......... .......... 27%  113M 1s\n",
      " 21750K .......... .......... .......... .......... .......... 28%  513M 1s\n",
      " 21800K .......... .......... .......... .......... .......... 28%  144M 1s\n",
      " 21850K .......... .......... .......... .......... .......... 28%  535M 1s\n",
      " 21900K .......... .......... .......... .......... .......... 28%  685M 1s\n",
      " 21950K .......... .......... .......... .......... .......... 28%  139M 1s\n",
      " 22000K .......... .......... .......... .......... .......... 28%  493M 1s\n",
      " 22050K .......... .......... .......... .......... .......... 28%  549M 1s\n",
      " 22100K .......... .......... .......... .......... .......... 28%  507M 1s\n",
      " 22150K .......... .......... .......... .......... .......... 28%  565M 1s\n",
      " 22200K .......... .......... .......... .......... .......... 28%  622M 1s\n",
      " 22250K .......... .......... .......... .......... .......... 28%  610M 1s\n",
      " 22300K .......... .......... .......... .......... .......... 28%  573M 1s\n",
      " 22350K .......... .......... .......... .......... .......... 28%  480M 1s\n",
      " 22400K .......... .......... .......... .......... .......... 28%  708M 1s\n",
      " 22450K .......... .......... .......... .......... .......... 28% 56.8M 1s\n",
      " 22500K .......... .......... .......... .......... .......... 28%  327M 1s\n",
      " 22550K .......... .......... .......... .......... .......... 29%  217M 1s\n",
      " 22600K .......... .......... .......... .......... .......... 29% 2.82M 1s\n",
      " 22650K .......... .......... .......... .......... .......... 29%  305M 1s\n",
      " 22700K .......... .......... .......... .......... .......... 29%  542M 1s\n",
      " 22750K .......... .......... .......... .......... .......... 29%  572M 1s\n",
      " 22800K .......... .......... .......... .......... .......... 29% 25.0M 1s\n",
      " 22850K .......... .......... .......... .......... .......... 29%  435M 1s\n",
      " 22900K .......... .......... .......... .......... .......... 29%  480M 1s\n",
      " 22950K .......... .......... .......... .......... .......... 29%  558M 1s\n",
      " 23000K .......... .......... .......... .......... .......... 29%  531M 1s\n",
      " 23050K .......... .......... .......... .......... .......... 29%  590M 1s\n",
      " 23100K .......... .......... .......... .......... .......... 29% 81.4M 1s\n",
      " 23150K .......... .......... .......... .......... .......... 29%  606M 1s\n",
      " 23200K .......... .......... .......... .......... .......... 29%  668M 1s\n",
      " 23250K .......... .......... .......... .......... .......... 29% 98.1M 1s\n",
      " 23300K .......... .......... .......... .......... .......... 30% 95.0M 1s\n",
      " 23350K .......... .......... .......... .......... .......... 30%  467M 1s\n",
      " 23400K .......... .......... .......... .......... .......... 30%  640M 1s\n",
      " 23450K .......... .......... .......... .......... .......... 30%  701M 1s\n",
      " 23500K .......... .......... .......... .......... .......... 30%  115M 1s\n",
      " 23550K .......... .......... .......... .......... .......... 30%  608M 1s\n",
      " 23600K .......... .......... .......... .......... .......... 30%  658M 1s\n",
      " 23650K .......... .......... .......... .......... .......... 30%  405M 1s\n",
      " 23700K .......... .......... .......... .......... .......... 30%  654M 1s\n",
      " 23750K .......... .......... .......... .......... .......... 30%  475M 1s\n",
      " 23800K .......... .......... .......... .......... .......... 30%  548M 1s\n",
      " 23850K .......... .......... .......... .......... .......... 30%  488M 1s\n",
      " 23900K .......... .......... .......... .......... .......... 30%  682M 1s\n",
      " 23950K .......... .......... .......... .......... .......... 30%  677M 1s\n",
      " 24000K .......... .......... .......... .......... .......... 30% 45.1M 1s\n",
      " 24050K .......... .......... .......... .......... .......... 30%  175M 1s\n",
      " 24100K .......... .......... .......... .......... .......... 31%  631M 1s\n",
      " 24150K .......... .......... .......... .......... .......... 31% 2.99M 1s\n",
      " 24200K .......... .......... .......... .......... .......... 31%  467M 1s\n",
      " 24250K .......... .......... .......... .......... .......... 31%  196M 1s\n",
      " 24300K .......... .......... .......... .......... .......... 31% 24.8M 1s\n",
      " 24350K .......... .......... .......... .......... .......... 31%  455M 1s\n",
      " 24400K .......... .......... .......... .......... .......... 31%  525M 1s\n",
      " 24450K .......... .......... .......... .......... .......... 31%  464M 1s\n",
      " 24500K .......... .......... .......... .......... .......... 31%  546M 1s\n",
      " 24550K .......... .......... .......... .......... .......... 31%  656M 1s\n",
      " 24600K .......... .......... .......... .......... .......... 31%  673M 1s\n",
      " 24650K .......... .......... .......... .......... .......... 31%  168M 1s\n",
      " 24700K .......... .......... .......... .......... .......... 31%  676M 1s\n",
      " 24750K .......... .......... .......... .......... .......... 31% 92.6M 1s\n",
      " 24800K .......... .......... .......... .......... .......... 31%  593M 1s\n",
      " 24850K .......... .......... .......... .......... .......... 31% 54.6M 1s\n",
      " 24900K .......... .......... .......... .......... .......... 32%  437M 1s\n",
      " 24950K .......... .......... .......... .......... .......... 32%  555M 1s\n",
      " 25000K .......... .......... .......... .......... .......... 32%  684M 1s\n",
      " 25050K .......... .......... .......... .......... .......... 32%  516M 1s\n",
      " 25100K .......... .......... .......... .......... .......... 32%  608M 1s\n",
      " 25150K .......... .......... .......... .......... .......... 32%  597M 1s\n",
      " 25200K .......... .......... .......... .......... .......... 32%  415M 1s\n",
      " 25250K .......... .......... .......... .......... .......... 32%  534M 1s\n",
      " 25300K .......... .......... .......... .......... .......... 32%  568M 1s\n",
      " 25350K .......... .......... .......... .......... .......... 32%  392M 1s\n",
      " 25400K .......... .......... .......... .......... .......... 32%  493M 1s\n",
      " 25450K .......... .......... .......... .......... .......... 32%  534M 1s\n",
      " 25500K .......... .......... .......... .......... .......... 32%  709M 1s\n",
      " 25550K .......... .......... .......... .......... .......... 32% 44.2M 1s\n",
      " 25600K .......... .......... .......... .......... .......... 32%  129M 1s\n",
      " 25650K .......... .......... .......... .......... .......... 33%  642M 1s\n",
      " 25700K .......... .......... .......... .......... .......... 33% 3.00M 1s\n",
      " 25750K .......... .......... .......... .......... .......... 33%  629M 1s\n",
      " 25800K .......... .......... .......... .......... .......... 33%  346M 1s\n",
      " 25850K .......... .......... .......... .......... .......... 33% 24.6M 1s\n",
      " 25900K .......... .......... .......... .......... .......... 33%  384M 1s\n",
      " 25950K .......... .......... .......... .......... .......... 33%  542M 1s\n",
      " 26000K .......... .......... .......... .......... .......... 33%  534M 1s\n",
      " 26050K .......... .......... .......... .......... .......... 33%  658M 1s\n",
      " 26100K .......... .......... .......... .......... .......... 33%  512M 1s\n",
      " 26150K .......... .......... .......... .......... .......... 33%  205M 1s\n",
      " 26200K .......... .......... .......... .......... .......... 33%  646M 1s\n",
      " 26250K .......... .......... .......... .......... .......... 33%  680M 1s\n",
      " 26300K .......... .......... .......... .......... .......... 33% 93.7M 1s\n",
      " 26350K .......... .......... .......... .......... .......... 33% 57.6M 1s\n",
      " 26400K .......... .......... .......... .......... .......... 33%  523M 1s\n",
      " 26450K .......... .......... .......... .......... .......... 34%  520M 1s\n",
      " 26500K .......... .......... .......... .......... .......... 34%  670M 1s\n",
      " 26550K .......... .......... .......... .......... .......... 34%  484M 1s\n",
      " 26600K .......... .......... .......... .......... .......... 34%  422M 1s\n",
      " 26650K .......... .......... .......... .......... .......... 34%  656M 1s\n",
      " 26700K .......... .......... .......... .......... .......... 34%  118M 1s\n",
      " 26750K .......... .......... .......... .......... .......... 34%  574M 1s\n",
      " 26800K .......... .......... .......... .......... .......... 34%  416M 1s\n",
      " 26850K .......... .......... .......... .......... .......... 34%  520M 1s\n",
      " 26900K .......... .......... .......... .......... .......... 34%  691M 1s\n",
      " 26950K .......... .......... .......... .......... .......... 34%  706M 1s\n",
      " 27000K .......... .......... .......... .......... .......... 34%  608M 1s\n",
      " 27050K .......... .......... .......... .......... .......... 34% 55.3M 1s\n",
      " 27100K .......... .......... .......... .......... .......... 34%  177M 1s\n",
      " 27150K .......... .......... .......... .......... .......... 34%  177M 1s\n",
      " 27200K .......... .......... .......... .......... .......... 35%  668M 1s\n",
      " 27250K .......... .......... .......... .......... .......... 35% 2.99M 1s\n",
      " 27300K .......... .......... .......... .......... .......... 35%  196M 1s\n",
      " 27350K .......... .......... .......... .......... .......... 35% 39.2M 1s\n",
      " 27400K .......... .......... .......... .......... .......... 35% 84.6M 1s\n",
      " 27450K .......... .......... .......... .......... .......... 35%  170M 1s\n",
      " 27500K .......... .......... .......... .......... .......... 35%  703M 1s\n",
      " 27550K .......... .......... .......... .......... .......... 35%  537M 1s\n",
      " 27600K .......... .......... .......... .......... .......... 35%  524M 1s\n",
      " 27650K .......... .......... .......... .......... .......... 35%  711M 1s\n",
      " 27700K .......... .......... .......... .......... .......... 35% 88.7M 1s\n",
      " 27750K .......... .......... .......... .......... .......... 35%  632M 1s\n",
      " 27800K .......... .......... .......... .......... .......... 35%  565M 1s\n",
      " 27850K .......... .......... .......... .......... .......... 35%  665M 1s\n",
      " 27900K .......... .......... .......... .......... .......... 35% 56.3M 1s\n",
      " 27950K .......... .......... .......... .......... .......... 35%  434M 1s\n",
      " 28000K .......... .......... .......... .......... .......... 36%  457M 1s\n",
      " 28050K .......... .......... .......... .......... .......... 36%  650M 1s\n",
      " 28100K .......... .......... .......... .......... .......... 36%  361M 1s\n",
      " 28150K .......... .......... .......... .......... .......... 36%  394M 1s\n",
      " 28200K .......... .......... .......... .......... .......... 36%  115M 1s\n",
      " 28250K .......... .......... .......... .......... .......... 36%  518M 1s\n",
      " 28300K .......... .......... .......... .......... .......... 36%  519M 1s\n",
      " 28350K .......... .......... .......... .......... .......... 36%  677M 1s\n",
      " 28400K .......... .......... .......... .......... .......... 36%  480M 1s\n",
      " 28450K .......... .......... .......... .......... .......... 36%  488M 1s\n",
      " 28500K .......... .......... .......... .......... .......... 36%  684M 1s\n",
      " 28550K .......... .......... .......... .......... .......... 36%  731M 1s\n",
      " 28600K .......... .......... .......... .......... .......... 36% 53.4M 1s\n",
      " 28650K .......... .......... .......... .......... .......... 36%  141M 1s\n",
      " 28700K .......... .......... .......... .......... .......... 36%  295M 1s\n",
      " 28750K .......... .......... .......... .......... .......... 37% 2.97M 1s\n",
      " 28800K .......... .......... .......... .......... .......... 37%  397M 1s\n",
      " 28850K .......... .......... .......... .......... .......... 37%  543M 1s\n",
      " 28900K .......... .......... .......... .......... .......... 37% 38.2M 1s\n",
      " 28950K .......... .......... .......... .......... .......... 37% 44.3M 1s\n",
      " 29000K .......... .......... .......... .......... .......... 37%  498M 1s\n",
      " 29050K .......... .......... .......... .......... .......... 37%  664M 1s\n",
      " 29100K .......... .......... .......... .......... .......... 37%  658M 1s\n",
      " 29150K .......... .......... .......... .......... .......... 37%  533M 1s\n",
      " 29200K .......... .......... .......... .......... .......... 37%  479M 1s\n",
      " 29250K .......... .......... .......... .......... .......... 37%  568M 1s\n",
      " 29300K .......... .......... .......... .......... .......... 37%  669M 1s\n",
      " 29350K .......... .......... .......... .......... .......... 37%  611M 1s\n",
      " 29400K .......... .......... .......... .......... .......... 37% 57.0M 1s\n",
      " 29450K .......... .......... .......... .......... .......... 37% 99.2M 1s\n",
      " 29500K .......... .......... .......... .......... .......... 37%  626M 1s\n",
      " 29550K .......... .......... .......... .......... .......... 38%  666M 1s\n",
      " 29600K .......... .......... .......... .......... .......... 38%  497M 1s\n",
      " 29650K .......... .......... .......... .......... .......... 38%  402M 1s\n",
      " 29700K .......... .......... .......... .......... .......... 38%  642M 1s\n",
      " 29750K .......... .......... .......... .......... .......... 38%  480M 1s\n",
      " 29800K .......... .......... .......... .......... .......... 38%  116M 1s\n",
      " 29850K .......... .......... .......... .......... .......... 38%  443M 1s\n",
      " 29900K .......... .......... .......... .......... .......... 38%  536M 1s\n",
      " 29950K .......... .......... .......... .......... .......... 38%  530M 1s\n",
      " 30000K .......... .......... .......... .......... .......... 38%  702M 1s\n",
      " 30050K .......... .......... .......... .......... .......... 38%  562M 1s\n",
      " 30100K .......... .......... .......... .......... .......... 38% 73.3M 1s\n",
      " 30150K .......... .......... .......... .......... .......... 38% 76.4M 1s\n",
      " 30200K .......... .......... .......... .......... .......... 38%  473M 1s\n",
      " 30250K .......... .......... .......... .......... .......... 38%  121M 1s\n",
      " 30300K .......... .......... .......... .......... .......... 39% 3.04M 1s\n",
      " 30350K .......... .......... .......... .......... .......... 39%  372M 1s\n",
      " 30400K .......... .......... .......... .......... .......... 39%  682M 1s\n",
      " 30450K .......... .......... .......... .......... .......... 39% 34.1M 1s\n",
      " 30500K .......... .......... .......... .......... .......... 39% 88.6M 1s\n",
      " 30550K .......... .......... .......... .......... .......... 39%  582M 1s\n",
      " 30600K .......... .......... .......... .......... .......... 39%  635M 1s\n",
      " 30650K .......... .......... .......... .......... .......... 39%  466M 1s\n",
      " 30700K .......... .......... .......... .......... .......... 39%  656M 1s\n",
      " 30750K .......... .......... .......... .......... .......... 39% 77.3M 1s\n",
      " 30800K .......... .......... .......... .......... .......... 39%  608M 1s\n",
      " 30850K .......... .......... .......... .......... .......... 39%  669M 1s\n",
      " 30900K .......... .......... .......... .......... .......... 39%  649M 1s\n",
      " 30950K .......... .......... .......... .......... .......... 39% 54.9M 1s\n",
      " 31000K .......... .......... .......... .......... .......... 39% 92.6M 1s\n",
      " 31050K .......... .......... .......... .......... .......... 39%  604M 1s\n",
      " 31100K .......... .......... .......... .......... .......... 40%  640M 1s\n",
      " 31150K .......... .......... .......... .......... .......... 40%  508M 1s\n",
      " 31200K .......... .......... .......... .......... .......... 40%  113M 1s\n",
      " 31250K .......... .......... .......... .......... .......... 40%  629M 1s\n",
      " 31300K .......... .......... .......... .......... .......... 40%  496M 1s\n",
      " 31350K .......... .......... .......... .......... .......... 40%  566M 1s\n",
      " 31400K .......... .......... .......... .......... .......... 40%  370M 1s\n",
      " 31450K .......... .......... .......... .......... .......... 40%  457M 1s\n",
      " 31500K .......... .......... .......... .......... .......... 40% 82.7M 1s\n",
      " 31550K .......... .......... .......... .......... .......... 40%  617M 1s\n",
      " 31600K .......... .......... .......... .......... .......... 40%  658M 1s\n",
      " 31650K .......... .......... .......... .......... .......... 40%  134M 1s\n",
      " 31700K .......... .......... .......... .......... .......... 40%  127M 1s\n",
      " 31750K .......... .......... .......... .......... .......... 40%  164M 1s\n",
      " 31800K .......... .......... .......... .......... .......... 40%  458M 1s\n",
      " 31850K .......... .......... .......... .......... .......... 40% 2.99M 1s\n",
      " 31900K .......... .......... .......... .......... .......... 41%  441M 1s\n",
      " 31950K .......... .......... .......... .......... .......... 41% 45.7M 1s\n",
      " 32000K .......... .......... .......... .......... .......... 41% 90.2M 1s\n",
      " 32050K .......... .......... .......... .......... .......... 41%  437M 1s\n",
      " 32100K .......... .......... .......... .......... .......... 41%  580M 1s\n",
      " 32150K .......... .......... .......... .......... .......... 41%  700M 1s\n",
      " 32200K .......... .......... .......... .......... .......... 41% 88.5M 1s\n",
      " 32250K .......... .......... .......... .......... .......... 41%  589M 1s\n",
      " 32300K .......... .......... .......... .......... .......... 41%  515M 1s\n",
      " 32350K .......... .......... .......... .......... .......... 41%  666M 1s\n",
      " 32400K .......... .......... .......... .......... .......... 41% 95.6M 1s\n",
      " 32450K .......... .......... .......... .......... .......... 41% 54.6M 1s\n",
      " 32500K .......... .......... .......... .......... .......... 41%  608M 1s\n",
      " 32550K .......... .......... .......... .......... .......... 41%  472M 1s\n",
      " 32600K .......... .......... .......... .......... .......... 41%  588M 1s\n",
      " 32650K .......... .......... .......... .......... .......... 42%  587M 1s\n",
      " 32700K .......... .......... .......... .......... .......... 42% 96.7M 1s\n",
      " 32750K .......... .......... .......... .......... .......... 42%  595M 1s\n",
      " 32800K .......... .......... .......... .......... .......... 42%  665M 1s\n",
      " 32850K .......... .......... .......... .......... .......... 42%  563M 1s\n",
      " 32900K .......... .......... .......... .......... .......... 42%  514M 1s\n",
      " 32950K .......... .......... .......... .......... .......... 42%  640M 1s\n",
      " 33000K .......... .......... .......... .......... .......... 42%  607M 1s\n",
      " 33050K .......... .......... .......... .......... .......... 42% 89.3M 1s\n",
      " 33100K .......... .......... .......... .......... .......... 42%  609M 1s\n",
      " 33150K .......... .......... .......... .......... .......... 42%  126M 1s\n",
      " 33200K .......... .......... .......... .......... .......... 42%  607M 1s\n",
      " 33250K .......... .......... .......... .......... .......... 42%  102M 1s\n",
      " 33300K .......... .......... .......... .......... .......... 42%  144M 1s\n",
      " 33350K .......... .......... .......... .......... .......... 42%  639M 1s\n",
      " 33400K .......... .......... .......... .......... .......... 42% 3.03M 1s\n",
      " 33450K .......... .......... .......... .......... .......... 43%  206M 1s\n",
      " 33500K .......... .......... .......... .......... .......... 43% 45.8M 1s\n",
      " 33550K .......... .......... .......... .......... .......... 43% 87.4M 1s\n",
      " 33600K .......... .......... .......... .......... .......... 43%  434M 1s\n",
      " 33650K .......... .......... .......... .......... .......... 43%  554M 1s\n",
      " 33700K .......... .......... .......... .......... .......... 43%  577M 1s\n",
      " 33750K .......... .......... .......... .......... .......... 43% 76.3M 1s\n",
      " 33800K .......... .......... .......... .......... .......... 43%  484M 1s\n",
      " 33850K .......... .......... .......... .......... .......... 43%  580M 1s\n",
      " 33900K .......... .......... .......... .......... .......... 43%  585M 1s\n",
      " 33950K .......... .......... .......... .......... .......... 43%  701M 1s\n",
      " 34000K .......... .......... .......... .......... .......... 43% 57.9M 1s\n",
      " 34050K .......... .......... .......... .......... .......... 43% 55.8M 1s\n",
      " 34100K .......... .......... .......... .......... .......... 43%  600M 1s\n",
      " 34150K .......... .......... .......... .......... .......... 43%  677M 1s\n",
      " 34200K .......... .......... .......... .......... .......... 44%  656M 1s\n",
      " 34250K .......... .......... .......... .......... .......... 44%  468M 1s\n",
      " 34300K .......... .......... .......... .......... .......... 44%  515M 1s\n",
      " 34350K .......... .......... .......... .......... .......... 44% 99.8M 1s\n",
      " 34400K .......... .......... .......... .......... .......... 44%  564M 1s\n",
      " 34450K .......... .......... .......... .......... .......... 44%  555M 1s\n",
      " 34500K .......... .......... .......... .......... .......... 44%  684M 1s\n",
      " 34550K .......... .......... .......... .......... .......... 44%  548M 1s\n",
      " 34600K .......... .......... .......... .......... .......... 44%  505M 1s\n",
      " 34650K .......... .......... .......... .......... .......... 44%  575M 1s\n",
      " 34700K .......... .......... .......... .......... .......... 44% 93.7M 1s\n",
      " 34750K .......... .......... .......... .......... .......... 44% 70.5M 1s\n",
      " 34800K .......... .......... .......... .......... .......... 44%  148M 1s\n",
      " 34850K .......... .......... .......... .......... .......... 44%  521M 1s\n",
      " 34900K .......... .......... .......... .......... .......... 44%  678M 1s\n",
      " 34950K .......... .......... .......... .......... .......... 44% 2.98M 1s\n",
      " 35000K .......... .......... .......... .......... .......... 45% 77.0M 1s\n",
      " 35050K .......... .......... .......... .......... .......... 45% 89.5M 1s\n",
      " 35100K .......... .......... .......... .......... .......... 45%  465M 1s\n",
      " 35150K .......... .......... .......... .......... .......... 45%  490M 1s\n",
      " 35200K .......... .......... .......... .......... .......... 45%  665M 1s\n",
      " 35250K .......... .......... .......... .......... .......... 45% 86.5M 1s\n",
      " 35300K .......... .......... .......... .......... .......... 45%  617M 1s\n",
      " 35350K .......... .......... .......... .......... .......... 45% 94.1M 1s\n",
      " 35400K .......... .......... .......... .......... .......... 45%  518M 1s\n",
      " 35450K .......... .......... .......... .......... .......... 45%  647M 1s\n",
      " 35500K .......... .......... .......... .......... .......... 45% 53.8M 1s\n",
      " 35550K .......... .......... .......... .......... .......... 45%  557M 1s\n",
      " 35600K .......... .......... .......... .......... .......... 45% 26.7M 1s\n",
      " 35650K .......... .......... .......... .......... .......... 45%  177M 1s\n",
      " 35700K .......... .......... .......... .......... .......... 45%  644M 1s\n",
      " 35750K .......... .......... .......... .......... .......... 46% 70.4M 1s\n",
      " 35800K .......... .......... .......... .......... .......... 46%  564M 1s\n",
      " 35850K .......... .......... .......... .......... .......... 46%  507M 1s\n",
      " 35900K .......... .......... .......... .......... .......... 46%  163M 1s\n",
      " 35950K .......... .......... .......... .......... .......... 46%  542M 1s\n",
      " 36000K .......... .......... .......... .......... .......... 46%  500M 1s\n",
      " 36050K .......... .......... .......... .......... .......... 46%  659M 1s\n",
      " 36100K .......... .......... .......... .......... .......... 46% 75.3M 1s\n",
      " 36150K .......... .......... .......... .......... .......... 46%  526M 1s\n",
      " 36200K .......... .......... .......... .......... .......... 46%  153M 1s\n",
      " 36250K .......... .......... .......... .......... .......... 46%  599M 1s\n",
      " 36300K .......... .......... .......... .......... .......... 46% 54.1M 1s\n",
      " 36350K .......... .......... .......... .......... .......... 46% 92.8M 1s\n",
      " 36400K .......... .......... .......... .......... .......... 46%  480M 1s\n",
      " 36450K .......... .......... .......... .......... .......... 46%  679M 1s\n",
      " 36500K .......... .......... .......... .......... .......... 46% 3.53M 1s\n",
      " 36550K .......... .......... .......... .......... .......... 47% 66.0M 1s\n",
      " 36600K .......... .......... .......... .......... .......... 47% 33.1M 1s\n",
      " 36650K .......... .......... .......... .......... .......... 47%  439M 1s\n",
      " 36700K .......... .......... .......... .......... .......... 47%  673M 1s\n",
      " 36750K .......... .......... .......... .......... .......... 47%  420M 1s\n",
      " 36800K .......... .......... .......... .......... .......... 47%  525M 1s\n",
      " 36850K .......... .......... .......... .......... .......... 47%  665M 1s\n",
      " 36900K .......... .......... .......... .......... .......... 47%  622M 1s\n",
      " 36950K .......... .......... .......... .......... .......... 47%  505M 1s\n",
      " 37000K .......... .......... .......... .......... .......... 47%  665M 1s\n",
      " 37050K .......... .......... .......... .......... .......... 47% 76.7M 1s\n",
      " 37100K .......... .......... .......... .......... .......... 47% 27.9M 1s\n",
      " 37150K .......... .......... .......... .......... .......... 47% 94.5M 1s\n",
      " 37200K .......... .......... .......... .......... .......... 47%  543M 1s\n",
      " 37250K .......... .......... .......... .......... .......... 47%  561M 1s\n",
      " 37300K .......... .......... .......... .......... .......... 47% 88.2M 1s\n",
      " 37350K .......... .......... .......... .......... .......... 48% 94.4M 1s\n",
      " 37400K .......... .......... .......... .......... .......... 48%  392M 1s\n",
      " 37450K .......... .......... .......... .......... .......... 48%  516M 1s\n",
      " 37500K .......... .......... .......... .......... .......... 48%  670M 1s\n",
      " 37550K .......... .......... .......... .......... .......... 48%  700M 1s\n",
      " 37600K .......... .......... .......... .......... .......... 48%  448M 1s\n",
      " 37650K .......... .......... .......... .......... .......... 48%  620M 1s\n",
      " 37700K .......... .......... .......... .......... .......... 48%  115M 1s\n",
      " 37750K .......... .......... .......... .......... .......... 48%  495M 1s\n",
      " 37800K .......... .......... .......... .......... .......... 48% 54.1M 1s\n",
      " 37850K .......... .......... .......... .......... .......... 48% 70.7M 1s\n",
      " 37900K .......... .......... .......... .......... .......... 48%  467M 1s\n",
      " 37950K .......... .......... .......... .......... .......... 48%  633M 1s\n",
      " 38000K .......... .......... .......... .......... .......... 48% 3.46M 1s\n",
      " 38050K .......... .......... .......... .......... .......... 48%  426M 1s\n",
      " 38100K .......... .......... .......... .......... .......... 49% 28.6M 1s\n",
      " 38150K .......... .......... .......... .......... .......... 49%  415M 1s\n",
      " 38200K .......... .......... .......... .......... .......... 49%  470M 1s\n",
      " 38250K .......... .......... .......... .......... .......... 49%  644M 1s\n",
      " 38300K .......... .......... .......... .......... .......... 49%  451M 1s\n",
      " 38350K .......... .......... .......... .......... .......... 49%  522M 1s\n",
      " 38400K .......... .......... .......... .......... .......... 49%  570M 1s\n",
      " 38450K .......... .......... .......... .......... .......... 49%  496M 1s\n",
      " 38500K .......... .......... .......... .......... .......... 49%  574M 1s\n",
      " 38550K .......... .......... .......... .......... .......... 49% 82.6M 1s\n",
      " 38600K .......... .......... .......... .......... .......... 49%  531M 1s\n",
      " 38650K .......... .......... .......... .......... .......... 49% 27.5M 1s\n",
      " 38700K .......... .......... .......... .......... .......... 49% 93.4M 1s\n",
      " 38750K .......... .......... .......... .......... .......... 49%  620M 1s\n",
      " 38800K .......... .......... .......... .......... .......... 49%  100M 1s\n",
      " 38850K .......... .......... .......... .......... .......... 49%  399M 1s\n",
      " 38900K .......... .......... .......... .......... .......... 50%  354M 1s\n",
      " 38950K .......... .......... .......... .......... .......... 50% 95.3M 1s\n",
      " 39000K .......... .......... .......... .......... .......... 50%  420M 1s\n",
      " 39050K .......... .......... .......... .......... .......... 50%  541M 1s\n",
      " 39100K .......... .......... .......... .......... .......... 50%  697M 1s\n",
      " 39150K .......... .......... .......... .......... .......... 50%  606M 1s\n",
      " 39200K .......... .......... .......... .......... .......... 50%  592M 1s\n",
      " 39250K .......... .......... .......... .......... .......... 50% 93.2M 1s\n",
      " 39300K .......... .......... .......... .......... .......... 50%  557M 1s\n",
      " 39350K .......... .......... .......... .......... .......... 50% 67.4M 1s\n",
      " 39400K .......... .......... .......... .......... .......... 50% 67.5M 1s\n",
      " 39450K .......... .......... .......... .......... .......... 50%  445M 1s\n",
      " 39500K .......... .......... .......... .......... .......... 50%  676M 1s\n",
      " 39550K .......... .......... .......... .......... .......... 50% 3.42M 1s\n",
      " 39600K .......... .......... .......... .......... .......... 50%  442M 1s\n",
      " 39650K .......... .......... .......... .......... .......... 51% 29.3M 1s\n",
      " 39700K .......... .......... .......... .......... .......... 51%  369M 1s\n",
      " 39750K .......... .......... .......... .......... .......... 51%  515M 1s\n",
      " 39800K .......... .......... .......... .......... .......... 51%  537M 1s\n",
      " 39850K .......... .......... .......... .......... .......... 51%  667M 1s\n",
      " 39900K .......... .......... .......... .......... .......... 51%  574M 1s\n",
      " 39950K .......... .......... .......... .......... .......... 51%  487M 1s\n",
      " 40000K .......... .......... .......... .......... .......... 51%  614M 1s\n",
      " 40050K .......... .......... .......... .......... .......... 51%  655M 1s\n",
      " 40100K .......... .......... .......... .......... .......... 51% 79.3M 1s\n",
      " 40150K .......... .......... .......... .......... .......... 51% 28.1M 1s\n",
      " 40200K .......... .......... .......... .......... .......... 51% 43.2M 1s\n",
      " 40250K .......... .......... .......... .......... .......... 51%  440M 1s\n",
      " 40300K .......... .......... .......... .......... .......... 51%  635M 1s\n",
      " 40350K .......... .......... .......... .......... .......... 51% 32.8M 1s\n",
      " 40400K .......... .......... .......... .......... .......... 51% 55.2M 1s\n",
      " 40450K .......... .......... .......... .......... .......... 52%  454M 1s\n",
      " 40500K .......... .......... .......... .......... .......... 52%  530M 1s\n",
      " 40550K .......... .......... .......... .......... .......... 52%  430M 1s\n",
      " 40600K .......... .......... .......... .......... .......... 52%  613M 1s\n",
      " 40650K .......... .......... .......... .......... .......... 52%  571M 1s\n",
      " 40700K .......... .......... .......... .......... .......... 52%  670M 1s\n",
      " 40750K .......... .......... .......... .......... .......... 52%  469M 1s\n",
      " 40800K .......... .......... .......... .......... .......... 52% 46.3M 1s\n",
      " 40850K .......... .......... .......... .......... .......... 52% 27.4M 1s\n",
      " 40900K .......... .......... .......... .......... .......... 52% 48.4M 1s\n",
      " 40950K .......... .......... .......... .......... .......... 52% 2.03M 1s\n",
      " 41000K .......... .......... .......... .......... .......... 52% 40.4M 1s\n",
      " 41050K .......... .......... .......... .......... .......... 52% 4.74M 1s\n",
      " 41100K .......... .......... .......... .......... .......... 52%  438M 1s\n",
      " 41150K .......... .......... .......... .......... .......... 52%  642M 1s\n",
      " 41200K .......... .......... .......... .......... .......... 53%  626M 1s\n",
      " 41250K .......... .......... .......... .......... .......... 53%  584M 1s\n",
      " 41300K .......... .......... .......... .......... .......... 53%  537M 1s\n",
      " 41350K .......... .......... .......... .......... .......... 53%  626M 1s\n",
      " 41400K .......... .......... .......... .......... .......... 53%  626M 1s\n",
      " 41450K .......... .......... .......... .......... .......... 53%  641M 1s\n",
      " 41500K .......... .......... .......... .......... .......... 53%  531M 1s\n",
      " 41550K .......... .......... .......... .......... .......... 53%  637M 1s\n",
      " 41600K .......... .......... .......... .......... .......... 53%  629M 1s\n",
      " 41650K .......... .......... .......... .......... .......... 53%  632M 1s\n",
      " 41700K .......... .......... .......... .......... .......... 53%  554M 1s\n",
      " 41750K .......... .......... .......... .......... .......... 53%  630M 1s\n",
      " 41800K .......... .......... .......... .......... .......... 53%  645M 1s\n",
      " 41850K .......... .......... .......... .......... .......... 53%  595M 1s\n",
      " 41900K .......... .......... .......... .......... .......... 53%  543M 1s\n",
      " 41950K .......... .......... .......... .......... .......... 53%  635M 1s\n",
      " 42000K .......... .......... .......... .......... .......... 54%  629M 1s\n",
      " 42050K .......... .......... .......... .......... .......... 54%  641M 1s\n",
      " 42100K .......... .......... .......... .......... .......... 54%  562M 1s\n",
      " 42150K .......... .......... .......... .......... .......... 54%  631M 1s\n",
      " 42200K .......... .......... .......... .......... .......... 54%  658M 1s\n",
      " 42250K .......... .......... .......... .......... .......... 54%  620M 1s\n",
      " 42300K .......... .......... .......... .......... .......... 54%  549M 1s\n",
      " 42350K .......... .......... .......... .......... .......... 54%  644M 1s\n",
      " 42400K .......... .......... .......... .......... .......... 54%  659M 1s\n",
      " 42450K .......... .......... .......... .......... .......... 54%  609M 1s\n",
      " 42500K .......... .......... .......... .......... .......... 54% 4.25M 1s\n",
      " 42550K .......... .......... .......... .......... .......... 54% 31.9M 1s\n",
      " 42600K .......... .......... .......... .......... .......... 54% 5.66M 1s\n",
      " 42650K .......... .......... .......... .......... .......... 54%  432M 1s\n",
      " 42700K .......... .......... .......... .......... .......... 54%  501M 1s\n",
      " 42750K .......... .......... .......... .......... .......... 55%  690M 1s\n",
      " 42800K .......... .......... .......... .......... .......... 55%  214M 1s\n",
      " 42850K .......... .......... .......... .......... .......... 55%  480M 1s\n",
      " 42900K .......... .......... .......... .......... .......... 55%  551M 1s\n",
      " 42950K .......... .......... .......... .......... .......... 55%  618M 1s\n",
      " 43000K .......... .......... .......... .......... .......... 55%  514M 1s\n",
      " 43050K .......... .......... .......... .......... .......... 55%  711M 1s\n",
      " 43100K .......... .......... .......... .......... .......... 55%  407M 1s\n",
      " 43150K .......... .......... .......... .......... .......... 55%  608M 1s\n",
      " 43200K .......... .......... .......... .......... .......... 55%  562M 1s\n",
      " 43250K .......... .......... .......... .......... .......... 55%  724M 1s\n",
      " 43300K .......... .......... .......... .......... .......... 55%  498M 1s\n",
      " 43350K .......... .......... .......... .......... .......... 55%  522M 1s\n",
      " 43400K .......... .......... .......... .......... .......... 55%  549M 1s\n",
      " 43450K .......... .......... .......... .......... .......... 55%  690M 1s\n",
      " 43500K .......... .......... .......... .......... .......... 55%  493M 1s\n",
      " 43550K .......... .......... .......... .......... .......... 56%  520M 1s\n",
      " 43600K .......... .......... .......... .......... .......... 56%  685M 1s\n",
      " 43650K .......... .......... .......... .......... .......... 56%  565M 1s\n",
      " 43700K .......... .......... .......... .......... .......... 56%  604M 1s\n",
      " 43750K .......... .......... .......... .......... .......... 56%  688M 1s\n",
      " 43800K .......... .......... .......... .......... .......... 56%  689M 1s\n",
      " 43850K .......... .......... .......... .......... .......... 56%  656M 1s\n",
      " 43900K .......... .......... .......... .......... .......... 56%  542M 1s\n",
      " 43950K .......... .......... .......... .......... .......... 56%  703M 1s\n",
      " 44000K .......... .......... .......... .......... .......... 56% 4.30M 1s\n",
      " 44050K .......... .......... .......... .......... .......... 56% 36.1M 1s\n",
      " 44100K .......... .......... .......... .......... .......... 56% 5.23M 1s\n",
      " 44150K .......... .......... .......... .......... .......... 56%  461M 1s\n",
      " 44200K .......... .......... .......... .......... .......... 56%  647M 1s\n",
      " 44250K .......... .......... .......... .......... .......... 56%  575M 1s\n",
      " 44300K .......... .......... .......... .......... .......... 56%  416M 1s\n",
      " 44350K .......... .......... .......... .......... .......... 57%  703M 1s\n",
      " 44400K .......... .......... .......... .......... .......... 57%  544M 1s\n",
      " 44450K .......... .......... .......... .......... .......... 57%  598M 1s\n",
      " 44500K .......... .......... .......... .......... .......... 57%  422M 1s\n",
      " 44550K .......... .......... .......... .......... .......... 57%  640M 1s\n",
      " 44600K .......... .......... .......... .......... .......... 57%  491M 1s\n",
      " 44650K .......... .......... .......... .......... .......... 57%  473M 1s\n",
      " 44700K .......... .......... .......... .......... .......... 57%  527M 1s\n",
      " 44750K .......... .......... .......... .......... .......... 57%  694M 1s\n",
      " 44800K .......... .......... .......... .......... .......... 57%  427M 1s\n",
      " 44850K .......... .......... .......... .......... .......... 57%  479M 1s\n",
      " 44900K .......... .......... .......... .......... .......... 57%  670M 1s\n",
      " 44950K .......... .......... .......... .......... .......... 57% 19.4M 1s\n",
      " 45000K .......... .......... .......... .......... .......... 57%  539M 1s\n",
      " 45050K .......... .......... .......... .......... .......... 57%  641M 1s\n",
      " 45100K .......... .......... .......... .......... .......... 58%  671M 1s\n",
      " 45150K .......... .......... .......... .......... .......... 58%  683M 1s\n",
      " 45200K .......... .......... .......... .......... .......... 58%  577M 1s\n",
      " 45250K .......... .......... .......... .......... .......... 58%  661M 1s\n",
      " 45300K .......... .......... .......... .......... .......... 58%  648M 1s\n",
      " 45350K .......... .......... .......... .......... .......... 58%  681M 1s\n",
      " 45400K .......... .......... .......... .......... .......... 58%  565M 1s\n",
      " 45450K .......... .......... .......... .......... .......... 58%  681M 1s\n",
      " 45500K .......... .......... .......... .......... .......... 58% 5.02M 1s\n",
      " 45550K .......... .......... .......... .......... .......... 58% 5.40M 1s\n",
      " 45600K .......... .......... .......... .......... .......... 58%  479M 1s\n",
      " 45650K .......... .......... .......... .......... .......... 58%  278M 1s\n",
      " 45700K .......... .......... .......... .......... .......... 58%  434M 1s\n",
      " 45750K .......... .......... .......... .......... .......... 58%  677M 1s\n",
      " 45800K .......... .......... .......... .......... .......... 58%  543M 1s\n",
      " 45850K .......... .......... .......... .......... .......... 58%  443M 1s\n",
      " 45900K .......... .......... .......... .......... .......... 59%  691M 1s\n",
      " 45950K .......... .......... .......... .......... .......... 59%  556M 1s\n",
      " 46000K .......... .......... .......... .......... .......... 59%  395M 1s\n",
      " 46050K .......... .......... .......... .......... .......... 59%  488M 1s\n",
      " 46100K .......... .......... .......... .......... .......... 59%  562M 1s\n",
      " 46150K .......... .......... .......... .......... .......... 59%  672M 1s\n",
      " 46200K .......... .......... .......... .......... .......... 59%  473M 1s\n",
      " 46250K .......... .......... .......... .......... .......... 59%  636M 1s\n",
      " 46300K .......... .......... .......... .......... .......... 59%  488M 1s\n",
      " 46350K .......... .......... .......... .......... .......... 59%  580M 1s\n",
      " 46400K .......... .......... .......... .......... .......... 59%  585M 1s\n",
      " 46450K .......... .......... .......... .......... .......... 59%  519M 1s\n",
      " 46500K .......... .......... .......... .......... .......... 59%  703M 1s\n",
      " 46550K .......... .......... .......... .......... .......... 59%  365M 1s\n",
      " 46600K .......... .......... .......... .......... .......... 59%  668M 1s\n",
      " 46650K .......... .......... .......... .......... .......... 60%  319M 1s\n",
      " 46700K .......... .......... .......... .......... .......... 60%  526M 1s\n",
      " 46750K .......... .......... .......... .......... .......... 60%  616M 1s\n",
      " 46800K .......... .......... .......... .......... .......... 60%  434M 1s\n",
      " 46850K .......... .......... .......... .......... .......... 60%  679M 1s\n",
      " 46900K .......... .......... .......... .......... .......... 60% 13.3M 1s\n",
      " 46950K .......... .......... .......... .......... .......... 60% 7.00M 1s\n",
      " 47000K .......... .......... .......... .......... .......... 60%  392M 1s\n",
      " 47050K .......... .......... .......... .......... .......... 60% 17.7M 1s\n",
      " 47100K .......... .......... .......... .......... .......... 60% 6.30M 1s\n",
      " 47150K .......... .......... .......... .......... .......... 60%  434M 1s\n",
      " 47200K .......... .......... .......... .......... .......... 60%  406M 1s\n",
      " 47250K .......... .......... .......... .......... .......... 60%  482M 1s\n",
      " 47300K .......... .......... .......... .......... .......... 60%  683M 1s\n",
      " 47350K .......... .......... .......... .......... .......... 60%  474M 1s\n",
      " 47400K .......... .......... .......... .......... .......... 60%  544M 1s\n",
      " 47450K .......... .......... .......... .......... .......... 61%  584M 1s\n",
      " 47500K .......... .......... .......... .......... .......... 61%  527M 1s\n",
      " 47550K .......... .......... .......... .......... .......... 61%  696M 1s\n",
      " 47600K .......... .......... .......... .......... .......... 61%  441M 1s\n",
      " 47650K .......... .......... .......... .......... .......... 61%  598M 1s\n",
      " 47700K .......... .......... .......... .......... .......... 61%  552M 1s\n",
      " 47750K .......... .......... .......... .......... .......... 61%  508M 1s\n",
      " 47800K .......... .......... .......... .......... .......... 61%  392M 1s\n",
      " 47850K .......... .......... .......... .......... .......... 61%  544M 1s\n",
      " 47900K .......... .......... .......... .......... .......... 61%  678M 1s\n",
      " 47950K .......... .......... .......... .......... .......... 61%  688M 1s\n",
      " 48000K .......... .......... .......... .......... .......... 61%  580M 1s\n",
      " 48050K .......... .......... .......... .......... .......... 61%  471M 1s\n",
      " 48100K .......... .......... .......... .......... .......... 61%  572M 1s\n",
      " 48150K .......... .......... .......... .......... .......... 61%  724M 1s\n",
      " 48200K .......... .......... .......... .......... .......... 62%  450M 1s\n",
      " 48250K .......... .......... .......... .......... .......... 62%  679M 1s\n",
      " 48300K .......... .......... .......... .......... .......... 62%  512M 1s\n",
      " 48350K .......... .......... .......... .......... .......... 62%  482M 1s\n",
      " 48400K .......... .......... .......... .......... .......... 62%  623M 1s\n",
      " 48450K .......... .......... .......... .......... .......... 62% 14.4M 1s\n",
      " 48500K .......... .......... .......... .......... .......... 62% 6.20M 1s\n",
      " 48550K .......... .......... .......... .......... .......... 62% 21.7M 1s\n",
      " 48600K .......... .......... .......... .......... .......... 62% 47.0M 1s\n",
      " 48650K .......... .......... .......... .......... .......... 62% 6.51M 1s\n",
      " 48700K .......... .......... .......... .......... .......... 62%  387M 1s\n",
      " 48750K .......... .......... .......... .......... .......... 62%  599M 1s\n",
      " 48800K .......... .......... .......... .......... .......... 62%  633M 1s\n",
      " 48850K .......... .......... .......... .......... .......... 62%  422M 1s\n",
      " 48900K .......... .......... .......... .......... .......... 62%  513M 1s\n",
      " 48950K .......... .......... .......... .......... .......... 62%  518M 1s\n",
      " 49000K .......... .......... .......... .......... .......... 63%  601M 1s\n",
      " 49050K .......... .......... .......... .......... .......... 63%  653M 1s\n",
      " 49100K .......... .......... .......... .......... .......... 63%  483M 1s\n",
      " 49150K .......... .......... .......... .......... .......... 63%  586M 1s\n",
      " 49200K .......... .......... .......... .......... .......... 63%  603M 1s\n",
      " 49250K .......... .......... .......... .......... .......... 63%  515M 1s\n",
      " 49300K .......... .......... .......... .......... .......... 63%  448M 1s\n",
      " 49350K .......... .......... .......... .......... .......... 63%  541M 1s\n",
      " 49400K .......... .......... .......... .......... .......... 63%  673M 1s\n",
      " 49450K .......... .......... .......... .......... .......... 63%  549M 1s\n",
      " 49500K .......... .......... .......... .......... .......... 63%  561M 1s\n",
      " 49550K .......... .......... .......... .......... .......... 63%  559M 1s\n",
      " 49600K .......... .......... .......... .......... .......... 63%  665M 1s\n",
      " 49650K .......... .......... .......... .......... .......... 63%  671M 1s\n",
      " 49700K .......... .......... .......... .......... .......... 63% 19.8M 1s\n",
      " 49750K .......... .......... .......... .......... .......... 63%  609M 1s\n",
      " 49800K .......... .......... .......... .......... .......... 64%  666M 1s\n",
      " 49850K .......... .......... .......... .......... .......... 64%  636M 1s\n",
      " 49900K .......... .......... .......... .......... .......... 64%  558M 1s\n",
      " 49950K .......... .......... .......... .......... .......... 64%  648M 1s\n",
      " 50000K .......... .......... .......... .......... .......... 64% 5.79M 1s\n",
      " 50050K .......... .......... .......... .......... .......... 64%  388M 1s\n",
      " 50100K .......... .......... .......... .......... .......... 64% 22.8M 1s\n",
      " 50150K .......... .......... .......... .......... .......... 64% 6.27M 1s\n",
      " 50200K .......... .......... .......... .......... .......... 64% 77.7M 1s\n",
      " 50250K .......... .......... .......... .......... .......... 64%  389M 1s\n",
      " 50300K .......... .......... .......... .......... .......... 64%  460M 1s\n",
      " 50350K .......... .......... .......... .......... .......... 64%  513M 1s\n",
      " 50400K .......... .......... .......... .......... .......... 64%  541M 1s\n",
      " 50450K .......... .......... .......... .......... .......... 64%  636M 1s\n",
      " 50500K .......... .......... .......... .......... .......... 64%  455M 1s\n",
      " 50550K .......... .......... .......... .......... .......... 65%  649M 1s\n",
      " 50600K .......... .......... .......... .......... .......... 65%  500M 1s\n",
      " 50650K .......... .......... .......... .......... .......... 65%  543M 1s\n",
      " 50700K .......... .......... .......... .......... .......... 65%  568M 1s\n",
      " 50750K .......... .......... .......... .......... .......... 65%  532M 1s\n",
      " 50800K .......... .......... .......... .......... .......... 65%  693M 1s\n",
      " 50850K .......... .......... .......... .......... .......... 65%  358M 1s\n",
      " 50900K .......... .......... .......... .......... .......... 65%  543M 1s\n",
      " 50950K .......... .......... .......... .......... .......... 65%  542M 1s\n",
      " 51000K .......... .......... .......... .......... .......... 65%  575M 1s\n",
      " 51050K .......... .......... .......... .......... .......... 65%  610M 1s\n",
      " 51100K .......... .......... .......... .......... .......... 65%  583M 1s\n",
      " 51150K .......... .......... .......... .......... .......... 65%  668M 1s\n",
      " 51200K .......... .......... .......... .......... .......... 65%  700M 1s\n",
      " 51250K .......... .......... .......... .......... .......... 65%  379M 1s\n",
      " 51300K .......... .......... .......... .......... .......... 65%  551M 1s\n",
      " 51350K .......... .......... .......... .......... .......... 66%  496M 1s\n",
      " 51400K .......... .......... .......... .......... .......... 66%  548M 1s\n",
      " 51450K .......... .......... .......... .......... .......... 66%  675M 1s\n",
      " 51500K .......... .......... .......... .......... .......... 66%  517M 1s\n",
      " 51550K .......... .......... .......... .......... .......... 66% 4.53M 1s\n",
      " 51600K .......... .......... .......... .......... .......... 66% 44.6M 1s\n",
      " 51650K .......... .......... .......... .......... .......... 66% 56.5M 1s\n",
      " 51700K .......... .......... .......... .......... .......... 66% 6.13M 1s\n",
      " 51750K .......... .......... .......... .......... .......... 66% 76.5M 1s\n",
      " 51800K .......... .......... .......... .......... .......... 66%  415M 1s\n",
      " 51850K .......... .......... .......... .......... .......... 66%  641M 1s\n",
      " 51900K .......... .......... .......... .......... .......... 66%  662M 1s\n",
      " 51950K .......... .......... .......... .......... .......... 66%  537M 1s\n",
      " 52000K .......... .......... .......... .......... .......... 66%  659M 0s\n",
      " 52050K .......... .......... .......... .......... .......... 66%  616M 0s\n",
      " 52100K .......... .......... .......... .......... .......... 67%  519M 0s\n",
      " 52150K .......... .......... .......... .......... .......... 67%  427M 0s\n",
      " 52200K .......... .......... .......... .......... .......... 67%  644M 0s\n",
      " 52250K .......... .......... .......... .......... .......... 67%  663M 0s\n",
      " 52300K .......... .......... .......... .......... .......... 67%  522M 0s\n",
      " 52350K .......... .......... .......... .......... .......... 67%  470M 0s\n",
      " 52400K .......... .......... .......... .......... .......... 67%  725M 0s\n",
      " 52450K .......... .......... .......... .......... .......... 67%  466M 0s\n",
      " 52500K .......... .......... .......... .......... .......... 67%  683M 0s\n",
      " 52550K .......... .......... .......... .......... .......... 67%  472M 0s\n",
      " 52600K .......... .......... .......... .......... .......... 67%  559M 0s\n",
      " 52650K .......... .......... .......... .......... .......... 67%  508M 0s\n",
      " 52700K .......... .......... .......... .......... .......... 67%  689M 0s\n",
      " 52750K .......... .......... .......... .......... .......... 67% 20.0M 0s\n",
      " 52800K .......... .......... .......... .......... .......... 67%  538M 0s\n",
      " 52850K .......... .......... .......... .......... .......... 67%  660M 0s\n",
      " 52900K .......... .......... .......... .......... .......... 68%  661M 0s\n",
      " 52950K .......... .......... .......... .......... .......... 68%  642M 0s\n",
      " 53000K .......... .......... .......... .......... .......... 68%  567M 0s\n",
      " 53050K .......... .......... .......... .......... .......... 68%  639M 0s\n",
      " 53100K .......... .......... .......... .......... .......... 68% 5.45M 0s\n",
      " 53150K .......... .......... .......... .......... .......... 68% 35.4M 0s\n",
      " 53200K .......... .......... .......... .......... .......... 68% 5.95M 0s\n",
      " 53250K .......... .......... .......... .......... .......... 68% 73.4M 0s\n",
      " 53300K .......... .......... .......... .......... .......... 68%  529M 0s\n",
      " 53350K .......... .......... .......... .......... .......... 68%  477M 0s\n",
      " 53400K .......... .......... .......... .......... .......... 68%  619M 0s\n",
      " 53450K .......... .......... .......... .......... .......... 68%  458M 0s\n",
      " 53500K .......... .......... .......... .......... .......... 68%  658M 0s\n",
      " 53550K .......... .......... .......... .......... .......... 68%  551M 0s\n",
      " 53600K .......... .......... .......... .......... .......... 68%  686M 0s\n",
      " 53650K .......... .......... .......... .......... .......... 69% 39.4M 0s\n",
      " 53700K .......... .......... .......... .......... .......... 69%  537M 0s\n",
      " 53750K .......... .......... .......... .......... .......... 69%  465M 0s\n",
      " 53800K .......... .......... .......... .......... .......... 69%  663M 0s\n",
      " 53850K .......... .......... .......... .......... .......... 69%  438M 0s\n",
      " 53900K .......... .......... .......... .......... .......... 69%  479M 0s\n",
      " 53950K .......... .......... .......... .......... .......... 69%  650M 0s\n",
      " 54000K .......... .......... .......... .......... .......... 69%  653M 0s\n",
      " 54050K .......... .......... .......... .......... .......... 69%  443M 0s\n",
      " 54100K .......... .......... .......... .......... .......... 69%  537M 0s\n",
      " 54150K .......... .......... .......... .......... .......... 69%  482M 0s\n",
      " 54200K .......... .......... .......... .......... .......... 69%  696M 0s\n",
      " 54250K .......... .......... .......... .......... .......... 69%  658M 0s\n",
      " 54300K .......... .......... .......... .......... .......... 69%  480M 0s\n",
      " 54350K .......... .......... .......... .......... .......... 69%  692M 0s\n",
      " 54400K .......... .......... .......... .......... .......... 69%  701M 0s\n",
      " 54450K .......... .......... .......... .......... .......... 70%  702M 0s\n",
      " 54500K .......... .......... .......... .......... .......... 70%  585M 0s\n",
      " 54550K .......... .......... .......... .......... .......... 70%  698M 0s\n",
      " 54600K .......... .......... .......... .......... .......... 70% 4.38M 0s\n",
      " 54650K .......... .......... .......... .......... .......... 70% 45.5M 0s\n",
      " 54700K .......... .......... .......... .......... .......... 70%  402M 0s\n",
      " 54750K .......... .......... .......... .......... .......... 70% 6.00M 0s\n",
      " 54800K .......... .......... .......... .......... .......... 70% 80.8M 0s\n",
      " 54850K .......... .......... .......... .......... .......... 70%  457M 0s\n",
      " 54900K .......... .......... .......... .......... .......... 70%  527M 0s\n",
      " 54950K .......... .......... .......... .......... .......... 70%  643M 0s\n",
      " 55000K .......... .......... .......... .......... .......... 70%  492M 0s\n",
      " 55050K .......... .......... .......... .......... .......... 70% 56.2M 0s\n",
      " 55100K .......... .......... .......... .......... .......... 70%  389M 0s\n",
      " 55150K .......... .......... .......... .......... .......... 70%  560M 0s\n",
      " 55200K .......... .......... .......... .......... .......... 70%  696M 0s\n",
      " 55250K .......... .......... .......... .......... .......... 71%  560M 0s\n",
      " 55300K .......... .......... .......... .......... .......... 71%  575M 0s\n",
      " 55350K .......... .......... .......... .......... .......... 71%  461M 0s\n",
      " 55400K .......... .......... .......... .......... .......... 71%  713M 0s\n",
      " 55450K .......... .......... .......... .......... .......... 71%  766M 0s\n",
      " 55500K .......... .......... .......... .......... .......... 71% 39.5M 0s\n",
      " 55550K .......... .......... .......... .......... .......... 71%  500M 0s\n",
      " 55600K .......... .......... .......... .......... .......... 71%  425M 0s\n",
      " 55650K .......... .......... .......... .......... .......... 71%  587M 0s\n",
      " 55700K .......... .......... .......... .......... .......... 71%  641M 0s\n",
      " 55750K .......... .......... .......... .......... .......... 71%  669M 0s\n",
      " 55800K .......... .......... .......... .......... .......... 71%  649M 0s\n",
      " 55850K .......... .......... .......... .......... .......... 71%  577M 0s\n",
      " 55900K .......... .......... .......... .......... .......... 71%  693M 0s\n",
      " 55950K .......... .......... .......... .......... .......... 71%  694M 0s\n",
      " 56000K .......... .......... .......... .......... .......... 72%  709M 0s\n",
      " 56050K .......... .......... .......... .......... .......... 72%  599M 0s\n",
      " 56100K .......... .......... .......... .......... .......... 72%  697M 0s\n",
      " 56150K .......... .......... .......... .......... .......... 72% 4.99M 0s\n",
      " 56200K .......... .......... .......... .......... .......... 72% 43.6M 0s\n",
      " 56250K .......... .......... .......... .......... .......... 72% 29.3M 0s\n",
      " 56300K .......... .......... .......... .......... .......... 72% 7.22M 0s\n",
      " 56350K .......... .......... .......... .......... .......... 72%  636M 0s\n",
      " 56400K .......... .......... .......... .......... .......... 72% 74.5M 0s\n",
      " 56450K .......... .......... .......... .......... .......... 72%  617M 0s\n",
      " 56500K .......... .......... .......... .......... .......... 72%  483M 0s\n",
      " 56550K .......... .......... .......... .......... .......... 72%  503M 0s\n",
      " 56600K .......... .......... .......... .......... .......... 72% 55.9M 0s\n",
      " 56650K .......... .......... .......... .......... .......... 72%  538M 0s\n",
      " 56700K .......... .......... .......... .......... .......... 72% 99.7M 0s\n",
      " 56750K .......... .......... .......... .......... .......... 72%  480M 0s\n",
      " 56800K .......... .......... .......... .......... .......... 73%  638M 0s\n",
      " 56850K .......... .......... .......... .......... .......... 73%  475M 0s\n",
      " 56900K .......... .......... .......... .......... .......... 73%  657M 0s\n",
      " 56950K .......... .......... .......... .......... .......... 73%  614M 0s\n",
      " 57000K .......... .......... .......... .......... .......... 73%  533M 0s\n",
      " 57050K .......... .......... .......... .......... .......... 73% 57.1M 0s\n",
      " 57100K .......... .......... .......... .......... .......... 73%  615M 0s\n",
      " 57150K .......... .......... .......... .......... .......... 73%  667M 0s\n",
      " 57200K .......... .......... .......... .......... .......... 73%  489M 0s\n",
      " 57250K .......... .......... .......... .......... .......... 73%  576M 0s\n",
      " 57300K .......... .......... .......... .......... .......... 73%  559M 0s\n",
      " 57350K .......... .......... .......... .......... .......... 73%  633M 0s\n",
      " 57400K .......... .......... .......... .......... .......... 73%  357M 0s\n",
      " 57450K .......... .......... .......... .......... .......... 73%  537M 0s\n",
      " 57500K .......... .......... .......... .......... .......... 73% 43.6M 0s\n",
      " 57550K .......... .......... .......... .......... .......... 74%  617M 0s\n",
      " 57600K .......... .......... .......... .......... .......... 74%  527M 0s\n",
      " 57650K .......... .......... .......... .......... .......... 74% 63.2M 0s\n",
      " 57700K .......... .......... .......... .......... .......... 74% 5.77M 0s\n",
      " 57750K .......... .......... .......... .......... .......... 74%  300M 0s\n",
      " 57800K .......... .......... .......... .......... .......... 74% 26.7M 0s\n",
      " 57850K .......... .......... .......... .......... .......... 74% 7.13M 0s\n",
      " 57900K .......... .......... .......... .......... .......... 74%  488M 0s\n",
      " 57950K .......... .......... .......... .......... .......... 74%  686M 0s\n",
      " 58000K .......... .......... .......... .......... .......... 74% 49.9M 0s\n",
      " 58050K .......... .......... .......... .......... .......... 74%  367M 0s\n",
      " 58100K .......... .......... .......... .......... .......... 74%  643M 0s\n",
      " 58150K .......... .......... .......... .......... .......... 74%  716M 0s\n",
      " 58200K .......... .......... .......... .......... .......... 74%  730M 0s\n",
      " 58250K .......... .......... .......... .......... .......... 74% 85.0M 0s\n",
      " 58300K .......... .......... .......... .......... .......... 74%  435M 0s\n",
      " 58350K .......... .......... .......... .......... .......... 75% 98.8M 0s\n",
      " 58400K .......... .......... .......... .......... .......... 75%  428M 0s\n",
      " 58450K .......... .......... .......... .......... .......... 75%  721M 0s\n",
      " 58500K .......... .......... .......... .......... .......... 75% 94.9M 0s\n",
      " 58550K .......... .......... .......... .......... .......... 75%  476M 0s\n",
      " 58600K .......... .......... .......... .......... .......... 75%  364M 0s\n",
      " 58650K .......... .......... .......... .......... .......... 75%  514M 0s\n",
      " 58700K .......... .......... .......... .......... .......... 75%  608M 0s\n",
      " 58750K .......... .......... .......... .......... .......... 75%  348M 0s\n",
      " 58800K .......... .......... .......... .......... .......... 75% 42.0M 0s\n",
      " 58850K .......... .......... .......... .......... .......... 75%  471M 0s\n",
      " 58900K .......... .......... .......... .......... .......... 75%  575M 0s\n",
      " 58950K .......... .......... .......... .......... .......... 75%  571M 0s\n",
      " 59000K .......... .......... .......... .......... .......... 75%  681M 0s\n",
      " 59050K .......... .......... .......... .......... .......... 75%  686M 0s\n",
      " 59100K .......... .......... .......... .......... .......... 76%  673M 0s\n",
      " 59150K .......... .......... .......... .......... .......... 76%  617M 0s\n",
      " 59200K .......... .......... .......... .......... .......... 76% 5.78M 0s\n",
      " 59250K .......... .......... .......... .......... .......... 76% 60.4M 0s\n",
      " 59300K .......... .......... .......... .......... .......... 76% 25.3M 0s\n",
      " 59350K .......... .......... .......... .......... .......... 76% 7.03M 0s\n",
      " 59400K .......... .......... .......... .......... .......... 76%  281M 0s\n",
      " 59450K .......... .......... .......... .......... .......... 76%  505M 0s\n",
      " 59500K .......... .......... .......... .......... .......... 76%  541M 0s\n",
      " 59550K .......... .......... .......... .......... .......... 76%  645M 0s\n",
      " 59600K .......... .......... .......... .......... .......... 76% 50.5M 0s\n",
      " 59650K .......... .......... .......... .......... .......... 76%  626M 0s\n",
      " 59700K .......... .......... .......... .......... .......... 76%  662M 0s\n",
      " 59750K .......... .......... .......... .......... .......... 76% 98.5M 0s\n",
      " 59800K .......... .......... .......... .......... .......... 76%  444M 0s\n",
      " 59850K .......... .......... .......... .......... .......... 76%  678M 0s\n",
      " 59900K .......... .......... .......... .......... .......... 77% 84.3M 0s\n",
      " 59950K .......... .......... .......... .......... .......... 77%  615M 0s\n",
      " 60000K .......... .......... .......... .......... .......... 77%  103M 0s\n",
      " 60050K .......... .......... .......... .......... .......... 77%  339M 0s\n",
      " 60100K .......... .......... .......... .......... .......... 77%  601M 0s\n",
      " 60150K .......... .......... .......... .......... .......... 77%  674M 0s\n",
      " 60200K .......... .......... .......... .......... .......... 77%  683M 0s\n",
      " 60250K .......... .......... .......... .......... .......... 77%  541M 0s\n",
      " 60300K .......... .......... .......... .......... .......... 77%  653M 0s\n",
      " 60350K .......... .......... .......... .......... .......... 77% 42.9M 0s\n",
      " 60400K .......... .......... .......... .......... .......... 77%  447M 0s\n",
      " 60450K .......... .......... .......... .......... .......... 77%  660M 0s\n",
      " 60500K .......... .......... .......... .......... .......... 77%  578M 0s\n",
      " 60550K .......... .......... .......... .......... .......... 77%  694M 0s\n",
      " 60600K .......... .......... .......... .......... .......... 77%  697M 0s\n",
      " 60650K .......... .......... .......... .......... .......... 78%  682M 0s\n",
      " 60700K .......... .......... .......... .......... .......... 78% 5.57M 0s\n",
      " 60750K .......... .......... .......... .......... .......... 78%  465M 0s\n",
      " 60800K .......... .......... .......... .......... .......... 78% 50.7M 0s\n",
      " 60850K .......... .......... .......... .......... .......... 78% 27.0M 0s\n",
      " 60900K .......... .......... .......... .......... .......... 78% 7.29M 0s\n",
      " 60950K .......... .......... .......... .......... .......... 78%  412M 0s\n",
      " 61000K .......... .......... .......... .......... .......... 78%  672M 0s\n",
      " 61050K .......... .......... .......... .......... .......... 78%  454M 0s\n",
      " 61100K .......... .......... .......... .......... .......... 78%  651M 0s\n",
      " 61150K .......... .......... .......... .......... .......... 78% 51.0M 0s\n",
      " 61200K .......... .......... .......... .......... .......... 78%  537M 0s\n",
      " 61250K .......... .......... .......... .......... .......... 78%  676M 0s\n",
      " 61300K .......... .......... .......... .......... .......... 78% 85.0M 0s\n",
      " 61350K .......... .......... .......... .......... .......... 78%  634M 0s\n",
      " 61400K .......... .......... .......... .......... .......... 78% 91.7M 0s\n",
      " 61450K .......... .......... .......... .......... .......... 79%  615M 0s\n",
      " 61500K .......... .......... .......... .......... .......... 79%  481M 0s\n",
      " 61550K .......... .......... .......... .......... .......... 79% 93.6M 0s\n",
      " 61600K .......... .......... .......... .......... .......... 79%  358M 0s\n",
      " 61650K .......... .......... .......... .......... .......... 79%  667M 0s\n",
      " 61700K .......... .......... .......... .......... .......... 79%  706M 0s\n",
      " 61750K .......... .......... .......... .......... .......... 79%  704M 0s\n",
      " 61800K .......... .......... .......... .......... .......... 79%  500M 0s\n",
      " 61850K .......... .......... .......... .......... .......... 79% 42.0M 0s\n",
      " 61900K .......... .......... .......... .......... .......... 79%  530M 0s\n",
      " 61950K .......... .......... .......... .......... .......... 79%  548M 0s\n",
      " 62000K .......... .......... .......... .......... .......... 79%  485M 0s\n",
      " 62050K .......... .......... .......... .......... .......... 79%  511M 0s\n",
      " 62100K .......... .......... .......... .......... .......... 79%  587M 0s\n",
      " 62150K .......... .......... .......... .......... .......... 79%  703M 0s\n",
      " 62200K .......... .......... .......... .......... .......... 79%  716M 0s\n",
      " 62250K .......... .......... .......... .......... .......... 80% 5.67M 0s\n",
      " 62300K .......... .......... .......... .......... .......... 80% 63.6M 0s\n",
      " 62350K .......... .......... .......... .......... .......... 80% 54.6M 0s\n",
      " 62400K .......... .......... .......... .......... .......... 80% 40.5M 0s\n",
      " 62450K .......... .......... .......... .......... .......... 80% 6.88M 0s\n",
      " 62500K .......... .......... .......... .......... .......... 80%  484M 0s\n",
      " 62550K .......... .......... .......... .......... .......... 80%  602M 0s\n",
      " 62600K .......... .......... .......... .......... .......... 80%  642M 0s\n",
      " 62650K .......... .......... .......... .......... .......... 80%  645M 0s\n",
      " 62700K .......... .......... .......... .......... .......... 80% 80.9M 0s\n",
      " 62750K .......... .......... .......... .......... .......... 80%  609M 0s\n",
      " 62800K .......... .......... .......... .......... .......... 80% 96.4M 0s\n",
      " 62850K .......... .......... .......... .......... .......... 80%  419M 0s\n",
      " 62900K .......... .......... .......... .......... .......... 80%  667M 0s\n",
      " 62950K .......... .......... .......... .......... .......... 80% 94.6M 0s\n",
      " 63000K .......... .......... .......... .......... .......... 81%  474M 0s\n",
      " 63050K .......... .......... .......... .......... .......... 81%  481M 0s\n",
      " 63100K .......... .......... .......... .......... .......... 81%  575M 0s\n",
      " 63150K .......... .......... .......... .......... .......... 81% 86.2M 0s\n",
      " 63200K .......... .......... .......... .......... .......... 81%  624M 0s\n",
      " 63250K .......... .......... .......... .......... .......... 81%  542M 0s\n",
      " 63300K .......... .......... .......... .......... .......... 81%  527M 0s\n",
      " 63350K .......... .......... .......... .......... .......... 81% 58.7M 0s\n",
      " 63400K .......... .......... .......... .......... .......... 81%  432M 0s\n",
      " 63450K .......... .......... .......... .......... .......... 81%  633M 0s\n",
      " 63500K .......... .......... .......... .......... .......... 81%  680M 0s\n",
      " 63550K .......... .......... .......... .......... .......... 81%  685M 0s\n",
      " 63600K .......... .......... .......... .......... .......... 81% 60.0M 0s\n",
      " 63650K .......... .......... .......... .......... .......... 81%  542M 0s\n",
      " 63700K .......... .......... .......... .......... .......... 81%  631M 0s\n",
      " 63750K .......... .......... .......... .......... .......... 81%  103M 0s\n",
      " 63800K .......... .......... .......... .......... .......... 82% 5.79M 0s\n",
      " 63850K .......... .......... .......... .......... .......... 82% 76.1M 0s\n",
      " 63900K .......... .......... .......... .......... .......... 82% 22.3M 0s\n",
      " 63950K .......... .......... .......... .......... .......... 82% 59.3M 0s\n",
      " 64000K .......... .......... .......... .......... .......... 82% 8.64M 0s\n",
      " 64050K .......... .......... .......... .......... .......... 82%  612M 0s\n",
      " 64100K .......... .......... .......... .......... .......... 82%  503M 0s\n",
      " 64150K .......... .......... .......... .......... .......... 82%  578M 0s\n",
      " 64200K .......... .......... .......... .......... .......... 82% 80.2M 0s\n",
      " 64250K .......... .......... .......... .......... .......... 82%  600M 0s\n",
      " 64300K .......... .......... .......... .......... .......... 82%  690M 0s\n",
      " 64350K .......... .......... .......... .......... .......... 82% 92.7M 0s\n",
      " 64400K .......... .......... .......... .......... .......... 82%  610M 0s\n",
      " 64450K .......... .......... .......... .......... .......... 82%  102M 0s\n",
      " 64500K .......... .......... .......... .......... .......... 82%  485M 0s\n",
      " 64550K .......... .......... .......... .......... .......... 83% 96.5M 0s\n",
      " 64600K .......... .......... .......... .......... .......... 83%  614M 0s\n",
      " 64650K .......... .......... .......... .......... .......... 83% 93.8M 0s\n",
      " 64700K .......... .......... .......... .......... .......... 83%  476M 0s\n",
      " 64750K .......... .......... .......... .......... .......... 83%  647M 0s\n",
      " 64800K .......... .......... .......... .......... .......... 83%  545M 0s\n",
      " 64850K .......... .......... .......... .......... .......... 83%  578M 0s\n",
      " 64900K .......... .......... .......... .......... .......... 83% 59.6M 0s\n",
      " 64950K .......... .......... .......... .......... .......... 83%  580M 0s\n",
      " 65000K .......... .......... .......... .......... .......... 83%  635M 0s\n",
      " 65050K .......... .......... .......... .......... .......... 83%  564M 0s\n",
      " 65100K .......... .......... .......... .......... .......... 83% 97.9M 0s\n",
      " 65150K .......... .......... .......... .......... .......... 83%  622M 0s\n",
      " 65200K .......... .......... .......... .......... .......... 83%  599M 0s\n",
      " 65250K .......... .......... .......... .......... .......... 83%  548M 0s\n",
      " 65300K .......... .......... .......... .......... .......... 83% 5.40M 0s\n",
      " 65350K .......... .......... .......... .......... .......... 84% 76.5M 0s\n",
      " 65400K .......... .......... .......... .......... .......... 84% 24.8M 0s\n",
      " 65450K .......... .......... .......... .......... .......... 84%  203M 0s\n",
      " 65500K .......... .......... .......... .......... .......... 84% 8.12M 0s\n",
      " 65550K .......... .......... .......... .......... .......... 84% 77.8M 0s\n",
      " 65600K .......... .......... .......... .......... .......... 84%  483M 0s\n",
      " 65650K .......... .......... .......... .......... .......... 84%  629M 0s\n",
      " 65700K .......... .......... .......... .......... .......... 84%  701M 0s\n",
      " 65750K .......... .......... .......... .......... .......... 84% 81.1M 0s\n",
      " 65800K .......... .......... .......... .......... .......... 84%  599M 0s\n",
      " 65850K .......... .......... .......... .......... .......... 84% 96.3M 0s\n",
      " 65900K .......... .......... .......... .......... .......... 84%  391M 0s\n",
      " 65950K .......... .......... .......... .......... .......... 84%  681M 0s\n",
      " 66000K .......... .......... .......... .......... .......... 84% 91.2M 0s\n",
      " 66050K .......... .......... .......... .......... .......... 84%  388M 0s\n",
      " 66100K .......... .......... .......... .......... .......... 85%  403M 0s\n",
      " 66150K .......... .......... .......... .......... .......... 85%  546M 0s\n",
      " 66200K .......... .......... .......... .......... .......... 85% 58.1M 0s\n",
      " 66250K .......... .......... .......... .......... .......... 85%  506M 0s\n",
      " 66300K .......... .......... .......... .......... .......... 85%  569M 0s\n",
      " 66350K .......... .......... .......... .......... .......... 85%  709M 0s\n",
      " 66400K .......... .......... .......... .......... .......... 85%  701M 0s\n",
      " 66450K .......... .......... .......... .......... .......... 85% 57.5M 0s\n",
      " 66500K .......... .......... .......... .......... .......... 85%  421M 0s\n",
      " 66550K .......... .......... .......... .......... .......... 85%  544M 0s\n",
      " 66600K .......... .......... .......... .......... .......... 85%  600M 0s\n",
      " 66650K .......... .......... .......... .......... .......... 85%  678M 0s\n",
      " 66700K .......... .......... .......... .......... .......... 85%  715M 0s\n",
      " 66750K .......... .......... .......... .......... .......... 85%  600M 0s\n",
      " 66800K .......... .......... .......... .......... .......... 85% 5.24M 0s\n",
      " 66850K .......... .......... .......... .......... .......... 85%  416M 0s\n",
      " 66900K .......... .......... .......... .......... .......... 86%  165M 0s\n",
      " 66950K .......... .......... .......... .......... .......... 86% 23.4M 0s\n",
      " 67000K .......... .......... .......... .......... .......... 86% 7.62M 0s\n",
      " 67050K .......... .......... .......... .......... .......... 86%  577M 0s\n",
      " 67100K .......... .......... .......... .......... .......... 86% 79.5M 0s\n",
      " 67150K .......... .......... .......... .......... .......... 86%  539M 0s\n",
      " 67200K .......... .......... .......... .......... .......... 86%  579M 0s\n",
      " 67250K .......... .......... .......... .......... .......... 86%  547M 0s\n",
      " 67300K .......... .......... .......... .......... .......... 86%  607M 0s\n",
      " 67350K .......... .......... .......... .......... .......... 86% 53.7M 0s\n",
      " 67400K .......... .......... .......... .......... .......... 86%  409M 0s\n",
      " 67450K .......... .......... .......... .......... .......... 86%  605M 0s\n",
      " 67500K .......... .......... .......... .......... .......... 86%  525M 0s\n",
      " 67550K .......... .......... .......... .......... .......... 86% 90.7M 0s\n",
      " 67600K .......... .......... .......... .......... .......... 86%  645M 0s\n",
      " 67650K .......... .......... .......... .......... .......... 86%  614M 0s\n",
      " 67700K .......... .......... .......... .......... .......... 87% 96.9M 0s\n",
      " 67750K .......... .......... .......... .......... .......... 87% 92.4M 0s\n",
      " 67800K .......... .......... .......... .......... .......... 87%  487M 0s\n",
      " 67850K .......... .......... .......... .......... .......... 87%  633M 0s\n",
      " 67900K .......... .......... .......... .......... .......... 87%  583M 0s\n",
      " 67950K .......... .......... .......... .......... .......... 87%  698M 0s\n",
      " 68000K .......... .......... .......... .......... .......... 87%  351M 0s\n",
      " 68050K .......... .......... .......... .......... .......... 87%  276M 0s\n",
      " 68100K .......... .......... .......... .......... .......... 87%  715M 0s\n",
      " 68150K .......... .......... .......... .......... .......... 87% 44.3M 0s\n",
      " 68200K .......... .......... .......... .......... .......... 87%  533M 0s\n",
      " 68250K .......... .......... .......... .......... .......... 87%  666M 0s\n",
      " 68300K .......... .......... .......... .......... .......... 87%  641M 0s\n",
      " 68350K .......... .......... .......... .......... .......... 87% 5.12M 0s\n",
      " 68400K .......... .......... .......... .......... .......... 87%  292M 0s\n",
      " 68450K .......... .......... .......... .......... .......... 88%  607M 0s\n",
      " 68500K .......... .......... .......... .......... .......... 88% 20.7M 0s\n",
      " 68550K .......... .......... .......... .......... .......... 88% 8.20M 0s\n",
      " 68600K .......... .......... .......... .......... .......... 88%  584M 0s\n",
      " 68650K .......... .......... .......... .......... .......... 88%  490M 0s\n",
      " 68700K .......... .......... .......... .......... .......... 88%  559M 0s\n",
      " 68750K .......... .......... .......... .......... .......... 88%  583M 0s\n",
      " 68800K .......... .......... .......... .......... .......... 88% 49.0M 0s\n",
      " 68850K .......... .......... .......... .......... .......... 88%  475M 0s\n",
      " 68900K .......... .......... .......... .......... .......... 88%  658M 0s\n",
      " 68950K .......... .......... .......... .......... .......... 88%  605M 0s\n",
      " 69000K .......... .......... .......... .......... .......... 88% 56.9M 0s\n",
      " 69050K .......... .......... .......... .......... .......... 88%  613M 0s\n",
      " 69100K .......... .......... .......... .......... .......... 88%  554M 0s\n",
      " 69150K .......... .......... .......... .......... .......... 88%  664M 0s\n",
      " 69200K .......... .......... .......... .......... .......... 88%  669M 0s\n",
      " 69250K .......... .......... .......... .......... .......... 89% 91.6M 0s\n",
      " 69300K .......... .......... .......... .......... .......... 89%  633M 0s\n",
      " 69350K .......... .......... .......... .......... .......... 89%  523M 0s\n",
      " 69400K .......... .......... .......... .......... .......... 89%  647M 0s\n",
      " 69450K .......... .......... .......... .......... .......... 89%  533M 0s\n",
      " 69500K .......... .......... .......... .......... .......... 89% 58.9M 0s\n",
      " 69550K .......... .......... .......... .......... .......... 89%  604M 0s\n",
      " 69600K .......... .......... .......... .......... .......... 89%  554M 0s\n",
      " 69650K .......... .......... .......... .......... .......... 89%  101M 0s\n",
      " 69700K .......... .......... .......... .......... .......... 89%  609M 0s\n",
      " 69750K .......... .......... .......... .......... .......... 89%  671M 0s\n",
      " 69800K .......... .......... .......... .......... .......... 89%  573M 0s\n",
      " 69850K .......... .......... .......... .......... .......... 89%  656M 0s\n",
      " 69900K .......... .......... .......... .......... .......... 89% 4.84M 0s\n",
      " 69950K .......... .......... .......... .......... .......... 89%  387M 0s\n",
      " 70000K .......... .......... .......... .......... .......... 90% 23.1M 0s\n",
      " 70050K .......... .......... .......... .......... .......... 90% 8.15M 0s\n",
      " 70100K .......... .......... .......... .......... .......... 90%  249M 0s\n",
      " 70150K .......... .......... .......... .......... .......... 90%  560M 0s\n",
      " 70200K .......... .......... .......... .......... .......... 90%  696M 0s\n",
      " 70250K .......... .......... .......... .......... .......... 90%  686M 0s\n",
      " 70300K .......... .......... .......... .......... .......... 90%  597M 0s\n",
      " 70350K .......... .......... .......... .......... .......... 90%  362M 0s\n",
      " 70400K .......... .......... .......... .......... .......... 90% 38.3M 0s\n",
      " 70450K .......... .......... .......... .......... .......... 90%  438M 0s\n",
      " 70500K .......... .......... .......... .......... .......... 90%  663M 0s\n",
      " 70550K .......... .......... .......... .......... .......... 90%  530M 0s\n",
      " 70600K .......... .......... .......... .......... .......... 90%  657M 0s\n",
      " 70650K .......... .......... .......... .......... .......... 90%  494M 0s\n",
      " 70700K .......... .......... .......... .......... .......... 90%  686M 0s\n",
      " 70750K .......... .......... .......... .......... .......... 90% 57.6M 0s\n",
      " 70800K .......... .......... .......... .......... .......... 91%  461M 0s\n",
      " 70850K .......... .......... .......... .......... .......... 91%  561M 0s\n",
      " 70900K .......... .......... .......... .......... .......... 91%  452M 0s\n",
      " 70950K .......... .......... .......... .......... .......... 91%  712M 0s\n",
      " 71000K .......... .......... .......... .......... .......... 91% 60.9M 0s\n",
      " 71050K .......... .......... .......... .......... .......... 91%  537M 0s\n",
      " 71100K .......... .......... .......... .......... .......... 91%  451M 0s\n",
      " 71150K .......... .......... .......... .......... .......... 91%  590M 0s\n",
      " 71200K .......... .......... .......... .......... .......... 91%  592M 0s\n",
      " 71250K .......... .......... .......... .......... .......... 91%  584M 0s\n",
      " 71300K .......... .......... .......... .......... .......... 91%  695M 0s\n",
      " 71350K .......... .......... .......... .......... .......... 91%  679M 0s\n",
      " 71400K .......... .......... .......... .......... .......... 91% 4.64M 0s\n",
      " 71450K .......... .......... .......... .......... .......... 91%  332M 0s\n",
      " 71500K .......... .......... .......... .......... .......... 91%  506M 0s\n",
      " 71550K .......... .......... .......... .......... .......... 92% 24.6M 0s\n",
      " 71600K .......... .......... .......... .......... .......... 92% 7.91M 0s\n",
      " 71650K .......... .......... .......... .......... .......... 92%  381M 0s\n",
      " 71700K .......... .......... .......... .......... .......... 92%  386M 0s\n",
      " 71750K .......... .......... .......... .......... .......... 92%  489M 0s\n",
      " 71800K .......... .......... .......... .......... .......... 92%  674M 0s\n",
      " 71850K .......... .......... .......... .......... .......... 92%  662M 0s\n",
      " 71900K .......... .......... .......... .......... .......... 92% 37.6M 0s\n",
      " 71950K .......... .......... .......... .......... .......... 92%  497M 0s\n",
      " 72000K .......... .......... .......... .......... .......... 92%  518M 0s\n",
      " 72050K .......... .......... .......... .......... .......... 92%  564M 0s\n",
      " 72100K .......... .......... .......... .......... .......... 92%  533M 0s\n",
      " 72150K .......... .......... .......... .......... .......... 92%  619M 0s\n",
      " 72200K .......... .......... .......... .......... .......... 92%  684M 0s\n",
      " 72250K .......... .......... .......... .......... .......... 92%  599M 0s\n",
      " 72300K .......... .......... .......... .......... .......... 92% 56.2M 0s\n",
      " 72350K .......... .......... .......... .......... .......... 93%  617M 0s\n",
      " 72400K .......... .......... .......... .......... .......... 93%  550M 0s\n",
      " 72450K .......... .......... .......... .......... .......... 93%  683M 0s\n",
      " 72500K .......... .......... .......... .......... .......... 93%  465M 0s\n",
      " 72550K .......... .......... .......... .......... .......... 93%  646M 0s\n",
      " 72600K .......... .......... .......... .......... .......... 93%  586M 0s\n",
      " 72650K .......... .......... .......... .......... .......... 93%  676M 0s\n",
      " 72700K .......... .......... .......... .......... .......... 93%  514M 0s\n",
      " 72750K .......... .......... .......... .......... .......... 93% 19.9M 0s\n",
      " 72800K .......... .......... .......... .......... .......... 93%  546M 0s\n",
      " 72850K .......... .......... .......... .......... .......... 93%  695M 0s\n",
      " 72900K .......... .......... .......... .......... .......... 93%  737M 0s\n",
      " 72950K .......... .......... .......... .......... .......... 93% 5.49M 0s\n",
      " 73000K .......... .......... .......... .......... .......... 93% 78.3M 0s\n",
      " 73050K .......... .......... .......... .......... .......... 93% 33.8M 0s\n",
      " 73100K .......... .......... .......... .......... .......... 94% 7.78M 0s\n",
      " 73150K .......... .......... .......... .......... .......... 94%  506M 0s\n",
      " 73200K .......... .......... .......... .......... .......... 94%  487M 0s\n",
      " 73250K .......... .......... .......... .......... .......... 94%  395M 0s\n",
      " 73300K .......... .......... .......... .......... .......... 94%  563M 0s\n",
      " 73350K .......... .......... .......... .......... .......... 94%  666M 0s\n",
      " 73400K .......... .......... .......... .......... .......... 94%  637M 0s\n",
      " 73450K .......... .......... .......... .......... .......... 94% 38.5M 0s\n",
      " 73500K .......... .......... .......... .......... .......... 94%  496M 0s\n",
      " 73550K .......... .......... .......... .......... .......... 94%  556M 0s\n",
      " 73600K .......... .......... .......... .......... .......... 94%  549M 0s\n",
      " 73650K .......... .......... .......... .......... .......... 94%  660M 0s\n",
      " 73700K .......... .......... .......... .......... .......... 94%  681M 0s\n",
      " 73750K .......... .......... .......... .......... .......... 94%  607M 0s\n",
      " 73800K .......... .......... .......... .......... .......... 94% 59.7M 0s\n",
      " 73850K .......... .......... .......... .......... .......... 94%  464M 0s\n",
      " 73900K .......... .......... .......... .......... .......... 95%  527M 0s\n",
      " 73950K .......... .......... .......... .......... .......... 95%  492M 0s\n",
      " 74000K .......... .......... .......... .......... .......... 95%  596M 0s\n",
      " 74050K .......... .......... .......... .......... .......... 95%  636M 0s\n",
      " 74100K .......... .......... .......... .......... .......... 95%  594M 0s\n",
      " 74150K .......... .......... .......... .......... .......... 95%  699M 0s\n",
      " 74200K .......... .......... .......... .......... .......... 95%  614M 0s\n",
      " 74250K .......... .......... .......... .......... .......... 95% 18.7M 0s\n",
      " 74300K .......... .......... .......... .......... .......... 95%  456M 0s\n",
      " 74350K .......... .......... .......... .......... .......... 95%  667M 0s\n",
      " 74400K .......... .......... .......... .......... .......... 95%  648M 0s\n",
      " 74450K .......... .......... .......... .......... .......... 95% 5.47M 0s\n",
      " 74500K .......... .......... .......... .......... .......... 95%  500M 0s\n",
      " 74550K .......... .......... .......... .......... .......... 95%  653M 0s\n",
      " 74600K .......... .......... .......... .......... .......... 95% 28.8M 0s\n",
      " 74650K .......... .......... .......... .......... .......... 95% 7.59M 0s\n",
      " 74700K .......... .......... .......... .......... .......... 96%  439M 0s\n",
      " 74750K .......... .......... .......... .......... .......... 96%  453M 0s\n",
      " 74800K .......... .......... .......... .......... .......... 96%  465M 0s\n",
      " 74850K .......... .......... .......... .......... .......... 96%  586M 0s\n",
      " 74900K .......... .......... .......... .......... .......... 96%  504M 0s\n",
      " 74950K .......... .......... .......... .......... .......... 96%  619M 0s\n",
      " 75000K .......... .......... .......... .......... .......... 96%  703M 0s\n",
      " 75050K .......... .......... .......... .......... .......... 96% 36.5M 0s\n",
      " 75100K .......... .......... .......... .......... .......... 96%  618M 0s\n",
      " 75150K .......... .......... .......... .......... .......... 96%  648M 0s\n",
      " 75200K .......... .......... .......... .......... .......... 96%  570M 0s\n",
      " 75250K .......... .......... .......... .......... .......... 96%  654M 0s\n",
      " 75300K .......... .......... .......... .......... .......... 96%  667M 0s\n",
      " 75350K .......... .......... .......... .......... .......... 96% 57.0M 0s\n",
      " 75400K .......... .......... .......... .......... .......... 96%  541M 0s\n",
      " 75450K .......... .......... .......... .......... .......... 97%  665M 0s\n",
      " 75500K .......... .......... .......... .......... .......... 97%  519M 0s\n",
      " 75550K .......... .......... .......... .......... .......... 97%  568M 0s\n",
      " 75600K .......... .......... .......... .......... .......... 97%  658M 0s\n",
      " 75650K .......... .......... .......... .......... .......... 97%  675M 0s\n",
      " 75700K .......... .......... .......... .......... .......... 97%  711M 0s\n",
      " 75750K .......... .......... .......... .......... .......... 97%  590M 0s\n",
      " 75800K .......... .......... .......... .......... .......... 97% 20.7M 0s\n",
      " 75850K .......... .......... .......... .......... .......... 97%  473M 0s\n",
      " 75900K .......... .......... .......... .......... .......... 97%  488M 0s\n",
      " 75950K .......... .......... .......... .......... .......... 97%  666M 0s\n",
      " 76000K .......... .......... .......... .......... .......... 97% 5.23M 0s\n",
      " 76050K .......... .......... .......... .......... .......... 97%  569M 0s\n",
      " 76100K .......... .......... .......... .......... .......... 97% 33.7M 0s\n",
      " 76150K .......... .......... .......... .......... .......... 97%  540M 0s\n",
      " 76200K .......... .......... .......... .......... .......... 97% 7.58M 0s\n",
      " 76250K .......... .......... .......... .......... .......... 98%  435M 0s\n",
      " 76300K .......... .......... .......... .......... .......... 98%  473M 0s\n",
      " 76350K .......... .......... .......... .......... .......... 98%  640M 0s\n",
      " 76400K .......... .......... .......... .......... .......... 98%  563M 0s\n",
      " 76450K .......... .......... .......... .......... .......... 98%  458M 0s\n",
      " 76500K .......... .......... .......... .......... .......... 98%  588M 0s\n",
      " 76550K .......... .......... .......... .......... .......... 98%  548M 0s\n",
      " 76600K .......... .......... .......... .......... .......... 98%  687M 0s\n",
      " 76650K .......... .......... .......... .......... .......... 98% 30.3M 0s\n",
      " 76700K .......... .......... .......... .......... .......... 98%  531M 0s\n",
      " 76750K .......... .......... .......... .......... .......... 98%  633M 0s\n",
      " 76800K .......... .......... .......... .......... .......... 98%  492M 0s\n",
      " 76850K .......... .......... .......... .......... .......... 98%  677M 0s\n",
      " 76900K .......... .......... .......... .......... .......... 98%  520M 0s\n",
      " 76950K .......... .......... .......... .......... .......... 98%  707M 0s\n",
      " 77000K .......... .......... .......... .......... .......... 99%  512M 0s\n",
      " 77050K .......... .......... .......... .......... .......... 99%  667M 0s\n",
      " 77100K .......... .......... .......... .......... .......... 99%  629M 0s\n",
      " 77150K .......... .......... .......... .......... .......... 99%  514M 0s\n",
      " 77200K .......... .......... .......... .......... .......... 99%  434M 0s\n",
      " 77250K .......... .......... .......... .......... .......... 99%  434M 0s\n",
      " 77300K .......... .......... .......... .......... .......... 99% 18.1M 0s\n",
      " 77350K .......... .......... .......... .......... .......... 99%  102M 0s\n",
      " 77400K .......... .......... .......... .......... .......... 99%  104M 0s\n",
      " 77450K .......... .......... .......... .......... .......... 99%  614M 0s\n",
      " 77500K .......... .......... .......... .......... .......... 99% 5.55M 0s\n",
      " 77550K .......... .......... .......... .......... .......... 99%  570M 0s\n",
      " 77600K .......... .......... .......... .......... .......... 99%  677M 0s\n",
      " 77650K .......... .......... .......... .......... .......... 99% 26.5M 0s\n",
      " 77700K .......... .......... .......... .......... .......... 99% 6.87M 0s\n",
      " 77750K .......... .......... .......... .......... .......... 99%  592M 0s\n",
      " 77800K .......... .......                                    100% 29.4M=1.4s\n",
      "\n",
      "2025-02-27 10:32:45 (53.6 MB/s) - 'data/Homo_sapiens.GRCh38.cdna.all.fa.gz' saved [79685174]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "wget ftp://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz -P data/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "628b5497-8a31-4415-b393-c5b8c97f5350",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Homo_sapiens.GRCh38.cdna.all.fa\n",
      "STAR_index\n",
      "alignment_Bowtie2_subset\n",
      "alignment_STAR_subset\n",
      "gene_counts_DESeq\n",
      "subset_rawdata_RNA-seq\n",
      "subset_trimmed_RNA-seq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls data\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "41402daf-553c-485b-b530-cf740db5a805",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"data/hg38_cdna_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  data/Homo_sapiens.GRCh38.cdna.all.fa\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:02\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:02\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:27\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:03\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:03\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 40214484 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:22\n",
      "Returning block of 40214485 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 66480457 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:18\n",
      "Returning block of 66480458 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 68586474 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:23\n",
      "Returning block of 68586475 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 49097504 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:43\n",
      "Returning block of 49097505 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 65349089 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:14\n",
      "Returning block of 65349090 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 28965054 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:57\n",
      "Returning block of 28965055 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 72809500 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:27\n",
      "Returning block of 72809501 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 3519685 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:06\n",
      "Returning block of 3519686 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/hg38_cdna_index.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/hg38_cdna_index.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:15:35\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:02\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:02\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 98755563\n",
      "Using parameters --bmax 74066673 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 74066673 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:26\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:03\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:04\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 5; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 4.93778e+07 (target: 74066672)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 8\n",
      "  Reserving size (74066673) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 69612405 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:25\n",
      "Returning block of 69612406 for bucket 1\n",
      "Getting block 2 of 8\n",
      "  Reserving size (74066673) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 48012854 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:38\n",
      "Returning block of 48012855 for bucket 2\n",
      "Getting block 3 of 8\n",
      "  Reserving size (74066673) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 53601703 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:46\n",
      "Returning block of 53601704 for bucket 3\n",
      "Getting block 4 of 8\n",
      "  Reserving size (74066673) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 50816782 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:43\n",
      "Returning block of 50816783 for bucket 4\n",
      "Getting block 5 of 8\n",
      "  Reserving size (74066673) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 49006051 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:35\n",
      "Returning block of 49006052 for bucket 5\n",
      "Getting block 6 of 8\n",
      "  Reserving size (74066673) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 25671179 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:53\n",
      "Returning block of 25671180 for bucket 6\n",
      "Getting block 7 of 8\n",
      "  Reserving size (74066673) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 58667016 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:00\n",
      "Returning block of 58667017 for bucket 7\n",
      "Getting block 8 of 8\n",
      "  Reserving size (74066673) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 8: 70%\n",
      "  bucket 8: 80%\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 39634257 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:20\n",
      "Returning block of 39634258 for bucket 8\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 103412150\n",
      "fchr[G]: 199191248\n",
      "fchr[T]: 297049552\n",
      "fchr[$]: 395022254\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 191817752 bytes to primary EBWT file: data/hg38_cdna_index.rev.1.bt2.tmp\n",
      "Wrote 98755568 bytes to secondary EBWT file: data/hg38_cdna_index.rev.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 395022254\n",
      "    bwtLen: 395022255\n",
      "    sz: 98755564\n",
      "    bwtSz: 98755564\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 24688891\n",
      "    offsSz: 98755564\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 2057408\n",
      "    numLines: 2057408\n",
      "    ebwtTotLen: 131674112\n",
      "    ebwtTotSz: 131674112\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:15:30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Renaming data/hg38_cdna_index.3.bt2.tmp to data/hg38_cdna_index.3.bt2\n",
      "Renaming data/hg38_cdna_index.4.bt2.tmp to data/hg38_cdna_index.4.bt2\n",
      "Renaming data/hg38_cdna_index.1.bt2.tmp to data/hg38_cdna_index.1.bt2\n",
      "Renaming data/hg38_cdna_index.2.bt2.tmp to data/hg38_cdna_index.2.bt2\n",
      "Renaming data/hg38_cdna_index.rev.1.bt2.tmp to data/hg38_cdna_index.rev.1.bt2\n",
      "Renaming data/hg38_cdna_index.rev.2.bt2.tmp to data/hg38_cdna_index.rev.2.bt2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bowtie2-build data/Homo_sapiens.GRCh38.cdna.all.fa data/hg38_cdna_index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "dedd7cb4-4bcc-47d5-9cd9-66f29d6ec971",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Homo_sapiens.GRCh38.cdna.all.fa\n",
      "STAR_index\n",
      "alignment_Bowtie2_subset\n",
      "alignment_STAR_subset\n",
      "gene_counts_DESeq\n",
      "hg38_cdna_index.1.bt2\n",
      "hg38_cdna_index.2.bt2\n",
      "hg38_cdna_index.3.bt2\n",
      "hg38_cdna_index.4.bt2\n",
      "hg38_cdna_index.rev.1.bt2\n",
      "hg38_cdna_index.rev.2.bt2\n",
      "subset_rawdata_RNA-seq\n",
      "subset_trimmed_RNA-seq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls data/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "48a43be5-1743-404d-9785-b9f5e201722f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERR950158_trimmed_1.fastq.gz\n",
      "ERR950158_trimmed_2.fastq.gz\n",
      "ERR950159_trimmed_1.fastq.gz\n",
      "ERR950159_trimmed_2.fastq.gz\n",
      "ERR950160_trimmed_1.fastq.gz\n",
      "ERR950160_trimmed_2.fastq.gz\n",
      "ERR950161_trimmed_1.fastq.gz\n",
      "ERR950161_trimmed_2.fastq.gz\n",
      "ERR950162_trimmed_1.fastq.gz\n",
      "ERR950162_trimmed_2.fastq.gz\n",
      "ERR950163_trimmed_1.fastq.gz\n",
      "ERR950163_trimmed_2.fastq.gz\n",
      "ERR950177_trimmed_1.fastq.gz\n",
      "ERR950177_trimmed_2.fastq.gz\n",
      "ERR950178_trimmed_1.fastq.gz\n",
      "ERR950178_trimmed_2.fastq.gz\n",
      "ERR950179_trimmed_1.fastq.gz\n",
      "ERR950179_trimmed_2.fastq.gz\n",
      "ERR950180_trimmed_1.fastq.gz\n",
      "ERR950180_trimmed_2.fastq.gz\n",
      "ERR950181_trimmed_1.fastq.gz\n",
      "ERR950181_trimmed_2.fastq.gz\n",
      "ERR950182_trimmed_1.fastq.gz\n",
      "ERR950182_trimmed_2.fastq.gz\n",
      "ERR950186_trimmed_1.fastq.gz\n",
      "ERR950186_trimmed_2.fastq.gz\n",
      "ERR950187_trimmed_1.fastq.gz\n",
      "ERR950187_trimmed_2.fastq.gz\n",
      "fastqc_subset_trimmed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls data/subset_trimmed_RNA-seq/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bc571cc3-585c-444a-a60a-a7b35d590787",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning ERR950158_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950158_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950158_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950158_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950158_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950159_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950159_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950159_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950159_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950159_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950160_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950160_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950160_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950160_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950160_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950161_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950161_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950161_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950161_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950161_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950162_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950162_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950162_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950162_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950162_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950163_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950163_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950163_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950163_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950163_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950177_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950177_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950177_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950177_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950177_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950178_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950178_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950178_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950178_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950178_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950179_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950179_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950179_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950179_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950179_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950180_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950180_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950180_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950180_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950180_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950181_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950181_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950181_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950181_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950181_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950182_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950182_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950182_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950182_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950182_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950186_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950186_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950186_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950186_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950186_trimmed\n",
      "-----------------------------------------\n",
      "Aligning ERR950187_trimmed with Bowtie2...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\n",
      "Bowtie 2 version 2.5.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\n",
      "Usage: \n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r> | --interleaved <i> | -b <bam>} [-S <sam>]\n",
      "\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <i>        Files with interleaved paired-end FASTQ/FASTA reads\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <bam>      Files are unaligned BAM sorted by read name.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --tab5             query input files are TAB5 .tab5\n",
      "  --tab6             query input files are TAB6 .tab6\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -F k:<int>,i:<int> query input files are continuous FASTA where reads\n",
      "                     are substrings (k-mers) extracted from the FASTA file\n",
      "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --trim-to [3:|5:]<int> trim reads exceeding <int> bases from either 3' or 5' end\n",
      "                     If the read end is not specified then it defaults to 3 (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "\n",
      " Presets:                 Same as:\n",
      "  For --end-to-end:\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      "  For --local:\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\n",
      "\n",
      " Alignment:\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\n",
      "                     scan for the optimal seeded alignments\n",
      "  --end-to-end       entire read must align; no clipping (on)\n",
      "   OR\n",
      "  --local            local alignment; ends might be soft clipped (off)\n",
      "\n",
      " Scoring:\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\n",
      "\n",
      " Reporting:\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\n",
      "   OR\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\n",
      "   OR\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\n",
      "\n",
      " Effort:\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0)\n",
      "  -X/--maxins <int>  maximum fragment length (500)\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "  --dovetail         concordant when mates extend past each other\n",
      "  --no-contain       not concordant when one mate alignment contains other\n",
      "  --no-overlap       not concordant when mates overlap at all\n",
      "\n",
      " BAM:\n",
      "  --align-paired-reads\n",
      "                     Bowtie2 will, by default, attempt to align unpaired BAM reads.\n",
      "                     Use this option to align paired-end reads instead.\n",
      "  --preserve-tags    Preserve tags from the original BAM record by\n",
      "                     appending them to the end of the corresponding SAM output.\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>        write unpaired reads that didn't align to <path>\n",
      "  --al <path>        write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>   write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>   write pairs that aligned concordantly at least once to <path>\n",
      "    (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "    --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --quiet            print nothing to stderr except serious errors\n",
      "  --met-file <path>  send metrics to file at <path> (off)\n",
      "  --met-stderr       send metrics to stderr (off)\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\n",
      "  --no-unal          suppress SAM records for unaligned reads\n",
      "  --no-head          suppress header lines, i.e. lines starting with @\n",
      "  --no-sq            suppress @SQ header lines\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                     Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "  --sam-no-qname-trunc\n",
      "                     Suppress standard behavior of truncating readname at first whitespace \n",
      "                     at the expense of generating non-standard SAM.\n",
      "  --xeq              Use '='/'X', instead of 'M,' to specify matches/mismatches in SAM record.\n",
      "  --soft-clipped-unmapped-tlen\n",
      "                     Exclude soft-clipped bases when reporting TLEN.\n",
      "  --sam-append-comment\n",
      "                     Append FASTA/FASTQ comment to SAM record.\n",
      "  --sam-opt-config <config>\n",
      "                     Use <config>, example '-MD,YP,-AS', to toggle SAM Optional fields.\n",
      "\n",
      " Performance:\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic\n",
      "                     seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "(ERR): bowtie2-align exited with value 1\n",
      "bash: line 20: -x: command not found\n",
      "bash: line 21: -1: command not found\n",
      "bash: line 22: -2: command not found\n",
      "bash: line 23: -S: command not found\n",
      "bash: line 24: --threads: command not found\n",
      "[E::hts_open_format] Failed to open file \"alignment_Bowtie2_subset/ERR950187_trimmed.sam\" : No such file or directory\n",
      "samtools view: failed to open \"alignment_Bowtie2_subset/ERR950187_trimmed.sam\" for reading: No such file or directory\n",
      "[W::hts_set_opt] Cannot change block size for this format\n",
      "samtools sort: failed to read header from \"-\"\n",
      "rm: cannot remove 'alignment_Bowtie2_subset/ERR950187_trimmed.sam': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done aligning ERR950187_trimmed\n",
      "-----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p alignment_Bowtie2_subset\n",
    "\n",
    "# Loop over all FASTQ files ending in _1.fastq.gz in the subset_trimmed_RNA-seq folder\n",
    "for R1 in data/subset_trimmed_RNA-seq/*_1.fastq.gz; do\n",
    "  \n",
    "  # Extract the base name \n",
    "  BN=$(basename \"$R1\")\n",
    "  \n",
    "  # Remove the _1.fastq.gz suffix to get the sample prefix \n",
    "  SAMPLE=${BN%%_1.fastq.gz}\n",
    "  \n",
    "  # Construct the path to the mate 2 file \n",
    "  R2=\"data/subset_trimmed_RNA-seq/${SAMPLE}_2.fastq.gz\"\n",
    "  \n",
    "  echo \"Aligning $SAMPLE with Bowtie2...\"\n",
    "  \n",
    "  # Run Bowtie2 in local alignment mode for the paired-end reads\n",
    "  bowtie2 --local \n",
    "    -x data/hg38_cdna_index \n",
    "    -1 \"$R1\" \n",
    "    -2 \"$R2\" \n",
    "    -S alignment_Bowtie2_subset/${SAMPLE}.sam \n",
    "    --threads 4\n",
    "  \n",
    "  # Convert the SAM to a sorted BAM and create an index using samtools\n",
    "  samtools view -bS alignment_Bowtie2_subset/${SAMPLE}.sam | samtools sort -o alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  samtools index alignment_Bowtie2_subset/${SAMPLE}.bam\n",
    "  \n",
    "  # Remove the SAM file to save disk space\n",
    "  rm alignment_Bowtie2_subset/${SAMPLE}.sam\n",
    "  \n",
    "  echo \"Done aligning $SAMPLE\"\n",
    "  echo \"-----------------------------------------\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4f529bb9-c1bd-4769-9490-a0e799b23fac",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1464931\n",
      "-rw-r--r-- 1 ankitap nogroup  83164222 Feb 25 20:59 ERR950158_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   6049096 Feb 26 10:08 ERR950158_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  82982403 Feb 25 21:02 ERR950159_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5178176 Feb 26 10:08 ERR950159_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  83563426 Feb 25 21:05 ERR950160_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5964464 Feb 26 10:08 ERR950160_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  80792165 Feb 25 21:08 ERR950161_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5849536 Feb 26 10:08 ERR950161_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  69514274 Feb 25 21:11 ERR950162_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   4070304 Feb 26 10:08 ERR950162_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  84502479 Feb 25 21:14 ERR950163_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5878184 Feb 26 10:08 ERR950163_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  79448036 Feb 25 21:18 ERR950177_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5347400 Feb 26 10:08 ERR950177_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  81779036 Feb 25 21:21 ERR950178_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5591712 Feb 26 10:08 ERR950178_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  82932454 Feb 25 21:24 ERR950179_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5707192 Feb 26 10:08 ERR950179_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  82446219 Feb 25 21:27 ERR950180_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5620408 Feb 26 10:08 ERR950180_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  82012189 Feb 25 21:31 ERR950181_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5575584 Feb 26 10:08 ERR950181_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  80420591 Feb 25 21:34 ERR950182_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5504848 Feb 26 10:08 ERR950182_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup  81349387 Feb 25 21:38 ERR950186_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   5316016 Feb 26 10:09 ERR950186_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup       616 Feb 26 10:29 ERR950187.log\n",
      "-rw-r--r-- 1 ankitap nogroup  76373426 Feb 25 21:41 ERR950187_trimmed.bam\n",
      "-rw-r--r-- 1 ankitap nogroup   4899720 Feb 26 10:09 ERR950187_trimmed.bam.bai\n",
      "-rw-r--r-- 1 ankitap nogroup 292248577 Feb 26 10:29 ERR950187_trimmed_1.sam\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l alignment_Bowtie2_subset/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8553eeaa-f81f-4dfc-9e51-e65e897bd0df",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "998450 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "998450 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "450566 + 0 mapped (45.13% : N/A)\n",
      "450566 + 0 primary mapped (45.13% : N/A)\n",
      "998450 + 0 paired in sequencing\n",
      "499225 + 0 read1\n",
      "499225 + 0 read2\n",
      "422530 + 0 properly paired (42.32% : N/A)\n",
      "438398 + 0 with itself and mate mapped\n",
      "12168 + 0 singletons (1.22% : N/A)\n",
      "11068 + 0 with mate mapped to a different chr\n",
      "4064 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samtools flagstat alignment_Bowtie2_subset/ERR950187_trimmed.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3885de00-1837-489b-9ea9-3fb8e6e992a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERR950158_trimmed.bam\n",
      "ERR950158_trimmed.bam.bai\n",
      "ERR950159_trimmed.bam\n",
      "ERR950159_trimmed.bam.bai\n",
      "ERR950160_trimmed.bam\n",
      "ERR950160_trimmed.bam.bai\n",
      "ERR950161_trimmed.bam\n",
      "ERR950161_trimmed.bam.bai\n",
      "ERR950162_trimmed.bam\n",
      "ERR950162_trimmed.bam.bai\n",
      "ERR950163_trimmed.bam\n",
      "ERR950163_trimmed.bam.bai\n",
      "ERR950177_trimmed.bam\n",
      "ERR950177_trimmed.bam.bai\n",
      "ERR950178_trimmed.bam\n",
      "ERR950178_trimmed.bam.bai\n",
      "ERR950179_trimmed.bam\n",
      "ERR950179_trimmed.bam.bai\n",
      "ERR950180_trimmed.bam\n",
      "ERR950180_trimmed.bam.bai\n",
      "ERR950181_trimmed.bam\n",
      "ERR950181_trimmed.bam.bai\n",
      "ERR950182_trimmed.bam\n",
      "ERR950182_trimmed.bam.bai\n",
      "ERR950186_trimmed.bam\n",
      "ERR950186_trimmed.bam.bai\n",
      "ERR950187.log\n",
      "ERR950187_trimmed.bam\n",
      "ERR950187_trimmed.bam.bai\n",
      "ERR950187_trimmed_1.sam\n"
     ]
    }
   ],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8879a271-b191-4f8b-87e4-5c5ca4e0439c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
