{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "20df84a3-9f00-4ca0-b7c3-4eee5aa137bb",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1259212017.py, line 4)",
     "output_type": "error",
     "traceback": [
      "  \u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 4\u001b[39m\n\u001b[31m    \u001b[39m\u001b[31mchmod -R 777 group_Bjork\u001b[39m\n             ^\n\u001b[31mSyntaxError\u001b[39m\u001b[31m:\u001b[39m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#Run this window everytime you create a new directory or file\n",
    "#Gives permission for everyone to work with the files created\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups\n",
    "chmod -R 777 group_Bjork\n",
    "cd group_Bjork\n",
    "chmod -R 755 containers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a740629c-edaa-459f-bf11-eb55ee396764",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERR950177\n",
      "ERR950178\n",
      "ERR950179\n",
      "ERR950186\n",
      "ERR950187\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "grep -P 'Hp' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0abc39e9-6263-4523-ba58-75b8769c6f57",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Extract IDs from file\n",
    "Hp_ids=($(grep -P 'Hp' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'))\n",
    "\n",
    "# Define directories\n",
    "source_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq\"\n",
    "destination_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/test\"\n",
    "\n",
    "# Loop through extracted IDs\n",
    "for id in \"${Hp_ids[@]}\"; do\n",
    "    # Store both file paths in an array\n",
    "    files=(\"$source_dir/${id}_1.fastq.gz\" \"$source_dir/${id}_2.fastq.gz\")\n",
    "\n",
    "    # Loop through each file\n",
    "    for file in \"${files[@]}\"; do\n",
    "        if [[ -f \"$file\" ]]; then \n",
    "            cp \"$file\" \"$destination_dir\"\n",
    "            echo \"Copied $file to $destination_dir\"\n",
    "        else\n",
    "            echo \"File $file not found!\"\n",
    "        fi\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0b2ab41f-1f38-49c4-8dab-0b714072b0cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERR950158\n",
      "ERR950159\n",
      "ERR950162\n",
      "ERR950161\n",
      "ERR950177\n",
      "ERR950181\n",
      "ERR950178\n",
      "ERR950179\n",
      "ERR950180\n",
      "ERR950182\n",
      "ERR950186\n",
      "ERR950187\n",
      "ERR950160\n",
      "ERR950163\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "grep -P '(Atr|Hp)' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "babfde3a-e91b-413c-980b-599c6b3b5b0a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950158_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950158_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950159_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950159_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950162_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950162_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950161_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950161_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950181_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950181_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950180_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950180_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950182_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950182_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950160_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950160_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950163_1.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n",
      "Linked /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950163_2.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Extract IDs from file\n",
    "Hp_ids=($(grep -P '(Atr|Hp)' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'))\n",
    "\n",
    "# Define directories\n",
    "source_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq\"\n",
    "destination_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/TEST_data\"\n",
    "\n",
    "# Loop through extracted IDs\n",
    "for id in \"${Hp_ids[@]}\"; do\n",
    "    # Store both file paths in an array\n",
    "    files=(\"$source_dir/${id}_1.fastq.gz\" \"$source_dir/${id}_2.fastq.gz\")\n",
    "\n",
    "    # Loop through each file\n",
    "    for file in \"${files[@]}\"; do\n",
    "        if [[ -f \"$file\" ]]; then \n",
    "            ln -s -t \"$destination_dir\" \"$file\"\n",
    "            echo \"Linked $file to $destination_dir\" \n",
    "        else\n",
    "            echo \"File $file not found!\"\n",
    "        fi\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5fdf24ab-4dc2-4d60-8767-a3a3b03b91a6",
   "metadata": {},
   "source": [
    "# Adding RNA-seq subset data to a directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3dba3b2b-5a05-4989-982c-893aef1989bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950158_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950158_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950159_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950159_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950162_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950162_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950161_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950161_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950177_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950181_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950181_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950178_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950179_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950180_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950180_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950182_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950182_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950186_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950187_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950160_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950160_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950163_1.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
      "Copied /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq/ERR950163_2.fastq.gz.subset.fastq.gz to /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Extract IDs from file\n",
    "Hp_ids=($(grep -P '(Atr|Hp)' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'))\n",
    "\n",
    "# Define directories\n",
    "source_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq\"\n",
    "destination_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\"\n",
    "\n",
    "# Loop through extracted IDs\n",
    "for id in \"${Hp_ids[@]}\"; do\n",
    "    # Store both file paths in an array\n",
    "    files=(\"$source_dir/${id}_1.fastq.gz.subset.fastq.gz\" \"$source_dir/${id}_2.fastq.gz.subset.fastq.gz\")\n",
    "\n",
    "    # Loop through each file\n",
    "    for file in \"${files[@]}\"; do\n",
    "        if [[ -f \"$file\" ]]; then \n",
    "            cp \"$file\" \"$destination_dir\"\n",
    "            echo \"Copied $file to $destination_dir\"\n",
    "        else\n",
    "            echo \"File $file not found!\"\n",
    "        fi\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4c09bf14-c3ef-4618-aac2-a423a5f1fd2f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950158_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950158_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950159_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950159_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950160_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950160_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950161_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950161_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950162_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950162_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950163_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950163_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950177_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950177_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950178_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950178_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950179_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950179_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950180_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950180_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950181_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950181_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950182_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950182_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950186_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950186_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950187_1.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950187_2.fastq.gz.subset.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
    "mkdir -p fastqc\n",
    "fastqc -o fastqc *.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a4b6e8f4-a890-47df-88d5-bd74606d1c29",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=629624;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\  \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq/fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90mm\u001b[90m\u001b[0m \u001b[35m 93\u001b[\u001b[0m \u001b[35m  0%\u001b[0m \u001b[32m0/56\u001b[0m  35m 18%\u001b[0m \u001b[32m10/56\u001b[0m \u001b[2mERR950159_1.fastq.gz.subset_fastqc.html\u001b[0m35m 38%\u001b[0m \u001b[32m21/56\u001b[0m \u001b[2mERR950178_2.fastq.gz.subset_fastqc.html\u001b[0m35m 59%\u001b[0m \u001b[32m33/56\u001b[0m \u001b[2mERR950158_1.fastq.gz.subset_fastqc.html\u001b[0m35m 75%\u001b[0m \u001b[32m42/56\u001b[0m \u001b[2mERR950187_1.fastq.gz.subset_fastqc.html\u001b[0m%\u001b[0m \u001b[32m52/56\u001b[0m \u001b[2mERR950160_1.fastq.gz.subset_fastqc.zip\u001b[0m\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m56/56\u001b[0m  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 28 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq/fastqc\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a31ec54-52cf-4642-b0c3-8a4639be2f58",
   "metadata": {},
   "source": [
    "# Trimming and removal of duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0a153972-d0d6-49d0-a8ab-25531afd7fbc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt version 5.0\n",
      "\n",
      "Copyright (C) 2010 Marcel Martin <marcel.martin@scilifelab.se> and contributors\n",
      "\n",
      "Cutadapt removes adapter sequences from high-throughput sequencing reads.\n",
      "\n",
      "Usage:\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\n",
      "\n",
      "For paired-end reads:\n",
      "    cutadapt -a ADAPT1 -A ADAPT2 [options] -o out1.fastq -p out2.fastq in1.fastq in2.fastq\n",
      "\n",
      "Replace \"ADAPTER\" with the actual sequence of your 3' adapter. IUPAC wildcard\n",
      "characters are supported. All reads from input.fastq will be written to\n",
      "output.fastq with the adapter sequence removed. Adapter matching is\n",
      "error-tolerant. Multiple adapter sequences can be given (use further -a\n",
      "options), but only the best-matching adapter will be removed.\n",
      "\n",
      "Input may also be in FASTA format. Compressed input and output is supported and\n",
      "auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for\n",
      "standard input/output. Without the -o option, output is sent to standard output.\n",
      "\n",
      "Citation:\n",
      "\n",
      "Marcel Martin. Cutadapt removes adapter sequences from high-throughput\n",
      "sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.\n",
      "http://dx.doi.org/10.14806/ej.17.1.200\n",
      "\n",
      "Run \"cutadapt --help\" to see all command-line options.\n",
      "See https://cutadapt.readthedocs.io/ for full documentation.\n",
      "\n",
      "Options:\n",
      "  -h, --help            Show this help message and exit\n",
      "  --version             Show version number and exit\n",
      "  --debug               Print debug log. Use twice to also print DP matrices\n",
      "  -j CORES, --cores CORES\n",
      "                        Number of CPU cores to use. Use 0 to auto-detect.\n",
      "                        Default: 1\n",
      "\n",
      "Finding adapters:\n",
      "  Parameters -a, -g, -b specify adapters to be removed from each read (or from\n",
      "  R1 if data is paired-end. If specified multiple times, only the best\n",
      "  matching adapter is trimmed (but see the --times option). Use notation\n",
      "  'file:FILE' to read adapter sequences from a FASTA file.\n",
      "\n",
      "  -a ADAPTER, --adapter ADAPTER\n",
      "                        Sequence of an adapter ligated to the 3' end (paired\n",
      "                        data: of the first read). The adapter and subsequent\n",
      "                        bases are trimmed. If a '$' character is appended\n",
      "                        ('anchoring'), the adapter is only found if it is a\n",
      "                        suffix of the read.\n",
      "  -g ADAPTER, --front ADAPTER\n",
      "                        Sequence of an adapter ligated to the 5' end (paired\n",
      "                        data: of the first read). The adapter and any preceding\n",
      "                        bases are trimmed. Partial matches at the 5' end are\n",
      "                        allowed. If a '^' character is prepended ('anchoring'),\n",
      "                        the adapter is only found if it is a prefix of the read.\n",
      "  -b ADAPTER, --anywhere ADAPTER\n",
      "                        Sequence of an adapter that may be ligated to the 5' or\n",
      "                        3' end (paired data: of the first read). Both types of\n",
      "                        matches as described under -a and -g are allowed. If the\n",
      "                        first base of the read is part of the match, the\n",
      "                        behavior is as with -g, otherwise as with -a. This\n",
      "                        option is mostly for rescuing failed library\n",
      "                        preparations - do not use if you know which end your\n",
      "                        adapter was ligated to!\n",
      "  -e E, --error-rate E, --errors E\n",
      "                        Maximum allowed error rate (if 0 <= E < 1), or absolute\n",
      "                        number of errors for full-length adapter match (if E is\n",
      "                        an integer >= 1). Error rate = no. of errors divided by\n",
      "                        length of matching region. Default: 0.1 (10%)\n",
      "  --no-indels           Allow only mismatches in alignments. Default: allow both\n",
      "                        mismatches and indels\n",
      "  -n COUNT, --times COUNT\n",
      "                        Remove up to COUNT adapters from each read. Default: 1\n",
      "  -O MINLENGTH, --overlap MINLENGTH\n",
      "                        Require MINLENGTH overlap between read and adapter for\n",
      "                        an adapter to be found. Default: 3\n",
      "  --match-read-wildcards\n",
      "                        Interpret IUPAC wildcards in reads. Default: False\n",
      "  -N, --no-match-adapter-wildcards\n",
      "                        Do not interpret IUPAC wildcards in adapters.\n",
      "  --action {trim,retain,mask,lowercase,crop,none}\n",
      "                        What to do if a match was found. trim: trim adapter and\n",
      "                        up- or downstream sequence; retain: trim, but retain\n",
      "                        adapter; mask: replace with 'N' characters; lowercase:\n",
      "                        convert to lowercase; crop: trim up and downstream\n",
      "                        sequence; none: leave unchanged. Default: trim\n",
      "  --rc, --revcomp       Check both the read and its reverse complement for\n",
      "                        adapter matches. If match is on reverse-complemented\n",
      "                        version, output that one. Default: check only read\n",
      "\n",
      "Additional read modifications:\n",
      "  -u LEN, --cut LEN     Remove LEN bases from each read (or R1 if paired; use -U\n",
      "                        option for R2). If LEN is positive, remove bases from\n",
      "                        the beginning. If LEN is negative, remove bases from the\n",
      "                        end. Can be used twice if LENs have different signs.\n",
      "                        Applied *before* adapter trimming.\n",
      "  --nextseq-trim 3'CUTOFF\n",
      "                        NextSeq-specific quality trimming (each read). Trims\n",
      "                        also dark cycles appearing as high-quality G bases.\n",
      "  -q [5'CUTOFF,]3'CUTOFF, --quality-cutoff [5'CUTOFF,]3'CUTOFF\n",
      "                        Trim low-quality bases from 5' and/or 3' ends of each\n",
      "                        read before adapter removal. Applied to both reads if\n",
      "                        data is paired. If one value is given, only the 3' end\n",
      "                        is trimmed. If two comma-separated cutoffs are given,\n",
      "                        the 5' end is trimmed with the first cutoff, the 3' end\n",
      "                        with the second.\n",
      "  --quality-base N      Assume that quality values in FASTQ are encoded as\n",
      "                        ascii(quality + N). This needs to be set to 64 for some\n",
      "                        old Illumina FASTQ files. Default: 33\n",
      "  --poly-a              Trim poly-A tails\n",
      "  --length LENGTH, -l LENGTH\n",
      "                        Shorten reads to LENGTH. Positive values remove bases at\n",
      "                        the end while negative ones remove bases at the\n",
      "                        beginning. This and the following modifications are\n",
      "                        applied after adapter trimming.\n",
      "  --trim-n              Trim N's on ends of reads.\n",
      "  --length-tag TAG      Search for TAG followed by a decimal number in the\n",
      "                        description field of the read. Replace the decimal\n",
      "                        number with the correct length of the trimmed read. For\n",
      "                        example, use --length-tag 'length=' to correct fields\n",
      "                        like 'length=123'.\n",
      "  --strip-suffix STRIP_SUFFIX\n",
      "                        Remove this suffix from read names if present. Can be\n",
      "                        given multiple times.\n",
      "  -x PREFIX, --prefix PREFIX\n",
      "                        Add this prefix to read names. Use {name} to insert the\n",
      "                        name of the matching adapter.\n",
      "  -y SUFFIX, --suffix SUFFIX\n",
      "                        Add this suffix to read names; can also include {name}\n",
      "  --rename TEMPLATE     Rename reads using TEMPLATE containing variables such as\n",
      "                        {id}, {adapter_name} etc. (see documentation)\n",
      "  --zero-cap, -z        Change negative quality values to zero.\n",
      "\n",
      "Filtering of processed reads:\n",
      "  Filters are applied after above read modifications. Paired-end reads are\n",
      "  always discarded pairwise (see also --pair-filter).\n",
      "\n",
      "  -m LEN[:LEN2], --minimum-length LEN[:LEN2]\n",
      "                        Discard reads shorter than LEN. Default: 0\n",
      "  -M LEN[:LEN2], --maximum-length LEN[:LEN2]\n",
      "                        Discard reads longer than LEN. Default: no limit\n",
      "  --max-n COUNT         Discard reads with more than COUNT 'N' bases. If COUNT\n",
      "                        is a number between 0 and 1, it is interpreted as a\n",
      "                        fraction of the read length.\n",
      "  --max-expected-errors ERRORS, --max-ee ERRORS\n",
      "                        Discard reads whose expected number of errors (computed\n",
      "                        from quality values) exceeds ERRORS.\n",
      "  --max-average-error-rate ERROR_RATE, --max-aer ERROR_RATE\n",
      "                        as --max-expected-errors (see above), but divided by\n",
      "                        length to account for reads of varying length.\n",
      "  --discard-trimmed, --discard\n",
      "                        Discard reads that contain an adapter. Use also -O to\n",
      "                        avoid discarding too many randomly matching reads.\n",
      "  --discard-untrimmed, --trimmed-only\n",
      "                        Discard reads that do not contain an adapter.\n",
      "  --discard-casava      Discard reads that did not pass CASAVA filtering (header\n",
      "                        has :Y:).\n",
      "\n",
      "Output:\n",
      "  --quiet               Print only error messages.\n",
      "  --report {full,minimal}\n",
      "                        Which type of report to print: 'full' or 'minimal'.\n",
      "                        Default: full\n",
      "  --json FILE           Dump report in JSON format to FILE\n",
      "  -o FILE, --output FILE\n",
      "                        Write trimmed reads to FILE. FASTQ or FASTA format is\n",
      "                        chosen depending on input. Summary report is sent to\n",
      "                        standard output. Use '{name}' for demultiplexing (see\n",
      "                        docs). Default: write to standard output\n",
      "  --fasta               Output FASTA to standard output even on FASTQ input.\n",
      "  --compression-level N\n",
      "                        Compression level for compressed output files. Default:\n",
      "                        1\n",
      "  -Z                    DEPRECATED because compression level 1 is now the\n",
      "                        default.\n",
      "  --info-file FILE      Write information about each read and its adapter\n",
      "                        matches into FILE. See the documentation for the file\n",
      "                        format.\n",
      "  -r FILE, --rest-file FILE\n",
      "                        When the adapter matches in the middle of a read, write\n",
      "                        the rest (after the adapter) to FILE.\n",
      "  --wildcard-file FILE  When the adapter has N wildcard bases, write adapter\n",
      "                        bases matching wildcard positions to FILE. (Inaccurate\n",
      "                        with indels.)\n",
      "  --too-short-output FILE\n",
      "                        Write reads that are too short (according to length\n",
      "                        specified by -m) to FILE. Default: discard reads\n",
      "  --too-long-output FILE\n",
      "                        Write reads that are too long (according to length\n",
      "                        specified by -M) to FILE. Default: discard reads\n",
      "  --untrimmed-output FILE\n",
      "                        Write reads that do not contain any adapter to FILE.\n",
      "                        Default: output to same file as trimmed reads\n",
      "\n",
      "Paired-end options:\n",
      "  The -A/-G/-B/-U/-Q options work like their lowercase counterparts, but are\n",
      "  applied to R2 (second read in pair)\n",
      "\n",
      "  -A ADAPTER            3' adapter to be removed from R2\n",
      "  -G ADAPTER            5' adapter to be removed from R2\n",
      "  -B ADAPTER            5'/3 adapter to be removed from R2\n",
      "  -U LENGTH             Remove LENGTH bases from R2\n",
      "  -Q [5'CUTOFF,]3'CUTOFF\n",
      "                        Quality-trimming cutoff for R2. Default: same as for R1\n",
      "  -L LENGTH             Shorten R2 to LENGTH. Default: same as for R1\n",
      "  -p FILE, --paired-output FILE\n",
      "                        Write R2 to FILE.\n",
      "  --pair-adapters       Treat adapters given with -a/-A etc. as pairs. Either\n",
      "                        both or none are removed from each read pair.\n",
      "  --pair-filter {any,both,first}\n",
      "                        Which of the reads in a paired-end read have to match\n",
      "                        the filtering criterion in order for the pair to be\n",
      "                        filtered. Default: any\n",
      "  --interleaved         Read and/or write interleaved paired-end reads.\n",
      "  --untrimmed-paired-output FILE\n",
      "                        Write second read in a pair to this FILE when no adapter\n",
      "                        was found. Use with --untrimmed-output. Default: output\n",
      "                        to same file as trimmed reads\n",
      "  --too-short-paired-output FILE\n",
      "                        Write second read in a pair to this file if pair is too\n",
      "                        short.\n",
      "  --too-long-paired-output FILE\n",
      "                        Write second read in a pair to this file if pair is too\n",
      "                        long.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cutadapt --help"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03c9717e-a84b-4c17-9396-8104831b98ce",
   "metadata": {},
   "source": [
    "# Removing adapters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4688ea10-4690-4999-8f0c-b4f2291db462",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950158_trimmed_R1.fastq.gz -p trimmed/ERR950158_trimmed_R2.fastq.gz ERR950158_1.fastq.gz.subset.fastq.gz ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 109,516 (21.9%)\n",
      "  Read 2 with adapter:                 107,125 (21.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 468 (0.1%)\n",
      "Pairs written (passing filters):       499,532 (99.9%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,310,392 bp (95.4%)\n",
      "  Read 1:    48,132,940 bp\n",
      "  Read 2:    48,177,452 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 109516 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.1%\n",
      "  C: 38.6%\n",
      "  G: 23.7%\n",
      "  T: 12.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11551\t7812.5\t0\t11551\n",
      "4\t5103\t1953.1\t0\t5103\n",
      "5\t3067\t488.3\t0\t3067\n",
      "6\t3290\t122.1\t0\t3290\n",
      "7\t2974\t30.5\t0\t2974\n",
      "8\t2837\t7.6\t0\t2837\n",
      "9\t2671\t1.9\t0\t2605 66\n",
      "10\t3270\t0.5\t1\t3094 176\n",
      "11\t2820\t0.1\t1\t2645 175\n",
      "12\t2785\t0.0\t1\t2600 185\n",
      "13\t2742\t0.0\t1\t2581 161\n",
      "14\t2755\t0.0\t1\t2595 160\n",
      "15\t2577\t0.0\t1\t2426 151\n",
      "16\t2536\t0.0\t1\t2377 159\n",
      "17\t2459\t0.0\t1\t2348 111\n",
      "18\t2244\t0.0\t1\t2131 113\n",
      "19\t2319\t0.0\t1\t2191 128\n",
      "20\t2385\t0.0\t1\t2254 131\n",
      "21\t2263\t0.0\t1\t2174 89\n",
      "22\t2068\t0.0\t1\t1971 97\n",
      "23\t2082\t0.0\t1\t1992 90\n",
      "24\t2280\t0.0\t1\t2168 112\n",
      "25\t2071\t0.0\t1\t1966 105\n",
      "26\t2124\t0.0\t1\t2041 83\n",
      "27\t1960\t0.0\t1\t1881 79\n",
      "28\t1976\t0.0\t1\t1890 86\n",
      "29\t1837\t0.0\t1\t1750 87\n",
      "30\t1803\t0.0\t1\t1731 72\n",
      "31\t2031\t0.0\t1\t1948 83\n",
      "32\t1558\t0.0\t1\t1488 70\n",
      "33\t1516\t0.0\t1\t1458 58\n",
      "34\t1459\t0.0\t1\t1406 53\n",
      "35\t1344\t0.0\t1\t1294 50\n",
      "36\t1255\t0.0\t1\t1186 69\n",
      "37\t1233\t0.0\t1\t1182 51\n",
      "38\t1141\t0.0\t1\t1093 48\n",
      "39\t1105\t0.0\t1\t1058 47\n",
      "40\t1118\t0.0\t1\t1069 49\n",
      "41\t966\t0.0\t1\t924 42\n",
      "42\t948\t0.0\t1\t909 39\n",
      "43\t882\t0.0\t1\t842 40\n",
      "44\t856\t0.0\t1\t810 46\n",
      "45\t851\t0.0\t1\t815 36\n",
      "46\t825\t0.0\t1\t802 23\n",
      "47\t771\t0.0\t1\t736 35\n",
      "48\t742\t0.0\t1\t708 34\n",
      "49\t812\t0.0\t1\t777 35\n",
      "50\t639\t0.0\t1\t623 16\n",
      "51\t610\t0.0\t1\t587 23\n",
      "52\t573\t0.0\t1\t551 22\n",
      "53\t443\t0.0\t1\t422 21\n",
      "54\t465\t0.0\t1\t438 27\n",
      "55\t419\t0.0\t1\t400 19\n",
      "56\t327\t0.0\t1\t311 16\n",
      "57\t337\t0.0\t1\t321 16\n",
      "58\t289\t0.0\t1\t270 19\n",
      "59\t271\t0.0\t1\t262 9\n",
      "60\t259\t0.0\t1\t248 11\n",
      "61\t228\t0.0\t1\t216 12\n",
      "62\t210\t0.0\t1\t199 11\n",
      "63\t203\t0.0\t1\t199 4\n",
      "64\t198\t0.0\t1\t193 5\n",
      "65\t174\t0.0\t1\t166 8\n",
      "66\t150\t0.0\t1\t140 10\n",
      "67\t161\t0.0\t1\t153 8\n",
      "68\t138\t0.0\t1\t131 7\n",
      "69\t114\t0.0\t1\t106 8\n",
      "70\t105\t0.0\t1\t101 4\n",
      "71\t89\t0.0\t1\t86 3\n",
      "72\t96\t0.0\t1\t92 4\n",
      "73\t98\t0.0\t1\t92 6\n",
      "74\t98\t0.0\t1\t93 5\n",
      "75\t60\t0.0\t1\t50 10\n",
      "76\t48\t0.0\t1\t47 1\n",
      "77\t40\t0.0\t1\t39 1\n",
      "78\t36\t0.0\t1\t33 3\n",
      "79\t26\t0.0\t1\t24 2\n",
      "80\t37\t0.0\t1\t32 5\n",
      "81\t18\t0.0\t1\t17 1\n",
      "82\t15\t0.0\t1\t13 2\n",
      "83\t20\t0.0\t1\t15 5\n",
      "84\t11\t0.0\t1\t10 1\n",
      "85\t12\t0.0\t1\t11 1\n",
      "86\t8\t0.0\t1\t7 1\n",
      "87\t5\t0.0\t1\t5\n",
      "88\t6\t0.0\t1\t3 3\n",
      "89\t8\t0.0\t1\t5 3\n",
      "90\t5\t0.0\t1\t4 1\n",
      "91\t7\t0.0\t1\t6 1\n",
      "92\t5\t0.0\t1\t4 1\n",
      "93\t18\t0.0\t1\t13 5\n",
      "94\t26\t0.0\t1\t21 5\n",
      "95\t34\t0.0\t1\t25 9\n",
      "96\t35\t0.0\t1\t26 9\n",
      "97\t26\t0.0\t1\t20 6\n",
      "98\t8\t0.0\t1\t8\n",
      "99\t6\t0.0\t1\t2 4\n",
      "100\t2\t0.0\t1\t1 1\n",
      "101\t38\t0.0\t1\t22 16\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 107125 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.4%\n",
      "  C: 22.7%\n",
      "  G: 28.5%\n",
      "  T: 30.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10833\t7812.5\t0\t10833\n",
      "4\t4555\t1953.1\t0\t4555\n",
      "5\t3155\t488.3\t0\t3155\n",
      "6\t3249\t122.1\t0\t3249\n",
      "7\t3055\t30.5\t0\t3055\n",
      "8\t2822\t7.6\t0\t2822\n",
      "9\t2698\t1.9\t0\t2666 32\n",
      "10\t3235\t0.5\t1\t3092 143\n",
      "11\t2769\t0.1\t1\t2633 136\n",
      "12\t2760\t0.0\t1\t2644 116\n",
      "13\t2717\t0.0\t1\t2607 110\n",
      "14\t2722\t0.0\t1\t2620 102\n",
      "15\t2568\t0.0\t1\t2474 94\n",
      "16\t2512\t0.0\t1\t2411 101\n",
      "17\t2441\t0.0\t1\t2330 111\n",
      "18\t2223\t0.0\t1\t2140 83\n",
      "19\t2280\t0.0\t1\t2181 99\n",
      "20\t2359\t0.0\t1\t2243 116\n",
      "21\t2254\t0.0\t1\t2165 89\n",
      "22\t2038\t0.0\t1\t1957 81\n",
      "23\t2063\t0.0\t1\t1974 89\n",
      "24\t2249\t0.0\t1\t2153 96\n",
      "25\t2043\t0.0\t1\t1972 71\n",
      "26\t2107\t0.0\t1\t2020 87\n",
      "27\t1948\t0.0\t1\t1865 83\n",
      "28\t1935\t0.0\t1\t1878 57\n",
      "29\t1809\t0.0\t1\t1732 77\n",
      "30\t1784\t0.0\t1\t1712 72\n",
      "31\t1995\t0.0\t1\t1926 69\n",
      "32\t1524\t0.0\t1\t1444 80\n",
      "33\t1498\t0.0\t1\t1423 75\n",
      "34\t1445\t0.0\t1\t1389 56\n",
      "35\t1326\t0.0\t1\t1284 42\n",
      "36\t1228\t0.0\t1\t1174 54\n",
      "37\t1207\t0.0\t1\t1167 40\n",
      "38\t1115\t0.0\t1\t1076 39\n",
      "39\t1092\t0.0\t1\t1045 47\n",
      "40\t1091\t0.0\t1\t1035 56\n",
      "41\t940\t0.0\t1\t907 33\n",
      "42\t934\t0.0\t1\t904 30\n",
      "43\t865\t0.0\t1\t831 34\n",
      "44\t831\t0.0\t1\t792 39\n",
      "45\t835\t0.0\t1\t794 41\n",
      "46\t811\t0.0\t1\t777 34\n",
      "47\t745\t0.0\t1\t716 29\n",
      "48\t728\t0.0\t1\t701 27\n",
      "49\t791\t0.0\t1\t753 38\n",
      "50\t627\t0.0\t1\t597 30\n",
      "51\t597\t0.0\t1\t574 23\n",
      "52\t557\t0.0\t1\t535 22\n",
      "53\t429\t0.0\t1\t409 20\n",
      "54\t448\t0.0\t1\t425 23\n",
      "55\t404\t0.0\t1\t387 17\n",
      "56\t316\t0.0\t1\t302 14\n",
      "57\t323\t0.0\t1\t317 6\n",
      "58\t280\t0.0\t1\t269 11\n",
      "59\t265\t0.0\t1\t249 16\n",
      "60\t252\t0.0\t1\t241 11\n",
      "61\t219\t0.0\t1\t210 9\n",
      "62\t203\t0.0\t1\t192 11\n",
      "63\t199\t0.0\t1\t185 14\n",
      "64\t193\t0.0\t1\t181 12\n",
      "65\t171\t0.0\t1\t163 8\n",
      "66\t142\t0.0\t1\t132 10\n",
      "67\t154\t0.0\t1\t147 7\n",
      "68\t134\t0.0\t1\t127 7\n",
      "69\t107\t0.0\t1\t103 4\n",
      "70\t99\t0.0\t1\t94 5\n",
      "71\t84\t0.0\t1\t80 4\n",
      "72\t98\t0.0\t1\t87 11\n",
      "73\t87\t0.0\t1\t82 5\n",
      "74\t94\t0.0\t1\t88 6\n",
      "75\t50\t0.0\t1\t48 2\n",
      "76\t47\t0.0\t1\t43 4\n",
      "77\t34\t0.0\t1\t32 2\n",
      "78\t34\t0.0\t1\t31 3\n",
      "79\t23\t0.0\t1\t22 1\n",
      "80\t34\t0.0\t1\t32 2\n",
      "81\t17\t0.0\t1\t15 2\n",
      "82\t13\t0.0\t1\t13\n",
      "83\t14\t0.0\t1\t12 2\n",
      "84\t10\t0.0\t1\t10\n",
      "85\t10\t0.0\t1\t9 1\n",
      "86\t6\t0.0\t1\t6\n",
      "87\t2\t0.0\t1\t2\n",
      "88\t4\t0.0\t1\t1 3\n",
      "89\t3\t0.0\t1\t3\n",
      "90\t2\t0.0\t1\t2\n",
      "91\t2\t0.0\t1\t1 1\n",
      "92\t3\t0.0\t1\t2 1\n",
      "93\t12\t0.0\t1\t10 2\n",
      "94\t22\t0.0\t1\t22\n",
      "95\t28\t0.0\t1\t27 1\n",
      "96\t29\t0.0\t1\t24 5\n",
      "97\t20\t0.0\t1\t18 2\n",
      "98\t7\t0.0\t1\t4 3\n",
      "99\t3\t0.0\t1\t2 1\n",
      "100\t1\t0.0\t1\t0 1\n",
      "101\t29\t0.0\t1\t23 6\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950159_trimmed_R1.fastq.gz -p trimmed/ERR950159_trimmed_R2.fastq.gz ERR950159_1.fastq.gz.subset.fastq.gz ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  92,288 (18.5%)\n",
      "  Read 2 with adapter:                  91,775 (18.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 296 (0.1%)\n",
      "Pairs written (passing filters):       499,704 (99.9%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     97,417,128 bp (96.5%)\n",
      "  Read 1:    48,700,085 bp\n",
      "  Read 2:    48,717,043 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 92288 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.7%\n",
      "  C: 41.0%\n",
      "  G: 21.4%\n",
      "  T: 12.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10665\t7812.5\t0\t10665\n",
      "4\t4671\t1953.1\t0\t4671\n",
      "5\t2781\t488.3\t0\t2781\n",
      "6\t2908\t122.1\t0\t2908\n",
      "7\t2894\t30.5\t0\t2894\n",
      "8\t2589\t7.6\t0\t2589\n",
      "9\t2626\t1.9\t0\t2600 26\n",
      "10\t3005\t0.5\t1\t2838 167\n",
      "11\t2617\t0.1\t1\t2442 175\n",
      "12\t2555\t0.0\t1\t2393 162\n",
      "13\t2483\t0.0\t1\t2310 173\n",
      "14\t2709\t0.0\t1\t2534 175\n",
      "15\t2586\t0.0\t1\t2418 168\n",
      "16\t2450\t0.0\t1\t2261 189\n",
      "17\t2163\t0.0\t1\t2017 146\n",
      "18\t1991\t0.0\t1\t1874 117\n",
      "19\t1981\t0.0\t1\t1847 134\n",
      "20\t1933\t0.0\t1\t1823 110\n",
      "21\t1781\t0.0\t1\t1687 94\n",
      "22\t1688\t0.0\t1\t1584 104\n",
      "23\t1785\t0.0\t1\t1696 89\n",
      "24\t1929\t0.0\t1\t1800 129\n",
      "25\t1750\t0.0\t1\t1640 110\n",
      "26\t1899\t0.0\t1\t1812 87\n",
      "27\t1752\t0.0\t1\t1673 79\n",
      "28\t1664\t0.0\t1\t1559 105\n",
      "29\t1440\t0.0\t1\t1346 94\n",
      "30\t1536\t0.0\t1\t1465 71\n",
      "31\t1915\t0.0\t1\t1817 98\n",
      "32\t1208\t0.0\t1\t1146 62\n",
      "33\t1121\t0.0\t1\t1078 43\n",
      "34\t1014\t0.0\t1\t965 49\n",
      "35\t934\t0.0\t1\t885 49\n",
      "36\t891\t0.0\t1\t840 51\n",
      "37\t905\t0.0\t1\t858 47\n",
      "38\t833\t0.0\t1\t779 54\n",
      "39\t747\t0.0\t1\t708 39\n",
      "40\t946\t0.0\t1\t889 57\n",
      "41\t668\t0.0\t1\t638 30\n",
      "42\t678\t0.0\t1\t654 24\n",
      "43\t610\t0.0\t1\t585 25\n",
      "44\t613\t0.0\t1\t581 32\n",
      "45\t490\t0.0\t1\t476 14\n",
      "46\t459\t0.0\t1\t447 12\n",
      "47\t438\t0.0\t1\t412 26\n",
      "48\t447\t0.0\t1\t433 14\n",
      "49\t545\t0.0\t1\t515 30\n",
      "50\t394\t0.0\t1\t378 16\n",
      "51\t403\t0.0\t1\t390 13\n",
      "52\t353\t0.0\t1\t332 21\n",
      "53\t309\t0.0\t1\t292 17\n",
      "54\t254\t0.0\t1\t244 10\n",
      "55\t187\t0.0\t1\t178 9\n",
      "56\t174\t0.0\t1\t161 13\n",
      "57\t196\t0.0\t1\t184 12\n",
      "58\t157\t0.0\t1\t146 11\n",
      "59\t139\t0.0\t1\t133 6\n",
      "60\t145\t0.0\t1\t138 7\n",
      "61\t112\t0.0\t1\t106 6\n",
      "62\t110\t0.0\t1\t106 4\n",
      "63\t94\t0.0\t1\t88 6\n",
      "64\t96\t0.0\t1\t91 5\n",
      "65\t93\t0.0\t1\t88 5\n",
      "66\t70\t0.0\t1\t63 7\n",
      "67\t62\t0.0\t1\t61 1\n",
      "68\t63\t0.0\t1\t59 4\n",
      "69\t51\t0.0\t1\t47 4\n",
      "70\t50\t0.0\t1\t48 2\n",
      "71\t54\t0.0\t1\t49 5\n",
      "72\t47\t0.0\t1\t41 6\n",
      "73\t48\t0.0\t1\t48\n",
      "74\t30\t0.0\t1\t29 1\n",
      "75\t15\t0.0\t1\t13 2\n",
      "76\t14\t0.0\t1\t13 1\n",
      "77\t21\t0.0\t1\t20 1\n",
      "78\t9\t0.0\t1\t5 4\n",
      "79\t16\t0.0\t1\t12 4\n",
      "80\t14\t0.0\t1\t13 1\n",
      "81\t12\t0.0\t1\t7 5\n",
      "82\t15\t0.0\t1\t9 6\n",
      "83\t5\t0.0\t1\t3 2\n",
      "84\t4\t0.0\t1\t3 1\n",
      "85\t2\t0.0\t1\t2\n",
      "86\t5\t0.0\t1\t4 1\n",
      "87\t1\t0.0\t1\t0 1\n",
      "88\t5\t0.0\t1\t1 4\n",
      "89\t6\t0.0\t1\t4 2\n",
      "90\t5\t0.0\t1\t0 5\n",
      "91\t8\t0.0\t1\t8\n",
      "92\t7\t0.0\t1\t4 3\n",
      "93\t7\t0.0\t1\t4 3\n",
      "94\t22\t0.0\t1\t21 1\n",
      "95\t22\t0.0\t1\t19 3\n",
      "96\t29\t0.0\t1\t23 6\n",
      "97\t22\t0.0\t1\t19 3\n",
      "98\t6\t0.0\t1\t5 1\n",
      "99\t6\t0.0\t1\t5 1\n",
      "100\t5\t0.0\t1\t0 5\n",
      "101\t21\t0.0\t1\t0 21\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 91775 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.9%\n",
      "  C: 20.5%\n",
      "  G: 30.7%\n",
      "  T: 30.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10522\t7812.5\t0\t10522\n",
      "4\t4347\t1953.1\t0\t4347\n",
      "5\t2898\t488.3\t0\t2898\n",
      "6\t3025\t122.1\t0\t3025\n",
      "7\t2944\t30.5\t0\t2944\n",
      "8\t2670\t7.6\t0\t2670\n",
      "9\t2681\t1.9\t0\t2652 29\n",
      "10\t3010\t0.5\t1\t2898 112\n",
      "11\t2613\t0.1\t1\t2484 129\n",
      "12\t2551\t0.0\t1\t2437 114\n",
      "13\t2503\t0.0\t1\t2400 103\n",
      "14\t2717\t0.0\t1\t2615 102\n",
      "15\t2570\t0.0\t1\t2472 98\n",
      "16\t2453\t0.0\t1\t2360 93\n",
      "17\t2160\t0.0\t1\t2076 84\n",
      "18\t1996\t0.0\t1\t1920 76\n",
      "19\t1981\t0.0\t1\t1889 92\n",
      "20\t1925\t0.0\t1\t1855 70\n",
      "21\t1772\t0.0\t1\t1707 65\n",
      "22\t1692\t0.0\t1\t1623 69\n",
      "23\t1776\t0.0\t1\t1702 74\n",
      "24\t1910\t0.0\t1\t1826 84\n",
      "25\t1741\t0.0\t1\t1680 61\n",
      "26\t1889\t0.0\t1\t1809 80\n",
      "27\t1744\t0.0\t1\t1668 76\n",
      "28\t1651\t0.0\t1\t1592 59\n",
      "29\t1421\t0.0\t1\t1358 63\n",
      "30\t1523\t0.0\t1\t1475 48\n",
      "31\t1908\t0.0\t1\t1826 82\n",
      "32\t1193\t0.0\t1\t1148 45\n",
      "33\t1110\t0.0\t1\t1057 53\n",
      "34\t1008\t0.0\t1\t962 46\n",
      "35\t926\t0.0\t1\t890 36\n",
      "36\t882\t0.0\t1\t847 35\n",
      "37\t900\t0.0\t1\t866 34\n",
      "38\t819\t0.0\t1\t775 44\n",
      "39\t741\t0.0\t1\t715 26\n",
      "40\t941\t0.0\t1\t891 50\n",
      "41\t658\t0.0\t1\t635 23\n",
      "42\t668\t0.0\t1\t641 27\n",
      "43\t603\t0.0\t1\t585 18\n",
      "44\t595\t0.0\t1\t573 22\n",
      "45\t491\t0.0\t1\t464 27\n",
      "46\t453\t0.0\t1\t436 17\n",
      "47\t429\t0.0\t1\t411 18\n",
      "48\t438\t0.0\t1\t409 29\n",
      "49\t528\t0.0\t1\t510 18\n",
      "50\t390\t0.0\t1\t368 22\n",
      "51\t396\t0.0\t1\t372 24\n",
      "52\t346\t0.0\t1\t332 14\n",
      "53\t298\t0.0\t1\t288 10\n",
      "54\t245\t0.0\t1\t236 9\n",
      "55\t183\t0.0\t1\t176 7\n",
      "56\t168\t0.0\t1\t161 7\n",
      "57\t186\t0.0\t1\t179 7\n",
      "58\t150\t0.0\t1\t147 3\n",
      "59\t132\t0.0\t1\t124 8\n",
      "60\t140\t0.0\t1\t131 9\n",
      "61\t105\t0.0\t1\t100 5\n",
      "62\t107\t0.0\t1\t104 3\n",
      "63\t88\t0.0\t1\t87 1\n",
      "64\t88\t0.0\t1\t86 2\n",
      "65\t88\t0.0\t1\t85 3\n",
      "66\t68\t0.0\t1\t61 7\n",
      "67\t59\t0.0\t1\t56 3\n",
      "68\t58\t0.0\t1\t56 2\n",
      "69\t48\t0.0\t1\t46 2\n",
      "70\t48\t0.0\t1\t48\n",
      "71\t53\t0.0\t1\t49 4\n",
      "72\t43\t0.0\t1\t41 2\n",
      "73\t46\t0.0\t1\t44 2\n",
      "74\t27\t0.0\t1\t26 1\n",
      "75\t14\t0.0\t1\t13 1\n",
      "76\t14\t0.0\t1\t13 1\n",
      "77\t15\t0.0\t1\t13 2\n",
      "78\t4\t0.0\t1\t3 1\n",
      "79\t12\t0.0\t1\t9 3\n",
      "80\t11\t0.0\t1\t10 1\n",
      "81\t7\t0.0\t1\t7\n",
      "82\t7\t0.0\t1\t7\n",
      "83\t2\t0.0\t1\t1 1\n",
      "84\t3\t0.0\t1\t3\n",
      "86\t4\t0.0\t1\t3 1\n",
      "87\t1\t0.0\t1\t0 1\n",
      "88\t1\t0.0\t1\t1\n",
      "89\t4\t0.0\t1\t4\n",
      "91\t7\t0.0\t1\t6 1\n",
      "92\t4\t0.0\t1\t4\n",
      "93\t2\t0.0\t1\t2\n",
      "94\t18\t0.0\t1\t18\n",
      "95\t21\t0.0\t1\t21\n",
      "96\t26\t0.0\t1\t25 1\n",
      "97\t18\t0.0\t1\t16 2\n",
      "98\t5\t0.0\t1\t4 1\n",
      "99\t5\t0.0\t1\t5\n",
      "100\t6\t0.0\t1\t4 2\n",
      "101\t28\t0.0\t1\t15 13\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950160_trimmed_R1.fastq.gz -p trimmed/ERR950160_trimmed_R2.fastq.gz ERR950160_1.fastq.gz.subset.fastq.gz ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  98,842 (19.8%)\n",
      "  Read 2 with adapter:                  95,762 (19.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,075 (0.2%)\n",
      "Pairs written (passing filters):       498,925 (99.8%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,759,841 bp (95.8%)\n",
      "  Read 1:    48,358,398 bp\n",
      "  Read 2:    48,401,443 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 98842 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.2%\n",
      "  C: 38.8%\n",
      "  G: 22.9%\n",
      "  T: 12.9%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11345\t7812.5\t0\t11345\n",
      "4\t4891\t1953.1\t0\t4891\n",
      "5\t2557\t488.3\t0\t2557\n",
      "6\t2913\t122.1\t0\t2913\n",
      "7\t2820\t30.5\t0\t2820\n",
      "8\t2578\t7.6\t0\t2578\n",
      "9\t2433\t1.9\t0\t2338 95\n",
      "10\t3016\t0.5\t1\t2824 192\n",
      "11\t2351\t0.1\t1\t2164 187\n",
      "12\t2421\t0.0\t1\t2233 188\n",
      "13\t2384\t0.0\t1\t2235 149\n",
      "14\t2685\t0.0\t1\t2525 160\n",
      "15\t2502\t0.0\t1\t2334 168\n",
      "16\t2210\t0.0\t1\t2064 146\n",
      "17\t2161\t0.0\t1\t2028 133\n",
      "18\t2038\t0.0\t1\t1918 120\n",
      "19\t2055\t0.0\t1\t1939 116\n",
      "20\t2023\t0.0\t1\t1920 103\n",
      "21\t1903\t0.0\t1\t1801 102\n",
      "22\t1904\t0.0\t1\t1802 102\n",
      "23\t1781\t0.0\t1\t1695 86\n",
      "24\t1832\t0.0\t1\t1754 78\n",
      "25\t1810\t0.0\t1\t1724 86\n",
      "26\t2022\t0.0\t1\t1932 90\n",
      "27\t1804\t0.0\t1\t1714 90\n",
      "28\t1770\t0.0\t1\t1696 74\n",
      "29\t1497\t0.0\t1\t1410 87\n",
      "30\t1511\t0.0\t1\t1436 75\n",
      "31\t2163\t0.0\t1\t2079 84\n",
      "32\t1381\t0.0\t1\t1308 73\n",
      "33\t1294\t0.0\t1\t1249 45\n",
      "34\t1217\t0.0\t1\t1164 53\n",
      "35\t1102\t0.0\t1\t1045 57\n",
      "36\t1052\t0.0\t1\t1004 48\n",
      "37\t1166\t0.0\t1\t1122 44\n",
      "38\t991\t0.0\t1\t953 38\n",
      "39\t923\t0.0\t1\t888 35\n",
      "40\t1100\t0.0\t1\t1057 43\n",
      "41\t774\t0.0\t1\t737 37\n",
      "42\t751\t0.0\t1\t730 21\n",
      "43\t736\t0.0\t1\t706 30\n",
      "44\t719\t0.0\t1\t685 34\n",
      "45\t687\t0.0\t1\t663 24\n",
      "46\t882\t0.0\t1\t849 33\n",
      "47\t588\t0.0\t1\t553 35\n",
      "48\t531\t0.0\t1\t511 20\n",
      "49\t758\t0.0\t1\t732 26\n",
      "50\t485\t0.0\t1\t466 19\n",
      "51\t528\t0.0\t1\t506 22\n",
      "52\t473\t0.0\t1\t454 19\n",
      "53\t368\t0.0\t1\t345 23\n",
      "54\t334\t0.0\t1\t311 23\n",
      "55\t365\t0.0\t1\t353 12\n",
      "56\t306\t0.0\t1\t290 16\n",
      "57\t261\t0.0\t1\t250 11\n",
      "58\t253\t0.0\t1\t238 15\n",
      "59\t202\t0.0\t1\t192 10\n",
      "60\t276\t0.0\t1\t259 17\n",
      "61\t223\t0.0\t1\t212 11\n",
      "62\t149\t0.0\t1\t143 6\n",
      "63\t170\t0.0\t1\t161 9\n",
      "64\t168\t0.0\t1\t160 8\n",
      "65\t155\t0.0\t1\t150 5\n",
      "66\t158\t0.0\t1\t149 9\n",
      "67\t155\t0.0\t1\t147 8\n",
      "68\t110\t0.0\t1\t102 8\n",
      "69\t108\t0.0\t1\t106 2\n",
      "70\t92\t0.0\t1\t85 7\n",
      "71\t93\t0.0\t1\t91 2\n",
      "72\t83\t0.0\t1\t80 3\n",
      "73\t86\t0.0\t1\t85 1\n",
      "74\t63\t0.0\t1\t55 8\n",
      "75\t50\t0.0\t1\t44 6\n",
      "76\t35\t0.0\t1\t34 1\n",
      "77\t33\t0.0\t1\t27 6\n",
      "78\t20\t0.0\t1\t20\n",
      "79\t17\t0.0\t1\t17\n",
      "80\t16\t0.0\t1\t12 4\n",
      "81\t15\t0.0\t1\t15\n",
      "82\t30\t0.0\t1\t26 4\n",
      "83\t25\t0.0\t1\t20 5\n",
      "84\t8\t0.0\t1\t5 3\n",
      "85\t8\t0.0\t1\t6 2\n",
      "86\t17\t0.0\t1\t10 7\n",
      "87\t6\t0.0\t1\t2 4\n",
      "88\t6\t0.0\t1\t4 2\n",
      "89\t6\t0.0\t1\t4 2\n",
      "90\t9\t0.0\t1\t9\n",
      "91\t5\t0.0\t1\t4 1\n",
      "92\t9\t0.0\t1\t8 1\n",
      "93\t33\t0.0\t1\t28 5\n",
      "94\t132\t0.0\t1\t124 8\n",
      "95\t226\t0.0\t1\t210 16\n",
      "96\t152\t0.0\t1\t134 18\n",
      "97\t85\t0.0\t1\t73 12\n",
      "98\t31\t0.0\t1\t26 5\n",
      "99\t23\t0.0\t1\t21 2\n",
      "100\t22\t0.0\t1\t17 5\n",
      "101\t127\t0.0\t1\t101 26\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 95762 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 18.5%\n",
      "  C: 23.0%\n",
      "  G: 29.4%\n",
      "  T: 29.0%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10186\t7812.5\t0\t10186\n",
      "4\t4056\t1953.1\t0\t4056\n",
      "5\t2748\t488.3\t0\t2748\n",
      "6\t2880\t122.1\t0\t2880\n",
      "7\t2884\t30.5\t0\t2884\n",
      "8\t2581\t7.6\t0\t2581\n",
      "9\t2455\t1.9\t0\t2431 24\n",
      "10\t2959\t0.5\t1\t2845 114\n",
      "11\t2312\t0.1\t1\t2188 124\n",
      "12\t2404\t0.0\t1\t2299 105\n",
      "13\t2368\t0.0\t1\t2254 114\n",
      "14\t2654\t0.0\t1\t2544 110\n",
      "15\t2468\t0.0\t1\t2378 90\n",
      "16\t2169\t0.0\t1\t2083 86\n",
      "17\t2116\t0.0\t1\t2029 87\n",
      "18\t1997\t0.0\t1\t1911 86\n",
      "19\t2032\t0.0\t1\t1971 61\n",
      "20\t2002\t0.0\t1\t1926 76\n",
      "21\t1877\t0.0\t1\t1808 69\n",
      "22\t1859\t0.0\t1\t1778 81\n",
      "23\t1755\t0.0\t1\t1707 48\n",
      "24\t1811\t0.0\t1\t1746 65\n",
      "25\t1797\t0.0\t1\t1732 65\n",
      "26\t1995\t0.0\t1\t1920 75\n",
      "27\t1752\t0.0\t1\t1691 61\n",
      "28\t1744\t0.0\t1\t1688 56\n",
      "29\t1465\t0.0\t1\t1416 49\n",
      "30\t1470\t0.0\t1\t1416 54\n",
      "31\t2127\t0.0\t1\t2060 67\n",
      "32\t1348\t0.0\t1\t1302 46\n",
      "33\t1271\t0.0\t1\t1214 57\n",
      "34\t1193\t0.0\t1\t1145 48\n",
      "35\t1078\t0.0\t1\t1030 48\n",
      "36\t1029\t0.0\t1\t993 36\n",
      "37\t1142\t0.0\t1\t1106 36\n",
      "38\t971\t0.0\t1\t935 36\n",
      "39\t905\t0.0\t1\t866 39\n",
      "40\t1074\t0.0\t1\t1019 55\n",
      "41\t753\t0.0\t1\t715 38\n",
      "42\t748\t0.0\t1\t717 31\n",
      "43\t719\t0.0\t1\t693 26\n",
      "44\t707\t0.0\t1\t680 27\n",
      "45\t679\t0.0\t1\t660 19\n",
      "46\t858\t0.0\t1\t827 31\n",
      "47\t573\t0.0\t1\t546 27\n",
      "48\t519\t0.0\t1\t504 15\n",
      "49\t749\t0.0\t1\t729 20\n",
      "50\t480\t0.0\t1\t456 24\n",
      "51\t511\t0.0\t1\t498 13\n",
      "52\t465\t0.0\t1\t447 18\n",
      "53\t358\t0.0\t1\t347 11\n",
      "54\t318\t0.0\t1\t302 16\n",
      "55\t360\t0.0\t1\t340 20\n",
      "56\t298\t0.0\t1\t282 16\n",
      "57\t256\t0.0\t1\t246 10\n",
      "58\t241\t0.0\t1\t233 8\n",
      "59\t200\t0.0\t1\t188 12\n",
      "60\t267\t0.0\t1\t256 11\n",
      "61\t216\t0.0\t1\t209 7\n",
      "62\t146\t0.0\t1\t144 2\n",
      "63\t165\t0.0\t1\t153 12\n",
      "64\t165\t0.0\t1\t151 14\n",
      "65\t150\t0.0\t1\t145 5\n",
      "66\t150\t0.0\t1\t145 5\n",
      "67\t150\t0.0\t1\t147 3\n",
      "68\t106\t0.0\t1\t101 5\n",
      "69\t110\t0.0\t1\t107 3\n",
      "70\t87\t0.0\t1\t85 2\n",
      "71\t89\t0.0\t1\t86 3\n",
      "72\t81\t0.0\t1\t79 2\n",
      "73\t82\t0.0\t1\t78 4\n",
      "74\t57\t0.0\t1\t53 4\n",
      "75\t46\t0.0\t1\t42 4\n",
      "76\t33\t0.0\t1\t30 3\n",
      "77\t29\t0.0\t1\t25 4\n",
      "78\t18\t0.0\t1\t17 1\n",
      "79\t17\t0.0\t1\t17\n",
      "80\t13\t0.0\t1\t10 3\n",
      "81\t15\t0.0\t1\t15\n",
      "82\t28\t0.0\t1\t24 4\n",
      "83\t21\t0.0\t1\t17 4\n",
      "84\t5\t0.0\t1\t4 1\n",
      "85\t6\t0.0\t1\t5 1\n",
      "86\t9\t0.0\t1\t8 1\n",
      "87\t2\t0.0\t1\t2\n",
      "88\t4\t0.0\t1\t4\n",
      "89\t5\t0.0\t1\t4 1\n",
      "90\t7\t0.0\t1\t7\n",
      "91\t5\t0.0\t1\t4 1\n",
      "92\t7\t0.0\t1\t6 1\n",
      "93\t27\t0.0\t1\t25 2\n",
      "94\t124\t0.0\t1\t121 3\n",
      "95\t211\t0.0\t1\t203 8\n",
      "96\t137\t0.0\t1\t127 10\n",
      "97\t76\t0.0\t1\t72 4\n",
      "98\t24\t0.0\t1\t22 2\n",
      "99\t20\t0.0\t1\t19 1\n",
      "100\t19\t0.0\t1\t18 1\n",
      "101\t107\t0.0\t1\t103 4\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950161_trimmed_R1.fastq.gz -p trimmed/ERR950161_trimmed_R2.fastq.gz ERR950161_1.fastq.gz.subset.fastq.gz ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 149,410 (29.9%)\n",
      "  Read 2 with adapter:                 147,335 (29.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,150 (0.2%)\n",
      "Pairs written (passing filters):       498,850 (99.8%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     93,109,163 bp (92.2%)\n",
      "  Read 1:    46,527,839 bp\n",
      "  Read 2:    46,581,324 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 149410 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 23.9%\n",
      "  C: 40.0%\n",
      "  G: 23.1%\n",
      "  T: 12.9%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10795\t7812.5\t0\t10795\n",
      "4\t4872\t1953.1\t0\t4872\n",
      "5\t3169\t488.3\t0\t3169\n",
      "6\t3345\t122.1\t0\t3345\n",
      "7\t3138\t30.5\t0\t3138\n",
      "8\t3011\t7.6\t0\t3011\n",
      "9\t3002\t1.9\t0\t2941 61\n",
      "10\t3462\t0.5\t1\t3305 157\n",
      "11\t3029\t0.1\t1\t2858 171\n",
      "12\t3107\t0.0\t1\t2939 168\n",
      "13\t3065\t0.0\t1\t2930 135\n",
      "14\t3169\t0.0\t1\t3020 149\n",
      "15\t3233\t0.0\t1\t3093 140\n",
      "16\t2857\t0.0\t1\t2740 117\n",
      "17\t2929\t0.0\t1\t2813 116\n",
      "18\t2739\t0.0\t1\t2631 108\n",
      "19\t2756\t0.0\t1\t2664 92\n",
      "20\t2977\t0.0\t1\t2871 106\n",
      "21\t2763\t0.0\t1\t2667 96\n",
      "22\t2810\t0.0\t1\t2705 105\n",
      "23\t2752\t0.0\t1\t2671 81\n",
      "24\t2857\t0.0\t1\t2767 90\n",
      "25\t2870\t0.0\t1\t2765 105\n",
      "26\t3034\t0.0\t1\t2946 88\n",
      "27\t2851\t0.0\t1\t2750 101\n",
      "28\t2737\t0.0\t1\t2641 96\n",
      "29\t2687\t0.0\t1\t2595 92\n",
      "30\t2744\t0.0\t1\t2658 86\n",
      "31\t3830\t0.0\t1\t3738 92\n",
      "32\t2476\t0.0\t1\t2396 80\n",
      "33\t2395\t0.0\t1\t2340 55\n",
      "34\t2362\t0.0\t1\t2304 58\n",
      "35\t2249\t0.0\t1\t2185 64\n",
      "36\t2117\t0.0\t1\t2063 54\n",
      "37\t2142\t0.0\t1\t2080 62\n",
      "38\t1927\t0.0\t1\t1870 57\n",
      "39\t1945\t0.0\t1\t1896 49\n",
      "40\t2031\t0.0\t1\t1986 45\n",
      "41\t1767\t0.0\t1\t1721 46\n",
      "42\t1800\t0.0\t1\t1758 42\n",
      "43\t1806\t0.0\t1\t1761 45\n",
      "44\t1825\t0.0\t1\t1773 52\n",
      "45\t1658\t0.0\t1\t1614 44\n",
      "46\t1887\t0.0\t1\t1840 47\n",
      "47\t1460\t0.0\t1\t1421 39\n",
      "48\t1507\t0.0\t1\t1462 45\n",
      "49\t1704\t0.0\t1\t1675 29\n",
      "50\t1426\t0.0\t1\t1382 44\n",
      "51\t1454\t0.0\t1\t1414 40\n",
      "52\t1286\t0.0\t1\t1258 28\n",
      "53\t1152\t0.0\t1\t1131 21\n",
      "54\t1037\t0.0\t1\t1012 25\n",
      "55\t914\t0.0\t1\t890 24\n",
      "56\t855\t0.0\t1\t835 20\n",
      "57\t784\t0.0\t1\t771 13\n",
      "58\t756\t0.0\t1\t732 24\n",
      "59\t650\t0.0\t1\t635 15\n",
      "60\t721\t0.0\t1\t702 19\n",
      "61\t599\t0.0\t1\t576 23\n",
      "62\t563\t0.0\t1\t547 16\n",
      "63\t542\t0.0\t1\t522 20\n",
      "64\t472\t0.0\t1\t456 16\n",
      "65\t434\t0.0\t1\t419 15\n",
      "66\t412\t0.0\t1\t394 18\n",
      "67\t414\t0.0\t1\t405 9\n",
      "68\t312\t0.0\t1\t298 14\n",
      "69\t323\t0.0\t1\t312 11\n",
      "70\t280\t0.0\t1\t267 13\n",
      "71\t293\t0.0\t1\t286 7\n",
      "72\t250\t0.0\t1\t240 10\n",
      "73\t307\t0.0\t1\t294 13\n",
      "74\t158\t0.0\t1\t151 7\n",
      "75\t133\t0.0\t1\t126 7\n",
      "76\t118\t0.0\t1\t110 8\n",
      "77\t88\t0.0\t1\t84 4\n",
      "78\t68\t0.0\t1\t62 6\n",
      "79\t72\t0.0\t1\t68 4\n",
      "80\t75\t0.0\t1\t70 5\n",
      "81\t70\t0.0\t1\t62 8\n",
      "82\t101\t0.0\t1\t97 4\n",
      "83\t58\t0.0\t1\t52 6\n",
      "84\t35\t0.0\t1\t29 6\n",
      "85\t30\t0.0\t1\t28 2\n",
      "86\t28\t0.0\t1\t26 2\n",
      "87\t15\t0.0\t1\t13 2\n",
      "88\t25\t0.0\t1\t18 7\n",
      "89\t23\t0.0\t1\t21 2\n",
      "90\t15\t0.0\t1\t15\n",
      "91\t24\t0.0\t1\t20 4\n",
      "92\t33\t0.0\t1\t27 6\n",
      "93\t23\t0.0\t1\t20 3\n",
      "94\t40\t0.0\t1\t36 4\n",
      "95\t63\t0.0\t1\t58 5\n",
      "96\t68\t0.0\t1\t56 12\n",
      "97\t30\t0.0\t1\t20 10\n",
      "98\t9\t0.0\t1\t7 2\n",
      "99\t4\t0.0\t1\t3 1\n",
      "100\t18\t0.0\t1\t7 11\n",
      "101\t102\t0.0\t1\t60 42\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 147335 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.6%\n",
      "  C: 21.9%\n",
      "  G: 29.5%\n",
      "  T: 31.0%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9933\t7812.5\t0\t9933\n",
      "4\t4413\t1953.1\t0\t4413\n",
      "5\t3385\t488.3\t0\t3385\n",
      "6\t3321\t122.1\t0\t3321\n",
      "7\t3185\t30.5\t0\t3185\n",
      "8\t3038\t7.6\t0\t3038\n",
      "9\t3051\t1.9\t0\t3017 34\n",
      "10\t3460\t0.5\t1\t3350 110\n",
      "11\t3014\t0.1\t1\t2890 124\n",
      "12\t3109\t0.0\t1\t2993 116\n",
      "13\t3069\t0.0\t1\t2954 115\n",
      "14\t3167\t0.0\t1\t3082 85\n",
      "15\t3227\t0.0\t1\t3117 110\n",
      "16\t2843\t0.0\t1\t2735 108\n",
      "17\t2938\t0.0\t1\t2833 105\n",
      "18\t2757\t0.0\t1\t2668 89\n",
      "19\t2748\t0.0\t1\t2648 100\n",
      "20\t2976\t0.0\t1\t2866 110\n",
      "21\t2764\t0.0\t1\t2675 89\n",
      "22\t2798\t0.0\t1\t2709 89\n",
      "23\t2766\t0.0\t1\t2671 95\n",
      "24\t2842\t0.0\t1\t2746 96\n",
      "25\t2856\t0.0\t1\t2750 106\n",
      "26\t3036\t0.0\t1\t2922 114\n",
      "27\t2835\t0.0\t1\t2739 96\n",
      "28\t2740\t0.0\t1\t2646 94\n",
      "29\t2659\t0.0\t1\t2550 109\n",
      "30\t2722\t0.0\t1\t2643 79\n",
      "31\t3810\t0.0\t1\t3696 114\n",
      "32\t2465\t0.0\t1\t2377 88\n",
      "33\t2388\t0.0\t1\t2299 89\n",
      "34\t2356\t0.0\t1\t2267 89\n",
      "35\t2227\t0.0\t1\t2169 58\n",
      "36\t2121\t0.0\t1\t2046 75\n",
      "37\t2139\t0.0\t1\t2067 72\n",
      "38\t1909\t0.0\t1\t1864 45\n",
      "39\t1921\t0.0\t1\t1865 56\n",
      "40\t2008\t0.0\t1\t1935 73\n",
      "41\t1742\t0.0\t1\t1678 64\n",
      "42\t1792\t0.0\t1\t1744 48\n",
      "43\t1793\t0.0\t1\t1731 62\n",
      "44\t1809\t0.0\t1\t1761 48\n",
      "45\t1651\t0.0\t1\t1601 50\n",
      "46\t1875\t0.0\t1\t1814 61\n",
      "47\t1454\t0.0\t1\t1397 57\n",
      "48\t1479\t0.0\t1\t1424 55\n",
      "49\t1671\t0.0\t1\t1628 43\n",
      "50\t1417\t0.0\t1\t1369 48\n",
      "51\t1436\t0.0\t1\t1382 54\n",
      "52\t1285\t0.0\t1\t1254 31\n",
      "53\t1133\t0.0\t1\t1094 39\n",
      "54\t1021\t0.0\t1\t977 44\n",
      "55\t903\t0.0\t1\t866 37\n",
      "56\t833\t0.0\t1\t807 26\n",
      "57\t768\t0.0\t1\t748 20\n",
      "58\t743\t0.0\t1\t707 36\n",
      "59\t631\t0.0\t1\t611 20\n",
      "60\t692\t0.0\t1\t675 17\n",
      "61\t583\t0.0\t1\t562 21\n",
      "62\t551\t0.0\t1\t531 20\n",
      "63\t527\t0.0\t1\t508 19\n",
      "64\t461\t0.0\t1\t442 19\n",
      "65\t423\t0.0\t1\t409 14\n",
      "66\t394\t0.0\t1\t375 19\n",
      "67\t407\t0.0\t1\t392 15\n",
      "68\t288\t0.0\t1\t281 7\n",
      "69\t316\t0.0\t1\t302 14\n",
      "70\t259\t0.0\t1\t253 6\n",
      "71\t276\t0.0\t1\t265 11\n",
      "72\t227\t0.0\t1\t214 13\n",
      "73\t276\t0.0\t1\t269 7\n",
      "74\t138\t0.0\t1\t127 11\n",
      "75\t114\t0.0\t1\t108 6\n",
      "76\t98\t0.0\t1\t95 3\n",
      "77\t85\t0.0\t1\t82 3\n",
      "78\t59\t0.0\t1\t55 4\n",
      "79\t67\t0.0\t1\t64 3\n",
      "80\t60\t0.0\t1\t59 1\n",
      "81\t51\t0.0\t1\t50 1\n",
      "82\t88\t0.0\t1\t82 6\n",
      "83\t48\t0.0\t1\t47 1\n",
      "84\t29\t0.0\t1\t25 4\n",
      "85\t21\t0.0\t1\t21\n",
      "86\t22\t0.0\t1\t21 1\n",
      "87\t10\t0.0\t1\t8 2\n",
      "88\t11\t0.0\t1\t9 2\n",
      "89\t14\t0.0\t1\t14\n",
      "90\t7\t0.0\t1\t7\n",
      "91\t12\t0.0\t1\t11 1\n",
      "92\t18\t0.0\t1\t16 2\n",
      "93\t17\t0.0\t1\t15 2\n",
      "94\t34\t0.0\t1\t29 5\n",
      "95\t58\t0.0\t1\t57 1\n",
      "96\t62\t0.0\t1\t54 8\n",
      "97\t21\t0.0\t1\t16 5\n",
      "98\t6\t0.0\t1\t5 1\n",
      "99\t4\t0.0\t1\t3 1\n",
      "100\t5\t0.0\t1\t3 2\n",
      "101\t64\t0.0\t1\t59 5\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950162_trimmed_R1.fastq.gz -p trimmed/ERR950162_trimmed_R2.fastq.gz ERR950162_1.fastq.gz.subset.fastq.gz ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 172,561 (34.5%)\n",
      "  Read 2 with adapter:                 168,523 (33.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:              18,487 (3.7%)\n",
      "Pairs written (passing filters):       481,513 (96.3%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     89,565,545 bp (88.7%)\n",
      "  Read 1:    44,743,359 bp\n",
      "  Read 2:    44,822,186 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 172561 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 21.3%\n",
      "  C: 47.3%\n",
      "  G: 16.8%\n",
      "  T: 13.7%\n",
      "  none/other: 0.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t8654\t7812.5\t0\t8654\n",
      "4\t5085\t1953.1\t0\t5085\n",
      "5\t3233\t488.3\t0\t3233\n",
      "6\t3972\t122.1\t0\t3972\n",
      "7\t4644\t30.5\t0\t4644\n",
      "8\t3682\t7.6\t0\t3682\n",
      "9\t4875\t1.9\t0\t4686 189\n",
      "10\t5006\t0.5\t1\t4705 301\n",
      "11\t3604\t0.1\t1\t3317 287\n",
      "12\t3667\t0.0\t1\t3335 332\n",
      "13\t3456\t0.0\t1\t3240 216\n",
      "14\t4352\t0.0\t1\t4079 273\n",
      "15\t3767\t0.0\t1\t3516 251\n",
      "16\t3253\t0.0\t1\t3059 194\n",
      "17\t2978\t0.0\t1\t2813 165\n",
      "18\t2670\t0.0\t1\t2540 130\n",
      "19\t2914\t0.0\t1\t2760 154\n",
      "20\t3216\t0.0\t1\t3052 164\n",
      "21\t2802\t0.0\t1\t2664 138\n",
      "22\t2490\t0.0\t1\t2374 116\n",
      "23\t2486\t0.0\t1\t2369 117\n",
      "24\t2753\t0.0\t1\t2633 120\n",
      "25\t2807\t0.0\t1\t2655 152\n",
      "26\t3105\t0.0\t1\t2963 142\n",
      "27\t3114\t0.0\t1\t2981 133\n",
      "28\t3124\t0.0\t1\t2994 130\n",
      "29\t2406\t0.0\t1\t2283 123\n",
      "30\t2492\t0.0\t1\t2394 98\n",
      "31\t4258\t0.0\t1\t4111 147\n",
      "32\t2219\t0.0\t1\t2128 91\n",
      "33\t2085\t0.0\t1\t1992 93\n",
      "34\t1981\t0.0\t1\t1886 95\n",
      "35\t1860\t0.0\t1\t1783 77\n",
      "36\t1787\t0.0\t1\t1708 79\n",
      "37\t1873\t0.0\t1\t1782 91\n",
      "38\t1478\t0.0\t1\t1414 64\n",
      "39\t1494\t0.0\t1\t1425 69\n",
      "40\t2493\t0.0\t1\t2420 73\n",
      "41\t1440\t0.0\t1\t1380 60\n",
      "42\t1503\t0.0\t1\t1438 65\n",
      "43\t1790\t0.0\t1\t1726 64\n",
      "44\t1788\t0.0\t1\t1728 60\n",
      "45\t1849\t0.0\t1\t1779 70\n",
      "46\t2909\t0.0\t1\t2815 94\n",
      "47\t1317\t0.0\t1\t1263 54\n",
      "48\t1271\t0.0\t1\t1218 53\n",
      "49\t1821\t0.0\t1\t1767 54\n",
      "50\t1353\t0.0\t1\t1294 59\n",
      "51\t1179\t0.0\t1\t1143 36\n",
      "52\t1219\t0.0\t1\t1173 46\n",
      "53\t1016\t0.0\t1\t972 44\n",
      "54\t836\t0.0\t1\t807 29\n",
      "55\t671\t0.0\t1\t646 25\n",
      "56\t700\t0.0\t1\t682 18\n",
      "57\t802\t0.0\t1\t775 27\n",
      "58\t630\t0.0\t1\t598 32\n",
      "59\t581\t0.0\t1\t566 15\n",
      "60\t710\t0.0\t1\t682 28\n",
      "61\t509\t0.0\t1\t481 28\n",
      "62\t518\t0.0\t1\t494 24\n",
      "63\t491\t0.0\t1\t470 21\n",
      "64\t595\t0.0\t1\t569 26\n",
      "65\t470\t0.0\t1\t449 21\n",
      "66\t471\t0.0\t1\t452 19\n",
      "67\t516\t0.0\t1\t498 18\n",
      "68\t392\t0.0\t1\t374 18\n",
      "69\t422\t0.0\t1\t407 15\n",
      "70\t505\t0.0\t1\t492 13\n",
      "71\t390\t0.0\t1\t368 22\n",
      "72\t376\t0.0\t1\t362 14\n",
      "73\t356\t0.0\t1\t339 17\n",
      "74\t250\t0.0\t1\t240 10\n",
      "75\t200\t0.0\t1\t192 8\n",
      "76\t141\t0.0\t1\t133 8\n",
      "77\t140\t0.0\t1\t132 8\n",
      "78\t122\t0.0\t1\t117 5\n",
      "79\t111\t0.0\t1\t104 7\n",
      "80\t77\t0.0\t1\t75 2\n",
      "81\t88\t0.0\t1\t81 7\n",
      "82\t104\t0.0\t1\t102 2\n",
      "83\t97\t0.0\t1\t90 7\n",
      "84\t52\t0.0\t1\t51 1\n",
      "85\t44\t0.0\t1\t41 3\n",
      "86\t54\t0.0\t1\t51 3\n",
      "87\t55\t0.0\t1\t52 3\n",
      "88\t25\t0.0\t1\t23 2\n",
      "89\t36\t0.0\t1\t28 8\n",
      "90\t46\t0.0\t1\t45 1\n",
      "91\t110\t0.0\t1\t102 8\n",
      "92\t191\t0.0\t1\t181 10\n",
      "93\t683\t0.0\t1\t655 28\n",
      "94\t3265\t0.0\t1\t3136 129\n",
      "95\t5277\t0.0\t1\t5059 218\n",
      "96\t2822\t0.0\t1\t2674 148\n",
      "97\t1504\t0.0\t1\t1427 77\n",
      "98\t596\t0.0\t1\t559 37\n",
      "99\t775\t0.0\t1\t742 33\n",
      "100\t537\t0.0\t1\t517 20\n",
      "101\t1628\t0.0\t1\t1383 245\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 168523 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.6%\n",
      "  C: 17.8%\n",
      "  G: 29.6%\n",
      "  T: 35.1%\n",
      "  none/other: 0.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7474\t7812.5\t0\t7474\n",
      "4\t4678\t1953.1\t0\t4678\n",
      "5\t3372\t488.3\t0\t3372\n",
      "6\t3943\t122.1\t0\t3943\n",
      "7\t4703\t30.5\t0\t4703\n",
      "8\t3684\t7.6\t0\t3684\n",
      "9\t4965\t1.9\t0\t4916 49\n",
      "10\t4957\t0.5\t1\t4766 191\n",
      "11\t3558\t0.1\t1\t3306 252\n",
      "12\t3627\t0.0\t1\t3442 185\n",
      "13\t3441\t0.0\t1\t3276 165\n",
      "14\t4324\t0.0\t1\t4137 187\n",
      "15\t3727\t0.0\t1\t3573 154\n",
      "16\t3211\t0.0\t1\t3079 132\n",
      "17\t2957\t0.0\t1\t2822 135\n",
      "18\t2651\t0.0\t1\t2523 128\n",
      "19\t2907\t0.0\t1\t2778 129\n",
      "20\t3187\t0.0\t1\t3040 147\n",
      "21\t2768\t0.0\t1\t2667 101\n",
      "22\t2462\t0.0\t1\t2375 87\n",
      "23\t2458\t0.0\t1\t2369 89\n",
      "24\t2703\t0.0\t1\t2579 124\n",
      "25\t2758\t0.0\t1\t2648 110\n",
      "26\t3060\t0.0\t1\t2934 126\n",
      "27\t3048\t0.0\t1\t2930 118\n",
      "28\t3074\t0.0\t1\t2949 125\n",
      "29\t2360\t0.0\t1\t2265 95\n",
      "30\t2438\t0.0\t1\t2359 79\n",
      "31\t4218\t0.0\t1\t4063 155\n",
      "32\t2174\t0.0\t1\t2090 84\n",
      "33\t2038\t0.0\t1\t1953 85\n",
      "34\t1946\t0.0\t1\t1848 98\n",
      "35\t1831\t0.0\t1\t1731 100\n",
      "36\t1754\t0.0\t1\t1669 85\n",
      "37\t1828\t0.0\t1\t1741 87\n",
      "38\t1441\t0.0\t1\t1383 58\n",
      "39\t1456\t0.0\t1\t1407 49\n",
      "40\t2451\t0.0\t1\t2374 77\n",
      "41\t1407\t0.0\t1\t1341 66\n",
      "42\t1462\t0.0\t1\t1401 61\n",
      "43\t1755\t0.0\t1\t1687 68\n",
      "44\t1747\t0.0\t1\t1682 65\n",
      "45\t1819\t0.0\t1\t1758 61\n",
      "46\t2872\t0.0\t1\t2763 109\n",
      "47\t1284\t0.0\t1\t1227 57\n",
      "48\t1238\t0.0\t1\t1191 47\n",
      "49\t1786\t0.0\t1\t1720 66\n",
      "50\t1308\t0.0\t1\t1243 65\n",
      "51\t1152\t0.0\t1\t1118 34\n",
      "52\t1186\t0.0\t1\t1150 36\n",
      "53\t995\t0.0\t1\t952 43\n",
      "54\t818\t0.0\t1\t789 29\n",
      "55\t657\t0.0\t1\t624 33\n",
      "56\t678\t0.0\t1\t653 25\n",
      "57\t774\t0.0\t1\t738 36\n",
      "58\t610\t0.0\t1\t587 23\n",
      "59\t566\t0.0\t1\t537 29\n",
      "60\t690\t0.0\t1\t667 23\n",
      "61\t495\t0.0\t1\t471 24\n",
      "62\t503\t0.0\t1\t482 21\n",
      "63\t476\t0.0\t1\t459 17\n",
      "64\t573\t0.0\t1\t544 29\n",
      "65\t454\t0.0\t1\t434 20\n",
      "66\t456\t0.0\t1\t444 12\n",
      "67\t503\t0.0\t1\t481 22\n",
      "68\t380\t0.0\t1\t371 9\n",
      "69\t405\t0.0\t1\t390 15\n",
      "70\t494\t0.0\t1\t470 24\n",
      "71\t367\t0.0\t1\t354 13\n",
      "72\t363\t0.0\t1\t343 20\n",
      "73\t337\t0.0\t1\t325 12\n",
      "74\t237\t0.0\t1\t222 15\n",
      "75\t196\t0.0\t1\t188 8\n",
      "76\t134\t0.0\t1\t127 7\n",
      "77\t134\t0.0\t1\t131 3\n",
      "78\t119\t0.0\t1\t115 4\n",
      "79\t106\t0.0\t1\t103 3\n",
      "80\t73\t0.0\t1\t67 6\n",
      "81\t84\t0.0\t1\t78 6\n",
      "82\t102\t0.0\t1\t99 3\n",
      "83\t90\t0.0\t1\t87 3\n",
      "84\t49\t0.0\t1\t46 3\n",
      "85\t42\t0.0\t1\t41 1\n",
      "86\t50\t0.0\t1\t47 3\n",
      "87\t49\t0.0\t1\t48 1\n",
      "88\t23\t0.0\t1\t21 2\n",
      "89\t27\t0.0\t1\t27\n",
      "90\t43\t0.0\t1\t43\n",
      "91\t103\t0.0\t1\t96 7\n",
      "92\t185\t0.0\t1\t178 7\n",
      "93\t670\t0.0\t1\t644 26\n",
      "94\t3191\t0.0\t1\t3073 118\n",
      "95\t5134\t0.0\t1\t4932 202\n",
      "96\t2714\t0.0\t1\t2606 108\n",
      "97\t1451\t0.0\t1\t1388 63\n",
      "98\t569\t0.0\t1\t545 24\n",
      "99\t751\t0.0\t1\t719 32\n",
      "100\t519\t0.0\t1\t477 42\n",
      "101\t1436\t0.0\t1\t1350 86\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950163_trimmed_R1.fastq.gz -p trimmed/ERR950163_trimmed_R2.fastq.gz ERR950163_1.fastq.gz.subset.fastq.gz ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 106,074 (21.2%)\n",
      "  Read 2 with adapter:                 104,358 (20.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,530 (0.3%)\n",
      "Pairs written (passing filters):       498,470 (99.7%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,188,902 bp (95.2%)\n",
      "  Read 1:    48,073,739 bp\n",
      "  Read 2:    48,115,163 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 106074 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.1%\n",
      "  C: 38.2%\n",
      "  G: 23.9%\n",
      "  T: 12.6%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11910\t7812.5\t0\t11910\n",
      "4\t4782\t1953.1\t0\t4782\n",
      "5\t2673\t488.3\t0\t2673\n",
      "6\t2780\t122.1\t0\t2780\n",
      "7\t2704\t30.5\t0\t2704\n",
      "8\t2612\t7.6\t0\t2612\n",
      "9\t2655\t1.9\t0\t2586 69\n",
      "10\t2877\t0.5\t1\t2687 190\n",
      "11\t2460\t0.1\t1\t2288 172\n",
      "12\t2577\t0.0\t1\t2327 250\n",
      "13\t2455\t0.0\t1\t2290 165\n",
      "14\t2590\t0.0\t1\t2400 190\n",
      "15\t2545\t0.0\t1\t2346 199\n",
      "16\t2627\t0.0\t1\t2464 163\n",
      "17\t2307\t0.0\t1\t2194 113\n",
      "18\t2152\t0.0\t1\t2012 140\n",
      "19\t2081\t0.0\t1\t1960 121\n",
      "20\t2151\t0.0\t1\t2031 120\n",
      "21\t2072\t0.0\t1\t1947 125\n",
      "22\t1980\t0.0\t1\t1878 102\n",
      "23\t2120\t0.0\t1\t2021 99\n",
      "24\t1961\t0.0\t1\t1868 93\n",
      "25\t2029\t0.0\t1\t1920 109\n",
      "26\t2122\t0.0\t1\t2019 103\n",
      "27\t1966\t0.0\t1\t1890 76\n",
      "28\t1726\t0.0\t1\t1645 81\n",
      "29\t1643\t0.0\t1\t1566 77\n",
      "30\t1605\t0.0\t1\t1522 83\n",
      "31\t2000\t0.0\t1\t1907 93\n",
      "32\t1401\t0.0\t1\t1342 59\n",
      "33\t1416\t0.0\t1\t1362 54\n",
      "34\t1326\t0.0\t1\t1255 71\n",
      "35\t1282\t0.0\t1\t1220 62\n",
      "36\t1284\t0.0\t1\t1228 56\n",
      "37\t1430\t0.0\t1\t1374 56\n",
      "38\t1119\t0.0\t1\t1073 46\n",
      "39\t1022\t0.0\t1\t985 37\n",
      "40\t1174\t0.0\t1\t1122 52\n",
      "41\t930\t0.0\t1\t877 53\n",
      "42\t914\t0.0\t1\t888 26\n",
      "43\t912\t0.0\t1\t873 39\n",
      "44\t914\t0.0\t1\t873 41\n",
      "45\t823\t0.0\t1\t792 31\n",
      "46\t871\t0.0\t1\t842 29\n",
      "47\t691\t0.0\t1\t665 26\n",
      "48\t676\t0.0\t1\t650 26\n",
      "49\t752\t0.0\t1\t724 28\n",
      "50\t634\t0.0\t1\t596 38\n",
      "51\t586\t0.0\t1\t555 31\n",
      "52\t582\t0.0\t1\t559 23\n",
      "53\t524\t0.0\t1\t504 20\n",
      "54\t498\t0.0\t1\t477 21\n",
      "55\t453\t0.0\t1\t427 26\n",
      "56\t365\t0.0\t1\t348 17\n",
      "57\t421\t0.0\t1\t401 20\n",
      "58\t336\t0.0\t1\t330 6\n",
      "59\t296\t0.0\t1\t281 15\n",
      "60\t310\t0.0\t1\t302 8\n",
      "61\t267\t0.0\t1\t256 11\n",
      "62\t274\t0.0\t1\t259 15\n",
      "63\t229\t0.0\t1\t219 10\n",
      "64\t221\t0.0\t1\t212 9\n",
      "65\t173\t0.0\t1\t167 6\n",
      "66\t174\t0.0\t1\t162 12\n",
      "67\t193\t0.0\t1\t180 13\n",
      "68\t156\t0.0\t1\t152 4\n",
      "69\t137\t0.0\t1\t129 8\n",
      "70\t124\t0.0\t1\t111 13\n",
      "71\t117\t0.0\t1\t108 9\n",
      "72\t118\t0.0\t1\t107 11\n",
      "73\t97\t0.0\t1\t95 2\n",
      "74\t75\t0.0\t1\t71 4\n",
      "75\t61\t0.0\t1\t59 2\n",
      "76\t46\t0.0\t1\t44 2\n",
      "77\t37\t0.0\t1\t36 1\n",
      "78\t36\t0.0\t1\t34 2\n",
      "79\t29\t0.0\t1\t25 4\n",
      "80\t26\t0.0\t1\t23 3\n",
      "81\t14\t0.0\t1\t12 2\n",
      "82\t24\t0.0\t1\t21 3\n",
      "83\t25\t0.0\t1\t21 4\n",
      "84\t14\t0.0\t1\t13 1\n",
      "85\t20\t0.0\t1\t20\n",
      "86\t7\t0.0\t1\t5 2\n",
      "87\t7\t0.0\t1\t5 2\n",
      "88\t4\t0.0\t1\t4\n",
      "89\t9\t0.0\t1\t7 2\n",
      "90\t12\t0.0\t1\t9 3\n",
      "91\t8\t0.0\t1\t6 2\n",
      "92\t14\t0.0\t1\t11 3\n",
      "93\t56\t0.0\t1\t48 8\n",
      "94\t208\t0.0\t1\t191 17\n",
      "95\t327\t0.0\t1\t304 23\n",
      "96\t233\t0.0\t1\t222 11\n",
      "97\t119\t0.0\t1\t108 11\n",
      "98\t43\t0.0\t1\t36 7\n",
      "99\t56\t0.0\t1\t53 3\n",
      "100\t42\t0.0\t1\t39 3\n",
      "101\t138\t0.0\t1\t108 30\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 104358 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 20.7%\n",
      "  C: 19.9%\n",
      "  G: 28.8%\n",
      "  T: 30.4%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11438\t7812.5\t0\t11438\n",
      "4\t4599\t1953.1\t0\t4599\n",
      "5\t2830\t488.3\t0\t2830\n",
      "6\t2749\t122.1\t0\t2749\n",
      "7\t2748\t30.5\t0\t2748\n",
      "8\t2613\t7.6\t0\t2613\n",
      "9\t2710\t1.9\t0\t2671 39\n",
      "10\t2856\t0.5\t1\t2738 118\n",
      "11\t2462\t0.1\t1\t2315 147\n",
      "12\t2570\t0.0\t1\t2454 116\n",
      "13\t2427\t0.0\t1\t2340 87\n",
      "14\t2554\t0.0\t1\t2430 124\n",
      "15\t2507\t0.0\t1\t2392 115\n",
      "16\t2609\t0.0\t1\t2473 136\n",
      "17\t2294\t0.0\t1\t2201 93\n",
      "18\t2132\t0.0\t1\t2028 104\n",
      "19\t2059\t0.0\t1\t1962 97\n",
      "20\t2116\t0.0\t1\t2029 87\n",
      "21\t2054\t0.0\t1\t1977 77\n",
      "22\t1953\t0.0\t1\t1884 69\n",
      "23\t2105\t0.0\t1\t2024 81\n",
      "24\t1918\t0.0\t1\t1849 69\n",
      "25\t1992\t0.0\t1\t1913 79\n",
      "26\t2099\t0.0\t1\t2021 78\n",
      "27\t1948\t0.0\t1\t1865 83\n",
      "28\t1700\t0.0\t1\t1630 70\n",
      "29\t1608\t0.0\t1\t1546 62\n",
      "30\t1579\t0.0\t1\t1520 59\n",
      "31\t1967\t0.0\t1\t1897 70\n",
      "32\t1381\t0.0\t1\t1329 52\n",
      "33\t1395\t0.0\t1\t1341 54\n",
      "34\t1298\t0.0\t1\t1244 54\n",
      "35\t1256\t0.0\t1\t1219 37\n",
      "36\t1249\t0.0\t1\t1200 49\n",
      "37\t1408\t0.0\t1\t1354 54\n",
      "38\t1097\t0.0\t1\t1061 36\n",
      "39\t1013\t0.0\t1\t980 33\n",
      "40\t1142\t0.0\t1\t1099 43\n",
      "41\t911\t0.0\t1\t869 42\n",
      "42\t898\t0.0\t1\t858 40\n",
      "43\t896\t0.0\t1\t862 34\n",
      "44\t907\t0.0\t1\t871 36\n",
      "45\t810\t0.0\t1\t789 21\n",
      "46\t862\t0.0\t1\t820 42\n",
      "47\t677\t0.0\t1\t654 23\n",
      "48\t662\t0.0\t1\t646 16\n",
      "49\t739\t0.0\t1\t712 27\n",
      "50\t612\t0.0\t1\t592 20\n",
      "51\t570\t0.0\t1\t550 20\n",
      "52\t565\t0.0\t1\t549 16\n",
      "53\t512\t0.0\t1\t489 23\n",
      "54\t492\t0.0\t1\t465 27\n",
      "55\t440\t0.0\t1\t419 21\n",
      "56\t352\t0.0\t1\t334 18\n",
      "57\t402\t0.0\t1\t389 13\n",
      "58\t335\t0.0\t1\t321 14\n",
      "59\t285\t0.0\t1\t278 7\n",
      "60\t304\t0.0\t1\t294 10\n",
      "61\t260\t0.0\t1\t245 15\n",
      "62\t264\t0.0\t1\t258 6\n",
      "63\t221\t0.0\t1\t208 13\n",
      "64\t212\t0.0\t1\t196 16\n",
      "65\t164\t0.0\t1\t156 8\n",
      "66\t166\t0.0\t1\t161 5\n",
      "67\t180\t0.0\t1\t175 5\n",
      "68\t151\t0.0\t1\t146 5\n",
      "69\t132\t0.0\t1\t123 9\n",
      "70\t113\t0.0\t1\t106 7\n",
      "71\t106\t0.0\t1\t103 3\n",
      "72\t109\t0.0\t1\t100 9\n",
      "73\t95\t0.0\t1\t88 7\n",
      "74\t68\t0.0\t1\t64 4\n",
      "75\t58\t0.0\t1\t56 2\n",
      "76\t44\t0.0\t1\t39 5\n",
      "77\t34\t0.0\t1\t30 4\n",
      "78\t33\t0.0\t1\t33\n",
      "79\t23\t0.0\t1\t20 3\n",
      "80\t22\t0.0\t1\t21 1\n",
      "81\t11\t0.0\t1\t10 1\n",
      "82\t19\t0.0\t1\t17 2\n",
      "83\t21\t0.0\t1\t20 1\n",
      "84\t13\t0.0\t1\t12 1\n",
      "85\t14\t0.0\t1\t13 1\n",
      "86\t5\t0.0\t1\t4 1\n",
      "87\t6\t0.0\t1\t4 2\n",
      "88\t2\t0.0\t1\t2\n",
      "89\t5\t0.0\t1\t5\n",
      "90\t5\t0.0\t1\t4 1\n",
      "91\t7\t0.0\t1\t5 2\n",
      "92\t9\t0.0\t1\t8 1\n",
      "93\t49\t0.0\t1\t46 3\n",
      "94\t193\t0.0\t1\t187 6\n",
      "95\t306\t0.0\t1\t289 17\n",
      "96\t223\t0.0\t1\t208 15\n",
      "97\t113\t0.0\t1\t108 5\n",
      "98\t37\t0.0\t1\t35 2\n",
      "99\t50\t0.0\t1\t49 1\n",
      "100\t37\t0.0\t1\t31 6\n",
      "101\t112\t0.0\t1\t103 9\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950177_trimmed_R1.fastq.gz -p trimmed/ERR950177_trimmed_R2.fastq.gz ERR950177_1.fastq.gz.subset.fastq.gz ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  83,629 (16.7%)\n",
      "  Read 2 with adapter:                  81,923 (16.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               6,273 (1.3%)\n",
      "Pairs written (passing filters):       493,727 (98.7%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,433,009 bp (95.5%)\n",
      "  Read 1:    48,196,842 bp\n",
      "  Read 2:    48,236,167 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 83629 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 20.4%\n",
      "  C: 42.6%\n",
      "  G: 21.7%\n",
      "  T: 14.0%\n",
      "  none/other: 1.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9407\t7812.5\t0\t9407\n",
      "4\t3828\t1953.1\t0\t3828\n",
      "5\t2226\t488.3\t0\t2226\n",
      "6\t2055\t122.1\t0\t2055\n",
      "7\t2098\t30.5\t0\t2098\n",
      "8\t1784\t7.6\t0\t1784\n",
      "9\t2957\t1.9\t0\t2868 89\n",
      "10\t2528\t0.5\t1\t2380 148\n",
      "11\t1739\t0.1\t1\t1585 154\n",
      "12\t1569\t0.0\t1\t1426 143\n",
      "13\t1756\t0.0\t1\t1618 138\n",
      "14\t1912\t0.0\t1\t1789 123\n",
      "15\t1777\t0.0\t1\t1641 136\n",
      "16\t1599\t0.0\t1\t1487 112\n",
      "17\t1567\t0.0\t1\t1487 80\n",
      "18\t1327\t0.0\t1\t1249 78\n",
      "19\t1417\t0.0\t1\t1336 81\n",
      "20\t1453\t0.0\t1\t1372 81\n",
      "21\t1381\t0.0\t1\t1310 71\n",
      "22\t1201\t0.0\t1\t1132 69\n",
      "23\t1243\t0.0\t1\t1175 68\n",
      "24\t1407\t0.0\t1\t1342 65\n",
      "25\t1204\t0.0\t1\t1137 67\n",
      "26\t1338\t0.0\t1\t1270 68\n",
      "27\t1285\t0.0\t1\t1232 53\n",
      "28\t1228\t0.0\t1\t1160 68\n",
      "29\t1149\t0.0\t1\t1095 54\n",
      "30\t1152\t0.0\t1\t1088 64\n",
      "31\t2263\t0.0\t1\t2175 88\n",
      "32\t1121\t0.0\t1\t1054 67\n",
      "33\t920\t0.0\t1\t885 35\n",
      "34\t877\t0.0\t1\t842 35\n",
      "35\t819\t0.0\t1\t782 37\n",
      "36\t841\t0.0\t1\t807 34\n",
      "37\t760\t0.0\t1\t716 44\n",
      "38\t768\t0.0\t1\t732 36\n",
      "39\t802\t0.0\t1\t776 26\n",
      "40\t947\t0.0\t1\t909 38\n",
      "41\t663\t0.0\t1\t633 30\n",
      "42\t669\t0.0\t1\t641 28\n",
      "43\t682\t0.0\t1\t644 38\n",
      "44\t632\t0.0\t1\t602 30\n",
      "45\t596\t0.0\t1\t570 26\n",
      "46\t587\t0.0\t1\t562 25\n",
      "47\t466\t0.0\t1\t448 18\n",
      "48\t514\t0.0\t1\t486 28\n",
      "49\t919\t0.0\t1\t891 28\n",
      "50\t555\t0.0\t1\t530 25\n",
      "51\t584\t0.0\t1\t560 24\n",
      "52\t516\t0.0\t1\t498 18\n",
      "53\t397\t0.0\t1\t383 14\n",
      "54\t391\t0.0\t1\t374 17\n",
      "55\t310\t0.0\t1\t297 13\n",
      "56\t306\t0.0\t1\t301 5\n",
      "57\t280\t0.0\t1\t268 12\n",
      "58\t258\t0.0\t1\t244 14\n",
      "59\t216\t0.0\t1\t208 8\n",
      "60\t296\t0.0\t1\t284 12\n",
      "61\t186\t0.0\t1\t173 13\n",
      "62\t168\t0.0\t1\t161 7\n",
      "63\t165\t0.0\t1\t156 9\n",
      "64\t158\t0.0\t1\t151 7\n",
      "65\t154\t0.0\t1\t144 10\n",
      "66\t121\t0.0\t1\t114 7\n",
      "67\t138\t0.0\t1\t129 9\n",
      "68\t88\t0.0\t1\t84 4\n",
      "69\t131\t0.0\t1\t120 11\n",
      "70\t87\t0.0\t1\t80 7\n",
      "71\t120\t0.0\t1\t113 7\n",
      "72\t91\t0.0\t1\t83 8\n",
      "73\t116\t0.0\t1\t107 9\n",
      "74\t56\t0.0\t1\t54 2\n",
      "75\t55\t0.0\t1\t50 5\n",
      "76\t39\t0.0\t1\t32 7\n",
      "77\t40\t0.0\t1\t36 4\n",
      "78\t24\t0.0\t1\t20 4\n",
      "79\t30\t0.0\t1\t28 2\n",
      "80\t17\t0.0\t1\t14 3\n",
      "81\t27\t0.0\t1\t25 2\n",
      "82\t28\t0.0\t1\t27 1\n",
      "83\t49\t0.0\t1\t48 1\n",
      "84\t18\t0.0\t1\t18\n",
      "85\t15\t0.0\t1\t12 3\n",
      "86\t11\t0.0\t1\t10 1\n",
      "87\t9\t0.0\t1\t7 2\n",
      "88\t9\t0.0\t1\t7 2\n",
      "89\t7\t0.0\t1\t6 1\n",
      "90\t8\t0.0\t1\t7 1\n",
      "91\t22\t0.0\t1\t21 1\n",
      "92\t36\t0.0\t1\t33 3\n",
      "93\t172\t0.0\t1\t162 10\n",
      "94\t845\t0.0\t1\t809 36\n",
      "95\t1518\t0.0\t1\t1445 73\n",
      "96\t900\t0.0\t1\t833 67\n",
      "97\t581\t0.0\t1\t546 35\n",
      "98\t234\t0.0\t1\t218 16\n",
      "99\t273\t0.0\t1\t253 20\n",
      "100\t264\t0.0\t1\t245 19\n",
      "101\t1077\t0.0\t1\t914 163\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 81923 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.5%\n",
      "  C: 27.4%\n",
      "  G: 30.1%\n",
      "  T: 24.7%\n",
      "  none/other: 1.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9328\t7812.5\t0\t9328\n",
      "4\t3395\t1953.1\t0\t3395\n",
      "5\t2404\t488.3\t0\t2404\n",
      "6\t2040\t122.1\t0\t2040\n",
      "7\t2142\t30.5\t0\t2142\n",
      "8\t1765\t7.6\t0\t1765\n",
      "9\t3025\t1.9\t0\t2989 36\n",
      "10\t2494\t0.5\t1\t2408 86\n",
      "11\t1726\t0.1\t1\t1612 114\n",
      "12\t1559\t0.0\t1\t1482 77\n",
      "13\t1738\t0.0\t1\t1665 73\n",
      "14\t1872\t0.0\t1\t1794 78\n",
      "15\t1737\t0.0\t1\t1679 58\n",
      "16\t1567\t0.0\t1\t1497 70\n",
      "17\t1552\t0.0\t1\t1503 49\n",
      "18\t1299\t0.0\t1\t1246 53\n",
      "19\t1392\t0.0\t1\t1329 63\n",
      "20\t1439\t0.0\t1\t1384 55\n",
      "21\t1372\t0.0\t1\t1322 50\n",
      "22\t1178\t0.0\t1\t1124 54\n",
      "23\t1212\t0.0\t1\t1164 48\n",
      "24\t1370\t0.0\t1\t1313 57\n",
      "25\t1181\t0.0\t1\t1132 49\n",
      "26\t1319\t0.0\t1\t1243 76\n",
      "27\t1264\t0.0\t1\t1215 49\n",
      "28\t1186\t0.0\t1\t1139 47\n",
      "29\t1104\t0.0\t1\t1074 30\n",
      "30\t1114\t0.0\t1\t1078 36\n",
      "31\t2218\t0.0\t1\t2151 67\n",
      "32\t1085\t0.0\t1\t1036 49\n",
      "33\t900\t0.0\t1\t856 44\n",
      "34\t852\t0.0\t1\t816 36\n",
      "35\t792\t0.0\t1\t767 25\n",
      "36\t829\t0.0\t1\t802 27\n",
      "37\t737\t0.0\t1\t708 29\n",
      "38\t745\t0.0\t1\t711 34\n",
      "39\t783\t0.0\t1\t760 23\n",
      "40\t938\t0.0\t1\t902 36\n",
      "41\t642\t0.0\t1\t619 23\n",
      "42\t643\t0.0\t1\t616 27\n",
      "43\t655\t0.0\t1\t630 25\n",
      "44\t608\t0.0\t1\t587 21\n",
      "45\t578\t0.0\t1\t557 21\n",
      "46\t576\t0.0\t1\t559 17\n",
      "47\t460\t0.0\t1\t441 19\n",
      "48\t501\t0.0\t1\t481 20\n",
      "49\t894\t0.0\t1\t874 20\n",
      "50\t539\t0.0\t1\t514 25\n",
      "51\t564\t0.0\t1\t545 19\n",
      "52\t504\t0.0\t1\t485 19\n",
      "53\t389\t0.0\t1\t374 15\n",
      "54\t383\t0.0\t1\t368 15\n",
      "55\t300\t0.0\t1\t285 15\n",
      "56\t303\t0.0\t1\t288 15\n",
      "57\t270\t0.0\t1\t254 16\n",
      "58\t250\t0.0\t1\t239 11\n",
      "59\t212\t0.0\t1\t199 13\n",
      "60\t286\t0.0\t1\t279 7\n",
      "61\t178\t0.0\t1\t170 8\n",
      "62\t163\t0.0\t1\t160 3\n",
      "63\t155\t0.0\t1\t152 3\n",
      "64\t153\t0.0\t1\t150 3\n",
      "65\t151\t0.0\t1\t137 14\n",
      "66\t116\t0.0\t1\t112 4\n",
      "67\t131\t0.0\t1\t122 9\n",
      "68\t84\t0.0\t1\t79 5\n",
      "69\t123\t0.0\t1\t118 5\n",
      "70\t83\t0.0\t1\t79 4\n",
      "71\t116\t0.0\t1\t106 10\n",
      "72\t85\t0.0\t1\t81 4\n",
      "73\t111\t0.0\t1\t101 10\n",
      "74\t54\t0.0\t1\t49 5\n",
      "75\t48\t0.0\t1\t45 3\n",
      "76\t35\t0.0\t1\t34 1\n",
      "77\t38\t0.0\t1\t36 2\n",
      "78\t22\t0.0\t1\t21 1\n",
      "79\t28\t0.0\t1\t27 1\n",
      "80\t19\t0.0\t1\t13 6\n",
      "81\t25\t0.0\t1\t21 4\n",
      "82\t28\t0.0\t1\t25 3\n",
      "83\t46\t0.0\t1\t42 4\n",
      "84\t18\t0.0\t1\t17 1\n",
      "85\t12\t0.0\t1\t12\n",
      "86\t12\t0.0\t1\t11 1\n",
      "87\t9\t0.0\t1\t8 1\n",
      "88\t14\t0.0\t1\t13 1\n",
      "89\t8\t0.0\t1\t6 2\n",
      "90\t10\t0.0\t1\t9 1\n",
      "91\t23\t0.0\t1\t23\n",
      "92\t36\t0.0\t1\t35 1\n",
      "93\t169\t0.0\t1\t157 12\n",
      "94\t824\t0.0\t1\t800 24\n",
      "95\t1469\t0.0\t1\t1404 65\n",
      "96\t863\t0.0\t1\t820 43\n",
      "97\t555\t0.0\t1\t532 23\n",
      "98\t222\t0.0\t1\t211 11\n",
      "99\t256\t0.0\t1\t251 5\n",
      "100\t256\t0.0\t1\t225 31\n",
      "101\t965\t0.0\t1\t899 66\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950178_trimmed_R1.fastq.gz -p trimmed/ERR950178_trimmed_R2.fastq.gz ERR950178_1.fastq.gz.subset.fastq.gz ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 151,453 (30.3%)\n",
      "  Read 2 with adapter:                 148,240 (29.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               4,447 (0.9%)\n",
      "Pairs written (passing filters):       495,553 (99.1%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     92,624,117 bp (91.7%)\n",
      "  Read 1:    46,282,487 bp\n",
      "  Read 2:    46,341,630 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 151453 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.1%\n",
      "  C: 39.3%\n",
      "  G: 23.5%\n",
      "  T: 13.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11352\t7812.5\t0\t11352\n",
      "4\t5140\t1953.1\t0\t5140\n",
      "5\t3167\t488.3\t0\t3167\n",
      "6\t3371\t122.1\t0\t3371\n",
      "7\t3341\t30.5\t0\t3341\n",
      "8\t3150\t7.6\t0\t3150\n",
      "9\t3232\t1.9\t0\t3136 96\n",
      "10\t3603\t0.5\t1\t3399 204\n",
      "11\t3047\t0.1\t1\t2816 231\n",
      "12\t3075\t0.0\t1\t2812 263\n",
      "13\t2995\t0.0\t1\t2798 197\n",
      "14\t3387\t0.0\t1\t3187 200\n",
      "15\t3194\t0.0\t1\t2999 195\n",
      "16\t3029\t0.0\t1\t2867 162\n",
      "17\t2890\t0.0\t1\t2756 134\n",
      "18\t2829\t0.0\t1\t2689 140\n",
      "19\t2841\t0.0\t1\t2706 135\n",
      "20\t3002\t0.0\t1\t2855 147\n",
      "21\t2858\t0.0\t1\t2737 121\n",
      "22\t2716\t0.0\t1\t2615 101\n",
      "23\t2800\t0.0\t1\t2680 120\n",
      "24\t2768\t0.0\t1\t2635 133\n",
      "25\t2758\t0.0\t1\t2634 124\n",
      "26\t2964\t0.0\t1\t2852 112\n",
      "27\t2765\t0.0\t1\t2660 105\n",
      "28\t2618\t0.0\t1\t2512 106\n",
      "29\t2487\t0.0\t1\t2396 91\n",
      "30\t2558\t0.0\t1\t2459 99\n",
      "31\t3389\t0.0\t1\t3285 104\n",
      "32\t2427\t0.0\t1\t2343 84\n",
      "33\t2243\t0.0\t1\t2159 84\n",
      "34\t2174\t0.0\t1\t2089 85\n",
      "35\t2010\t0.0\t1\t1950 60\n",
      "36\t2006\t0.0\t1\t1924 82\n",
      "37\t2065\t0.0\t1\t1996 69\n",
      "38\t1894\t0.0\t1\t1830 64\n",
      "39\t1818\t0.0\t1\t1749 69\n",
      "40\t1779\t0.0\t1\t1714 65\n",
      "41\t1665\t0.0\t1\t1604 61\n",
      "42\t1711\t0.0\t1\t1650 61\n",
      "43\t1650\t0.0\t1\t1595 55\n",
      "44\t1701\t0.0\t1\t1644 57\n",
      "45\t1576\t0.0\t1\t1529 47\n",
      "46\t1803\t0.0\t1\t1746 57\n",
      "47\t1415\t0.0\t1\t1368 47\n",
      "48\t1445\t0.0\t1\t1405 40\n",
      "49\t1561\t0.0\t1\t1506 55\n",
      "50\t1315\t0.0\t1\t1266 49\n",
      "51\t1273\t0.0\t1\t1232 41\n",
      "52\t1190\t0.0\t1\t1146 44\n",
      "53\t1049\t0.0\t1\t1020 29\n",
      "54\t1065\t0.0\t1\t1026 39\n",
      "55\t875\t0.0\t1\t848 27\n",
      "56\t856\t0.0\t1\t828 28\n",
      "57\t769\t0.0\t1\t738 31\n",
      "58\t726\t0.0\t1\t701 25\n",
      "59\t671\t0.0\t1\t650 21\n",
      "60\t742\t0.0\t1\t711 31\n",
      "61\t613\t0.0\t1\t596 17\n",
      "62\t572\t0.0\t1\t549 23\n",
      "63\t567\t0.0\t1\t545 22\n",
      "64\t504\t0.0\t1\t479 25\n",
      "65\t452\t0.0\t1\t431 21\n",
      "66\t488\t0.0\t1\t466 22\n",
      "67\t455\t0.0\t1\t431 24\n",
      "68\t415\t0.0\t1\t396 19\n",
      "69\t369\t0.0\t1\t350 19\n",
      "70\t347\t0.0\t1\t329 18\n",
      "71\t337\t0.0\t1\t313 24\n",
      "72\t286\t0.0\t1\t269 17\n",
      "73\t272\t0.0\t1\t264 8\n",
      "74\t209\t0.0\t1\t201 8\n",
      "75\t197\t0.0\t1\t188 9\n",
      "76\t148\t0.0\t1\t140 8\n",
      "77\t121\t0.0\t1\t113 8\n",
      "78\t93\t0.0\t1\t87 6\n",
      "79\t74\t0.0\t1\t68 6\n",
      "80\t71\t0.0\t1\t70 1\n",
      "81\t58\t0.0\t1\t54 4\n",
      "82\t66\t0.0\t1\t56 10\n",
      "83\t60\t0.0\t1\t55 5\n",
      "84\t40\t0.0\t1\t39 1\n",
      "85\t28\t0.0\t1\t27 1\n",
      "86\t25\t0.0\t1\t22 3\n",
      "87\t23\t0.0\t1\t22 1\n",
      "88\t19\t0.0\t1\t13 6\n",
      "89\t17\t0.0\t1\t15 2\n",
      "90\t31\t0.0\t1\t28 3\n",
      "91\t41\t0.0\t1\t40 1\n",
      "92\t58\t0.0\t1\t52 6\n",
      "93\t175\t0.0\t1\t168 7\n",
      "94\t807\t0.0\t1\t743 64\n",
      "95\t1110\t0.0\t1\t1025 85\n",
      "96\t599\t0.0\t1\t551 48\n",
      "97\t269\t0.0\t1\t240 29\n",
      "98\t121\t0.0\t1\t116 5\n",
      "99\t147\t0.0\t1\t138 9\n",
      "100\t96\t0.0\t1\t90 6\n",
      "101\t273\t0.0\t1\t250 23\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 148240 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.8%\n",
      "  C: 19.7%\n",
      "  G: 28.3%\n",
      "  T: 32.1%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9955\t7812.5\t0\t9955\n",
      "4\t4657\t1953.1\t0\t4657\n",
      "5\t3319\t488.3\t0\t3319\n",
      "6\t3372\t122.1\t0\t3372\n",
      "7\t3380\t30.5\t0\t3380\n",
      "8\t3159\t7.6\t0\t3159\n",
      "9\t3314\t1.9\t0\t3271 43\n",
      "10\t3582\t0.5\t1\t3431 151\n",
      "11\t3025\t0.1\t1\t2863 162\n",
      "12\t3074\t0.0\t1\t2946 128\n",
      "13\t2983\t0.0\t1\t2849 134\n",
      "14\t3371\t0.0\t1\t3230 141\n",
      "15\t3185\t0.0\t1\t3057 128\n",
      "16\t3013\t0.0\t1\t2895 118\n",
      "17\t2856\t0.0\t1\t2751 105\n",
      "18\t2810\t0.0\t1\t2696 114\n",
      "19\t2827\t0.0\t1\t2720 107\n",
      "20\t2971\t0.0\t1\t2862 109\n",
      "21\t2848\t0.0\t1\t2740 108\n",
      "22\t2707\t0.0\t1\t2604 103\n",
      "23\t2786\t0.0\t1\t2681 105\n",
      "24\t2723\t0.0\t1\t2640 83\n",
      "25\t2735\t0.0\t1\t2646 89\n",
      "26\t2932\t0.0\t1\t2825 107\n",
      "27\t2732\t0.0\t1\t2619 113\n",
      "28\t2591\t0.0\t1\t2513 78\n",
      "29\t2468\t0.0\t1\t2383 85\n",
      "30\t2524\t0.0\t1\t2431 93\n",
      "31\t3375\t0.0\t1\t3246 129\n",
      "32\t2400\t0.0\t1\t2317 83\n",
      "33\t2213\t0.0\t1\t2114 99\n",
      "34\t2146\t0.0\t1\t2082 64\n",
      "35\t1989\t0.0\t1\t1931 58\n",
      "36\t1967\t0.0\t1\t1890 77\n",
      "37\t2043\t0.0\t1\t1971 72\n",
      "38\t1872\t0.0\t1\t1804 68\n",
      "39\t1793\t0.0\t1\t1737 56\n",
      "40\t1752\t0.0\t1\t1683 69\n",
      "41\t1631\t0.0\t1\t1577 54\n",
      "42\t1689\t0.0\t1\t1629 60\n",
      "43\t1631\t0.0\t1\t1565 66\n",
      "44\t1688\t0.0\t1\t1626 62\n",
      "45\t1553\t0.0\t1\t1502 51\n",
      "46\t1781\t0.0\t1\t1720 61\n",
      "47\t1394\t0.0\t1\t1353 41\n",
      "48\t1419\t0.0\t1\t1381 38\n",
      "49\t1528\t0.0\t1\t1473 55\n",
      "50\t1303\t0.0\t1\t1247 56\n",
      "51\t1250\t0.0\t1\t1203 47\n",
      "52\t1167\t0.0\t1\t1136 31\n",
      "53\t1022\t0.0\t1\t978 44\n",
      "54\t1044\t0.0\t1\t1001 43\n",
      "55\t859\t0.0\t1\t834 25\n",
      "56\t842\t0.0\t1\t813 29\n",
      "57\t743\t0.0\t1\t708 35\n",
      "58\t712\t0.0\t1\t686 26\n",
      "59\t659\t0.0\t1\t637 22\n",
      "60\t719\t0.0\t1\t693 26\n",
      "61\t607\t0.0\t1\t586 21\n",
      "62\t559\t0.0\t1\t536 23\n",
      "63\t556\t0.0\t1\t532 24\n",
      "64\t491\t0.0\t1\t472 19\n",
      "65\t436\t0.0\t1\t417 19\n",
      "66\t478\t0.0\t1\t463 15\n",
      "67\t434\t0.0\t1\t413 21\n",
      "68\t404\t0.0\t1\t391 13\n",
      "69\t355\t0.0\t1\t344 11\n",
      "70\t330\t0.0\t1\t320 10\n",
      "71\t322\t0.0\t1\t310 12\n",
      "72\t271\t0.0\t1\t259 12\n",
      "73\t260\t0.0\t1\t245 15\n",
      "74\t206\t0.0\t1\t193 13\n",
      "75\t185\t0.0\t1\t178 7\n",
      "76\t135\t0.0\t1\t135\n",
      "77\t115\t0.0\t1\t113 2\n",
      "78\t88\t0.0\t1\t84 4\n",
      "79\t69\t0.0\t1\t66 3\n",
      "80\t74\t0.0\t1\t70 4\n",
      "81\t53\t0.0\t1\t49 4\n",
      "82\t57\t0.0\t1\t53 4\n",
      "83\t53\t0.0\t1\t52 1\n",
      "84\t38\t0.0\t1\t34 4\n",
      "85\t26\t0.0\t1\t26\n",
      "86\t25\t0.0\t1\t21 4\n",
      "87\t22\t0.0\t1\t21 1\n",
      "88\t11\t0.0\t1\t10 1\n",
      "89\t14\t0.0\t1\t11 3\n",
      "90\t28\t0.0\t1\t26 2\n",
      "91\t39\t0.0\t1\t38 1\n",
      "92\t53\t0.0\t1\t50 3\n",
      "93\t169\t0.0\t1\t165 4\n",
      "94\t758\t0.0\t1\t731 27\n",
      "95\t1045\t0.0\t1\t992 53\n",
      "96\t546\t0.0\t1\t504 42\n",
      "97\t246\t0.0\t1\t237 9\n",
      "98\t116\t0.0\t1\t110 6\n",
      "99\t133\t0.0\t1\t128 5\n",
      "100\t93\t0.0\t1\t82 11\n",
      "101\t257\t0.0\t1\t244 13\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950179_trimmed_R1.fastq.gz -p trimmed/ERR950179_trimmed_R2.fastq.gz ERR950179_1.fastq.gz.subset.fastq.gz ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  95,778 (19.2%)\n",
      "  Read 2 with adapter:                  93,698 (18.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 457 (0.1%)\n",
      "Pairs written (passing filters):       499,543 (99.9%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     97,036,101 bp (96.1%)\n",
      "  Read 1:    48,499,177 bp\n",
      "  Read 2:    48,536,924 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 95778 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.1%\n",
      "  C: 39.4%\n",
      "  G: 23.2%\n",
      "  T: 13.2%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11414\t7812.5\t0\t11414\n",
      "4\t4851\t1953.1\t0\t4851\n",
      "5\t2809\t488.3\t0\t2809\n",
      "6\t2846\t122.1\t0\t2846\n",
      "7\t2569\t30.5\t0\t2569\n",
      "8\t2416\t7.6\t0\t2416\n",
      "9\t2397\t1.9\t0\t2350 47\n",
      "10\t3000\t0.5\t1\t2835 165\n",
      "11\t2404\t0.1\t1\t2270 134\n",
      "12\t2352\t0.0\t1\t2194 158\n",
      "13\t2302\t0.0\t1\t2170 132\n",
      "14\t2512\t0.0\t1\t2383 129\n",
      "15\t2358\t0.0\t1\t2222 136\n",
      "16\t2218\t0.0\t1\t2078 140\n",
      "17\t2134\t0.0\t1\t2023 111\n",
      "18\t1977\t0.0\t1\t1888 89\n",
      "19\t1849\t0.0\t1\t1760 89\n",
      "20\t2057\t0.0\t1\t1957 100\n",
      "21\t1874\t0.0\t1\t1786 88\n",
      "22\t1866\t0.0\t1\t1794 72\n",
      "23\t1745\t0.0\t1\t1676 69\n",
      "24\t1899\t0.0\t1\t1808 91\n",
      "25\t1715\t0.0\t1\t1640 75\n",
      "26\t1846\t0.0\t1\t1780 66\n",
      "27\t1630\t0.0\t1\t1561 69\n",
      "28\t1671\t0.0\t1\t1597 74\n",
      "29\t1495\t0.0\t1\t1424 71\n",
      "30\t1553\t0.0\t1\t1496 57\n",
      "31\t2180\t0.0\t1\t2112 68\n",
      "32\t1337\t0.0\t1\t1285 52\n",
      "33\t1229\t0.0\t1\t1178 51\n",
      "34\t1177\t0.0\t1\t1140 37\n",
      "35\t1117\t0.0\t1\t1079 38\n",
      "36\t1028\t0.0\t1\t998 30\n",
      "37\t1039\t0.0\t1\t1005 34\n",
      "38\t923\t0.0\t1\t882 41\n",
      "39\t946\t0.0\t1\t909 37\n",
      "40\t954\t0.0\t1\t916 38\n",
      "41\t807\t0.0\t1\t783 24\n",
      "42\t781\t0.0\t1\t754 27\n",
      "43\t819\t0.0\t1\t798 21\n",
      "44\t743\t0.0\t1\t715 28\n",
      "45\t646\t0.0\t1\t623 23\n",
      "46\t887\t0.0\t1\t861 26\n",
      "47\t642\t0.0\t1\t619 23\n",
      "48\t536\t0.0\t1\t517 19\n",
      "49\t668\t0.0\t1\t649 19\n",
      "50\t546\t0.0\t1\t526 20\n",
      "51\t497\t0.0\t1\t478 19\n",
      "52\t429\t0.0\t1\t414 15\n",
      "53\t404\t0.0\t1\t386 18\n",
      "54\t350\t0.0\t1\t338 12\n",
      "55\t311\t0.0\t1\t294 17\n",
      "56\t264\t0.0\t1\t255 9\n",
      "57\t236\t0.0\t1\t230 6\n",
      "58\t234\t0.0\t1\t227 7\n",
      "59\t203\t0.0\t1\t194 9\n",
      "60\t194\t0.0\t1\t184 10\n",
      "61\t144\t0.0\t1\t137 7\n",
      "62\t144\t0.0\t1\t136 8\n",
      "63\t153\t0.0\t1\t148 5\n",
      "64\t147\t0.0\t1\t141 6\n",
      "65\t117\t0.0\t1\t106 11\n",
      "66\t88\t0.0\t1\t79 9\n",
      "67\t105\t0.0\t1\t100 5\n",
      "68\t77\t0.0\t1\t74 3\n",
      "69\t84\t0.0\t1\t74 10\n",
      "70\t74\t0.0\t1\t68 6\n",
      "71\t72\t0.0\t1\t69 3\n",
      "72\t76\t0.0\t1\t69 7\n",
      "73\t61\t0.0\t1\t56 5\n",
      "74\t57\t0.0\t1\t52 5\n",
      "75\t55\t0.0\t1\t47 8\n",
      "76\t32\t0.0\t1\t29 3\n",
      "77\t23\t0.0\t1\t22 1\n",
      "78\t23\t0.0\t1\t21 2\n",
      "79\t16\t0.0\t1\t14 2\n",
      "80\t21\t0.0\t1\t13 8\n",
      "81\t16\t0.0\t1\t15 1\n",
      "82\t17\t0.0\t1\t14 3\n",
      "83\t19\t0.0\t1\t12 7\n",
      "84\t9\t0.0\t1\t8 1\n",
      "85\t7\t0.0\t1\t4 3\n",
      "86\t9\t0.0\t1\t9\n",
      "87\t6\t0.0\t1\t5 1\n",
      "88\t8\t0.0\t1\t6 2\n",
      "89\t9\t0.0\t1\t8 1\n",
      "90\t6\t0.0\t1\t4 2\n",
      "91\t8\t0.0\t1\t8\n",
      "92\t7\t0.0\t1\t5 2\n",
      "93\t15\t0.0\t1\t11 4\n",
      "94\t26\t0.0\t1\t22 4\n",
      "95\t31\t0.0\t1\t28 3\n",
      "96\t36\t0.0\t1\t26 10\n",
      "97\t25\t0.0\t1\t21 4\n",
      "98\t12\t0.0\t1\t7 5\n",
      "99\t5\t0.0\t1\t4 1\n",
      "100\t3\t0.0\t1\t1 2\n",
      "101\t49\t0.0\t1\t25 24\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 93698 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.4%\n",
      "  C: 23.9%\n",
      "  G: 30.3%\n",
      "  T: 28.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10787\t7812.5\t0\t10787\n",
      "4\t4223\t1953.1\t0\t4223\n",
      "5\t2916\t488.3\t0\t2916\n",
      "6\t2808\t122.1\t0\t2808\n",
      "7\t2630\t30.5\t0\t2630\n",
      "8\t2420\t7.6\t0\t2420\n",
      "9\t2409\t1.9\t0\t2381 28\n",
      "10\t2977\t0.5\t1\t2871 106\n",
      "11\t2398\t0.1\t1\t2286 112\n",
      "12\t2344\t0.0\t1\t2237 107\n",
      "13\t2268\t0.0\t1\t2171 97\n",
      "14\t2485\t0.0\t1\t2362 123\n",
      "15\t2353\t0.0\t1\t2241 112\n",
      "16\t2205\t0.0\t1\t2111 94\n",
      "17\t2135\t0.0\t1\t2032 103\n",
      "18\t1966\t0.0\t1\t1879 87\n",
      "19\t1833\t0.0\t1\t1770 63\n",
      "20\t2023\t0.0\t1\t1950 73\n",
      "21\t1860\t0.0\t1\t1796 64\n",
      "22\t1845\t0.0\t1\t1778 67\n",
      "23\t1713\t0.0\t1\t1651 62\n",
      "24\t1876\t0.0\t1\t1803 73\n",
      "25\t1712\t0.0\t1\t1644 68\n",
      "26\t1821\t0.0\t1\t1764 57\n",
      "27\t1623\t0.0\t1\t1554 69\n",
      "28\t1646\t0.0\t1\t1572 74\n",
      "29\t1468\t0.0\t1\t1404 64\n",
      "30\t1523\t0.0\t1\t1469 54\n",
      "31\t2161\t0.0\t1\t2072 89\n",
      "32\t1334\t0.0\t1\t1276 58\n",
      "33\t1217\t0.0\t1\t1160 57\n",
      "34\t1160\t0.0\t1\t1112 48\n",
      "35\t1096\t0.0\t1\t1059 37\n",
      "36\t1017\t0.0\t1\t981 36\n",
      "37\t1031\t0.0\t1\t990 41\n",
      "38\t910\t0.0\t1\t882 28\n",
      "39\t916\t0.0\t1\t880 36\n",
      "40\t935\t0.0\t1\t885 50\n",
      "41\t802\t0.0\t1\t772 30\n",
      "42\t770\t0.0\t1\t741 29\n",
      "43\t806\t0.0\t1\t769 37\n",
      "44\t728\t0.0\t1\t697 31\n",
      "45\t633\t0.0\t1\t601 32\n",
      "46\t861\t0.0\t1\t827 34\n",
      "47\t618\t0.0\t1\t605 13\n",
      "48\t523\t0.0\t1\t497 26\n",
      "49\t656\t0.0\t1\t621 35\n",
      "50\t519\t0.0\t1\t494 25\n",
      "51\t489\t0.0\t1\t471 18\n",
      "52\t407\t0.0\t1\t392 15\n",
      "53\t399\t0.0\t1\t378 21\n",
      "54\t343\t0.0\t1\t327 16\n",
      "55\t298\t0.0\t1\t285 13\n",
      "56\t256\t0.0\t1\t244 12\n",
      "57\t226\t0.0\t1\t221 5\n",
      "58\t224\t0.0\t1\t212 12\n",
      "59\t197\t0.0\t1\t191 6\n",
      "60\t181\t0.0\t1\t171 10\n",
      "61\t136\t0.0\t1\t127 9\n",
      "62\t143\t0.0\t1\t133 10\n",
      "63\t149\t0.0\t1\t143 6\n",
      "64\t131\t0.0\t1\t129 2\n",
      "65\t108\t0.0\t1\t104 4\n",
      "66\t87\t0.0\t1\t82 5\n",
      "67\t107\t0.0\t1\t96 11\n",
      "68\t70\t0.0\t1\t66 4\n",
      "69\t77\t0.0\t1\t72 5\n",
      "70\t70\t0.0\t1\t67 3\n",
      "71\t68\t0.0\t1\t62 6\n",
      "72\t67\t0.0\t1\t64 3\n",
      "73\t55\t0.0\t1\t50 5\n",
      "74\t55\t0.0\t1\t49 6\n",
      "75\t43\t0.0\t1\t39 4\n",
      "76\t28\t0.0\t1\t26 2\n",
      "77\t21\t0.0\t1\t19 2\n",
      "78\t19\t0.0\t1\t19\n",
      "79\t13\t0.0\t1\t10 3\n",
      "80\t14\t0.0\t1\t10 4\n",
      "81\t14\t0.0\t1\t11 3\n",
      "82\t15\t0.0\t1\t13 2\n",
      "83\t12\t0.0\t1\t11 1\n",
      "84\t11\t0.0\t1\t5 6\n",
      "85\t5\t0.0\t1\t3 2\n",
      "86\t6\t0.0\t1\t6\n",
      "87\t3\t0.0\t1\t3\n",
      "88\t3\t0.0\t1\t3\n",
      "89\t6\t0.0\t1\t3 3\n",
      "90\t4\t0.0\t1\t2 2\n",
      "91\t5\t0.0\t1\t5\n",
      "92\t4\t0.0\t1\t3 1\n",
      "93\t10\t0.0\t1\t10\n",
      "94\t23\t0.0\t1\t20 3\n",
      "95\t29\t0.0\t1\t27 2\n",
      "96\t28\t0.0\t1\t23 5\n",
      "97\t29\t0.0\t1\t19 10\n",
      "98\t10\t0.0\t1\t6 4\n",
      "99\t3\t0.0\t1\t2 1\n",
      "100\t2\t0.0\t1\t1 1\n",
      "101\t35\t0.0\t1\t26 9\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950180_trimmed_R1.fastq.gz -p trimmed/ERR950180_trimmed_R2.fastq.gz ERR950180_1.fastq.gz.subset.fastq.gz ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  98,458 (19.7%)\n",
      "  Read 2 with adapter:                  96,100 (19.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               3,491 (0.7%)\n",
      "Pairs written (passing filters):       496,509 (99.3%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,215,617 bp (95.3%)\n",
      "  Read 1:    48,082,239 bp\n",
      "  Read 2:    48,133,378 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 98458 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.0%\n",
      "  C: 40.6%\n",
      "  G: 21.4%\n",
      "  T: 13.7%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10919\t7812.5\t0\t10919\n",
      "4\t4525\t1953.1\t0\t4525\n",
      "5\t2514\t488.3\t0\t2514\n",
      "6\t2531\t122.1\t0\t2531\n",
      "7\t2765\t30.5\t0\t2765\n",
      "8\t2283\t7.6\t0\t2283\n",
      "9\t2705\t1.9\t0\t2628 77\n",
      "10\t2745\t0.5\t1\t2570 175\n",
      "11\t2266\t0.1\t1\t2094 172\n",
      "12\t2302\t0.0\t1\t2070 232\n",
      "13\t2254\t0.0\t1\t2099 155\n",
      "14\t2482\t0.0\t1\t2298 184\n",
      "15\t2232\t0.0\t1\t2038 194\n",
      "16\t2123\t0.0\t1\t1982 141\n",
      "17\t2054\t0.0\t1\t1937 117\n",
      "18\t1921\t0.0\t1\t1796 125\n",
      "19\t1836\t0.0\t1\t1733 103\n",
      "20\t1938\t0.0\t1\t1807 131\n",
      "21\t1852\t0.0\t1\t1730 122\n",
      "22\t1645\t0.0\t1\t1558 87\n",
      "23\t1719\t0.0\t1\t1623 96\n",
      "24\t1687\t0.0\t1\t1598 89\n",
      "25\t1620\t0.0\t1\t1520 100\n",
      "26\t1837\t0.0\t1\t1741 96\n",
      "27\t1838\t0.0\t1\t1746 92\n",
      "28\t1545\t0.0\t1\t1464 81\n",
      "29\t1407\t0.0\t1\t1337 70\n",
      "30\t1421\t0.0\t1\t1330 91\n",
      "31\t1868\t0.0\t1\t1769 99\n",
      "32\t1281\t0.0\t1\t1224 57\n",
      "33\t1214\t0.0\t1\t1162 52\n",
      "34\t1192\t0.0\t1\t1130 62\n",
      "35\t1128\t0.0\t1\t1072 56\n",
      "36\t1065\t0.0\t1\t1029 36\n",
      "37\t1135\t0.0\t1\t1090 45\n",
      "38\t995\t0.0\t1\t943 52\n",
      "39\t961\t0.0\t1\t926 35\n",
      "40\t1057\t0.0\t1\t1016 41\n",
      "41\t803\t0.0\t1\t759 44\n",
      "42\t825\t0.0\t1\t791 34\n",
      "43\t805\t0.0\t1\t767 38\n",
      "44\t761\t0.0\t1\t732 29\n",
      "45\t760\t0.0\t1\t711 49\n",
      "46\t870\t0.0\t1\t835 35\n",
      "47\t621\t0.0\t1\t596 25\n",
      "48\t658\t0.0\t1\t627 31\n",
      "49\t758\t0.0\t1\t725 33\n",
      "50\t599\t0.0\t1\t578 21\n",
      "51\t614\t0.0\t1\t590 24\n",
      "52\t557\t0.0\t1\t535 22\n",
      "53\t463\t0.0\t1\t445 18\n",
      "54\t469\t0.0\t1\t443 26\n",
      "55\t342\t0.0\t1\t331 11\n",
      "56\t395\t0.0\t1\t369 26\n",
      "57\t362\t0.0\t1\t342 20\n",
      "58\t314\t0.0\t1\t294 20\n",
      "59\t302\t0.0\t1\t282 20\n",
      "60\t345\t0.0\t1\t325 20\n",
      "61\t256\t0.0\t1\t243 13\n",
      "62\t241\t0.0\t1\t231 10\n",
      "63\t246\t0.0\t1\t229 17\n",
      "64\t215\t0.0\t1\t200 15\n",
      "65\t196\t0.0\t1\t183 13\n",
      "66\t156\t0.0\t1\t145 11\n",
      "67\t185\t0.0\t1\t174 11\n",
      "68\t163\t0.0\t1\t157 6\n",
      "69\t138\t0.0\t1\t127 11\n",
      "70\t117\t0.0\t1\t115 2\n",
      "71\t138\t0.0\t1\t129 9\n",
      "72\t114\t0.0\t1\t108 6\n",
      "73\t115\t0.0\t1\t108 7\n",
      "74\t94\t0.0\t1\t86 8\n",
      "75\t73\t0.0\t1\t61 12\n",
      "76\t62\t0.0\t1\t59 3\n",
      "77\t43\t0.0\t1\t42 1\n",
      "78\t44\t0.0\t1\t42 2\n",
      "79\t38\t0.0\t1\t35 3\n",
      "80\t29\t0.0\t1\t29\n",
      "81\t32\t0.0\t1\t30 2\n",
      "82\t38\t0.0\t1\t37 1\n",
      "83\t34\t0.0\t1\t31 3\n",
      "84\t17\t0.0\t1\t13 4\n",
      "85\t8\t0.0\t1\t7 1\n",
      "86\t8\t0.0\t1\t8\n",
      "87\t10\t0.0\t1\t7 3\n",
      "88\t8\t0.0\t1\t6 2\n",
      "89\t4\t0.0\t1\t4\n",
      "90\t7\t0.0\t1\t7\n",
      "91\t23\t0.0\t1\t21 2\n",
      "92\t44\t0.0\t1\t41 3\n",
      "93\t152\t0.0\t1\t141 11\n",
      "94\t620\t0.0\t1\t580 40\n",
      "95\t957\t0.0\t1\t888 69\n",
      "96\t545\t0.0\t1\t500 45\n",
      "97\t258\t0.0\t1\t230 28\n",
      "98\t107\t0.0\t1\t98 9\n",
      "99\t124\t0.0\t1\t116 8\n",
      "100\t81\t0.0\t1\t78 3\n",
      "101\t233\t0.0\t1\t182 51\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 96100 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.3%\n",
      "  C: 19.6%\n",
      "  G: 29.4%\n",
      "  T: 31.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10384\t7812.5\t0\t10384\n",
      "4\t4211\t1953.1\t0\t4211\n",
      "5\t2648\t488.3\t0\t2648\n",
      "6\t2508\t122.1\t0\t2508\n",
      "7\t2847\t30.5\t0\t2847\n",
      "8\t2287\t7.6\t0\t2287\n",
      "9\t2753\t1.9\t0\t2717 36\n",
      "10\t2692\t0.5\t1\t2565 127\n",
      "11\t2225\t0.1\t1\t2106 119\n",
      "12\t2278\t0.0\t1\t2166 112\n",
      "13\t2214\t0.0\t1\t2113 101\n",
      "14\t2428\t0.0\t1\t2315 113\n",
      "15\t2179\t0.0\t1\t2078 101\n",
      "16\t2088\t0.0\t1\t1998 90\n",
      "17\t2044\t0.0\t1\t1963 81\n",
      "18\t1887\t0.0\t1\t1812 75\n",
      "19\t1790\t0.0\t1\t1727 63\n",
      "20\t1903\t0.0\t1\t1828 75\n",
      "21\t1812\t0.0\t1\t1736 76\n",
      "22\t1602\t0.0\t1\t1538 64\n",
      "23\t1693\t0.0\t1\t1608 85\n",
      "24\t1646\t0.0\t1\t1580 66\n",
      "25\t1574\t0.0\t1\t1519 55\n",
      "26\t1807\t0.0\t1\t1734 73\n",
      "27\t1805\t0.0\t1\t1728 77\n",
      "28\t1519\t0.0\t1\t1459 60\n",
      "29\t1372\t0.0\t1\t1314 58\n",
      "30\t1389\t0.0\t1\t1334 55\n",
      "31\t1823\t0.0\t1\t1757 66\n",
      "32\t1261\t0.0\t1\t1224 37\n",
      "33\t1188\t0.0\t1\t1156 32\n",
      "34\t1165\t0.0\t1\t1119 46\n",
      "35\t1106\t0.0\t1\t1068 38\n",
      "36\t1046\t0.0\t1\t1006 40\n",
      "37\t1108\t0.0\t1\t1062 46\n",
      "38\t974\t0.0\t1\t920 54\n",
      "39\t927\t0.0\t1\t900 27\n",
      "40\t1031\t0.0\t1\t988 43\n",
      "41\t777\t0.0\t1\t742 35\n",
      "42\t806\t0.0\t1\t772 34\n",
      "43\t783\t0.0\t1\t749 34\n",
      "44\t751\t0.0\t1\t732 19\n",
      "45\t730\t0.0\t1\t700 30\n",
      "46\t847\t0.0\t1\t812 35\n",
      "47\t604\t0.0\t1\t571 33\n",
      "48\t645\t0.0\t1\t612 33\n",
      "49\t735\t0.0\t1\t716 19\n",
      "50\t589\t0.0\t1\t558 31\n",
      "51\t595\t0.0\t1\t568 27\n",
      "52\t536\t0.0\t1\t516 20\n",
      "53\t453\t0.0\t1\t435 18\n",
      "54\t448\t0.0\t1\t426 22\n",
      "55\t332\t0.0\t1\t319 13\n",
      "56\t380\t0.0\t1\t362 18\n",
      "57\t345\t0.0\t1\t336 9\n",
      "58\t299\t0.0\t1\t282 17\n",
      "59\t290\t0.0\t1\t277 13\n",
      "60\t337\t0.0\t1\t327 10\n",
      "61\t242\t0.0\t1\t235 7\n",
      "62\t236\t0.0\t1\t228 8\n",
      "63\t237\t0.0\t1\t225 12\n",
      "64\t205\t0.0\t1\t201 4\n",
      "65\t186\t0.0\t1\t175 11\n",
      "66\t150\t0.0\t1\t143 7\n",
      "67\t177\t0.0\t1\t167 10\n",
      "68\t156\t0.0\t1\t148 8\n",
      "69\t128\t0.0\t1\t122 6\n",
      "70\t114\t0.0\t1\t110 4\n",
      "71\t131\t0.0\t1\t125 6\n",
      "72\t110\t0.0\t1\t101 9\n",
      "73\t112\t0.0\t1\t107 5\n",
      "74\t88\t0.0\t1\t83 5\n",
      "75\t65\t0.0\t1\t60 5\n",
      "76\t58\t0.0\t1\t54 4\n",
      "77\t40\t0.0\t1\t39 1\n",
      "78\t44\t0.0\t1\t34 10\n",
      "79\t36\t0.0\t1\t31 5\n",
      "80\t33\t0.0\t1\t30 3\n",
      "81\t30\t0.0\t1\t30\n",
      "82\t36\t0.0\t1\t35 1\n",
      "83\t30\t0.0\t1\t30\n",
      "84\t15\t0.0\t1\t15\n",
      "85\t7\t0.0\t1\t7\n",
      "86\t8\t0.0\t1\t7 1\n",
      "87\t8\t0.0\t1\t6 2\n",
      "88\t6\t0.0\t1\t6\n",
      "89\t3\t0.0\t1\t3\n",
      "90\t6\t0.0\t1\t4 2\n",
      "91\t19\t0.0\t1\t17 2\n",
      "92\t43\t0.0\t1\t38 5\n",
      "93\t143\t0.0\t1\t138 5\n",
      "94\t586\t0.0\t1\t554 32\n",
      "95\t898\t0.0\t1\t868 30\n",
      "96\t504\t0.0\t1\t472 32\n",
      "97\t228\t0.0\t1\t217 11\n",
      "98\t99\t0.0\t1\t96 3\n",
      "99\t114\t0.0\t1\t107 7\n",
      "100\t79\t0.0\t1\t69 10\n",
      "101\t194\t0.0\t1\t177 17\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950181_trimmed_R1.fastq.gz -p trimmed/ERR950181_trimmed_R2.fastq.gz ERR950181_1.fastq.gz.subset.fastq.gz ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                 111,242 (22.2%)\n",
      "  Read 2 with adapter:                 111,126 (22.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 506 (0.1%)\n",
      "Pairs written (passing filters):       499,494 (99.9%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,053,772 bp (95.1%)\n",
      "  Read 1:    48,018,585 bp\n",
      "  Read 2:    48,035,187 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 111242 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.4%\n",
      "  C: 42.3%\n",
      "  G: 20.6%\n",
      "  T: 12.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10041\t7812.5\t0\t10041\n",
      "4\t4996\t1953.1\t0\t4996\n",
      "5\t3081\t488.3\t0\t3081\n",
      "6\t3137\t122.1\t0\t3137\n",
      "7\t3008\t30.5\t0\t3008\n",
      "8\t2789\t7.6\t0\t2789\n",
      "9\t2765\t1.9\t0\t2738 27\n",
      "10\t3416\t0.5\t1\t3254 162\n",
      "11\t2784\t0.1\t1\t2591 193\n",
      "12\t2745\t0.0\t1\t2560 185\n",
      "13\t2715\t0.0\t1\t2502 213\n",
      "14\t2992\t0.0\t1\t2833 159\n",
      "15\t2820\t0.0\t1\t2640 180\n",
      "16\t2561\t0.0\t1\t2378 183\n",
      "17\t2537\t0.0\t1\t2382 155\n",
      "18\t2190\t0.0\t1\t2049 141\n",
      "19\t2152\t0.0\t1\t2035 117\n",
      "20\t2351\t0.0\t1\t2212 139\n",
      "21\t2193\t0.0\t1\t2070 123\n",
      "22\t2071\t0.0\t1\t1972 99\n",
      "23\t2158\t0.0\t1\t2055 103\n",
      "24\t2428\t0.0\t1\t2296 132\n",
      "25\t2250\t0.0\t1\t2121 129\n",
      "26\t2345\t0.0\t1\t2229 116\n",
      "27\t2126\t0.0\t1\t2036 90\n",
      "28\t2127\t0.0\t1\t2054 73\n",
      "29\t1926\t0.0\t1\t1850 76\n",
      "30\t1806\t0.0\t1\t1726 80\n",
      "31\t2525\t0.0\t1\t2434 91\n",
      "32\t1447\t0.0\t1\t1399 48\n",
      "33\t1651\t0.0\t1\t1584 67\n",
      "34\t1420\t0.0\t1\t1361 59\n",
      "35\t1436\t0.0\t1\t1370 66\n",
      "36\t1397\t0.0\t1\t1344 53\n",
      "37\t1383\t0.0\t1\t1334 49\n",
      "38\t1182\t0.0\t1\t1135 47\n",
      "39\t1099\t0.0\t1\t1062 37\n",
      "40\t1293\t0.0\t1\t1245 48\n",
      "41\t942\t0.0\t1\t908 34\n",
      "42\t988\t0.0\t1\t958 30\n",
      "43\t967\t0.0\t1\t927 40\n",
      "44\t936\t0.0\t1\t907 29\n",
      "45\t863\t0.0\t1\t834 29\n",
      "46\t1064\t0.0\t1\t1025 39\n",
      "47\t789\t0.0\t1\t765 24\n",
      "48\t773\t0.0\t1\t750 23\n",
      "49\t926\t0.0\t1\t904 22\n",
      "50\t659\t0.0\t1\t633 26\n",
      "51\t646\t0.0\t1\t617 29\n",
      "52\t587\t0.0\t1\t572 15\n",
      "53\t479\t0.0\t1\t461 18\n",
      "54\t465\t0.0\t1\t444 21\n",
      "55\t433\t0.0\t1\t413 20\n",
      "56\t361\t0.0\t1\t347 14\n",
      "57\t338\t0.0\t1\t326 12\n",
      "58\t304\t0.0\t1\t292 12\n",
      "59\t268\t0.0\t1\t259 9\n",
      "60\t263\t0.0\t1\t252 11\n",
      "61\t250\t0.0\t1\t245 5\n",
      "62\t230\t0.0\t1\t219 11\n",
      "63\t209\t0.0\t1\t204 5\n",
      "64\t194\t0.0\t1\t186 8\n",
      "65\t204\t0.0\t1\t193 11\n",
      "66\t144\t0.0\t1\t138 6\n",
      "67\t177\t0.0\t1\t170 7\n",
      "68\t157\t0.0\t1\t146 11\n",
      "69\t138\t0.0\t1\t132 6\n",
      "70\t117\t0.0\t1\t110 7\n",
      "71\t109\t0.0\t1\t108 1\n",
      "72\t110\t0.0\t1\t107 3\n",
      "73\t157\t0.0\t1\t147 10\n",
      "74\t67\t0.0\t1\t63 4\n",
      "75\t45\t0.0\t1\t43 2\n",
      "76\t50\t0.0\t1\t48 2\n",
      "77\t34\t0.0\t1\t32 2\n",
      "78\t32\t0.0\t1\t31 1\n",
      "79\t25\t0.0\t1\t23 2\n",
      "80\t40\t0.0\t1\t32 8\n",
      "81\t30\t0.0\t1\t29 1\n",
      "82\t39\t0.0\t1\t33 6\n",
      "83\t30\t0.0\t1\t28 2\n",
      "84\t12\t0.0\t1\t10 2\n",
      "85\t6\t0.0\t1\t6\n",
      "86\t18\t0.0\t1\t17 1\n",
      "87\t6\t0.0\t1\t5 1\n",
      "88\t4\t0.0\t1\t4\n",
      "89\t4\t0.0\t1\t3 1\n",
      "90\t4\t0.0\t1\t2 2\n",
      "91\t9\t0.0\t1\t7 2\n",
      "92\t11\t0.0\t1\t10 1\n",
      "93\t19\t0.0\t1\t18 1\n",
      "94\t32\t0.0\t1\t29 3\n",
      "95\t43\t0.0\t1\t41 2\n",
      "96\t35\t0.0\t1\t32 3\n",
      "97\t22\t0.0\t1\t21 1\n",
      "98\t9\t0.0\t1\t5 4\n",
      "99\t5\t0.0\t1\t3 2\n",
      "100\t2\t0.0\t1\t0 2\n",
      "101\t19\t0.0\t1\t0 19\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 111126 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.4%\n",
      "  C: 21.1%\n",
      "  G: 30.0%\n",
      "  T: 31.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10296\t7812.5\t0\t10296\n",
      "4\t4541\t1953.1\t0\t4541\n",
      "5\t3212\t488.3\t0\t3212\n",
      "6\t3253\t122.1\t0\t3253\n",
      "7\t3068\t30.5\t0\t3068\n",
      "8\t2897\t7.6\t0\t2897\n",
      "9\t2852\t1.9\t0\t2827 25\n",
      "10\t3394\t0.5\t1\t3293 101\n",
      "11\t2795\t0.1\t1\t2679 116\n",
      "12\t2762\t0.0\t1\t2653 109\n",
      "13\t2767\t0.0\t1\t2676 91\n",
      "14\t3012\t0.0\t1\t2895 117\n",
      "15\t2824\t0.0\t1\t2746 78\n",
      "16\t2565\t0.0\t1\t2487 78\n",
      "17\t2521\t0.0\t1\t2407 114\n",
      "18\t2189\t0.0\t1\t2112 77\n",
      "19\t2147\t0.0\t1\t2073 74\n",
      "20\t2334\t0.0\t1\t2251 83\n",
      "21\t2187\t0.0\t1\t2094 93\n",
      "22\t2073\t0.0\t1\t2004 69\n",
      "23\t2151\t0.0\t1\t2077 74\n",
      "24\t2417\t0.0\t1\t2341 76\n",
      "25\t2260\t0.0\t1\t2182 78\n",
      "26\t2330\t0.0\t1\t2234 96\n",
      "27\t2118\t0.0\t1\t2042 76\n",
      "28\t2130\t0.0\t1\t2061 69\n",
      "29\t1919\t0.0\t1\t1848 71\n",
      "30\t1793\t0.0\t1\t1741 52\n",
      "31\t2509\t0.0\t1\t2419 90\n",
      "32\t1437\t0.0\t1\t1374 63\n",
      "33\t1650\t0.0\t1\t1577 73\n",
      "34\t1406\t0.0\t1\t1355 51\n",
      "35\t1419\t0.0\t1\t1376 43\n",
      "36\t1399\t0.0\t1\t1337 62\n",
      "37\t1379\t0.0\t1\t1327 52\n",
      "38\t1173\t0.0\t1\t1133 40\n",
      "39\t1085\t0.0\t1\t1052 33\n",
      "40\t1283\t0.0\t1\t1234 49\n",
      "41\t933\t0.0\t1\t900 33\n",
      "42\t994\t0.0\t1\t944 50\n",
      "43\t956\t0.0\t1\t930 26\n",
      "44\t931\t0.0\t1\t893 38\n",
      "45\t851\t0.0\t1\t823 28\n",
      "46\t1054\t0.0\t1\t1010 44\n",
      "47\t781\t0.0\t1\t760 21\n",
      "48\t760\t0.0\t1\t735 25\n",
      "49\t914\t0.0\t1\t883 31\n",
      "50\t657\t0.0\t1\t624 33\n",
      "51\t637\t0.0\t1\t617 20\n",
      "52\t589\t0.0\t1\t566 23\n",
      "53\t472\t0.0\t1\t456 16\n",
      "54\t449\t0.0\t1\t434 15\n",
      "55\t426\t0.0\t1\t408 18\n",
      "56\t361\t0.0\t1\t343 18\n",
      "57\t334\t0.0\t1\t317 17\n",
      "58\t296\t0.0\t1\t291 5\n",
      "59\t259\t0.0\t1\t252 7\n",
      "60\t252\t0.0\t1\t247 5\n",
      "61\t246\t0.0\t1\t240 6\n",
      "62\t226\t0.0\t1\t218 8\n",
      "63\t205\t0.0\t1\t196 9\n",
      "64\t186\t0.0\t1\t174 12\n",
      "65\t196\t0.0\t1\t187 9\n",
      "66\t140\t0.0\t1\t134 6\n",
      "67\t169\t0.0\t1\t164 5\n",
      "68\t143\t0.0\t1\t136 7\n",
      "69\t132\t0.0\t1\t125 7\n",
      "70\t111\t0.0\t1\t106 5\n",
      "71\t104\t0.0\t1\t102 2\n",
      "72\t107\t0.0\t1\t102 5\n",
      "73\t150\t0.0\t1\t140 10\n",
      "74\t61\t0.0\t1\t59 2\n",
      "75\t42\t0.0\t1\t39 3\n",
      "76\t43\t0.0\t1\t39 4\n",
      "77\t31\t0.0\t1\t27 4\n",
      "78\t30\t0.0\t1\t28 2\n",
      "79\t22\t0.0\t1\t22\n",
      "80\t27\t0.0\t1\t25 2\n",
      "81\t29\t0.0\t1\t27 2\n",
      "82\t31\t0.0\t1\t30 1\n",
      "83\t28\t0.0\t1\t26 2\n",
      "84\t8\t0.0\t1\t7 1\n",
      "85\t6\t0.0\t1\t6\n",
      "86\t12\t0.0\t1\t12\n",
      "87\t6\t0.0\t1\t5 1\n",
      "88\t3\t0.0\t1\t3\n",
      "89\t2\t0.0\t1\t1 1\n",
      "90\t2\t0.0\t1\t2\n",
      "91\t4\t0.0\t1\t3 1\n",
      "92\t4\t0.0\t1\t4\n",
      "93\t17\t0.0\t1\t14 3\n",
      "94\t28\t0.0\t1\t25 3\n",
      "95\t39\t0.0\t1\t37 2\n",
      "96\t31\t0.0\t1\t29 2\n",
      "97\t21\t0.0\t1\t17 4\n",
      "98\t6\t0.0\t1\t5 1\n",
      "99\t1\t0.0\t1\t1\n",
      "100\t2\t0.0\t1\t2\n",
      "101\t22\t0.0\t1\t19 3\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950182_trimmed_R1.fastq.gz -p trimmed/ERR950182_trimmed_R2.fastq.gz ERR950182_1.fastq.gz.subset.fastq.gz ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  82,333 (16.5%)\n",
      "  Read 2 with adapter:                  80,595 (16.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 787 (0.2%)\n",
      "Pairs written (passing filters):       499,213 (99.8%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     97,501,259 bp (96.5%)\n",
      "  Read 1:    48,728,368 bp\n",
      "  Read 2:    48,772,891 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 82333 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 23.2%\n",
      "  C: 41.2%\n",
      "  G: 22.0%\n",
      "  T: 13.5%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10463\t7812.5\t0\t10463\n",
      "4\t4496\t1953.1\t0\t4496\n",
      "5\t2313\t488.3\t0\t2313\n",
      "6\t2308\t122.1\t0\t2308\n",
      "7\t2356\t30.5\t0\t2356\n",
      "8\t1968\t7.6\t0\t1968\n",
      "9\t2097\t1.9\t0\t2054 43\n",
      "10\t2492\t0.5\t1\t2374 118\n",
      "11\t2011\t0.1\t1\t1878 133\n",
      "12\t1954\t0.0\t1\t1839 115\n",
      "13\t1886\t0.0\t1\t1777 109\n",
      "14\t1959\t0.0\t1\t1848 111\n",
      "15\t1907\t0.0\t1\t1793 114\n",
      "16\t1764\t0.0\t1\t1665 99\n",
      "17\t1678\t0.0\t1\t1597 81\n",
      "18\t1664\t0.0\t1\t1573 91\n",
      "19\t1563\t0.0\t1\t1473 90\n",
      "20\t1628\t0.0\t1\t1553 75\n",
      "21\t1518\t0.0\t1\t1443 75\n",
      "22\t1378\t0.0\t1\t1328 50\n",
      "23\t1412\t0.0\t1\t1349 63\n",
      "24\t1508\t0.0\t1\t1450 58\n",
      "25\t1492\t0.0\t1\t1429 63\n",
      "26\t1532\t0.0\t1\t1459 73\n",
      "27\t1453\t0.0\t1\t1393 60\n",
      "28\t1357\t0.0\t1\t1302 55\n",
      "29\t1239\t0.0\t1\t1185 54\n",
      "30\t1321\t0.0\t1\t1264 57\n",
      "31\t1937\t0.0\t1\t1850 87\n",
      "32\t1073\t0.0\t1\t1036 37\n",
      "33\t1012\t0.0\t1\t962 50\n",
      "34\t948\t0.0\t1\t911 37\n",
      "35\t878\t0.0\t1\t848 30\n",
      "36\t890\t0.0\t1\t851 39\n",
      "37\t1020\t0.0\t1\t977 43\n",
      "38\t776\t0.0\t1\t732 44\n",
      "39\t767\t0.0\t1\t728 39\n",
      "40\t830\t0.0\t1\t797 33\n",
      "41\t646\t0.0\t1\t622 24\n",
      "42\t648\t0.0\t1\t616 32\n",
      "43\t602\t0.0\t1\t583 19\n",
      "44\t648\t0.0\t1\t623 25\n",
      "45\t574\t0.0\t1\t548 26\n",
      "46\t691\t0.0\t1\t662 29\n",
      "47\t466\t0.0\t1\t447 19\n",
      "48\t497\t0.0\t1\t479 18\n",
      "49\t610\t0.0\t1\t589 21\n",
      "50\t507\t0.0\t1\t490 17\n",
      "51\t456\t0.0\t1\t439 17\n",
      "52\t400\t0.0\t1\t381 19\n",
      "53\t361\t0.0\t1\t352 9\n",
      "54\t357\t0.0\t1\t336 21\n",
      "55\t315\t0.0\t1\t300 15\n",
      "56\t275\t0.0\t1\t263 12\n",
      "57\t220\t0.0\t1\t208 12\n",
      "58\t239\t0.0\t1\t226 13\n",
      "59\t219\t0.0\t1\t202 17\n",
      "60\t255\t0.0\t1\t237 18\n",
      "61\t169\t0.0\t1\t161 8\n",
      "62\t160\t0.0\t1\t148 12\n",
      "63\t142\t0.0\t1\t138 4\n",
      "64\t148\t0.0\t1\t140 8\n",
      "65\t144\t0.0\t1\t136 8\n",
      "66\t127\t0.0\t1\t117 10\n",
      "67\t151\t0.0\t1\t144 7\n",
      "68\t119\t0.0\t1\t109 10\n",
      "69\t104\t0.0\t1\t102 2\n",
      "70\t73\t0.0\t1\t68 5\n",
      "71\t87\t0.0\t1\t78 9\n",
      "72\t79\t0.0\t1\t72 7\n",
      "73\t89\t0.0\t1\t88 1\n",
      "74\t52\t0.0\t1\t49 3\n",
      "75\t46\t0.0\t1\t44 2\n",
      "76\t51\t0.0\t1\t43 8\n",
      "77\t26\t0.0\t1\t22 4\n",
      "78\t38\t0.0\t1\t32 6\n",
      "79\t28\t0.0\t1\t25 3\n",
      "80\t28\t0.0\t1\t24 4\n",
      "81\t19\t0.0\t1\t15 4\n",
      "82\t30\t0.0\t1\t27 3\n",
      "83\t22\t0.0\t1\t19 3\n",
      "84\t11\t0.0\t1\t11\n",
      "85\t13\t0.0\t1\t10 3\n",
      "86\t6\t0.0\t1\t5 1\n",
      "87\t4\t0.0\t1\t3 1\n",
      "88\t11\t0.0\t1\t9 2\n",
      "89\t3\t0.0\t1\t3\n",
      "90\t7\t0.0\t1\t7\n",
      "91\t10\t0.0\t1\t7 3\n",
      "92\t11\t0.0\t1\t9 2\n",
      "93\t18\t0.0\t1\t15 3\n",
      "94\t72\t0.0\t1\t66 6\n",
      "95\t103\t0.0\t1\t99 4\n",
      "96\t69\t0.0\t1\t58 11\n",
      "97\t57\t0.0\t1\t44 13\n",
      "98\t17\t0.0\t1\t14 3\n",
      "99\t20\t0.0\t1\t19 1\n",
      "100\t22\t0.0\t1\t18 4\n",
      "101\t113\t0.0\t1\t75 38\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 80595 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.1%\n",
      "  C: 21.1%\n",
      "  G: 30.5%\n",
      "  T: 29.3%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10409\t7812.5\t0\t10409\n",
      "4\t3919\t1953.1\t0\t3919\n",
      "5\t2381\t488.3\t0\t2381\n",
      "6\t2302\t122.1\t0\t2302\n",
      "7\t2412\t30.5\t0\t2412\n",
      "8\t1971\t7.6\t0\t1971\n",
      "9\t2111\t1.9\t0\t2094 17\n",
      "10\t2468\t0.5\t1\t2387 81\n",
      "11\t1979\t0.1\t1\t1900 79\n",
      "12\t1941\t0.0\t1\t1864 77\n",
      "13\t1870\t0.0\t1\t1786 84\n",
      "14\t1935\t0.0\t1\t1867 68\n",
      "15\t1897\t0.0\t1\t1820 77\n",
      "16\t1766\t0.0\t1\t1690 76\n",
      "17\t1667\t0.0\t1\t1593 74\n",
      "18\t1633\t0.0\t1\t1577 56\n",
      "19\t1551\t0.0\t1\t1492 59\n",
      "20\t1598\t0.0\t1\t1535 63\n",
      "21\t1500\t0.0\t1\t1444 56\n",
      "22\t1361\t0.0\t1\t1319 42\n",
      "23\t1399\t0.0\t1\t1339 60\n",
      "24\t1485\t0.0\t1\t1449 36\n",
      "25\t1488\t0.0\t1\t1416 72\n",
      "26\t1502\t0.0\t1\t1436 66\n",
      "27\t1432\t0.0\t1\t1370 62\n",
      "28\t1327\t0.0\t1\t1275 52\n",
      "29\t1216\t0.0\t1\t1166 50\n",
      "30\t1282\t0.0\t1\t1238 44\n",
      "31\t1894\t0.0\t1\t1827 67\n",
      "32\t1059\t0.0\t1\t1023 36\n",
      "33\t981\t0.0\t1\t938 43\n",
      "34\t927\t0.0\t1\t899 28\n",
      "35\t865\t0.0\t1\t832 33\n",
      "36\t866\t0.0\t1\t836 30\n",
      "37\t987\t0.0\t1\t949 38\n",
      "38\t743\t0.0\t1\t706 37\n",
      "39\t739\t0.0\t1\t705 34\n",
      "40\t815\t0.0\t1\t786 29\n",
      "41\t639\t0.0\t1\t616 23\n",
      "42\t638\t0.0\t1\t609 29\n",
      "43\t587\t0.0\t1\t566 21\n",
      "44\t630\t0.0\t1\t614 16\n",
      "45\t560\t0.0\t1\t540 20\n",
      "46\t670\t0.0\t1\t636 34\n",
      "47\t442\t0.0\t1\t428 14\n",
      "48\t478\t0.0\t1\t461 17\n",
      "49\t589\t0.0\t1\t566 23\n",
      "50\t503\t0.0\t1\t470 33\n",
      "51\t446\t0.0\t1\t418 28\n",
      "52\t384\t0.0\t1\t363 21\n",
      "53\t352\t0.0\t1\t336 16\n",
      "54\t336\t0.0\t1\t324 12\n",
      "55\t308\t0.0\t1\t289 19\n",
      "56\t264\t0.0\t1\t253 11\n",
      "57\t209\t0.0\t1\t203 6\n",
      "58\t226\t0.0\t1\t218 8\n",
      "59\t203\t0.0\t1\t191 12\n",
      "60\t234\t0.0\t1\t224 10\n",
      "61\t167\t0.0\t1\t153 14\n",
      "62\t141\t0.0\t1\t135 6\n",
      "63\t135\t0.0\t1\t126 9\n",
      "64\t133\t0.0\t1\t127 6\n",
      "65\t131\t0.0\t1\t124 7\n",
      "66\t120\t0.0\t1\t110 10\n",
      "67\t142\t0.0\t1\t135 7\n",
      "68\t109\t0.0\t1\t102 7\n",
      "69\t98\t0.0\t1\t95 3\n",
      "70\t73\t0.0\t1\t68 5\n",
      "71\t72\t0.0\t1\t68 4\n",
      "72\t72\t0.0\t1\t68 4\n",
      "73\t81\t0.0\t1\t76 5\n",
      "74\t45\t0.0\t1\t41 4\n",
      "75\t42\t0.0\t1\t38 4\n",
      "76\t44\t0.0\t1\t41 3\n",
      "77\t20\t0.0\t1\t19 1\n",
      "78\t29\t0.0\t1\t23 6\n",
      "79\t25\t0.0\t1\t22 3\n",
      "80\t19\t0.0\t1\t18 1\n",
      "81\t15\t0.0\t1\t15\n",
      "82\t23\t0.0\t1\t20 3\n",
      "83\t18\t0.0\t1\t17 1\n",
      "84\t4\t0.0\t1\t4\n",
      "85\t11\t0.0\t1\t11\n",
      "86\t1\t0.0\t1\t1\n",
      "87\t2\t0.0\t1\t0 2\n",
      "88\t5\t0.0\t1\t5\n",
      "89\t3\t0.0\t1\t3\n",
      "90\t5\t0.0\t1\t4 1\n",
      "91\t6\t0.0\t1\t6\n",
      "92\t7\t0.0\t1\t7\n",
      "93\t15\t0.0\t1\t14 1\n",
      "94\t66\t0.0\t1\t62 4\n",
      "95\t94\t0.0\t1\t91 3\n",
      "96\t61\t0.0\t1\t54 7\n",
      "97\t45\t0.0\t1\t36 9\n",
      "98\t14\t0.0\t1\t13 1\n",
      "99\t18\t0.0\t1\t15 3\n",
      "100\t20\t0.0\t1\t18 2\n",
      "101\t88\t0.0\t1\t74 14\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950186_trimmed_R1.fastq.gz -p trimmed/ERR950186_trimmed_R2.fastq.gz ERR950186_1.fastq.gz.subset.fastq.gz ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  92,888 (18.6%)\n",
      "  Read 2 with adapter:                  93,160 (18.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,491 (0.3%)\n",
      "Pairs written (passing filters):       498,509 (99.7%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     96,804,306 bp (95.8%)\n",
      "  Read 1:    48,391,688 bp\n",
      "  Read 2:    48,412,618 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 92888 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.9%\n",
      "  C: 42.1%\n",
      "  G: 21.4%\n",
      "  T: 13.4%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9305\t7812.5\t0\t9305\n",
      "4\t4473\t1953.1\t0\t4473\n",
      "5\t2532\t488.3\t0\t2532\n",
      "6\t2580\t122.1\t0\t2580\n",
      "7\t2787\t30.5\t0\t2787\n",
      "8\t2246\t7.6\t0\t2246\n",
      "9\t2582\t1.9\t0\t2548 34\n",
      "10\t2904\t0.5\t1\t2764 140\n",
      "11\t2228\t0.1\t1\t2069 159\n",
      "12\t2215\t0.0\t1\t2048 167\n",
      "13\t2112\t0.0\t1\t1962 150\n",
      "14\t2629\t0.0\t1\t2454 175\n",
      "15\t2464\t0.0\t1\t2269 195\n",
      "16\t2154\t0.0\t1\t2015 139\n",
      "17\t1975\t0.0\t1\t1843 132\n",
      "18\t1740\t0.0\t1\t1625 115\n",
      "19\t1759\t0.0\t1\t1654 105\n",
      "20\t1948\t0.0\t1\t1839 109\n",
      "21\t1792\t0.0\t1\t1678 114\n",
      "22\t1614\t0.0\t1\t1514 100\n",
      "23\t1602\t0.0\t1\t1515 87\n",
      "24\t1690\t0.0\t1\t1602 88\n",
      "25\t1703\t0.0\t1\t1613 90\n",
      "26\t1931\t0.0\t1\t1834 97\n",
      "27\t1551\t0.0\t1\t1455 96\n",
      "28\t1574\t0.0\t1\t1492 82\n",
      "29\t1427\t0.0\t1\t1329 98\n",
      "30\t1456\t0.0\t1\t1390 66\n",
      "31\t2435\t0.0\t1\t2340 95\n",
      "32\t1259\t0.0\t1\t1195 64\n",
      "33\t1171\t0.0\t1\t1116 55\n",
      "34\t1128\t0.0\t1\t1084 44\n",
      "35\t1051\t0.0\t1\t1001 50\n",
      "36\t1039\t0.0\t1\t991 48\n",
      "37\t1159\t0.0\t1\t1102 57\n",
      "38\t920\t0.0\t1\t880 40\n",
      "39\t909\t0.0\t1\t864 45\n",
      "40\t999\t0.0\t1\t938 61\n",
      "41\t674\t0.0\t1\t631 43\n",
      "42\t732\t0.0\t1\t701 31\n",
      "43\t765\t0.0\t1\t727 38\n",
      "44\t767\t0.0\t1\t734 33\n",
      "45\t692\t0.0\t1\t668 24\n",
      "46\t783\t0.0\t1\t754 29\n",
      "47\t578\t0.0\t1\t553 25\n",
      "48\t576\t0.0\t1\t545 31\n",
      "49\t766\t0.0\t1\t735 31\n",
      "50\t599\t0.0\t1\t573 26\n",
      "51\t582\t0.0\t1\t556 26\n",
      "52\t511\t0.0\t1\t483 28\n",
      "53\t426\t0.0\t1\t399 27\n",
      "54\t383\t0.0\t1\t370 13\n",
      "55\t334\t0.0\t1\t319 15\n",
      "56\t336\t0.0\t1\t315 21\n",
      "57\t271\t0.0\t1\t258 13\n",
      "58\t249\t0.0\t1\t237 12\n",
      "59\t239\t0.0\t1\t229 10\n",
      "60\t275\t0.0\t1\t261 14\n",
      "61\t161\t0.0\t1\t152 9\n",
      "62\t196\t0.0\t1\t188 8\n",
      "63\t169\t0.0\t1\t157 12\n",
      "64\t160\t0.0\t1\t155 5\n",
      "65\t148\t0.0\t1\t137 11\n",
      "66\t127\t0.0\t1\t119 8\n",
      "67\t144\t0.0\t1\t137 7\n",
      "68\t100\t0.0\t1\t96 4\n",
      "69\t112\t0.0\t1\t107 5\n",
      "70\t85\t0.0\t1\t79 6\n",
      "71\t88\t0.0\t1\t81 7\n",
      "72\t91\t0.0\t1\t88 3\n",
      "73\t116\t0.0\t1\t112 4\n",
      "74\t82\t0.0\t1\t77 5\n",
      "75\t46\t0.0\t1\t43 3\n",
      "76\t30\t0.0\t1\t29 1\n",
      "77\t30\t0.0\t1\t28 2\n",
      "78\t27\t0.0\t1\t23 4\n",
      "79\t27\t0.0\t1\t25 2\n",
      "80\t27\t0.0\t1\t23 4\n",
      "81\t18\t0.0\t1\t11 7\n",
      "82\t27\t0.0\t1\t25 2\n",
      "83\t15\t0.0\t1\t15\n",
      "84\t6\t0.0\t1\t6\n",
      "85\t8\t0.0\t1\t5 3\n",
      "86\t15\t0.0\t1\t11 4\n",
      "87\t5\t0.0\t1\t4 1\n",
      "88\t7\t0.0\t1\t6 1\n",
      "89\t6\t0.0\t1\t5 1\n",
      "90\t6\t0.0\t1\t5 1\n",
      "91\t10\t0.0\t1\t10\n",
      "92\t15\t0.0\t1\t12 3\n",
      "93\t35\t0.0\t1\t33 2\n",
      "94\t201\t0.0\t1\t184 17\n",
      "95\t305\t0.0\t1\t286 19\n",
      "96\t230\t0.0\t1\t212 18\n",
      "97\t116\t0.0\t1\t105 11\n",
      "98\t56\t0.0\t1\t47 9\n",
      "99\t70\t0.0\t1\t69 1\n",
      "100\t40\t0.0\t1\t0 40\n",
      "101\t150\t0.0\t1\t0 150\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 93160 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.0%\n",
      "  C: 20.4%\n",
      "  G: 30.5%\n",
      "  T: 30.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10284\t7812.5\t0\t10284\n",
      "4\t4025\t1953.1\t0\t4025\n",
      "5\t2668\t488.3\t0\t2668\n",
      "6\t2666\t122.1\t0\t2666\n",
      "7\t2825\t30.5\t0\t2825\n",
      "8\t2328\t7.6\t0\t2328\n",
      "9\t2644\t1.9\t0\t2607 37\n",
      "10\t2896\t0.5\t1\t2807 89\n",
      "11\t2235\t0.1\t1\t2131 104\n",
      "12\t2226\t0.0\t1\t2135 91\n",
      "13\t2129\t0.0\t1\t2049 80\n",
      "14\t2626\t0.0\t1\t2554 72\n",
      "15\t2450\t0.0\t1\t2366 84\n",
      "16\t2151\t0.0\t1\t2079 72\n",
      "17\t1962\t0.0\t1\t1890 72\n",
      "18\t1735\t0.0\t1\t1675 60\n",
      "19\t1761\t0.0\t1\t1715 46\n",
      "20\t1948\t0.0\t1\t1879 69\n",
      "21\t1775\t0.0\t1\t1712 63\n",
      "22\t1583\t0.0\t1\t1532 51\n",
      "23\t1581\t0.0\t1\t1536 45\n",
      "24\t1674\t0.0\t1\t1622 52\n",
      "25\t1693\t0.0\t1\t1649 44\n",
      "26\t1909\t0.0\t1\t1848 61\n",
      "27\t1539\t0.0\t1\t1478 61\n",
      "28\t1563\t0.0\t1\t1502 61\n",
      "29\t1414\t0.0\t1\t1361 53\n",
      "30\t1449\t0.0\t1\t1395 54\n",
      "31\t2411\t0.0\t1\t2345 66\n",
      "32\t1249\t0.0\t1\t1193 56\n",
      "33\t1161\t0.0\t1\t1105 56\n",
      "34\t1125\t0.0\t1\t1085 40\n",
      "35\t1045\t0.0\t1\t1008 37\n",
      "36\t1032\t0.0\t1\t994 38\n",
      "37\t1145\t0.0\t1\t1101 44\n",
      "38\t910\t0.0\t1\t879 31\n",
      "39\t899\t0.0\t1\t862 37\n",
      "40\t979\t0.0\t1\t936 43\n",
      "41\t653\t0.0\t1\t630 23\n",
      "42\t727\t0.0\t1\t702 25\n",
      "43\t752\t0.0\t1\t725 27\n",
      "44\t759\t0.0\t1\t732 27\n",
      "45\t683\t0.0\t1\t658 25\n",
      "46\t775\t0.0\t1\t758 17\n",
      "47\t572\t0.0\t1\t553 19\n",
      "48\t556\t0.0\t1\t530 26\n",
      "49\t749\t0.0\t1\t724 25\n",
      "50\t592\t0.0\t1\t564 28\n",
      "51\t565\t0.0\t1\t540 25\n",
      "52\t497\t0.0\t1\t480 17\n",
      "53\t416\t0.0\t1\t400 16\n",
      "54\t373\t0.0\t1\t363 10\n",
      "55\t324\t0.0\t1\t308 16\n",
      "56\t329\t0.0\t1\t314 15\n",
      "57\t261\t0.0\t1\t249 12\n",
      "58\t242\t0.0\t1\t231 11\n",
      "59\t232\t0.0\t1\t224 8\n",
      "60\t273\t0.0\t1\t262 11\n",
      "61\t153\t0.0\t1\t150 3\n",
      "62\t187\t0.0\t1\t178 9\n",
      "63\t159\t0.0\t1\t154 5\n",
      "64\t158\t0.0\t1\t154 4\n",
      "65\t141\t0.0\t1\t138 3\n",
      "66\t121\t0.0\t1\t109 12\n",
      "67\t142\t0.0\t1\t137 5\n",
      "68\t93\t0.0\t1\t89 4\n",
      "69\t109\t0.0\t1\t102 7\n",
      "70\t84\t0.0\t1\t80 4\n",
      "71\t78\t0.0\t1\t76 2\n",
      "72\t93\t0.0\t1\t88 5\n",
      "73\t113\t0.0\t1\t106 7\n",
      "74\t76\t0.0\t1\t71 5\n",
      "75\t41\t0.0\t1\t37 4\n",
      "76\t27\t0.0\t1\t25 2\n",
      "77\t26\t0.0\t1\t25 1\n",
      "78\t21\t0.0\t1\t19 2\n",
      "79\t26\t0.0\t1\t26\n",
      "80\t20\t0.0\t1\t19 1\n",
      "81\t11\t0.0\t1\t10 1\n",
      "82\t25\t0.0\t1\t25\n",
      "83\t14\t0.0\t1\t12 2\n",
      "84\t7\t0.0\t1\t7\n",
      "85\t4\t0.0\t1\t4\n",
      "86\t9\t0.0\t1\t8 1\n",
      "87\t4\t0.0\t1\t4\n",
      "88\t4\t0.0\t1\t4\n",
      "89\t6\t0.0\t1\t4 2\n",
      "90\t5\t0.0\t1\t3 2\n",
      "91\t9\t0.0\t1\t9\n",
      "92\t11\t0.0\t1\t10 1\n",
      "93\t35\t0.0\t1\t32 3\n",
      "94\t191\t0.0\t1\t186 5\n",
      "95\t296\t0.0\t1\t284 12\n",
      "96\t210\t0.0\t1\t204 6\n",
      "97\t108\t0.0\t1\t101 7\n",
      "98\t47\t0.0\t1\t45 2\n",
      "99\t67\t0.0\t1\t65 2\n",
      "100\t40\t0.0\t1\t38 2\n",
      "101\t164\t0.0\t1\t146 18\n",
      "This is cutadapt 5.0 with Python 3.11.11\n",
      "Command line parameters: -a AGATCGGAAGAG -A AGATCGGAAGAG --minimum-length 25 -o trimmed/ERR950187_trimmed_R1.fastq.gz -p trimmed/ERR950187_trimmed_R2.fastq.gz ERR950187_1.fastq.gz.subset.fastq.gz ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Processing paired-end reads on 1 core ...\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            500,000\n",
      "  Read 1 with adapter:                  82,710 (16.5%)\n",
      "  Read 2 with adapter:                  79,811 (16.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 775 (0.2%)\n",
      "Pairs written (passing filters):       499,225 (99.8%)\n",
      "\n",
      "Total basepairs processed:   101,000,000 bp\n",
      "  Read 1:    50,500,000 bp\n",
      "  Read 2:    50,500,000 bp\n",
      "Total written (filtered):     97,758,175 bp (96.8%)\n",
      "  Read 1:    48,868,636 bp\n",
      "  Read 2:    48,889,539 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 82710 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.5%\n",
      "  C: 36.3%\n",
      "  G: 29.2%\n",
      "  T: 11.7%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13619\t7812.5\t0\t13619\n",
      "4\t4068\t1953.1\t0\t4068\n",
      "5\t2032\t488.3\t0\t2032\n",
      "6\t2048\t122.1\t0\t2048\n",
      "7\t2299\t30.5\t0\t2299\n",
      "8\t1839\t7.6\t0\t1839\n",
      "9\t3041\t1.9\t0\t2964 77\n",
      "10\t2335\t0.5\t1\t2219 116\n",
      "11\t1746\t0.1\t1\t1641 105\n",
      "12\t1795\t0.0\t1\t1673 122\n",
      "13\t1920\t0.0\t1\t1789 131\n",
      "14\t2419\t0.0\t1\t2293 126\n",
      "15\t2515\t0.0\t1\t2392 123\n",
      "16\t3712\t0.0\t1\t3507 205\n",
      "17\t1592\t0.0\t1\t1517 75\n",
      "18\t1393\t0.0\t1\t1313 80\n",
      "19\t1472\t0.0\t1\t1393 79\n",
      "20\t1456\t0.0\t1\t1398 58\n",
      "21\t1430\t0.0\t1\t1363 67\n",
      "22\t1352\t0.0\t1\t1306 46\n",
      "23\t1635\t0.0\t1\t1570 65\n",
      "24\t1274\t0.0\t1\t1214 60\n",
      "25\t1322\t0.0\t1\t1253 69\n",
      "26\t1427\t0.0\t1\t1367 60\n",
      "27\t1978\t0.0\t1\t1890 88\n",
      "28\t1232\t0.0\t1\t1182 50\n",
      "29\t1137\t0.0\t1\t1101 36\n",
      "30\t919\t0.0\t1\t881 38\n",
      "31\t1081\t0.0\t1\t1045 36\n",
      "32\t822\t0.0\t1\t795 27\n",
      "33\t857\t0.0\t1\t822 35\n",
      "34\t779\t0.0\t1\t747 32\n",
      "35\t669\t0.0\t1\t643 26\n",
      "36\t868\t0.0\t1\t845 23\n",
      "37\t932\t0.0\t1\t908 24\n",
      "38\t675\t0.0\t1\t650 25\n",
      "39\t593\t0.0\t1\t567 26\n",
      "40\t690\t0.0\t1\t669 21\n",
      "41\t543\t0.0\t1\t519 24\n",
      "42\t560\t0.0\t1\t541 19\n",
      "43\t589\t0.0\t1\t573 16\n",
      "44\t580\t0.0\t1\t559 21\n",
      "45\t555\t0.0\t1\t536 19\n",
      "46\t514\t0.0\t1\t502 12\n",
      "47\t428\t0.0\t1\t411 17\n",
      "48\t406\t0.0\t1\t389 17\n",
      "49\t508\t0.0\t1\t486 22\n",
      "50\t387\t0.0\t1\t368 19\n",
      "51\t339\t0.0\t1\t325 14\n",
      "52\t331\t0.0\t1\t321 10\n",
      "53\t272\t0.0\t1\t260 12\n",
      "54\t289\t0.0\t1\t277 12\n",
      "55\t286\t0.0\t1\t280 6\n",
      "56\t220\t0.0\t1\t214 6\n",
      "57\t184\t0.0\t1\t176 8\n",
      "58\t244\t0.0\t1\t233 11\n",
      "59\t167\t0.0\t1\t163 4\n",
      "60\t201\t0.0\t1\t199 2\n",
      "61\t121\t0.0\t1\t116 5\n",
      "62\t138\t0.0\t1\t134 4\n",
      "63\t121\t0.0\t1\t114 7\n",
      "64\t141\t0.0\t1\t134 7\n",
      "65\t108\t0.0\t1\t100 8\n",
      "66\t112\t0.0\t1\t108 4\n",
      "67\t93\t0.0\t1\t92 1\n",
      "68\t74\t0.0\t1\t73 1\n",
      "69\t76\t0.0\t1\t70 6\n",
      "70\t50\t0.0\t1\t49 1\n",
      "71\t71\t0.0\t1\t65 6\n",
      "72\t63\t0.0\t1\t60 3\n",
      "73\t90\t0.0\t1\t85 5\n",
      "74\t51\t0.0\t1\t48 3\n",
      "75\t38\t0.0\t1\t36 2\n",
      "76\t21\t0.0\t1\t21\n",
      "77\t26\t0.0\t1\t26\n",
      "78\t19\t0.0\t1\t18 1\n",
      "79\t15\t0.0\t1\t11 4\n",
      "80\t12\t0.0\t1\t10 2\n",
      "81\t12\t0.0\t1\t11 1\n",
      "82\t12\t0.0\t1\t11 1\n",
      "83\t8\t0.0\t1\t7 1\n",
      "84\t6\t0.0\t1\t3 3\n",
      "85\t3\t0.0\t1\t2 1\n",
      "86\t7\t0.0\t1\t5 2\n",
      "87\t4\t0.0\t1\t3 1\n",
      "88\t5\t0.0\t1\t3 2\n",
      "90\t1\t0.0\t1\t1\n",
      "91\t9\t0.0\t1\t9\n",
      "92\t6\t0.0\t1\t6\n",
      "93\t14\t0.0\t1\t13 1\n",
      "94\t70\t0.0\t1\t67 3\n",
      "95\t114\t0.0\t1\t111 3\n",
      "96\t67\t0.0\t1\t61 6\n",
      "97\t47\t0.0\t1\t41 6\n",
      "98\t28\t0.0\t1\t23 5\n",
      "99\t24\t0.0\t1\t19 5\n",
      "100\t44\t0.0\t1\t32 12\n",
      "101\t213\t0.0\t1\t168 45\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAG; Type: regular 3'; Length: 12; Trimmed: 79811 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 21.3%\n",
      "  C: 18.3%\n",
      "  G: 32.7%\n",
      "  T: 27.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11367\t7812.5\t0\t11367\n",
      "4\t3486\t1953.1\t0\t3486\n",
      "5\t2353\t488.3\t0\t2353\n",
      "6\t2035\t122.1\t0\t2035\n",
      "7\t2328\t30.5\t0\t2328\n",
      "8\t1833\t7.6\t0\t1833\n",
      "9\t3115\t1.9\t0\t3077 38\n",
      "10\t2324\t0.5\t1\t2245 79\n",
      "11\t1748\t0.1\t1\t1651 97\n",
      "12\t1782\t0.0\t1\t1698 84\n",
      "13\t1912\t0.0\t1\t1839 73\n",
      "14\t2430\t0.0\t1\t2325 105\n",
      "15\t2499\t0.0\t1\t2413 86\n",
      "16\t3697\t0.0\t1\t3552 145\n",
      "17\t1572\t0.0\t1\t1491 81\n",
      "18\t1381\t0.0\t1\t1321 60\n",
      "19\t1463\t0.0\t1\t1409 54\n",
      "20\t1452\t0.0\t1\t1397 55\n",
      "21\t1431\t0.0\t1\t1360 71\n",
      "22\t1344\t0.0\t1\t1281 63\n",
      "23\t1627\t0.0\t1\t1525 102\n",
      "24\t1264\t0.0\t1\t1219 45\n",
      "25\t1302\t0.0\t1\t1261 41\n",
      "26\t1414\t0.0\t1\t1362 52\n",
      "27\t1956\t0.0\t1\t1864 92\n",
      "28\t1220\t0.0\t1\t1179 41\n",
      "29\t1133\t0.0\t1\t1096 37\n",
      "30\t917\t0.0\t1\t879 38\n",
      "31\t1075\t0.0\t1\t1032 43\n",
      "32\t813\t0.0\t1\t791 22\n",
      "33\t845\t0.0\t1\t808 37\n",
      "34\t766\t0.0\t1\t732 34\n",
      "35\t660\t0.0\t1\t636 24\n",
      "36\t858\t0.0\t1\t830 28\n",
      "37\t919\t0.0\t1\t889 30\n",
      "38\t664\t0.0\t1\t625 39\n",
      "39\t586\t0.0\t1\t568 18\n",
      "40\t690\t0.0\t1\t661 29\n",
      "41\t532\t0.0\t1\t514 18\n",
      "42\t553\t0.0\t1\t536 17\n",
      "43\t585\t0.0\t1\t568 17\n",
      "44\t572\t0.0\t1\t544 28\n",
      "45\t547\t0.0\t1\t522 25\n",
      "46\t510\t0.0\t1\t486 24\n",
      "47\t422\t0.0\t1\t412 10\n",
      "48\t403\t0.0\t1\t379 24\n",
      "49\t497\t0.0\t1\t479 18\n",
      "50\t386\t0.0\t1\t360 26\n",
      "51\t337\t0.0\t1\t327 10\n",
      "52\t326\t0.0\t1\t320 6\n",
      "53\t268\t0.0\t1\t251 17\n",
      "54\t284\t0.0\t1\t271 13\n",
      "55\t286\t0.0\t1\t270 16\n",
      "56\t220\t0.0\t1\t214 6\n",
      "57\t182\t0.0\t1\t172 10\n",
      "58\t237\t0.0\t1\t229 8\n",
      "59\t167\t0.0\t1\t160 7\n",
      "60\t203\t0.0\t1\t198 5\n",
      "61\t121\t0.0\t1\t119 2\n",
      "62\t136\t0.0\t1\t128 8\n",
      "63\t116\t0.0\t1\t112 4\n",
      "64\t138\t0.0\t1\t134 4\n",
      "65\t105\t0.0\t1\t102 3\n",
      "66\t110\t0.0\t1\t106 4\n",
      "67\t92\t0.0\t1\t86 6\n",
      "68\t71\t0.0\t1\t67 4\n",
      "69\t72\t0.0\t1\t67 5\n",
      "70\t53\t0.0\t1\t49 4\n",
      "71\t69\t0.0\t1\t65 4\n",
      "72\t60\t0.0\t1\t52 8\n",
      "73\t90\t0.0\t1\t83 7\n",
      "74\t51\t0.0\t1\t49 2\n",
      "75\t36\t0.0\t1\t34 2\n",
      "76\t21\t0.0\t1\t18 3\n",
      "77\t26\t0.0\t1\t24 2\n",
      "78\t18\t0.0\t1\t17 1\n",
      "79\t9\t0.0\t1\t8 1\n",
      "80\t8\t0.0\t1\t8\n",
      "81\t11\t0.0\t1\t11\n",
      "82\t11\t0.0\t1\t10 1\n",
      "83\t6\t0.0\t1\t6\n",
      "84\t4\t0.0\t1\t3 1\n",
      "85\t1\t0.0\t1\t1\n",
      "86\t5\t0.0\t1\t5\n",
      "87\t3\t0.0\t1\t3\n",
      "88\t4\t0.0\t1\t3 1\n",
      "89\t1\t0.0\t1\t0 1\n",
      "90\t1\t0.0\t1\t1\n",
      "91\t8\t0.0\t1\t7 1\n",
      "92\t6\t0.0\t1\t6\n",
      "93\t13\t0.0\t1\t13\n",
      "94\t67\t0.0\t1\t65 2\n",
      "95\t113\t0.0\t1\t107 6\n",
      "96\t62\t0.0\t1\t58 4\n",
      "97\t44\t0.0\t1\t41 3\n",
      "98\t23\t0.0\t1\t20 3\n",
      "99\t21\t0.0\t1\t20 1\n",
      "100\t36\t0.0\t1\t34 2\n",
      "101\t191\t0.0\t1\t165 26\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq\n",
    "mkdir -p trimmed  # Create output directory if it doesn't exist\n",
    "\n",
    "for file in *_1.fastq.gz.subset.fastq.gz; do\n",
    "  # Extract the base filename (remove _1.fastq.gz.subset.fastq.gz)\n",
    "  base=\"${file%%_1.fastq.gz.subset.fastq.gz}\"\n",
    "  \n",
    "  # Define corresponding R2 file\n",
    "  file_R2=\"${base}_2.fastq.gz.subset.fastq.gz\"\n",
    "\n",
    "  # Run Cutadapt for paired-end reads\n",
    "  cutadapt \\\n",
    "    -a AGATCGGAAGAG -A AGATCGGAAGAG \\\n",
    "    --minimum-length 25 \\\n",
    "    -o trimmed/\"${base}_trimmed_R1.fastq.gz\" \\\n",
    "    -p trimmed/\"${base}_trimmed_R2.fastq.gz\" \\\n",
    "    \"$file\" \"$file_R2\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "abce0821-e2df-474f-8ff8-2afac2bb1fae",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n",
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950158_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "application/gzip\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Approx 5% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950158_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950158_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950158_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950158_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950158_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950158_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950159_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950159_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950159_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950159_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950159_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950159_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950160_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950160_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950160_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950160_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950160_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950160_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950161_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950161_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950161_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950161_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950161_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950161_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950162_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950162_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950162_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950162_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950162_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950162_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950163_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950163_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950163_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950163_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950163_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950163_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950177_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950177_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950177_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950177_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950177_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950177_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950178_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950178_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950178_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950178_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950178_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950178_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950179_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950179_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950179_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950179_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950179_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950179_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950180_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950180_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950180_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950180_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950180_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950180_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950181_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950181_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950181_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950181_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950181_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950181_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950182_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950182_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950182_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950182_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950182_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950182_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950186_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950186_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950186_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950186_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950186_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950186_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 5% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 10% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 15% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 20% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 25% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 30% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 35% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 40% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 45% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 50% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 55% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 60% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 65% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 70% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 75% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 80% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 85% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 90% complete for ERR950187_trimmed_R1.fastq.gz\n",
      "Approx 95% complete for ERR950187_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950187_trimmed_R1.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 5% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 10% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 15% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 20% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 25% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 30% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 35% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 40% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 45% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 50% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 55% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 60% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 65% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 70% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 75% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 80% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 85% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 90% complete for ERR950187_trimmed_R2.fastq.gz\n",
      "Approx 95% complete for ERR950187_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR950187_trimmed_R2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=318029;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\  \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq/trimmed/fastqc\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m\u001b[0m\u001b[9m0m[9m\u001b[0m \u001b[35m  0%\u001b[0m \u001b[32m0/56\u001b[0m  \u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  5%\u001b[0m \u001b[32m3/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m  7%\u001b[0m \u001b[32m4/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 12%\u001b[0m \u001b[32m7/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m 12%\u001b[0m \u001b[32m7/56\u001b[0m \u001b[2mERR950160_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m 14%\u001b[0m \u001b[32m8/56\u001b[0m \u001b[2mERR950161_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 20%\u001b[0m \u001b[32m11/56\u001b[0m \u001b[2mERR950161_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m 21%\u001b[0m \u001b[32m12/56\u001b[0m \u001b[2mERR950162_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m 23%\u001b[0m \u001b[32m13/56\u001b[0m \u001b[2mERR950163_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 27%\u001b[0m \u001b[32m15/56\u001b[0m \u001b[2mERR950181_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 36%\u001b[0m \u001b[32m20/56\u001b[0m \u001b[2mERR950180_trimmed_R1_fastqc.html\u001b[0m\u001b[0m \u001b[35m 43%\u001b[0m \u001b[32m24/56\u001b[0m \u001b[2mERR950160_trimmed_R1_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 48%\u001b[0m \u001b[32m27/56\u001b[0m \u001b[2mERR950158_trimmed_R1_fastqc.html\u001b[0m\u001b[0m \u001b[35m 50%\u001b[0m \u001b[32m28/56\u001b[0m \u001b[2mERR950159_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m 55%\u001b[0m \u001b[32m31/56\u001b[0m \u001b[2mERR950162_trimmed_R2_fastqc.zip\u001b[0m\u001b[0m \u001b[35m 59%\u001b[0m \u001b[32m33/56\u001b[0m \u001b[2mERR950163_trimmed_R1_fastqc.html\u001b[0m\u001b[0m \u001b[35m 64%\u001b[0m \u001b[32m36/56\u001b[0m \u001b[2mERR950161_trimmed_R1_fastqc.html\u001b[0m0m\u001b[0m \u001b[35m 70%\u001b[0m \u001b[32m39/56\u001b[0m \u001b[2mERR950181_trimmed_R2_fastqc.html\u001b[0m\u001b[90m\u001b[0m \u001b[35m 71%\u001b[0m \u001b[32m40/56\u001b[0m \u001b[2mERR950161_trimmed_R1_fastqc.zip\u001b[0m\u001b[90m\u001b[0m \u001b[35m 77%\u001b[0m \u001b[32m43/56\u001b[0m \u001b[2mERR950160_trimmed_R1_fastqc.html\u001b[0m[0m\u001b[90m\u001b[0m \u001b[35m 80%\u001b[0m \u001b[32m45/56\u001b[0m \u001b[2mERR950162_trimmed_R1_fastqc.html\u001b[0m[0m\u001b[90m\u001b[0m \u001b[35m 86%\u001b[0m \u001b[32m48/56\u001b[0m \u001b[2mERR950158_trimmed_R2_fastqc.html\u001b[0m\u001b[0m\u001b[90m\u001b[0m \u001b[35m 88%\u001b[0m \u001b[32m49/56\u001b[0m \u001b[2mERR950187_trimmed_R2_fastqc.html\u001b[0m\u001b[0m\u001b[90m\u001b[0m \u001b[35m 89%\u001b[0m \u001b[32m50/56\u001b[0m \u001b[2mERR950163_trimmed_R2_fastqc.html\u001b[0m1m\u001b[0m\u001b[90m\u001b[0m \u001b[35m 93%\u001b[0m \u001b[32m52/56\u001b[0m \u001b[2mERR950158_trimmed_R2_fastqc.zip\u001b[0m1m\u001b[0m\u001b[90m\u001b[0m \u001b[35m 93%\u001b[0m \u001b[32m52/56\u001b[0m \u001b[2mERR950158_trimmed_R2_fastqc.zip\u001b[0m\u001b[90m\u001b[0m \u001b[35m 95%\u001b[0m \u001b[32m53/56\u001b[0m \u001b[2mERR950162_trimmed_R1_fastqc.zip\u001b[0m0m\u001b[0m \u001b[35m 96%\u001b[0m \u001b[32m54/56\u001b[0m \u001b[2mERR950186_trimmed_R2_fastqc.html\u001b[0m\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m56/56\u001b[0m  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 28 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq/trimmed\n",
    "mkdir -p fastqc\n",
    "fastqc -o fastqc *.fastq.gz\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/subset_rawdata_RNA-seq/trimmed/fastqc\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "69d50cd5-b1d1-4280-a994-0132dfc5ddc6",
   "metadata": {},
   "source": [
    "# Test to create count per gene table from bowtie2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0c29d68c-5a05-43f2-a73c-a7f45f3bf5a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "htseq-count completed successfully. Output saved in /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_bowtie2_TEST/ERR950158_htseq_counts.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set variables\n",
    "gtf_file=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/human_genome/Homo_sapiens.GRCh38.113.gtf\" #!!!!!!\n",
    "output_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_bowtie2_TEST\"\n",
    "bam_file=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/alignment_Bowtie2_subset_old/ERR950158_trimmed.bam\" #!!!!!!\n",
    "output_file=\"$output_dir/ERR950158_htseq_counts.txt\"\n",
    "\n",
    "# Create output directory if it does not exist\n",
    "mkdir -p \"$output_dir\"\n",
    "\n",
    "# Sort BAM file by name for htseq-count\n",
    "sorted_bam_file=\"$output_dir/ERR950158_sorted.bam\"\n",
    "samtools sort -n -o \"$sorted_bam_file\" \"$bam_file\"\n",
    "\n",
    "# Run htseq-count\n",
    "htseq-count -f bam -r name -s no -t exon -i gene_id \"$sorted_bam_file\" \"$gtf_file\" > \"$output_file\" 2> \"$output_dir/htseq-count.log\"\n",
    "\n",
    "# Check if htseq-count ran successfully\n",
    "if [ $? -eq 0 ]; then\n",
    "    echo \"htseq-count completed successfully. Output saved in $output_file\"\n",
    "else\n",
    "    echo \"htseq-count failed. Check log file at $output_dir/htseq-count.log\" >&2\n",
    "    exit 1\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "50e3519e-f6f2-4351-a6dc-2ab49264e007",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.0.5\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "htseq-count --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "856ca7d8-d042-42f6-bcdc-89a98716d8a1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
