{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "27cbbf6e-b9f6-465f-b2b4-1ec8d713b746",
   "metadata": {},
   "source": [
    "Selecting files from the alignment and convert to counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "936303b4-a1d7-49c2-b465-8e4995a5a91e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950158_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950158_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950158_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950158_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1074660                                              ||\n",
      "||    Successfully assigned alignments : 492123 (45.8%)                       ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950158_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950159_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950159_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950159_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950159_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1425238                                              ||\n",
      "||    Successfully assigned alignments : 494281 (34.7%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950159_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950160_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950160_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950160_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950160_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1059580                                              ||\n",
      "||    Successfully assigned alignments : 539337 (50.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950160_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950161_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950161_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950161_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950161_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1117023                                              ||\n",
      "||    Successfully assigned alignments : 464415 (41.6%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950161_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950162_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950162_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950162_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950162_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1293714                                              ||\n",
      "||    Successfully assigned alignments : 581420 (44.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950162_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950163_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950163_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950163_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950163_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1403658                                              ||\n",
      "||    Successfully assigned alignments : 472356 (33.7%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950163_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950177_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950177_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950177_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950177_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1248097                                              ||\n",
      "||    Successfully assigned alignments : 527736 (42.3%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950177_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950178_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950178_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950178_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950178_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1062809                                              ||\n",
      "||    Successfully assigned alignments : 459354 (43.2%)                       ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950178_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950179_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950179_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950179_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950179_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1188651                                              ||\n",
      "||    Successfully assigned alignments : 475514 (40.0%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950179_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950180_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950180_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950180_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950180_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1204165                                              ||\n",
      "||    Successfully assigned alignments : 508288 (42.2%)                       ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950180_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950181_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950181_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950181_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950181_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1316958                                              ||\n",
      "||    Successfully assigned alignments : 591522 (44.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950181_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950182_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950182_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950182_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950182_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1268304                                              ||\n",
      "||    Successfully assigned alignments : 542894 (42.8%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950182_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950186_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950186_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950186_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950186_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1362679                                              ||\n",
      "||    Successfully assigned alignments : 600046 (44.0%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950186_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950187_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950187_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950187_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950187_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 2452737                                              ||\n",
      "||    Successfully assigned alignments : 473877 (19.3%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950187_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "  # featurecounts from bioconda subread\n",
    "# Define paths\n",
    "ALIGNED_DIR=\"data/alignment_STAR_subset/\"  \n",
    "OUTPUT_DIR=\"data/gene_counts_DESeq\"  # Define output directory\n",
    "GTF_FILE=\"/human_genome/cleaned_Homo_sapiens.GRCh38.113.gtf\"  # GTF annotation file\n",
    "\n",
    "# Create output directory \n",
    "mkdir -p \"$OUTPUT_DIR\"\n",
    "\n",
    "# Loop through BAM files\n",
    "for file in ${ALIGNED_DIR}/*sortedByCoord.out.bam; do\n",
    "    # Extract sample name\n",
    "    SAMPLE=$(basename \"$file\" _Aligned.sortedByCoord.out.bam)\n",
    "\n",
    "    # Run featureCounts\n",
    "    featureCounts -T 8 \\\n",
    "        -a \"human_genome/Homo_sapiens.GRCh38.113.gtf\" \\\n",
    "        -o \"${OUTPUT_DIR}/${SAMPLE}.counts.txt\" \\\n",
    "        -g gene_id \\\n",
    "        -t exon \\\n",
    "        -s 0 \\\n",
    "        -p \\\n",
    "        \"$file\"\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c36ba3d1-0880-472e-9237-591d76791654",
   "metadata": {},
   "source": [
    "merge all files into a count matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2bc8673c-0649-4bb5-b2c8-2950a184f48d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged count matrix saved as 'merged_counts_matrix.csv'.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Define directory\n",
    "count_dir = \"data/gene_counts_DESeq/\"\n",
    "\n",
    "# List all count files\n",
    "count_files = [f for f in os.listdir(count_dir) if f.endswith(\"trimmed.counts.txt\")]\n",
    "\n",
    "# Read and merge count files\n",
    "dfs = []\n",
    "for file in count_files:\n",
    "    df = pd.read_csv(os.path.join(count_dir, file), sep=\"\\t\", skiprows=1, usecols=[0, 6])  # Geneid & counts\n",
    "    sample_name = file.replace(\"trimmed.counts.txt\", \"\")\n",
    "    df.rename(columns={df.columns[1]: sample_name}, inplace=True)\n",
    "    dfs.append(df)\n",
    "\n",
    "# Merge all dataframes\n",
    "merged_counts = dfs[0]\n",
    "for df in dfs[1:]:\n",
    "    merged_counts = merged_counts.merge(df, on=\"Geneid\", how=\"outer\")\n",
    "\n",
    "# Save merged file\n",
    "merged_counts.to_csv(\"merged_counts_matrix.csv\", index=False)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0c76151-82af-4f3f-9ece-9a2a1f139a5f",
   "metadata": {},
   "source": [
    "Run DESeq2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e33c4aff-d114-4423-a93d-ba4c05471745",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Error in data.frame(row.names = colnames(count_matrix), condition = c(\"control\",  : \n",
      "  row names supplied are of the wrong length\n",
      "\n"
     ]
    },
    {
     "ename": "RRuntimeError",
     "evalue": "Error in data.frame(row.names = colnames(count_matrix), condition = c(\"control\",  : \n  row names supplied are of the wrong length\n",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRRuntimeError\u001b[0m                             Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 29\u001b[0m\n\u001b[1;32m      1\u001b[0m r_deseq_script \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;124mlibrary(DESeq2)\u001b[39m\n\u001b[1;32m      3\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     25\u001b[0m \u001b[38;5;124mwrite.csv(res, \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSTAR.DESeq2_results.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m)\u001b[39m\n\u001b[1;32m     26\u001b[0m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m     28\u001b[0m \u001b[38;5;66;03m# Run DESeq2 inside Python\u001b[39;00m\n\u001b[0;32m---> 29\u001b[0m \u001b[43mrobjects\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mr\u001b[49m\u001b[43m(\u001b[49m\u001b[43mr_deseq_script\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/__init__.py:459\u001b[0m, in \u001b[0;36mR.__call__\u001b[0;34m(self, string)\u001b[0m\n\u001b[1;32m    457\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mself\u001b[39m, string):\n\u001b[1;32m    458\u001b[0m     p \u001b[38;5;241m=\u001b[39m rinterface\u001b[38;5;241m.\u001b[39mparse(string)\n\u001b[0;32m--> 459\u001b[0m     res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43meval\u001b[49m\u001b[43m(\u001b[49m\u001b[43mp\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    460\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m conversion\u001b[38;5;241m.\u001b[39mget_conversion()\u001b[38;5;241m.\u001b[39mrpy2py(res)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/functions.py:208\u001b[0m, in \u001b[0;36mSignatureTranslatedFunction.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    206\u001b[0m         v \u001b[38;5;241m=\u001b[39m kwargs\u001b[38;5;241m.\u001b[39mpop(k)\n\u001b[1;32m    207\u001b[0m         kwargs[r_k] \u001b[38;5;241m=\u001b[39m v\n\u001b[0;32m--> 208\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m (\u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mSignatureTranslatedFunction\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m    209\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__call__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/functions.py:131\u001b[0m, in \u001b[0;36mFunction.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    129\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    130\u001b[0m         new_kwargs[k] \u001b[38;5;241m=\u001b[39m cv\u001b[38;5;241m.\u001b[39mpy2rpy(v)\n\u001b[0;32m--> 131\u001b[0m res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mFunction\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__call__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mnew_args\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mnew_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    132\u001b[0m res \u001b[38;5;241m=\u001b[39m cv\u001b[38;5;241m.\u001b[39mrpy2py(res)\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/rinterface_lib/conversion.py:45\u001b[0m, in \u001b[0;36m_cdata_res_to_rinterface.<locals>._\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     44\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21m_\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m---> 45\u001b[0m     cdata \u001b[38;5;241m=\u001b[39m \u001b[43mfunction\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     46\u001b[0m     \u001b[38;5;66;03m# TODO: test cdata is of the expected CType\u001b[39;00m\n\u001b[1;32m     47\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m _cdata_to_rinterface(cdata)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/rinterface.py:817\u001b[0m, in \u001b[0;36mSexpClosure.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    810\u001b[0m     res \u001b[38;5;241m=\u001b[39m rmemory\u001b[38;5;241m.\u001b[39mprotect(\n\u001b[1;32m    811\u001b[0m         openrlib\u001b[38;5;241m.\u001b[39mrlib\u001b[38;5;241m.\u001b[39mR_tryEval(\n\u001b[1;32m    812\u001b[0m             call_r,\n\u001b[1;32m    813\u001b[0m             call_context\u001b[38;5;241m.\u001b[39m__sexp__\u001b[38;5;241m.\u001b[39m_cdata,\n\u001b[1;32m    814\u001b[0m             error_occured)\n\u001b[1;32m    815\u001b[0m     )\n\u001b[1;32m    816\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m error_occured[\u001b[38;5;241m0\u001b[39m]:\n\u001b[0;32m--> 817\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m embedded\u001b[38;5;241m.\u001b[39mRRuntimeError(_rinterface\u001b[38;5;241m.\u001b[39m_geterrmessage())\n\u001b[1;32m    818\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res\n",
      "\u001b[0;31mRRuntimeError\u001b[0m: Error in data.frame(row.names = colnames(count_matrix), condition = c(\"control\",  : \n  row names supplied are of the wrong length\n"
     ]
    }
   ],
   "source": [
    "r_deseq_script = \"\"\"\n",
    "library(DESeq2)\n",
    "\n",
    "# Load count matrix\n",
    "count_matrix <- read.csv(\"data/merged_counts_matrix.csv\", row.names = 1)\n",
    "\n",
    "sample_info <- data.frame(\n",
    "    row.names = colnames(count_matrix),\n",
    "    condition = c(\"control\", \"control\", \"treated\", \"treated\")\n",
    ")\n",
    "\n",
    "# Convert to factors\n",
    "sample_info$condition <- factor(sample_info$condition, levels = c(\"control\", \"treated\"))\n",
    "\n",
    "# Create DESeq2 data\n",
    "dds <- DESeqDataSetFromMatrix(countData = as.matrix(count_matrix), colData = sample_info, design = ~ condition)\n",
    "\n",
    "# Run DESeq2\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "# Extract results\n",
    "res <- results(dds, contrast = c(\"condition\", \"treated\", \"control\"))\n",
    "\n",
    "# Save results\n",
    "write.csv(res, \"STAR.DESeq2_results.csv\")\n",
    "\"\"\"\n",
    "\n",
    "# Run DESeq2 inside Python\n",
    "robjects.r(r_deseq_script)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93e9bb9a-0517-4232-bf67-c7ace7d40f8b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
