{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "27cbbf6e-b9f6-465f-b2b4-1ec8d713b746",
   "metadata": {},
   "source": [
    "Selecting files from the alignment and convert to counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "936303b4-a1d7-49c2-b465-8e4995a5a91e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950158_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950158_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950158_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950158_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1074660                                              ||\n",
      "||    Successfully assigned alignments : 492123 (45.8%)                       ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950158_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950159_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950159_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950159_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950159_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1425238                                              ||\n",
      "||    Successfully assigned alignments : 494281 (34.7%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950159_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950160_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950160_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950160_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950160_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1059580                                              ||\n",
      "||    Successfully assigned alignments : 539337 (50.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950160_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950161_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950161_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950161_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950161_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1117023                                              ||\n",
      "||    Successfully assigned alignments : 464415 (41.6%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950161_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950162_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950162_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950162_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950162_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1293714                                              ||\n",
      "||    Successfully assigned alignments : 581420 (44.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950162_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950163_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950163_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950163_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950163_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1403658                                              ||\n",
      "||    Successfully assigned alignments : 472356 (33.7%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950163_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950177_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950177_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950177_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950177_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1248097                                              ||\n",
      "||    Successfully assigned alignments : 527736 (42.3%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950177_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950178_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950178_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950178_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950178_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1062809                                              ||\n",
      "||    Successfully assigned alignments : 459354 (43.2%)                       ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950178_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950179_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950179_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950179_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950179_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1188651                                              ||\n",
      "||    Successfully assigned alignments : 475514 (40.0%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950179_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950180_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950180_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950180_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950180_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1204165                                              ||\n",
      "||    Successfully assigned alignments : 508288 (42.2%)                       ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950180_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950181_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950181_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950181_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950181_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1316958                                              ||\n",
      "||    Successfully assigned alignments : 591522 (44.9%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950181_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950182_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950182_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950182_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950182_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1268304                                              ||\n",
      "||    Successfully assigned alignments : 542894 (42.8%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950182_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950186_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950186_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950186_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950186_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 1362679                                              ||\n",
      "||    Successfully assigned alignments : 600046 (44.0%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950186_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.8\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                                                                            ||\n",
      "||                           ERR950187_trimmed_Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : ERR950187_trimmed.counts.txt                     ||\n",
      "||                 Summary : ERR950187_trimmed.counts.txt.summary             ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.113.gtf (GTF)                ||\n",
      "||      Dir for temp files : data/gene_counts_DESeq                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 8                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.113.gtf ...                       ||\n",
      "||    Features : 2164410                                                      ||\n",
      "||    Meta-features : 78932                                                   ||\n",
      "||    Chromosomes/contigs : 70                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ERR950187_trimmed_Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    The reads are assigned on the single-end mode.                          ||\n",
      "||    Total alignments : 2452737                                              ||\n",
      "||    Successfully assigned alignments : 473877 (19.3%)                       ||\n",
      "||    Running time : 0.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"data/gene_counts_DESeq/  ||\n",
      "|| ERR950187_trimmed.counts.txt.summary\"                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "  # featurecounts from bioconda subread\n",
    "# Define paths\n",
    "ALIGNED_DIR=\"data/alignment_STAR_subset/\"  \n",
    "OUTPUT_DIR=\"data/gene_counts_DESeq\"  # Define output directory\n",
    "GTF_FILE=\"/human_genome/cleaned_Homo_sapiens.GRCh38.113.gtf\"  # GTF annotation file\n",
    "\n",
    "# Create output directory \n",
    "mkdir -p \"$OUTPUT_DIR\"\n",
    "\n",
    "# Loop through BAM files\n",
    "for file in ${ALIGNED_DIR}/*sortedByCoord.out.bam; do\n",
    "    # Extract sample name\n",
    "    SAMPLE=$(basename \"$file\" _Aligned.sortedByCoord.out.bam)\n",
    "\n",
    "    # Run featureCounts\n",
    "    featureCounts -T 8 \\\n",
    "        -a \"human_genome/Homo_sapiens.GRCh38.113.gtf\" \\\n",
    "        -o \"${OUTPUT_DIR}/${SAMPLE}.counts.txt\" \\\n",
    "        -g gene_id \\\n",
    "        -t exon \\\n",
    "        -s 0 \\\n",
    "        -p \\\n",
    "        \"$file\"\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c36ba3d1-0880-472e-9237-591d76791654",
   "metadata": {},
   "source": [
    "merge all files into a count matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2bc8673c-0649-4bb5-b2c8-2950a184f48d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged count matrix saved as 'merged_counts_matrix.csv'.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Define directory\n",
    "count_dir = \"data/gene_counts_DESeq/\"\n",
    "\n",
    "# List all count files\n",
    "count_files = [f for f in os.listdir(count_dir) if f.endswith(\"trimmed.counts.txt\")]\n",
    "\n",
    "# Read and merge count files\n",
    "dfs = []\n",
    "for file in count_files:\n",
    "    df = pd.read_csv(os.path.join(count_dir, file), sep=\"\\t\", skiprows=1, usecols=[0, 6])  # Geneid & counts\n",
    "    sample_name = file.replace(\"trimmed.counts.txt\", \"\")\n",
    "    df.rename(columns={df.columns[1]: sample_name}, inplace=True)\n",
    "    dfs.append(df)\n",
    "\n",
    "# Merge all dataframes\n",
    "merged_counts = dfs[0]\n",
    "for df in dfs[1:]:\n",
    "    merged_counts = merged_counts.merge(df, on=\"Geneid\", how=\"outer\")\n",
    "\n",
    "# Save merged file\n",
    "merged_counts.to_csv(\"merged_counts_matrix.csv\", index=False)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0c76151-82af-4f3f-9ece-9a2a1f139a5f",
   "metadata": {},
   "source": [
    "Run DESeq2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e33c4aff-d114-4423-a93d-ba4c05471745",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: -- replacing outliers and refitting for 413 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<rpy2.rinterface_lib.sexp.NULLType object at 0x1510060874d0> [0]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "import rpy2.robjects as robjects\n",
    "\n",
    "r_deseq_script = \"\"\"\n",
    "library(DESeq2)                                              \n",
    "\n",
    "# Load count matrix\n",
    "count_matrix <- read.table(\"data/STAR_DE_analysis/STAR_merged_counts.txt\",  header = TRUE, sep = \"\\t\", row.names=1)\n",
    "#count_matrix <- count_matrix[, -1]\n",
    "metadata <- read.table(\"data/STAR_DE_analysis/STAR_metadata.txt\", header = TRUE)\n",
    "count_matrix <- as.matrix(count_matrix)\n",
    "mode(count_matrix) <- \"numeric\"\n",
    "\n",
    "\n",
    "rownames(metadata) <- metadata$Sample_ID\n",
    "\n",
    "#create the factors from diagnosis\n",
    "metadata$Diagnosis <- factor(metadata$Diagnosis, levels = c(\"Atr\", \"Hp\"))\n",
    "\n",
    "samples <- rownames(metadata)\n",
    "\n",
    "# Keep only samples present in both metadata and count matrix\n",
    "samples <- intersect(samples, colnames(count_matrix))\n",
    "\n",
    "count_matrix <- count_matrix[, samples]\n",
    "metadata <- metadata[samples, , drop = FALSE]\n",
    "\n",
    "# Create DESeq2 data\n",
    "dds <- DESeqDataSetFromMatrix(countData = as.matrix(count_matrix), colData = metadata, design = ~ Diagnosis)\n",
    "\n",
    "# Run DESeq2\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "# Extract results\n",
    "res <- results(dds, contrast = c(\"Diagnosis\", \"Hp\", \"Atr\"))\n",
    "\n",
    "# Save results\n",
    "write.csv(res, \"STAR.DESeq2_results.csv\")\n",
    "\"\"\"\n",
    "\n",
    "# Run DESeq2 inside Python\n",
    "robjects.r(r_deseq_script)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b161fa86-5ed9-4f22-a55f-f5ef332b245b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: Error in DESeqDataSet(se, design = design, ignoreRank) : \n",
      "  all samples have 0 counts for all genes. check the counting script.\n",
      "\n"
     ]
    },
    {
     "ename": "RRuntimeError",
     "evalue": "Error in DESeqDataSet(se, design = design, ignoreRank) : \n  all samples have 0 counts for all genes. check the counting script.\n",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRRuntimeError\u001b[0m                             Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[33], line 42\u001b[0m\n\u001b[1;32m      3\u001b[0m r_deseq_script \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;124mlibrary(DESeq2)                                              \u001b[39m\n\u001b[1;32m      5\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     38\u001b[0m \u001b[38;5;124mwrite.csv(res, \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSTAR.DESeq2.subread_results.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m)\u001b[39m\n\u001b[1;32m     39\u001b[0m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m     41\u001b[0m \u001b[38;5;66;03m# Run DESeq2 inside Python\u001b[39;00m\n\u001b[0;32m---> 42\u001b[0m \u001b[43mrobjects\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mr\u001b[49m\u001b[43m(\u001b[49m\u001b[43mr_deseq_script\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/__init__.py:459\u001b[0m, in \u001b[0;36mR.__call__\u001b[0;34m(self, string)\u001b[0m\n\u001b[1;32m    457\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21m__call__\u001b[39m(\u001b[38;5;28mself\u001b[39m, string):\n\u001b[1;32m    458\u001b[0m     p \u001b[38;5;241m=\u001b[39m rinterface\u001b[38;5;241m.\u001b[39mparse(string)\n\u001b[0;32m--> 459\u001b[0m     res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43meval\u001b[49m\u001b[43m(\u001b[49m\u001b[43mp\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    460\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m conversion\u001b[38;5;241m.\u001b[39mget_conversion()\u001b[38;5;241m.\u001b[39mrpy2py(res)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/functions.py:208\u001b[0m, in \u001b[0;36mSignatureTranslatedFunction.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    206\u001b[0m         v \u001b[38;5;241m=\u001b[39m kwargs\u001b[38;5;241m.\u001b[39mpop(k)\n\u001b[1;32m    207\u001b[0m         kwargs[r_k] \u001b[38;5;241m=\u001b[39m v\n\u001b[0;32m--> 208\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m (\u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mSignatureTranslatedFunction\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m    209\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__call__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/robjects/functions.py:131\u001b[0m, in \u001b[0;36mFunction.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    129\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    130\u001b[0m         new_kwargs[k] \u001b[38;5;241m=\u001b[39m cv\u001b[38;5;241m.\u001b[39mpy2rpy(v)\n\u001b[0;32m--> 131\u001b[0m res \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mFunction\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__call__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mnew_args\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mnew_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    132\u001b[0m res \u001b[38;5;241m=\u001b[39m cv\u001b[38;5;241m.\u001b[39mrpy2py(res)\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/rinterface_lib/conversion.py:45\u001b[0m, in \u001b[0;36m_cdata_res_to_rinterface.<locals>._\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     44\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21m_\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m---> 45\u001b[0m     cdata \u001b[38;5;241m=\u001b[39m \u001b[43mfunction\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     46\u001b[0m     \u001b[38;5;66;03m# TODO: test cdata is of the expected CType\u001b[39;00m\n\u001b[1;32m     47\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m _cdata_to_rinterface(cdata)\n",
      "File \u001b[0;32m/opt/conda/lib/python3.11/site-packages/rpy2/rinterface.py:817\u001b[0m, in \u001b[0;36mSexpClosure.__call__\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    810\u001b[0m     res \u001b[38;5;241m=\u001b[39m rmemory\u001b[38;5;241m.\u001b[39mprotect(\n\u001b[1;32m    811\u001b[0m         openrlib\u001b[38;5;241m.\u001b[39mrlib\u001b[38;5;241m.\u001b[39mR_tryEval(\n\u001b[1;32m    812\u001b[0m             call_r,\n\u001b[1;32m    813\u001b[0m             call_context\u001b[38;5;241m.\u001b[39m__sexp__\u001b[38;5;241m.\u001b[39m_cdata,\n\u001b[1;32m    814\u001b[0m             error_occured)\n\u001b[1;32m    815\u001b[0m     )\n\u001b[1;32m    816\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m error_occured[\u001b[38;5;241m0\u001b[39m]:\n\u001b[0;32m--> 817\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m embedded\u001b[38;5;241m.\u001b[39mRRuntimeError(_rinterface\u001b[38;5;241m.\u001b[39m_geterrmessage())\n\u001b[1;32m    818\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m res\n",
      "\u001b[0;31mRRuntimeError\u001b[0m: Error in DESeqDataSet(se, design = design, ignoreRank) : \n  all samples have 0 counts for all genes. check the counting script.\n"
     ]
    }
   ],
   "source": [
    "import rpy2.robjects as robjects\n",
    "\n",
    "r_deseq_script = \"\"\"\n",
    "library(DESeq2)                                              \n",
    "\n",
    "#Try the deseq analysis using the subread data\n",
    "# Load count matrix\n",
    "count_matrix <- read.delim(\"data/merged_counts_matrix.csv\",  header = TRUE, sep = \"\\t\", row.names=1, check.names=FALSE)\n",
    "#count_matrix <- count_matrix[, -1]\n",
    "metadata <- read.table(\"data/STAR_DE_analysis/STAR_metadata.txt\", header = TRUE)\n",
    "count_matrix <- as.matrix(count_matrix)\n",
    "mode(count_matrix) <- \"numeric\"\n",
    "\n",
    "\n",
    "rownames(metadata) <- metadata$Sample_ID\n",
    "\n",
    "#create the factors from diagnosis\n",
    "metadata$Diagnosis <- factor(metadata$Diagnosis, levels = c(\"Atr\", \"Hp\"))\n",
    "\n",
    "samples <- rownames(metadata)\n",
    "\n",
    "# Keep only samples present in both metadata and count matrix\n",
    "samples <- intersect(samples, colnames(count_matrix))\n",
    "\n",
    "count_matrix <- count_matrix[, samples]\n",
    "metadata <- metadata[samples, , drop = FALSE]\n",
    "\n",
    "# Create DESeq2 data\n",
    "dds <- DESeqDataSetFromMatrix(countData = as.matrix(count_matrix), colData = metadata, design = ~ Diagnosis)\n",
    "\n",
    "# Run DESeq2\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "# Extract results\n",
    "res <- results(dds, contrast = c(\"Diagnosis\", \"Hp\", \"Atr\"))\n",
    "\n",
    "# Save results\n",
    "write.csv(res, \"STAR.DESeq2.subread_results.csv\")\n",
    "\"\"\"\n",
    "\n",
    "# Run DESeq2 inside Python\n",
    "robjects.r(r_deseq_script)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c359804-4beb-4df8-9778-952f3203c03c",
   "metadata": {},
   "source": [
    "filter results using python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "93e9bb9a-0517-4232-bf67-c7ace7d40f8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            Unnamed: 0     baseMean  log2FoldChange     lfcSE      stat  \\\n",
      "53956  ENSG00000211897  2344.407879       -4.478580  0.531624 -8.424334   \n",
      "62320  ENSG00000163735   123.902293       -7.465867  0.888871 -8.399267   \n",
      "38348  ENSG00000211893  2147.097202       -4.209961  0.515895 -8.160507   \n",
      "11211  ENSG00000145113   766.820801       -5.530088  0.706498 -7.827464   \n",
      "565    ENSG00000269404    65.987284       -5.324327  0.686340 -7.757569   \n",
      "...                ...          ...             ...       ...       ...   \n",
      "44565  ENSG00000198894   203.482200        0.402649  0.137751  2.923010   \n",
      "34318  ENSG00000175449    18.735403        1.000771  0.342404  2.922780   \n",
      "35466  ENSG00000254777    26.416927       -1.349131  0.461829 -2.921280   \n",
      "75771  ENSG00000007933    68.438163       -1.241131  0.424991 -2.920371   \n",
      "37991  ENSG00000247516   144.270115        0.801186  0.274420  2.919557   \n",
      "\n",
      "             pvalue          padj  FDR_corrected_padj  Significant  \n",
      "53956  3.628069e-17  6.590410e-13        9.667473e-09         True  \n",
      "62320  4.492747e-17  6.590410e-13        9.667473e-09         True  \n",
      "38348  3.336225e-16  3.262606e-12        3.190611e-08         True  \n",
      "11211  4.978102e-15  3.651189e-11        2.677965e-07         True  \n",
      "565    8.657260e-15  5.079734e-11        2.980585e-07         True  \n",
      "...             ...           ...                 ...          ...  \n",
      "44565  3.466658e-03  4.937126e-02        7.029696e-01        False  \n",
      "34318  3.469218e-03  4.938375e-02        7.029696e-01        False  \n",
      "35466  3.485967e-03  4.959811e-02        7.056786e-01        False  \n",
      "75771  3.496148e-03  4.971884e-02        7.070535e-01        False  \n",
      "37991  3.505298e-03  4.982483e-02        7.082174e-01        False  \n",
      "\n",
      "[2064 rows x 9 columns]\n",
      "420\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import statsmodels.stats.multitest as smm\n",
    "DESeq2_STAR_results = pd.read_csv(\"STAR.DESeq2_results.csv\") \n",
    "\n",
    "DESeq2_STAR_results = DESeq2_STAR_results.dropna(subset=[\"padj\"]) #drop all NAN in padj column\n",
    "significant, corrected_pvals = smm.fdrcorrection(DESeq2_STAR_results[\"padj\"], alpha=0.05, method='indep', is_sorted=False)\n",
    "DESeq2_STAR_results[\"FDR_corrected_padj\"] = corrected_pvals\n",
    "DESeq2_STAR_results[\"Significant\"] = significant \n",
    "\n",
    "DESeq2_STAR_filtered_results = DESeq2_STAR_results.sort_values(\"FDR_corrected_padj\", ascending=True) \n",
    "print(DESeq2_STAR_filtered_results.loc[DESeq2_STAR_filtered_results[\"padj\"]<=0.05])\n",
    "print(len(DESeq2_STAR_filtered_results[DESeq2_STAR_filtered_results[\"Significant\"]==True]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bac0d296-95b5-47bc-8458-d51c8f91bdce",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
