{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "84c963c2-b16d-46c4-8fb7-59704c5e6338",
   "metadata": {},
   "source": [
    "# Preprocessing the RNA-seq subset_data\n",
    "\n",
    "Setting up relevant directories for the subset RNA-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c2f099e8-01f3-4a4f-979f-83f215d821c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p data \n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data \n",
    "mkdir -p subset_rawdata_RNA-seq \n",
    "mkdir -p subset_trimmed_RNA-seq\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d47af25a-699b-443e-a57b-42480915cb03",
   "metadata": {},
   "source": [
    "# Selecting the RNA-seq files and creating a symlink to the data for the subset of Hp and Atr "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "286412d6-c7a6-4ec4-adb8-1a265bc76954",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Extract IDs for 'Atr' and 'Hp'nfrom the RNA-seq-list.txt file\n",
    "Hp_ids=($(grep -P '(Atr|Hp)' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'))\n",
    "\n",
    "# Define directories\n",
    "source_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq\"\n",
    "destination_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_rawdata_RNA-seq\"\n",
    "\n",
    "# Loop through extracted IDs\n",
    "for id in \"${Hp_ids[@]}\"; do\n",
    "    # Store both file paths in an array\n",
    "    files=(\"$source_dir/${id}_1.fastq.gz.subset.fastq.gz\" \"$source_dir/${id}_2.fastq.gz.subset.fastq.gz\")\n",
    "\n",
    "    # Loop through each file\n",
    "    for file in \"${files[@]}\"; do\n",
    "        if [[ -f \"$file\" ]]; then \n",
    "            ln -s -t \"$destination_dir\" \"$file\"\n",
    "        else\n",
    "            echo \"File $file not found!\"\n",
    "        fi\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29756a13-bd08-48b5-a68c-6c7422e53388",
   "metadata": {},
   "source": [
    "Skriv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a963da0d-adf0-4ba7-9f00-db5d7930128f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "Started analysis of ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950158_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950158_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950159_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950159_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950160_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950160_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950161_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950161_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950162_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950162_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950163_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950163_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950177_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950177_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950178_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950178_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950179_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950179_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950180_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950180_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950181_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950181_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950182_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950182_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950186_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950186_2.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950187_1.fastq.gz.subset.fastq.gz\n",
      "Started analysis of ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 5% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 10% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 15% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 20% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 25% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 30% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 35% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 40% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 45% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 50% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 55% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 60% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 65% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 70% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 75% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 80% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 85% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 90% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 95% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "Approx 100% complete for ERR950187_2.fastq.gz.subset.fastq.gz\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=941590;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_rawdata_RNA-seq/fastqc_subset_rawdata\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━\u001b[91m╸\u001b[0m \u001b[35m 95%\u001b[━━━━━━━━━\u001b[0m \u001b[35m  0%\u001b[0m \u001b[32m0/56\u001b[0m  35m 20%\u001b[0m \u001b[32m11/56\u001b[0m \u001b[2mERR950179_1.fastq.gz.subset_fastqc.html\u001b[0m35m 38%\u001b[0m \u001b[32m21/56\u001b[0m \u001b[2mERR950178_2.fastq.gz.subset_fastqc.html\u001b[0m35m 59%\u001b[0m \u001b[32m33/56\u001b[0m \u001b[2mERR950158_1.fastq.gz.subset_fastqc.html\u001b[0m \u001b[35m 77%\u001b[0m \u001b[32m43/56\u001b[0m \u001b[2mERR950159_2.fastq.gz.subset_fastqc.zip\u001b[0m0m \u001b[32m53/56\u001b[0m \u001b[2mERR950158_2.fastq.gz.subset_fastqc.html\u001b[0m━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m56/56\u001b[0m  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 28 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_rawdata_RNA-seq\n",
    "mkdir -p fastqc_subset_rawdata\n",
    "fastqc -o fastqc_subset_rawdata *.fastq.gz > /dev/null\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_rawdata_RNA-seq/fastqc_subset_rawdata\n",
    "multiqc . > /dev/null"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ee5bec2-2fa4-4a15-b1f0-b1953ff18f7d",
   "metadata": {},
   "source": [
    "skriv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f6ff4792-d150-4e0f-bbc2-392e9b25307f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_rawdata_RNA-seq\n",
    "mkdir -p ../subset_trimmed_RNA-seq  # Create output directory if it doesn't exist\n",
    "\n",
    "for file in *_1.fastq.gz.subset.fastq.gz; do\n",
    "  # Extract the base filename (remove _1.fastq.gz.subset.fastq.gz)\n",
    "  base=\"${file%%_1.fastq.gz.subset.fastq.gz}\"\n",
    "  \n",
    "  # Define corresponding R2 file\n",
    "  file_R2=\"${base}_2.fastq.gz.subset.fastq.gz\"\n",
    "\n",
    "  # Run Cutadapt for paired-end reads\n",
    "  cutadapt \\\n",
    "    -a AGATCGGAAGAG -A AGATCGGAAGAG \\\n",
    "    --minimum-length 25 \\\n",
    "    -o /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq/\"${base}_trimmed_1.fastq.gz\" \\\n",
    "    -p /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq/\"${base}_trimmed_2.fastq.gz\" \\\n",
    "    \"$file\" \"$file_R2\" > /dev/null\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0f14481-b89d-4e64-b635-785048fa8300",
   "metadata": {},
   "source": [
    "skriv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e7322686-5c96-4c0c-9dcc-9b412e028aea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "Started analysis of ERR950158_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950158_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950158_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950158_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950159_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950159_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950159_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950159_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950160_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950160_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950160_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950160_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950161_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950161_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950161_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950161_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950162_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950162_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950162_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950162_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950163_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950163_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950163_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950163_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950177_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950177_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950177_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950177_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950178_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950178_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950178_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950178_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950179_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950179_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950179_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950179_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950180_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950180_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950180_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950180_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950181_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950181_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950181_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950181_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950182_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950182_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950182_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950182_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950186_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950186_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950186_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950186_trimmed_2.fastq.gz\n",
      "Started analysis of ERR950187_trimmed_1.fastq.gz\n",
      "Approx 5% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 10% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 15% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 20% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 25% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 30% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 35% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 40% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 45% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 50% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 55% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 60% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 65% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 70% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 75% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 80% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 85% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 90% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Approx 95% complete for ERR950187_trimmed_1.fastq.gz\n",
      "Started analysis of ERR950187_trimmed_2.fastq.gz\n",
      "Approx 5% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 10% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 15% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 20% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 25% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 30% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 35% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 40% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 45% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 50% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 55% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 60% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 65% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 70% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 75% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 80% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 85% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 90% complete for ERR950187_trimmed_2.fastq.gz\n",
      "Approx 95% complete for ERR950187_trimmed_2.fastq.gz\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=421486;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.27.1\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq/fastqc_subset_trimmed\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[91m�[90m━�━━━━━━━━━━━━━\u001b[0m \u001b[35m  0%\u001b[0m \u001b[32m0/56\u001b[0m  ��━━━━━━\u001b[0m \u001b[35m 18%\u001b[0m \u001b[32m10/56\u001b[0m \u001b[2mERR950178_trimmed_1_fastqc.zip\u001b[0m��━━━━━\u001b[0m \u001b[35m 36%\u001b[0m \u001b[32m20/56\u001b[0m \u001b[2mERR950160_trimmed_1_fastqc.html\u001b[0m��━━━━━\u001b[0m \u001b[35m 54%\u001b[0m \u001b[32m30/56\u001b[0m \u001b[2mERR950158_trimmed_2_fastqc.html\u001b[0m[0m\u001b[90m━━━━\u001b[0m \u001b[35m 71%\u001b[0m \u001b[32m40/56\u001b[0m \u001b[2mERR950160_trimmed_1_fastqc.zip\u001b[0m��\u001b[0m\u001b[90m━━\u001b[0m \u001b[35m 86%\u001b[0m \u001b[32m48/56\u001b[0m \u001b[2mERR950187_trimmed_2_fastqc.html\u001b[0m━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m56/56\u001b[0m  \n",
      "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 28 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq\n",
    "mkdir -p fastqc_subset_trimmed\n",
    "fastqc -o fastqc_subset_trimmed *.fastq.gz > /dev/null\n",
    "\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq/fastqc_subset_trimmed\n",
    "multiqc . > /dev/null"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba04c2da-0a97-4d41-b793-27960e59ebbd",
   "metadata": {},
   "source": [
    "# Preprocessing for RNA-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "270c08d6-369c-4769-9589-575c55ea3262",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data \n",
    "mkdir -p rawdata_RNA-seq \n",
    "mkdir -p trimmed_RNA-seq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91acd5ce-f872-45c0-a7c5-8f51bbf8838c",
   "metadata": {},
   "source": [
    "# Selecting the RNA-seq files and creating a symlink to the data for Hp and Atr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fa866ca0-d41b-494c-aaed-1802ed9d9cfc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Extract IDs for 'Atr' and 'Hp'nfrom the RNA-seq-list.txt file\n",
    "Hp_ids=($(grep -P '(Atr|Hp)' /cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNA-seq-list.txt | awk '{print $1}'))\n",
    "\n",
    "# Define directories\n",
    "source_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/Projects/Data/RNAseq\"\n",
    "destination_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/rawdata_RNA-seq\"\n",
    "\n",
    "# Loop through extracted IDs\n",
    "for id in \"${Hp_ids[@]}\"; do\n",
    "    # Store both file paths in an array\n",
    "    files=(\"$source_dir/${id}_1.fastq.gz\" \"$source_dir/${id}_2.fastq.gz\")\n",
    "\n",
    "    # Loop through each file\n",
    "    for file in \"${files[@]}\"; do\n",
    "        if [[ -f \"$file\" ]]; then \n",
    "            ln -s -t \"$destination_dir\" \"$file\"\n",
    "        else\n",
    "            echo \"File $file not found!\"\n",
    "        fi\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34d05989-61df-451b-b715-55fe6d9575e3",
   "metadata": {},
   "source": [
    "# Write in a Sbatch scripts (below) - note to myself"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4c30a05-c7da-4d62-a40b-90b301b206b2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/lmod/lmod/init/bash: /usr/share/lmod/lmod/libexec/addto: /usr/bin/lua: bad interpreter: No such file or directory\n",
      "Started analysis of ERR950158_1.fastq.gz\n",
      "Approx 5% complete for ERR950158_1.fastq.gz\n",
      "Approx 10% complete for ERR950158_1.fastq.gz\n",
      "Approx 15% complete for ERR950158_1.fastq.gz\n",
      "Approx 20% complete for ERR950158_1.fastq.gz\n",
      "Approx 25% complete for ERR950158_1.fastq.gz\n",
      "Approx 30% complete for ERR950158_1.fastq.gz\n",
      "Approx 35% complete for ERR950158_1.fastq.gz\n",
      "Approx 40% complete for ERR950158_1.fastq.gz\n",
      "Approx 45% complete for ERR950158_1.fastq.gz\n",
      "Approx 50% complete for ERR950158_1.fastq.gz\n",
      "Approx 55% complete for ERR950158_1.fastq.gz\n",
      "Approx 60% complete for ERR950158_1.fastq.gz\n",
      "Approx 65% complete for ERR950158_1.fastq.gz\n",
      "Approx 70% complete for ERR950158_1.fastq.gz\n",
      "Approx 75% complete for ERR950158_1.fastq.gz\n",
      "Approx 80% complete for ERR950158_1.fastq.gz\n",
      "Approx 85% complete for ERR950158_1.fastq.gz\n",
      "Approx 90% complete for ERR950158_1.fastq.gz\n",
      "Approx 95% complete for ERR950158_1.fastq.gz\n",
      "Started analysis of ERR950158_2.fastq.gz\n",
      "Approx 5% complete for ERR950158_2.fastq.gz\n",
      "Approx 10% complete for ERR950158_2.fastq.gz\n",
      "Approx 15% complete for ERR950158_2.fastq.gz\n",
      "Approx 20% complete for ERR950158_2.fastq.gz\n",
      "Approx 25% complete for ERR950158_2.fastq.gz\n",
      "Approx 30% complete for ERR950158_2.fastq.gz\n",
      "Approx 35% complete for ERR950158_2.fastq.gz\n",
      "Approx 40% complete for ERR950158_2.fastq.gz\n",
      "Approx 45% complete for ERR950158_2.fastq.gz\n",
      "Approx 50% complete for ERR950158_2.fastq.gz\n",
      "Approx 55% complete for ERR950158_2.fastq.gz\n",
      "Approx 60% complete for ERR950158_2.fastq.gz\n",
      "Approx 65% complete for ERR950158_2.fastq.gz\n",
      "Approx 70% complete for ERR950158_2.fastq.gz\n",
      "Approx 75% complete for ERR950158_2.fastq.gz\n",
      "Approx 80% complete for ERR950158_2.fastq.gz\n",
      "Approx 85% complete for ERR950158_2.fastq.gz\n",
      "Approx 90% complete for ERR950158_2.fastq.gz\n",
      "Approx 95% complete for ERR950158_2.fastq.gz\n",
      "Started analysis of ERR950159_1.fastq.gz\n",
      "Approx 5% complete for ERR950159_1.fastq.gz\n",
      "Approx 10% complete for ERR950159_1.fastq.gz\n",
      "Approx 15% complete for ERR950159_1.fastq.gz\n",
      "Approx 20% complete for ERR950159_1.fastq.gz\n",
      "Approx 25% complete for ERR950159_1.fastq.gz\n",
      "Approx 30% complete for ERR950159_1.fastq.gz\n",
      "Approx 35% complete for ERR950159_1.fastq.gz\n",
      "Approx 40% complete for ERR950159_1.fastq.gz\n",
      "Approx 45% complete for ERR950159_1.fastq.gz\n",
      "Approx 50% complete for ERR950159_1.fastq.gz\n",
      "Approx 55% complete for ERR950159_1.fastq.gz\n",
      "Approx 60% complete for ERR950159_1.fastq.gz\n",
      "Approx 65% complete for ERR950159_1.fastq.gz\n",
      "Approx 70% complete for ERR950159_1.fastq.gz\n",
      "Approx 75% complete for ERR950159_1.fastq.gz\n",
      "Approx 80% complete for ERR950159_1.fastq.gz\n",
      "Approx 85% complete for ERR950159_1.fastq.gz\n",
      "Approx 90% complete for ERR950159_1.fastq.gz\n",
      "Approx 95% complete for ERR950159_1.fastq.gz\n",
      "Started analysis of ERR950159_2.fastq.gz\n",
      "Approx 5% complete for ERR950159_2.fastq.gz\n",
      "Approx 10% complete for ERR950159_2.fastq.gz\n",
      "Approx 15% complete for ERR950159_2.fastq.gz\n",
      "Approx 20% complete for ERR950159_2.fastq.gz\n",
      "Approx 25% complete for ERR950159_2.fastq.gz\n",
      "Approx 30% complete for ERR950159_2.fastq.gz\n",
      "Approx 35% complete for ERR950159_2.fastq.gz\n",
      "Approx 40% complete for ERR950159_2.fastq.gz\n",
      "Approx 45% complete for ERR950159_2.fastq.gz\n",
      "Approx 50% complete for ERR950159_2.fastq.gz\n",
      "Approx 55% complete for ERR950159_2.fastq.gz\n",
      "Approx 60% complete for ERR950159_2.fastq.gz\n",
      "Approx 65% complete for ERR950159_2.fastq.gz\n",
      "Approx 70% complete for ERR950159_2.fastq.gz\n",
      "Approx 75% complete for ERR950159_2.fastq.gz\n",
      "Approx 80% complete for ERR950159_2.fastq.gz\n",
      "Approx 85% complete for ERR950159_2.fastq.gz\n",
      "Approx 90% complete for ERR950159_2.fastq.gz\n",
      "Approx 95% complete for ERR950159_2.fastq.gz\n",
      "Started analysis of ERR950160_1.fastq.gz\n",
      "Approx 5% complete for ERR950160_1.fastq.gz\n",
      "Approx 10% complete for ERR950160_1.fastq.gz\n",
      "Approx 15% complete for ERR950160_1.fastq.gz\n",
      "Approx 20% complete for ERR950160_1.fastq.gz\n",
      "Approx 25% complete for ERR950160_1.fastq.gz\n",
      "Approx 30% complete for ERR950160_1.fastq.gz\n",
      "Approx 35% complete for ERR950160_1.fastq.gz\n",
      "Approx 40% complete for ERR950160_1.fastq.gz\n",
      "Approx 45% complete for ERR950160_1.fastq.gz\n",
      "Approx 50% complete for ERR950160_1.fastq.gz\n",
      "Approx 55% complete for ERR950160_1.fastq.gz\n",
      "Approx 60% complete for ERR950160_1.fastq.gz\n",
      "Approx 65% complete for ERR950160_1.fastq.gz\n",
      "Approx 70% complete for ERR950160_1.fastq.gz\n",
      "Approx 75% complete for ERR950160_1.fastq.gz\n",
      "Approx 80% complete for ERR950160_1.fastq.gz\n",
      "Approx 85% complete for ERR950160_1.fastq.gz\n",
      "Approx 90% complete for ERR950160_1.fastq.gz\n",
      "Approx 95% complete for ERR950160_1.fastq.gz\n",
      "Started analysis of ERR950160_2.fastq.gz\n",
      "Approx 5% complete for ERR950160_2.fastq.gz\n",
      "Approx 10% complete for ERR950160_2.fastq.gz\n",
      "Approx 15% complete for ERR950160_2.fastq.gz\n",
      "Approx 20% complete for ERR950160_2.fastq.gz\n",
      "Approx 25% complete for ERR950160_2.fastq.gz\n",
      "Approx 30% complete for ERR950160_2.fastq.gz\n",
      "Approx 35% complete for ERR950160_2.fastq.gz\n",
      "Approx 40% complete for ERR950160_2.fastq.gz\n",
      "Approx 45% complete for ERR950160_2.fastq.gz\n",
      "Approx 50% complete for ERR950160_2.fastq.gz\n",
      "Approx 55% complete for ERR950160_2.fastq.gz\n",
      "Approx 60% complete for ERR950160_2.fastq.gz\n",
      "Approx 65% complete for ERR950160_2.fastq.gz\n",
      "Approx 70% complete for ERR950160_2.fastq.gz\n",
      "Approx 75% complete for ERR950160_2.fastq.gz\n",
      "Approx 80% complete for ERR950160_2.fastq.gz\n",
      "Approx 85% complete for ERR950160_2.fastq.gz\n",
      "Approx 90% complete for ERR950160_2.fastq.gz\n",
      "Approx 95% complete for ERR950160_2.fastq.gz\n",
      "Started analysis of ERR950161_1.fastq.gz\n",
      "Approx 5% complete for ERR950161_1.fastq.gz\n",
      "Approx 10% complete for ERR950161_1.fastq.gz\n",
      "Approx 15% complete for ERR950161_1.fastq.gz\n",
      "Approx 20% complete for ERR950161_1.fastq.gz\n",
      "Approx 25% complete for ERR950161_1.fastq.gz\n",
      "Approx 30% complete for ERR950161_1.fastq.gz\n",
      "Approx 35% complete for ERR950161_1.fastq.gz\n",
      "Approx 40% complete for ERR950161_1.fastq.gz\n",
      "Approx 45% complete for ERR950161_1.fastq.gz\n",
      "Approx 50% complete for ERR950161_1.fastq.gz\n",
      "Approx 55% complete for ERR950161_1.fastq.gz\n",
      "Approx 60% complete for ERR950161_1.fastq.gz\n",
      "Approx 65% complete for ERR950161_1.fastq.gz\n",
      "Approx 70% complete for ERR950161_1.fastq.gz\n",
      "Approx 75% complete for ERR950161_1.fastq.gz\n",
      "Approx 80% complete for ERR950161_1.fastq.gz\n",
      "Approx 85% complete for ERR950161_1.fastq.gz\n",
      "Approx 90% complete for ERR950161_1.fastq.gz\n",
      "Approx 95% complete for ERR950161_1.fastq.gz\n",
      "Started analysis of ERR950161_2.fastq.gz\n",
      "Approx 5% complete for ERR950161_2.fastq.gz\n",
      "Approx 10% complete for ERR950161_2.fastq.gz\n",
      "Approx 15% complete for ERR950161_2.fastq.gz\n",
      "Approx 20% complete for ERR950161_2.fastq.gz\n",
      "Approx 25% complete for ERR950161_2.fastq.gz\n",
      "Approx 30% complete for ERR950161_2.fastq.gz\n",
      "Approx 35% complete for ERR950161_2.fastq.gz\n",
      "Approx 40% complete for ERR950161_2.fastq.gz\n",
      "Approx 45% complete for ERR950161_2.fastq.gz\n",
      "Approx 50% complete for ERR950161_2.fastq.gz\n",
      "Approx 55% complete for ERR950161_2.fastq.gz\n",
      "Approx 60% complete for ERR950161_2.fastq.gz\n",
      "Approx 65% complete for ERR950161_2.fastq.gz\n",
      "Approx 70% complete for ERR950161_2.fastq.gz\n",
      "Approx 75% complete for ERR950161_2.fastq.gz\n",
      "Approx 80% complete for ERR950161_2.fastq.gz\n",
      "Approx 85% complete for ERR950161_2.fastq.gz\n",
      "Approx 90% complete for ERR950161_2.fastq.gz\n",
      "Approx 95% complete for ERR950161_2.fastq.gz\n",
      "Started analysis of ERR950162_1.fastq.gz\n",
      "Approx 5% complete for ERR950162_1.fastq.gz\n",
      "Approx 10% complete for ERR950162_1.fastq.gz\n",
      "Approx 15% complete for ERR950162_1.fastq.gz\n",
      "Approx 20% complete for ERR950162_1.fastq.gz\n",
      "Approx 25% complete for ERR950162_1.fastq.gz\n",
      "Approx 30% complete for ERR950162_1.fastq.gz\n",
      "Approx 35% complete for ERR950162_1.fastq.gz\n",
      "Approx 40% complete for ERR950162_1.fastq.gz\n",
      "Approx 45% complete for ERR950162_1.fastq.gz\n",
      "Approx 50% complete for ERR950162_1.fastq.gz\n",
      "Approx 55% complete for ERR950162_1.fastq.gz\n",
      "Approx 60% complete for ERR950162_1.fastq.gz\n",
      "Approx 65% complete for ERR950162_1.fastq.gz\n",
      "Approx 70% complete for ERR950162_1.fastq.gz\n",
      "Approx 75% complete for ERR950162_1.fastq.gz\n",
      "Approx 80% complete for ERR950162_1.fastq.gz\n",
      "Approx 85% complete for ERR950162_1.fastq.gz\n",
      "Approx 90% complete for ERR950162_1.fastq.gz\n",
      "Approx 95% complete for ERR950162_1.fastq.gz\n",
      "Started analysis of ERR950162_2.fastq.gz\n",
      "Approx 5% complete for ERR950162_2.fastq.gz\n",
      "Approx 10% complete for ERR950162_2.fastq.gz\n",
      "Approx 15% complete for ERR950162_2.fastq.gz\n",
      "Approx 20% complete for ERR950162_2.fastq.gz\n",
      "Approx 25% complete for ERR950162_2.fastq.gz\n",
      "Approx 30% complete for ERR950162_2.fastq.gz\n",
      "Approx 35% complete for ERR950162_2.fastq.gz\n",
      "Approx 40% complete for ERR950162_2.fastq.gz\n",
      "Approx 45% complete for ERR950162_2.fastq.gz\n",
      "Approx 50% complete for ERR950162_2.fastq.gz\n",
      "Approx 55% complete for ERR950162_2.fastq.gz\n",
      "Approx 60% complete for ERR950162_2.fastq.gz\n",
      "Approx 65% complete for ERR950162_2.fastq.gz\n",
      "Approx 70% complete for ERR950162_2.fastq.gz\n",
      "Approx 75% complete for ERR950162_2.fastq.gz\n",
      "Approx 80% complete for ERR950162_2.fastq.gz\n",
      "Approx 85% complete for ERR950162_2.fastq.gz\n",
      "Approx 90% complete for ERR950162_2.fastq.gz\n",
      "Approx 95% complete for ERR950162_2.fastq.gz\n",
      "Started analysis of ERR950163_1.fastq.gz\n",
      "Approx 5% complete for ERR950163_1.fastq.gz\n",
      "Approx 10% complete for ERR950163_1.fastq.gz\n",
      "Approx 15% complete for ERR950163_1.fastq.gz\n",
      "Approx 20% complete for ERR950163_1.fastq.gz\n",
      "Approx 25% complete for ERR950163_1.fastq.gz\n",
      "Approx 30% complete for ERR950163_1.fastq.gz\n",
      "Approx 35% complete for ERR950163_1.fastq.gz\n",
      "Approx 40% complete for ERR950163_1.fastq.gz\n",
      "Approx 45% complete for ERR950163_1.fastq.gz\n",
      "Approx 50% complete for ERR950163_1.fastq.gz\n",
      "Approx 55% complete for ERR950163_1.fastq.gz\n",
      "Approx 60% complete for ERR950163_1.fastq.gz\n",
      "Approx 65% complete for ERR950163_1.fastq.gz\n",
      "Approx 70% complete for ERR950163_1.fastq.gz\n",
      "Approx 75% complete for ERR950163_1.fastq.gz\n",
      "Approx 80% complete for ERR950163_1.fastq.gz\n",
      "Approx 85% complete for ERR950163_1.fastq.gz\n",
      "Approx 90% complete for ERR950163_1.fastq.gz\n",
      "Approx 95% complete for ERR950163_1.fastq.gz\n",
      "Started analysis of ERR950163_2.fastq.gz\n",
      "Approx 5% complete for ERR950163_2.fastq.gz\n",
      "Approx 10% complete for ERR950163_2.fastq.gz\n",
      "Approx 15% complete for ERR950163_2.fastq.gz\n",
      "Approx 20% complete for ERR950163_2.fastq.gz\n",
      "Approx 25% complete for ERR950163_2.fastq.gz\n",
      "Approx 30% complete for ERR950163_2.fastq.gz\n",
      "Approx 35% complete for ERR950163_2.fastq.gz\n",
      "Approx 40% complete for ERR950163_2.fastq.gz\n",
      "Approx 45% complete for ERR950163_2.fastq.gz\n",
      "Approx 50% complete for ERR950163_2.fastq.gz\n",
      "Approx 55% complete for ERR950163_2.fastq.gz\n",
      "Approx 60% complete for ERR950163_2.fastq.gz\n",
      "Approx 65% complete for ERR950163_2.fastq.gz\n",
      "Approx 70% complete for ERR950163_2.fastq.gz\n",
      "Approx 75% complete for ERR950163_2.fastq.gz\n",
      "Approx 80% complete for ERR950163_2.fastq.gz\n",
      "Approx 85% complete for ERR950163_2.fastq.gz\n",
      "Approx 90% complete for ERR950163_2.fastq.gz\n",
      "Approx 95% complete for ERR950163_2.fastq.gz\n",
      "Started analysis of ERR950177_1.fastq.gz\n",
      "Approx 5% complete for ERR950177_1.fastq.gz\n",
      "Approx 10% complete for ERR950177_1.fastq.gz\n",
      "Approx 15% complete for ERR950177_1.fastq.gz\n",
      "Approx 20% complete for ERR950177_1.fastq.gz\n",
      "Approx 25% complete for ERR950177_1.fastq.gz\n",
      "Approx 30% complete for ERR950177_1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#fastqc for rawdata\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/rawdata_RNA-seq\n",
    "mkdir -p fastqc_rawdata\n",
    "fastqc -o fastqc_rawdata *.fastq.gz > /dev/null\n",
    "#multiqc for fastqc reports from rawdata\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/rawdata_RNA-seq/fastqc_rawdata\n",
    "multiqc . > /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69b90a78-02ec-4dd5-8c9f-6517e0893da8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Trimming off adaptors\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/rawdata_RNA-seq\n",
    "mkdir -p ../trimmed_RNA-seq  # Create output directory if it doesn't exist\n",
    "\n",
    "for file in *_1.fastq.gz; do\n",
    "  # Extract the base filename (remove _1.fastq.gz.subset.fastq.gz)\n",
    "  base=\"${file%%_1.fastq.gz}\"\n",
    "  \n",
    "  # Define corresponding R2 file\n",
    "  file_R2=\"${base}_2.fastq.gz\"\n",
    "\n",
    "  # Run Cutadapt for paired-end reads\n",
    "  cutadapt \\\n",
    "    -a AGATCGGAAGAG -A AGATCGGAAGAG \\\n",
    "    --minimum-length 25 \\\n",
    "    -o /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/trimmed_RNA-seq/\"${base}_trimmed_1.fastq.gz\" \\\n",
    "    -p /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/trimmed_RNA-seq/\"${base}_trimmed_2.fastq.gz\" \\\n",
    "    \"$file\" \"$file_R2\" > /dev/null\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "904ecfb8-f458-4860-86b1-30a48d9d31d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#fastqc for trimmed data\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/trimmed_RNA-seq\n",
    "mkdir -p fastqc_trimmed\n",
    "fastqc -o fastqc_trimmed *.fastq.gz > /dev/null\n",
    "#multiqc for fastqc reports from trimmed data\n",
    "cd /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/trimmed_RNA-seq/fastqc_trimmed\n",
    "multiqc . > /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9b6c12a-53d3-4648-9d3f-4bb442de8f2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define paths\n",
    "GENOME_DIR=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/STAR_index\"\n",
    "FASTQ_DIR=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/subset_trimmed_RNA-seq/\"\n",
    "OUTPUT_DIR=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/alignment_STAR_subset\"\n",
    "\n",
    "# Create output directory if it doesn't exist\n",
    "mkdir -p \"$OUTPUT_DIR\"\n",
    "\n",
    "# Loop through FASTQ files\n",
    "for file1 in ${FASTQ_DIR}/*_1.fastq.gz; do\n",
    "    # Extract sample name\n",
    "    SAMPLE=$(basename \"$file1\" _1.fastq.gz)\n",
    "    file2=\"${FASTQ_DIR}/${SAMPLE}_2.fastq.gz\"\n",
    "\n",
    "    # Run STAR alignment\n",
    "    STAR --runThreadN 8 \\\n",
    "         --genomeDir \"$GENOME_DIR\" \\\n",
    "         --readFilesIn \"$file1\" \"$file2\" \\\n",
    "         --readFilesCommand zcat \\\n",
    "         --outFileNamePrefix \"${OUTPUT_DIR}/${SAMPLE}_\" \\\n",
    "         --sjdbGTFfile /cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/human_genome/Homo_sapiens.GRCh38.113.gtf \\\n",
    "         --outSAMtype BAM SortedByCoordinate \\\n",
    "         --quantMode TranscriptomeSAM GeneCounts\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c1d0d65-c386-4b00-80d0-621b0badbb4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variables\n",
    "gtf_file=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/human_genome/Homo_sapiens.GRCh38.113.gtf\"\n",
    "output_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_TEST\"\n",
    "aligned_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/alignment_STAR_subset/\"\n",
    "\n",
    "# Create output directory if it does not exist\n",
    "mkdir -p \"$output_dir\"\n",
    "\n",
    "\n",
    "# Loop through BAM files\n",
    "for bam_file in ${aligned_dir}/*sortedByCoord.out.bam; do\n",
    "    # Extract sample name\n",
    "    SAMPLE=$(basename \"$bam_file\" _Aligned.sortedByCoord.out.bam)\n",
    "    \n",
    "    # Define output files\n",
    "    sorted_bam_file=\"$output_dir/${SAMPLE}_sorted.bam\"\n",
    "    output_file=\"$output_dir/${SAMPLE}_htseq_counts.txt\"\n",
    "    log_file=\"$output_dir/${SAMPLE}_htseq-count.log\"\n",
    "\n",
    "    # Sort BAM file by name for htseq-count\n",
    "    samtools sort -n -o \"$sorted_bam_file\" \"$bam_file\"\n",
    "\n",
    "    # Check if samtools sort was successful\n",
    "    if [ $? -ne 0 ]; then\n",
    "        echo \"Error: samtools sort failed for $bam_file\" >&2\n",
    "        continue\n",
    "    fi\n",
    "\n",
    "    # Run htseq-count\n",
    "    htseq-count -f bam -r name -s no -t exon -i gene_id \"$sorted_bam_file\" \"$gtf_file\" > \"$output_file\" 2> \"$log_file\"\n",
    "\n",
    "    # Check if htseq-count ran successfully\n",
    "    if [ $? -eq 0 ]; then\n",
    "        echo \"htseq-count completed successfully. Output saved in $output_file\"\n",
    "    else\n",
    "        echo \"htseq-count failed for $bam_file. Check log file at $log_file\" >&2\n",
    "        echo \"htseq-count log:\"\n",
    "        cat \"$log_file\"\n",
    "    fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e60d999-0f98-435a-9a83-e1a8d48692f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "count_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_TEST\"\n",
    "sub_dir_count=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_TEST/sub_dir_count\"\n",
    "\n",
    "mkdir -p \"$sub_dir_count\"\n",
    "\n",
    "# Move BAM and log files to the sub-directory\n",
    "for file in \"$count_dir\"/*_sorted.bam \"$count_dir\"/*.log; do\n",
    "    if [ -e \"$file\" ]; then  # Check if the file exists\n",
    "        mv \"$file\" \"$sub_dir_count/\"\n",
    "    fi\n",
    "done\n",
    "\n",
    "echo \"Files moved successfully to $sub_dir_count\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b08bc71-75ba-49ff-a245-3b5e4a48d854",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define input directory containing the count files\n",
    "input_dir=\"/cephyr/NOBACKUP/groups/bbt045_2025/groups/group_Bjork/data/CN_gene_counts_TEST\"\n",
    "\n",
    "# Define output file\n",
    "output_file=\"$input_dir/merged_counts.txt\"\n",
    "\n",
    "# Get list of count files\n",
    "count_files=(\"$input_dir\"/*_trimmed_htseq_counts.txt)\n",
    "\n",
    "# Check if there are count files\n",
    "if [ ${#count_files[@]} -eq 0 ]; then\n",
    "    echo \"Error: No count files found in $input_dir\" >&2\n",
    "    exit 1\n",
    "fi\n",
    "\n",
    "# Use the first file as a reference to check gene consistency\n",
    "first_file=\"${count_files[0]}\"\n",
    "awk '{print $1}' \"$first_file\" | sort > /tmp/gene_list_reference.txt\n",
    "\n",
    "# Merge files based on first column\n",
    "awk -v ref_file=\"/tmp/gene_list_reference.txt\" '\n",
    "    BEGIN { OFS=\"\\t\"; while ((getline < ref_file) > 0) ref_genes[$1] = 1; close(ref_file) }\n",
    "    FNR==1 { files++ }  # Track number of files\n",
    "    {\n",
    "        gene_id = $1\n",
    "        count = $2\n",
    "        if (!(gene_id in ref_genes)) {\n",
    "            print \"Error: Gene \" gene_id \" is missing from one or more files!\" > \"/dev/stderr\"\n",
    "            exit 1\n",
    "        }\n",
    "        counts[gene_id, FILENAME] = count\n",
    "        genes[gene_id]  # Store unique gene IDs\n",
    "    }\n",
    "    END {\n",
    "        # Print header\n",
    "        printf \"Gene_ID\"\n",
    "        for (i in ARGV) {\n",
    "            if (ARGV[i] ~ /_trimmed_htseq_counts.txt$/) {\n",
    "                sample_name = ARGV[i]\n",
    "                sub(\".*/\", \"\", sample_name)  # Remove path\n",
    "                sub(\"_trimmed_htseq_counts.txt\", \"\", sample_name)  # Extract sample ID\n",
    "                printf \"\\t%s\", sample_name\n",
    "            }\n",
    "        }\n",
    "        print \"\"\n",
    "\n",
    "        # Print merged counts\n",
    "        for (gene in genes) {\n",
    "            split(gene, arr, SUBSEP)\n",
    "            printf \"%s\", arr[1]  # Print gene ID\n",
    "            for (i in ARGV) {\n",
    "                if (ARGV[i] ~ /_trimmed_htseq_counts.txt$/) {\n",
    "                    file = ARGV[i]\n",
    "                    printf \"\\t%s\", (counts[arr[1], file] ? counts[arr[1], file] : 0)\n",
    "                }\n",
    "            }\n",
    "            print \"\"\n",
    "        }\n",
    "    }\n",
    "' \"${count_files[@]}\" > \"$output_file\"\n",
    "\n",
    "echo \"Merged file saved as: $output_file\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
